{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "441dbb2a",
   "metadata": {},
   "source": [
    "## This is just a playground. See finished standalone python script at ~/GitHub/YBD/genename_to_bedfile.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "bcf7c786",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curl 'https://data.orthodb.org/current/search?query=SAUR&level=4447' -L -o SAUR_at4447.dat\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  5896    0  5896    0     0   3234      0 --:--:--  0:00:01 --:--:--  3234\n"
     ]
    }
   ],
   "source": [
    "import subprocess, json\n",
    "import pandas as pd\n",
    "\n",
    "gene_name = 'SAUR'\n",
    "level = 4447\n",
    "# Function to parse the 'Attributes' column in gff file and convert it into a dictionary of key-value pairs\n",
    "def parse_attributes(attribute_str):\n",
    "    attribute_pairs = attribute_str.split(';')\n",
    "    attribute_dict = {}\n",
    "    for pair in attribute_pairs:\n",
    "        if len(pair.split('=')) == 2:\n",
    "            key, value = pair.split('=')\n",
    "        else:\n",
    "            key = pair.split('=')[0]\n",
    "            value = ''\n",
    "        attribute_dict[key] = value\n",
    "    return attribute_dict\n",
    "anno_file = \"/bioinfo2/palm/ref/orthoDBv11/dura{}_Results/dura{}.og.annotations\".format(level, level)\n",
    "gff_file = '/bioinfo2/palm/ref/dura/dura_ref.gff'\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "cmd = \"curl 'https://data.orthodb.org/current/search?query={}&level={}' -L -o {}_at{}.dat\".format(gene_name, level, gene_name, level)\n",
    "print(cmd)\n",
    "subprocess.run(cmd, shell=True, check=True)\n",
    "with open(\"{}_at{}.dat\".format(gene_name, level), \"r\") as json_file:\n",
    "    data_dict = json.load(json_file)\n",
    "\n",
    "df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "selected_rows = df[df['ODB_OG'].isin(data_dict['data'])]\n",
    "selected_rows = selected_rows.rename(columns={'#query': 'query'})\n",
    "selected_rows['gene'] = selected_rows['query'].str.split('-').str[0]\n",
    "\n",
    "# Filter rows where value from column A is in the list\n",
    "filtered_rows = df_gff[df_gff['ID'].isin(selected_rows['gene'])]\n",
    "\n",
    "# write bed file\n",
    "df = filtered_rows[['Seqid', 'Start','End','ID']]\n",
    "df_no_duplicates = df.drop_duplicates()\n",
    "df_sorted = df_no_duplicates.sort_values(by=['Seqid', 'Start'])\n",
    "# add pound(#) to the first column name - convention of bed file\n",
    "df_sorted.columns = ['#' + df_sorted.columns[0] if i == 0 else col for i, col in enumerate(df_sorted.columns)]\n",
    "bedfile = gene_name + '_at' + level + '_dura.bed'\n",
    "df_sorted.to_csv(bedfile, sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c575e40f",
   "metadata": {},
   "source": [
    "# Write bed file based on a list of transcript names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b5f4774d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "def parse_attributes(attribute_str):\n",
    "    attribute_pairs = attribute_str.split(';')\n",
    "    attribute_dict = {}\n",
    "    for pair in attribute_pairs:\n",
    "        if len(pair.split('=')) == 2:\n",
    "            key, value = pair.split('=')\n",
    "        else:\n",
    "            key = pair.split('=')[0]\n",
    "            value = ''\n",
    "        attribute_dict[key] = value\n",
    "    return attribute_dict\n",
    "with open('/bioinfo2/palm/accession/Apriyanto2023/DE_table.genes', 'r') as f:\n",
    "    genes = [line.strip() for line in f]\n",
    "gff_file = '/bioinfo2/palm/ref/dura/dura_ref.gff'\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "\n",
    "# Filter rows where value from column A is in the gene list\n",
    "filtered_rows = df_gff[df_gff['ID'].isin(genes)]\n",
    "\n",
    "# write bed file\n",
    "df = filtered_rows[['Seqid', 'Start','End','ID']]\n",
    "df_no_duplicates = df.drop_duplicates()\n",
    "df_sorted = df_no_duplicates.sort_values(by=['Seqid', 'Start'])\n",
    "# add pound(#) to the first column name - convention of bed file\n",
    "df_sorted.columns = ['#' + df_sorted.columns[0] if i == 0 else col for i, col in enumerate(df_sorted.columns)]\n",
    "bedfile = 'Apriyanto2023_DE100.bed'\n",
    "df_sorted.to_csv(bedfile, sep='\\t', index=False)\n",
    "# the rest of the analysis please see https://www.notion.so/Association-studies-e780ce66173e44849149291fed4c0e52?pvs=4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acd430c8",
   "metadata": {},
   "source": [
    "# To look up any SNP from the genome (slow)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d12a2eb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# first let's work with just one coord\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq, MutableSeq\n",
    "\n",
    "# eventually this will be a script that only takes the coord, and go look the exon and REF ALT by it self.\n",
    "# * means stop codon. When this is changed we need to look further down the sequence until another stop codon right?\n",
    "\n",
    "coord = 'chr1_8180009'\n",
    "gff_file = '/bioinfo2/palm/ref/dura/dura_ref.gff'\n",
    "\n",
    "CHR, POS = coord.split('_')\n",
    "POS = int(POS)\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "filtered_df = df_gff[(df_gff['Type'] == \"exon\") & (df['Seqid'] == CHR) & (df['Start'] <= POS) & (df['End'] >= POS)]\n",
    "gene = filtered_df['ID'].iloc[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "11eab720",
   "metadata": {},
   "outputs": [],
   "source": [
    "in_file = '/bioinfo2/palm/analysis/batch1/batch1_chr1_nosex.assoc.qassoc.tab'\n",
    "in_df = pd.read_csv(in_file, sep = '\\t', header=0)\n",
    "in_df['coord'] = 'chr' + in_df['CHR'].astype(str) + '_' + in_df['BP'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "a22bef38",
   "metadata": {},
   "outputs": [],
   "source": [
    "in_df1 = in_df[in_df['P'] < 0.01]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "d6c1418c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4c3daee0>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex', 'ins', 'del')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del', 'mnp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins', 'ins', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins', 'del', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('mnp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del', 'ins', 'del')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n"
     ]
    }
   ],
   "source": [
    "vcf_file = '/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100.exon.b1ch1.vcf'\n",
    "vcf = pysam.VariantFile(vcf_file)\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "f58fc08c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('del',)\n",
      "('ins', 'ins')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex', 'ins', 'del')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex', 'snp', 'ins')\n",
      "('snp',)\n",
      "('del', 'mnp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins', 'snp')\n",
      "('snp',)\n",
      "('ins', 'ins', 'ins', 'snp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins', 'ins', 'del', 'ins')\n",
      "('del',)\n",
      "('snp',)\n",
      "('del',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('snp',)\n",
      "('complex', 'snp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('mnp',)\n",
      "('complex',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex', 'snp')\n",
      "('snp',)\n",
      "('del',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('del', 'ins', 'del')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('complex', 'snp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins', 'ins')\n",
      "('snp',)\n",
      "('snp',)\n"
     ]
    }
   ],
   "source": [
    "for variant in vcf.fetch():\n",
    "    print(variant.info[\"TYPE\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "9d8ffc7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>UNADJ</th>\n",
       "      <th>GC</th>\n",
       "      <th>BONF</th>\n",
       "      <th>HOLM</th>\n",
       "      <th>SIDAK_SS</th>\n",
       "      <th>SIDAK_SD</th>\n",
       "      <th>FDR_BH</th>\n",
       "      <th>FDR_BY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_151950957</td>\n",
       "      <td>7.533e-42</td>\n",
       "      <td>1.863000e-07</td>\n",
       "      <td>5.036000e-35</td>\n",
       "      <td>5.036000e-35</td>\n",
       "      <td>inf</td>\n",
       "      <td>inf</td>\n",
       "      <td>5.036000e-35</td>\n",
       "      <td>8.206000e-34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_4763349</td>\n",
       "      <td>1.051e-38</td>\n",
       "      <td>2.387000e-17</td>\n",
       "      <td>7.026000e-32</td>\n",
       "      <td>7.026000e-32</td>\n",
       "      <td>inf</td>\n",
       "      <td>inf</td>\n",
       "      <td>2.387000e-32</td>\n",
       "      <td>3.889000e-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_1902364</td>\n",
       "      <td>1.687e-37</td>\n",
       "      <td>7.533000e-42</td>\n",
       "      <td>1.128000e-30</td>\n",
       "      <td>1.128000e-30</td>\n",
       "      <td>inf</td>\n",
       "      <td>inf</td>\n",
       "      <td>2.387000e-32</td>\n",
       "      <td>3.889000e-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_96141504</td>\n",
       "      <td>2.577e-37</td>\n",
       "      <td>5.654000e-34</td>\n",
       "      <td>1.723000e-30</td>\n",
       "      <td>1.723000e-30</td>\n",
       "      <td>inf</td>\n",
       "      <td>inf</td>\n",
       "      <td>2.387000e-32</td>\n",
       "      <td>3.889000e-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_158324180</td>\n",
       "      <td>2.604e-37</td>\n",
       "      <td>6.030000e-34</td>\n",
       "      <td>1.741000e-30</td>\n",
       "      <td>1.741000e-30</td>\n",
       "      <td>inf</td>\n",
       "      <td>inf</td>\n",
       "      <td>2.387000e-32</td>\n",
       "      <td>3.889000e-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_22292153</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_18289297</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_40755887</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_106500245</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_106501777</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6685964 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHR             SNP      UNADJ            GC          BONF          HOLM  \\\n",
       "NaN    2  chr2_151950957  7.533e-42  1.863000e-07  5.036000e-35  5.036000e-35   \n",
       "NaN    2    chr2_4763349  1.051e-38  2.387000e-17  7.026000e-32  7.026000e-32   \n",
       "NaN    2    chr2_1902364  1.687e-37  7.533000e-42  1.128000e-30  1.128000e-30   \n",
       "NaN    2   chr2_96141504  2.577e-37  5.654000e-34  1.723000e-30  1.723000e-30   \n",
       "NaN    2  chr2_158324180  2.604e-37  6.030000e-34  1.741000e-30  1.741000e-30   \n",
       "..   ...             ...        ...           ...           ...           ...   \n",
       "NaN    2   chr2_22292153          1  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "NaN    2   chr2_18289297          1  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "NaN    2   chr2_40755887          1  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "NaN    2  chr2_106500245          1  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "NaN    2  chr2_106501777          1  1.000000e+00  1.000000e+00  1.000000e+00   \n",
       "\n",
       "     SIDAK_SS  SIDAK_SD        FDR_BH        FDR_BY  \n",
       "NaN       inf       inf  5.036000e-35  8.206000e-34  \n",
       "NaN       inf       inf  2.387000e-32  3.889000e-31  \n",
       "NaN       inf       inf  2.387000e-32  3.889000e-31  \n",
       "NaN       inf       inf  2.387000e-32  3.889000e-31  \n",
       "NaN       inf       inf  2.387000e-32  3.889000e-31  \n",
       "..        ...       ...           ...           ...  \n",
       "NaN       1.0       1.0  1.000000e+00  1.000000e+00  \n",
       "NaN       1.0       1.0  1.000000e+00  1.000000e+00  \n",
       "NaN       1.0       1.0  1.000000e+00  1.000000e+00  \n",
       "NaN       1.0       1.0  1.000000e+00  1.000000e+00  \n",
       "NaN       1.0       1.0  1.000000e+00  1.000000e+00  \n",
       "\n",
       "[6685964 rows x 10 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "in_df_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "8546f234",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>R2</th>\n",
       "      <th>T</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_128</td>\n",
       "      <td>128</td>\n",
       "      <td>103</td>\n",
       "      <td>-37.200</td>\n",
       "      <td>22.27</td>\n",
       "      <td>0.026880</td>\n",
       "      <td>-1.67000</td>\n",
       "      <td>0.09794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_133</td>\n",
       "      <td>133</td>\n",
       "      <td>103</td>\n",
       "      <td>17.600</td>\n",
       "      <td>57.94</td>\n",
       "      <td>0.000912</td>\n",
       "      <td>0.30370</td>\n",
       "      <td>0.76200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_180</td>\n",
       "      <td>180</td>\n",
       "      <td>102</td>\n",
       "      <td>-31.630</td>\n",
       "      <td>21.00</td>\n",
       "      <td>0.022180</td>\n",
       "      <td>-1.50600</td>\n",
       "      <td>0.13520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_185</td>\n",
       "      <td>185</td>\n",
       "      <td>102</td>\n",
       "      <td>-31.630</td>\n",
       "      <td>21.00</td>\n",
       "      <td>0.022180</td>\n",
       "      <td>-1.50600</td>\n",
       "      <td>0.13520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_240</td>\n",
       "      <td>240</td>\n",
       "      <td>104</td>\n",
       "      <td>-1.024</td>\n",
       "      <td>43.88</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>-0.02333</td>\n",
       "      <td>0.98140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6698015</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_160141624</td>\n",
       "      <td>160141624</td>\n",
       "      <td>62</td>\n",
       "      <td>-18.050</td>\n",
       "      <td>41.85</td>\n",
       "      <td>0.003091</td>\n",
       "      <td>-0.43130</td>\n",
       "      <td>0.66780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6698016</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_160147416</td>\n",
       "      <td>160147416</td>\n",
       "      <td>23</td>\n",
       "      <td>-26.320</td>\n",
       "      <td>155.30</td>\n",
       "      <td>0.001366</td>\n",
       "      <td>-0.16950</td>\n",
       "      <td>0.86700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6698017</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_160147423</td>\n",
       "      <td>160147423</td>\n",
       "      <td>23</td>\n",
       "      <td>-33.490</td>\n",
       "      <td>111.80</td>\n",
       "      <td>0.004256</td>\n",
       "      <td>-0.29960</td>\n",
       "      <td>0.76740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6698018</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_160147477</td>\n",
       "      <td>160147477</td>\n",
       "      <td>27</td>\n",
       "      <td>-18.850</td>\n",
       "      <td>156.60</td>\n",
       "      <td>0.000579</td>\n",
       "      <td>-0.12040</td>\n",
       "      <td>0.90510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6698019</th>\n",
       "      <td>2</td>\n",
       "      <td>chr2_160147491</td>\n",
       "      <td>160147491</td>\n",
       "      <td>27</td>\n",
       "      <td>-18.850</td>\n",
       "      <td>156.60</td>\n",
       "      <td>0.000579</td>\n",
       "      <td>-0.12040</td>\n",
       "      <td>0.90510</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6698020 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHR             SNP         BP  NMISS    BETA      SE        R2  \\\n",
       "0         2        chr2_128        128    103 -37.200   22.27  0.026880   \n",
       "1         2        chr2_133        133    103  17.600   57.94  0.000912   \n",
       "2         2        chr2_180        180    102 -31.630   21.00  0.022180   \n",
       "3         2        chr2_185        185    102 -31.630   21.00  0.022180   \n",
       "4         2        chr2_240        240    104  -1.024   43.88  0.000005   \n",
       "...      ..             ...        ...    ...     ...     ...       ...   \n",
       "6698015   2  chr2_160141624  160141624     62 -18.050   41.85  0.003091   \n",
       "6698016   2  chr2_160147416  160147416     23 -26.320  155.30  0.001366   \n",
       "6698017   2  chr2_160147423  160147423     23 -33.490  111.80  0.004256   \n",
       "6698018   2  chr2_160147477  160147477     27 -18.850  156.60  0.000579   \n",
       "6698019   2  chr2_160147491  160147491     27 -18.850  156.60  0.000579   \n",
       "\n",
       "               T        P  \n",
       "0       -1.67000  0.09794  \n",
       "1        0.30370  0.76200  \n",
       "2       -1.50600  0.13520  \n",
       "3       -1.50600  0.13520  \n",
       "4       -0.02333  0.98140  \n",
       "...          ...      ...  \n",
       "6698015 -0.43130  0.66780  \n",
       "6698016 -0.16950  0.86700  \n",
       "6698017 -0.29960  0.76740  \n",
       "6698018 -0.12040  0.90510  \n",
       "6698019 -0.12040  0.90510  \n",
       "\n",
       "[6698020 rows x 9 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "in_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "c9bd95e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df1 = merged_df[merged_df['BONF'] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "cdf84c55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(38451, 17)"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "f7d63c79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/bioinfo2/palm/analysis/batch1/batch1_chr2_nosex.assoc.qassoc.adjusted.tab'"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "in_file_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "f65a7973",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_file = '/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100.exon.b1ch2.vcf'\n",
    "for variant in vcf.fetch():\n",
    "    TYPE = variant.info[\"TYPE\"]\n",
    "    type_out = '|'.join([i for i in TYPE])\n",
    "    if variant.id == coord:\n",
    "        REF = variant.ref\n",
    "        ALT = variant.alts[0]\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "c882dfff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHR          object\n",
       "SNP          object\n",
       "BP            int64\n",
       "NMISS         int64\n",
       "BETA        float64\n",
       "SE          float64\n",
       "R2          float64\n",
       "T           float64\n",
       "P           float64\n",
       "UNADJ       float64\n",
       "GC          float64\n",
       "BONF        float64\n",
       "HOLM        float64\n",
       "SIDAK_SS    float64\n",
       "SIDAK_SD    float64\n",
       "FDR_BH      float64\n",
       "FDR_BY      float64\n",
       "dtype: object"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "in_df.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "772ef95e",
   "metadata": {},
   "source": [
    "# Look up SNPs from a designated bed file (fast)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "a9a16937",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Columns must be same length as key",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[185], line 91\u001b[0m\n\u001b[1;32m     88\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcustom_series\u001b[39m(x):\n\u001b[1;32m     89\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m pd\u001b[38;5;241m.\u001b[39mSeries(x, dtype\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mobject\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m---> 91\u001b[0m \u001b[43min_df\u001b[49m\u001b[43m[\u001b[49m\u001b[43mcol_names\u001b[49m\u001b[43m]\u001b[49m \u001b[38;5;241m=\u001b[39m in_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSNP\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mapply(coord2codon)\u001b[38;5;241m.\u001b[39mapply(custom_series)\n\u001b[1;32m     92\u001b[0m \u001b[38;5;66;03m#in_df = in_df.drop(in_df.columns[0], axis=1)\u001b[39;00m\n\u001b[1;32m     93\u001b[0m in_df\u001b[38;5;241m.\u001b[39mto_csv(out_file, index\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/pandas/core/frame.py:3968\u001b[0m, in \u001b[0;36mDataFrame.__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3966\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_setitem_frame(key, value)\n\u001b[1;32m   3967\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(key, (Series, np\u001b[38;5;241m.\u001b[39mndarray, \u001b[38;5;28mlist\u001b[39m, Index)):\n\u001b[0;32m-> 3968\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_setitem_array\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3969\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(value, DataFrame):\n\u001b[1;32m   3970\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_set_item_frame_value(key, value)\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/pandas/core/frame.py:4010\u001b[0m, in \u001b[0;36mDataFrame._setitem_array\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   4005\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   4006\u001b[0m     \u001b[38;5;66;03m# Note: unlike self.iloc[:, indexer] = value, this will\u001b[39;00m\n\u001b[1;32m   4007\u001b[0m     \u001b[38;5;66;03m#  never try to overwrite values inplace\u001b[39;00m\n\u001b[1;32m   4009\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(value, DataFrame):\n\u001b[0;32m-> 4010\u001b[0m         \u001b[43mcheck_key_length\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   4011\u001b[0m         \u001b[38;5;28;01mfor\u001b[39;00m k1, k2 \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(key, value\u001b[38;5;241m.\u001b[39mcolumns):\n\u001b[1;32m   4012\u001b[0m             \u001b[38;5;28mself\u001b[39m[k1] \u001b[38;5;241m=\u001b[39m value[k2]\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/pandas/core/indexers/utils.py:401\u001b[0m, in \u001b[0;36mcheck_key_length\u001b[0;34m(columns, key, value)\u001b[0m\n\u001b[1;32m    399\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m columns\u001b[38;5;241m.\u001b[39mis_unique:\n\u001b[1;32m    400\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(value\u001b[38;5;241m.\u001b[39mcolumns) \u001b[38;5;241m!=\u001b[39m \u001b[38;5;28mlen\u001b[39m(key):\n\u001b[0;32m--> 401\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mColumns must be same length as key\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    402\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    403\u001b[0m     \u001b[38;5;66;03m# Missing keys in columns are represented as -1\u001b[39;00m\n\u001b[1;32m    404\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(columns\u001b[38;5;241m.\u001b[39mget_indexer_non_unique(key)[\u001b[38;5;241m0\u001b[39m]) \u001b[38;5;241m!=\u001b[39m \u001b[38;5;28mlen\u001b[39m(value\u001b[38;5;241m.\u001b[39mcolumns):\n",
      "\u001b[0;31mValueError\u001b[0m: Columns must be same length as key"
     ]
    }
   ],
   "source": [
    "import pysam\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq, MutableSeq\n",
    "from Bio import SeqIO\n",
    "\n",
    "# Input\n",
    "out_file = \"/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100_batch1_chr2.summary\"\n",
    "bed_file = '/bioinfo2/palm/analysis/yield/Apriyanto2023_DE100.exon.bed'\n",
    "# prepare vcf by: bcftools view -R /bioinfo2/palm/analysis/yield/Apriyanto2023_DE100.exon.bed /bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr2_lc200_C5.bcf > Apriyanto2023_DE100.exon.b1ch2.vcf\n",
    "vcf_file = '/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100.exon.b1ch2.vcf'\n",
    "# tab file is prepared by \n",
    "in_file = '/bioinfo2/palm/analysis/batch1/batch1_chr2_nosex.assoc.qassoc.tab'\n",
    "p = 0.01\n",
    "\n",
    "# Main code\n",
    "in_df = pd.read_csv(in_file, sep = '\\t', header=0, dtype={'CHR':str, 'SNP': str})\n",
    "in_df = in_df.drop([col for col in in_df.columns if col.startswith('Unnamed:')], axis=1)\n",
    "in_file_adj = in_file.split('.tab')[0] + '.adjusted.tab'\n",
    "in_df_adj = pd.read_csv(in_file_adj, sep = '\\t', header=0)\n",
    "colnames = list(in_df_adj.columns)\n",
    "in_df_adj = in_df_adj.iloc[:, :-1]\n",
    "in_df_adj.columns = colnames[1:]\n",
    "in_df_adj['CHR'] = in_df_adj['CHR'].astype(str)\n",
    "in_df = in_df.merge(in_df_adj)\n",
    "in_df = in_df[in_df['BONF'] < p]\n",
    "bed_column_names = ['Seqid', 'Start', 'End', 'Genes']\n",
    "bed_df = pd.read_csv(bed_file, sep='\\t',comment='#', names=bed_column_names)\n",
    "vcf = pysam.VariantFile(vcf_file, \"r\")\n",
    "genome_fa = '/bioinfo2/palm/ref/dura/dura_ref.fasta'\n",
    "anno_file = \"/bioinfo2/palm/ref/dura/dura_orthoDBv11_4447_3193_33090.og.annotations\"\n",
    "anno_df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "anno_df = anno_df.rename(columns={'#query': 'query'})\n",
    "\n",
    "def coord2codon(coord, bed_df = bed_df, vcf = vcf, genome_fa = genome_fa, anno_df = anno_df):\n",
    "    # this function takes a coordinate of a SNP in the dura genome and returns some useful info about it\n",
    "    CHR, POS = coord.split('_')\n",
    "    POS = int(POS)\n",
    "    # look up gene from bed file\n",
    "    snp_df = bed_df[(bed_df['Seqid'] == CHR) & (bed_df['Start'] <= POS) & (bed_df['End'] >= POS)]\n",
    "    if len(snp_df) > 0: # if snp is in the region we care\n",
    "        print(coord)\n",
    "        start = int(snp_df['Start'].iloc[0])\n",
    "        end = int(snp_df['End'].iloc[0])\n",
    "        gene = snp_df['Genes'].iloc[0]\n",
    "        anno_df = anno_df[anno_df['query'] == gene]\n",
    "        if len(anno_df) == 1:\n",
    "            og = anno_df['ODB_OG'].iloc[0]\n",
    "            gene_name = anno_df['Description'].iloc[0]\n",
    "        elif len(anno_df) == 0:\n",
    "            og = 'NA'\n",
    "            gene_name = 'NA'\n",
    "        else:\n",
    "            print(gene, \" has more than one og, recording the first one\")\n",
    "            og = anno_df['ODB_OG'].iloc[0]\n",
    "            gene_name = anno_df['Description'].iloc[0]\n",
    "        # look up REF and ALT from vcf\n",
    "        for variant in vcf.fetch():\n",
    "            TYPE = variant.info[\"TYPE\"]\n",
    "            type_out = '|'.join([i for i in TYPE])\n",
    "            if variant.id == coord:\n",
    "                REF = variant.ref\n",
    "                ALT = variant.alts[0]\n",
    "                break\n",
    "        # get the codon and amino acid\n",
    "        for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "            if record.id == CHR:\n",
    "                exon_seq = record.seq[start-1:end]\n",
    "                exon_seq_var = MutableSeq(str(exon_seq))\n",
    "                protein_seq = exon_seq.translate()\n",
    "                snp_position = POS - start\n",
    "                # Find the codon's start position\n",
    "                codon_start = snp_position - (snp_position % 3)\n",
    "                # Extract the codons\n",
    "                codon = exon_seq[codon_start: codon_start+3]\n",
    "                exon_seq_var[snp_position: snp_position+len(ALT)] = ALT\n",
    "                modified_codon = exon_seq_var[codon_start: codon_start+3]    \n",
    "                original_aa = codon.translate()\n",
    "                modified_aa = modified_codon.translate()\n",
    "                break\n",
    "\n",
    "        print(';'.join([coord, gene, og, gene_name, type_out, REF, ALT, str(codon), str(modified_codon), str(original_aa), str(modified_aa)]))\n",
    "        return [gene, og, gene_name, type_out, REF, ALT, str(codon), str(modified_codon), str(original_aa), str(modified_aa)]\n",
    "\n",
    "# new column names\n",
    "col_names = ['gene', 'og', 'gene_name', 'Type','REF', 'ALT', 'codon', 'modified_codon', 'original_aa', 'modified_aa']\n",
    "\n",
    "# Generate lists and split them into separate columns\n",
    "def custom_series(x):\n",
    "    return pd.Series(x, dtype='str')\n",
    "\n",
    "in_df[col_names] = in_df['SNP'].apply(coord2codon).apply(custom_series)\n",
    "#in_df = in_df.drop(in_df.columns[0], axis=1)\n",
    "in_df.to_csv(out_file, index=False)\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "f7b071c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1_86758000\n",
      "chr1_72869463 chr1_86758000\n",
      "chr1_72869477 chr1_86758000\n",
      "chr1_72869507 chr1_86758000\n",
      "chr1_72869527 chr1_86758000\n",
      "chr1_72869566 chr1_86758000\n",
      "chr1_72869593 chr1_86758000\n",
      "chr1_72869712 chr1_86758000\n",
      "chr1_72869721 chr1_86758000\n",
      "chr1_72869790 chr1_86758000\n",
      "chr1_72870090 chr1_86758000\n",
      "chr1_72870446 chr1_86758000\n",
      "chr1_72870533 chr1_86758000\n",
      "chr1_72870659 chr1_86758000\n",
      "chr1_72870738 chr1_86758000\n",
      "chr1_72870881 chr1_86758000\n",
      "chr1_72871070 chr1_86758000\n",
      "chr1_72871270 chr1_86758000\n",
      "chr1_72871276 chr1_86758000\n",
      "chr1_72871319 chr1_86758000\n",
      "chr1_72871462 chr1_86758000\n",
      "chr1_72871561 chr1_86758000\n",
      "chr1_72871675 chr1_86758000\n",
      "chr1_72871697 chr1_86758000\n",
      "chr1_72871709 chr1_86758000\n",
      "chr1_72871771 chr1_86758000\n",
      "chr1_72871920 chr1_86758000\n",
      "chr1_72872008 chr1_86758000\n",
      "chr1_72872324 chr1_86758000\n",
      "chr1_72872376 chr1_86758000\n",
      "chr1_72872392 chr1_86758000\n",
      "chr1_72872437 chr1_86758000\n",
      "chr1_72872615 chr1_86758000\n",
      "chr1_72882701 chr1_86758000\n",
      "chr1_72882829 chr1_86758000\n",
      "chr1_72883799 chr1_86758000\n",
      "chr1_72883850 chr1_86758000\n",
      "chr1_72890943 chr1_86758000\n",
      "chr1_72891132 chr1_86758000\n",
      "chr1_72891217 chr1_86758000\n",
      "chr1_72891238 chr1_86758000\n",
      "chr1_72891360 chr1_86758000\n",
      "chr1_72891372 chr1_86758000\n",
      "chr1_72891395 chr1_86758000\n",
      "chr1_72891464 chr1_86758000\n",
      "chr1_72891470 chr1_86758000\n",
      "chr1_72891513 chr1_86758000\n",
      "chr1_73734371 chr1_86758000\n",
      "chr1_73734389 chr1_86758000\n",
      "chr1_73734419 chr1_86758000\n",
      "chr1_73734513 chr1_86758000\n",
      "chr1_73734607 chr1_86758000\n",
      "chr1_73737522 chr1_86758000\n",
      "chr1_73737549 chr1_86758000\n",
      "chr1_73737610 chr1_86758000\n",
      "chr1_73739336 chr1_86758000\n",
      "chr1_73739381 chr1_86758000\n",
      "chr1_73739530 chr1_86758000\n",
      "chr1_73739558 chr1_86758000\n",
      "chr1_73739680 chr1_86758000\n",
      "chr1_82720351 chr1_86758000\n",
      "chr1_82720377 chr1_86758000\n",
      "chr1_82720405 chr1_86758000\n",
      "chr1_82720543 chr1_86758000\n",
      "chr1_82720566 chr1_86758000\n",
      "chr1_82720617 chr1_86758000\n",
      "chr1_82720622 chr1_86758000\n",
      "chr1_82720638 chr1_86758000\n",
      "chr1_82720659 chr1_86758000\n",
      "chr1_82720684 chr1_86758000\n",
      "chr1_82720796 chr1_86758000\n",
      "chr1_82720802 chr1_86758000\n",
      "chr1_82722154 chr1_86758000\n",
      "chr1_82722210 chr1_86758000\n",
      "chr1_82722625 chr1_86758000\n",
      "chr1_82723240 chr1_86758000\n",
      "chr1_82723943 chr1_86758000\n",
      "chr1_82723968 chr1_86758000\n",
      "chr1_82724185 chr1_86758000\n",
      "chr1_82724210 chr1_86758000\n",
      "chr1_82724219 chr1_86758000\n",
      "chr1_86751356 chr1_86758000\n",
      "chr1_86751428 chr1_86758000\n",
      "chr1_86751445 chr1_86758000\n",
      "chr1_86752542 chr1_86758000\n",
      "chr1_86754732 chr1_86758000\n",
      "chr1_86754740 chr1_86758000\n",
      "chr1_86754789 chr1_86758000\n",
      "chr1_86754810 chr1_86758000\n",
      "chr1_86754867 chr1_86758000\n",
      "chr1_86754879 chr1_86758000\n",
      "chr1_86754894 chr1_86758000\n",
      "chr1_86754900 chr1_86758000\n",
      "chr1_86754928 chr1_86758000\n",
      "chr1_86754937 chr1_86758000\n",
      "chr1_86754953 chr1_86758000\n",
      "chr1_86754969 chr1_86758000\n",
      "chr1_86754981 chr1_86758000\n",
      "chr1_86755336 chr1_86758000\n",
      "chr1_86755356 chr1_86758000\n",
      "chr1_86757875 chr1_86758000\n",
      "chr1_86757970 chr1_86758000\n",
      "chr1_86757985 chr1_86758000\n",
      "chr1_86758000 chr1_86758000\n",
      "chr1_86758000;Egu005191-mRNA-3;136145at4447;jacalin-related lectin 3;snp;C;G;CAG;GAG;Q;E\n",
      "chr1_93213103\n",
      "chr1_72869463 chr1_93213103\n",
      "chr1_72869477 chr1_93213103\n",
      "chr1_72869507 chr1_93213103\n",
      "chr1_72869527 chr1_93213103\n",
      "chr1_72869566 chr1_93213103\n",
      "chr1_72869593 chr1_93213103\n",
      "chr1_72869712 chr1_93213103\n",
      "chr1_72869721 chr1_93213103\n",
      "chr1_72869790 chr1_93213103\n",
      "chr1_72870090 chr1_93213103\n",
      "chr1_72870446 chr1_93213103\n",
      "chr1_72870533 chr1_93213103\n",
      "chr1_72870659 chr1_93213103\n",
      "chr1_72870738 chr1_93213103\n",
      "chr1_72870881 chr1_93213103\n",
      "chr1_72871070 chr1_93213103\n",
      "chr1_72871270 chr1_93213103\n",
      "chr1_72871276 chr1_93213103\n",
      "chr1_72871319 chr1_93213103\n",
      "chr1_72871462 chr1_93213103\n",
      "chr1_72871561 chr1_93213103\n",
      "chr1_72871675 chr1_93213103\n",
      "chr1_72871697 chr1_93213103\n",
      "chr1_72871709 chr1_93213103\n",
      "chr1_72871771 chr1_93213103\n",
      "chr1_72871920 chr1_93213103\n",
      "chr1_72872008 chr1_93213103\n",
      "chr1_72872324 chr1_93213103\n",
      "chr1_72872376 chr1_93213103\n",
      "chr1_72872392 chr1_93213103\n",
      "chr1_72872437 chr1_93213103\n",
      "chr1_72872615 chr1_93213103\n",
      "chr1_72882701 chr1_93213103\n",
      "chr1_72882829 chr1_93213103\n",
      "chr1_72883799 chr1_93213103\n",
      "chr1_72883850 chr1_93213103\n",
      "chr1_72890943 chr1_93213103\n",
      "chr1_72891132 chr1_93213103\n",
      "chr1_72891217 chr1_93213103\n",
      "chr1_72891238 chr1_93213103\n",
      "chr1_72891360 chr1_93213103\n",
      "chr1_72891372 chr1_93213103\n",
      "chr1_72891395 chr1_93213103\n",
      "chr1_72891464 chr1_93213103\n",
      "chr1_72891470 chr1_93213103\n",
      "chr1_72891513 chr1_93213103\n",
      "chr1_73734371 chr1_93213103\n",
      "chr1_73734389 chr1_93213103\n",
      "chr1_73734419 chr1_93213103\n",
      "chr1_73734513 chr1_93213103\n",
      "chr1_73734607 chr1_93213103\n",
      "chr1_73737522 chr1_93213103\n",
      "chr1_73737549 chr1_93213103\n",
      "chr1_73737610 chr1_93213103\n",
      "chr1_73739336 chr1_93213103\n",
      "chr1_73739381 chr1_93213103\n",
      "chr1_73739530 chr1_93213103\n",
      "chr1_73739558 chr1_93213103\n",
      "chr1_73739680 chr1_93213103\n",
      "chr1_82720351 chr1_93213103\n",
      "chr1_82720377 chr1_93213103\n",
      "chr1_82720405 chr1_93213103\n",
      "chr1_82720543 chr1_93213103\n",
      "chr1_82720566 chr1_93213103\n",
      "chr1_82720617 chr1_93213103\n",
      "chr1_82720622 chr1_93213103\n",
      "chr1_82720638 chr1_93213103\n",
      "chr1_82720659 chr1_93213103\n",
      "chr1_82720684 chr1_93213103\n",
      "chr1_82720796 chr1_93213103\n",
      "chr1_82720802 chr1_93213103\n",
      "chr1_82722154 chr1_93213103\n",
      "chr1_82722210 chr1_93213103\n",
      "chr1_82722625 chr1_93213103\n",
      "chr1_82723240 chr1_93213103\n",
      "chr1_82723943 chr1_93213103\n",
      "chr1_82723968 chr1_93213103\n",
      "chr1_82724185 chr1_93213103\n",
      "chr1_82724210 chr1_93213103\n",
      "chr1_82724219 chr1_93213103\n",
      "chr1_86751356 chr1_93213103\n",
      "chr1_86751428 chr1_93213103\n",
      "chr1_86751445 chr1_93213103\n",
      "chr1_86752542 chr1_93213103\n",
      "chr1_86754732 chr1_93213103\n",
      "chr1_86754740 chr1_93213103\n",
      "chr1_86754789 chr1_93213103\n",
      "chr1_86754810 chr1_93213103\n",
      "chr1_86754867 chr1_93213103\n",
      "chr1_86754879 chr1_93213103\n",
      "chr1_86754894 chr1_93213103\n",
      "chr1_86754900 chr1_93213103\n",
      "chr1_86754928 chr1_93213103\n",
      "chr1_86754937 chr1_93213103\n",
      "chr1_86754953 chr1_93213103\n",
      "chr1_86754969 chr1_93213103\n",
      "chr1_86754981 chr1_93213103\n",
      "chr1_86755336 chr1_93213103\n",
      "chr1_86755356 chr1_93213103\n",
      "chr1_86757875 chr1_93213103\n",
      "chr1_86757970 chr1_93213103\n",
      "chr1_86757985 chr1_93213103\n",
      "chr1_86758000 chr1_93213103\n",
      "chr1_86758058 chr1_93213103\n",
      "chr1_86761620 chr1_93213103\n",
      "chr1_86761670 chr1_93213103\n",
      "chr1_86764652 chr1_93213103\n",
      "chr1_86765098 chr1_93213103\n",
      "chr1_86765180 chr1_93213103\n",
      "chr1_86765327 chr1_93213103\n",
      "chr1_86765373 chr1_93213103\n",
      "chr1_92899718 chr1_93213103\n",
      "chr1_92899723 chr1_93213103\n",
      "chr1_92899770 chr1_93213103\n",
      "chr1_92899776 chr1_93213103\n",
      "chr1_92899790 chr1_93213103\n",
      "chr1_92899844 chr1_93213103\n",
      "chr1_92899870 chr1_93213103\n",
      "chr1_92899890 chr1_93213103\n",
      "chr1_92899902 chr1_93213103\n",
      "chr1_92899973 chr1_93213103\n",
      "chr1_92899989 chr1_93213103\n",
      "chr1_92900009 chr1_93213103\n",
      "chr1_92900028 chr1_93213103\n",
      "chr1_92900052 chr1_93213103\n",
      "chr1_92900071 chr1_93213103\n",
      "chr1_92900079 chr1_93213103\n",
      "chr1_92900093 chr1_93213103\n",
      "chr1_92900114 chr1_93213103\n",
      "chr1_92900131 chr1_93213103\n",
      "chr1_92900149 chr1_93213103\n",
      "chr1_92900197 chr1_93213103\n",
      "chr1_92900231 chr1_93213103\n",
      "chr1_92900248 chr1_93213103\n",
      "chr1_92900257 chr1_93213103\n",
      "chr1_92900267 chr1_93213103\n",
      "chr1_92900293 chr1_93213103\n",
      "chr1_92900304 chr1_93213103\n",
      "chr1_92900316 chr1_93213103\n",
      "chr1_92900393 chr1_93213103\n",
      "chr1_92900398 chr1_93213103\n",
      "chr1_92900404 chr1_93213103\n",
      "chr1_92900409 chr1_93213103\n",
      "chr1_92900431 chr1_93213103\n",
      "chr1_92900515 chr1_93213103\n",
      "chr1_92900552 chr1_93213103\n",
      "chr1_92901842 chr1_93213103\n",
      "chr1_92903367 chr1_93213103\n",
      "chr1_92903445 chr1_93213103\n",
      "chr1_92903484 chr1_93213103\n",
      "chr1_92903553 chr1_93213103\n",
      "chr1_92903756 chr1_93213103\n",
      "chr1_92903806 chr1_93213103\n",
      "chr1_92903840 chr1_93213103\n",
      "chr1_92918777 chr1_93213103\n",
      "chr1_92918807 chr1_93213103\n",
      "chr1_92918827 chr1_93213103\n",
      "chr1_92918856 chr1_93213103\n",
      "chr1_92918883 chr1_93213103\n",
      "chr1_92918888 chr1_93213103\n",
      "chr1_92918910 chr1_93213103\n",
      "chr1_92918925 chr1_93213103\n",
      "chr1_92918933 chr1_93213103\n",
      "chr1_92918947 chr1_93213103\n",
      "chr1_92918958 chr1_93213103\n",
      "chr1_92918965 chr1_93213103\n",
      "chr1_92918971 chr1_93213103\n",
      "chr1_92918980 chr1_93213103\n",
      "chr1_92918981 chr1_93213103\n",
      "chr1_92919044 chr1_93213103\n",
      "chr1_92919052 chr1_93213103\n",
      "chr1_92919073 chr1_93213103\n",
      "chr1_92919090 chr1_93213103\n",
      "chr1_92919123 chr1_93213103\n",
      "chr1_92919129 chr1_93213103\n",
      "chr1_92920383 chr1_93213103\n",
      "chr1_92920393 chr1_93213103\n",
      "chr1_92920402 chr1_93213103\n",
      "chr1_92920408 chr1_93213103\n",
      "chr1_92920409 chr1_93213103\n",
      "chr1_92920413 chr1_93213103\n",
      "chr1_92920429 chr1_93213103\n",
      "chr1_92920446 chr1_93213103\n",
      "chr1_92920451 chr1_93213103\n",
      "chr1_92920460 chr1_93213103\n",
      "chr1_92920469 chr1_93213103\n",
      "chr1_92920487 chr1_93213103\n",
      "chr1_93213083 chr1_93213103\n",
      "chr1_93213103 chr1_93213103\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1_93213103;Egu005507-mRNA-1;NA;NA;ins;TTC;TCTC;TTC;TCT;F;S\n",
      "chr1_93213371\n",
      "chr1_72869463 chr1_93213371\n",
      "chr1_72869477 chr1_93213371\n",
      "chr1_72869507 chr1_93213371\n",
      "chr1_72869527 chr1_93213371\n",
      "chr1_72869566 chr1_93213371\n",
      "chr1_72869593 chr1_93213371\n",
      "chr1_72869712 chr1_93213371\n",
      "chr1_72869721 chr1_93213371\n",
      "chr1_72869790 chr1_93213371\n",
      "chr1_72870090 chr1_93213371\n",
      "chr1_72870446 chr1_93213371\n",
      "chr1_72870533 chr1_93213371\n",
      "chr1_72870659 chr1_93213371\n",
      "chr1_72870738 chr1_93213371\n",
      "chr1_72870881 chr1_93213371\n",
      "chr1_72871070 chr1_93213371\n",
      "chr1_72871270 chr1_93213371\n",
      "chr1_72871276 chr1_93213371\n",
      "chr1_72871319 chr1_93213371\n",
      "chr1_72871462 chr1_93213371\n",
      "chr1_72871561 chr1_93213371\n",
      "chr1_72871675 chr1_93213371\n",
      "chr1_72871697 chr1_93213371\n",
      "chr1_72871709 chr1_93213371\n",
      "chr1_72871771 chr1_93213371\n",
      "chr1_72871920 chr1_93213371\n",
      "chr1_72872008 chr1_93213371\n",
      "chr1_72872324 chr1_93213371\n",
      "chr1_72872376 chr1_93213371\n",
      "chr1_72872392 chr1_93213371\n",
      "chr1_72872437 chr1_93213371\n",
      "chr1_72872615 chr1_93213371\n",
      "chr1_72882701 chr1_93213371\n",
      "chr1_72882829 chr1_93213371\n",
      "chr1_72883799 chr1_93213371\n",
      "chr1_72883850 chr1_93213371\n",
      "chr1_72890943 chr1_93213371\n",
      "chr1_72891132 chr1_93213371\n",
      "chr1_72891217 chr1_93213371\n",
      "chr1_72891238 chr1_93213371\n",
      "chr1_72891360 chr1_93213371\n",
      "chr1_72891372 chr1_93213371\n",
      "chr1_72891395 chr1_93213371\n",
      "chr1_72891464 chr1_93213371\n",
      "chr1_72891470 chr1_93213371\n",
      "chr1_72891513 chr1_93213371\n",
      "chr1_73734371 chr1_93213371\n",
      "chr1_73734389 chr1_93213371\n",
      "chr1_73734419 chr1_93213371\n",
      "chr1_73734513 chr1_93213371\n",
      "chr1_73734607 chr1_93213371\n",
      "chr1_73737522 chr1_93213371\n",
      "chr1_73737549 chr1_93213371\n",
      "chr1_73737610 chr1_93213371\n",
      "chr1_73739336 chr1_93213371\n",
      "chr1_73739381 chr1_93213371\n",
      "chr1_73739530 chr1_93213371\n",
      "chr1_73739558 chr1_93213371\n",
      "chr1_73739680 chr1_93213371\n",
      "chr1_82720351 chr1_93213371\n",
      "chr1_82720377 chr1_93213371\n",
      "chr1_82720405 chr1_93213371\n",
      "chr1_82720543 chr1_93213371\n",
      "chr1_82720566 chr1_93213371\n",
      "chr1_82720617 chr1_93213371\n",
      "chr1_82720622 chr1_93213371\n",
      "chr1_82720638 chr1_93213371\n",
      "chr1_82720659 chr1_93213371\n",
      "chr1_82720684 chr1_93213371\n",
      "chr1_82720796 chr1_93213371\n",
      "chr1_82720802 chr1_93213371\n",
      "chr1_82722154 chr1_93213371\n",
      "chr1_82722210 chr1_93213371\n",
      "chr1_82722625 chr1_93213371\n",
      "chr1_82723240 chr1_93213371\n",
      "chr1_82723943 chr1_93213371\n",
      "chr1_82723968 chr1_93213371\n",
      "chr1_82724185 chr1_93213371\n",
      "chr1_82724210 chr1_93213371\n",
      "chr1_82724219 chr1_93213371\n",
      "chr1_86751356 chr1_93213371\n",
      "chr1_86751428 chr1_93213371\n",
      "chr1_86751445 chr1_93213371\n",
      "chr1_86752542 chr1_93213371\n",
      "chr1_86754732 chr1_93213371\n",
      "chr1_86754740 chr1_93213371\n",
      "chr1_86754789 chr1_93213371\n",
      "chr1_86754810 chr1_93213371\n",
      "chr1_86754867 chr1_93213371\n",
      "chr1_86754879 chr1_93213371\n",
      "chr1_86754894 chr1_93213371\n",
      "chr1_86754900 chr1_93213371\n",
      "chr1_86754928 chr1_93213371\n",
      "chr1_86754937 chr1_93213371\n",
      "chr1_86754953 chr1_93213371\n",
      "chr1_86754969 chr1_93213371\n",
      "chr1_86754981 chr1_93213371\n",
      "chr1_86755336 chr1_93213371\n",
      "chr1_86755356 chr1_93213371\n",
      "chr1_86757875 chr1_93213371\n",
      "chr1_86757970 chr1_93213371\n",
      "chr1_86757985 chr1_93213371\n",
      "chr1_86758000 chr1_93213371\n",
      "chr1_86758058 chr1_93213371\n",
      "chr1_86761620 chr1_93213371\n",
      "chr1_86761670 chr1_93213371\n",
      "chr1_86764652 chr1_93213371\n",
      "chr1_86765098 chr1_93213371\n",
      "chr1_86765180 chr1_93213371\n",
      "chr1_86765327 chr1_93213371\n",
      "chr1_86765373 chr1_93213371\n",
      "chr1_92899718 chr1_93213371\n",
      "chr1_92899723 chr1_93213371\n",
      "chr1_92899770 chr1_93213371\n",
      "chr1_92899776 chr1_93213371\n",
      "chr1_92899790 chr1_93213371\n",
      "chr1_92899844 chr1_93213371\n",
      "chr1_92899870 chr1_93213371\n",
      "chr1_92899890 chr1_93213371\n",
      "chr1_92899902 chr1_93213371\n",
      "chr1_92899973 chr1_93213371\n",
      "chr1_92899989 chr1_93213371\n",
      "chr1_92900009 chr1_93213371\n",
      "chr1_92900028 chr1_93213371\n",
      "chr1_92900052 chr1_93213371\n",
      "chr1_92900071 chr1_93213371\n",
      "chr1_92900079 chr1_93213371\n",
      "chr1_92900093 chr1_93213371\n",
      "chr1_92900114 chr1_93213371\n",
      "chr1_92900131 chr1_93213371\n",
      "chr1_92900149 chr1_93213371\n",
      "chr1_92900197 chr1_93213371\n",
      "chr1_92900231 chr1_93213371\n",
      "chr1_92900248 chr1_93213371\n",
      "chr1_92900257 chr1_93213371\n",
      "chr1_92900267 chr1_93213371\n",
      "chr1_92900293 chr1_93213371\n",
      "chr1_92900304 chr1_93213371\n",
      "chr1_92900316 chr1_93213371\n",
      "chr1_92900393 chr1_93213371\n",
      "chr1_92900398 chr1_93213371\n",
      "chr1_92900404 chr1_93213371\n",
      "chr1_92900409 chr1_93213371\n",
      "chr1_92900431 chr1_93213371\n",
      "chr1_92900515 chr1_93213371\n",
      "chr1_92900552 chr1_93213371\n",
      "chr1_92901842 chr1_93213371\n",
      "chr1_92903367 chr1_93213371\n",
      "chr1_92903445 chr1_93213371\n",
      "chr1_92903484 chr1_93213371\n",
      "chr1_92903553 chr1_93213371\n",
      "chr1_92903756 chr1_93213371\n",
      "chr1_92903806 chr1_93213371\n",
      "chr1_92903840 chr1_93213371\n",
      "chr1_92918777 chr1_93213371\n",
      "chr1_92918807 chr1_93213371\n",
      "chr1_92918827 chr1_93213371\n",
      "chr1_92918856 chr1_93213371\n",
      "chr1_92918883 chr1_93213371\n",
      "chr1_92918888 chr1_93213371\n",
      "chr1_92918910 chr1_93213371\n",
      "chr1_92918925 chr1_93213371\n",
      "chr1_92918933 chr1_93213371\n",
      "chr1_92918947 chr1_93213371\n",
      "chr1_92918958 chr1_93213371\n",
      "chr1_92918965 chr1_93213371\n",
      "chr1_92918971 chr1_93213371\n",
      "chr1_92918980 chr1_93213371\n",
      "chr1_92918981 chr1_93213371\n",
      "chr1_92919044 chr1_93213371\n",
      "chr1_92919052 chr1_93213371\n",
      "chr1_92919073 chr1_93213371\n",
      "chr1_92919090 chr1_93213371\n",
      "chr1_92919123 chr1_93213371\n",
      "chr1_92919129 chr1_93213371\n",
      "chr1_92920383 chr1_93213371\n",
      "chr1_92920393 chr1_93213371\n",
      "chr1_92920402 chr1_93213371\n",
      "chr1_92920408 chr1_93213371\n",
      "chr1_92920409 chr1_93213371\n",
      "chr1_92920413 chr1_93213371\n",
      "chr1_92920429 chr1_93213371\n",
      "chr1_92920446 chr1_93213371\n",
      "chr1_92920451 chr1_93213371\n",
      "chr1_92920460 chr1_93213371\n",
      "chr1_92920469 chr1_93213371\n",
      "chr1_92920487 chr1_93213371\n",
      "chr1_93213083 chr1_93213371\n",
      "chr1_93213103 chr1_93213371\n",
      "chr1_93213113 chr1_93213371\n",
      "chr1_93213275 chr1_93213371\n",
      "chr1_93213310 chr1_93213371\n",
      "chr1_93213334 chr1_93213371\n",
      "chr1_93213348 chr1_93213371\n",
      "chr1_93213371 chr1_93213371\n",
      "chr1_93213371;Egu005507-mRNA-1;NA;NA;snp;C;T;CCG;CTG;P;L\n",
      "chr1_93215477\n",
      "chr1_72869463 chr1_93215477\n",
      "chr1_72869477 chr1_93215477\n",
      "chr1_72869507 chr1_93215477\n",
      "chr1_72869527 chr1_93215477\n",
      "chr1_72869566 chr1_93215477\n",
      "chr1_72869593 chr1_93215477\n",
      "chr1_72869712 chr1_93215477\n",
      "chr1_72869721 chr1_93215477\n",
      "chr1_72869790 chr1_93215477\n",
      "chr1_72870090 chr1_93215477\n",
      "chr1_72870446 chr1_93215477\n",
      "chr1_72870533 chr1_93215477\n",
      "chr1_72870659 chr1_93215477\n",
      "chr1_72870738 chr1_93215477\n",
      "chr1_72870881 chr1_93215477\n",
      "chr1_72871070 chr1_93215477\n",
      "chr1_72871270 chr1_93215477\n",
      "chr1_72871276 chr1_93215477\n",
      "chr1_72871319 chr1_93215477\n",
      "chr1_72871462 chr1_93215477\n",
      "chr1_72871561 chr1_93215477\n",
      "chr1_72871675 chr1_93215477\n",
      "chr1_72871697 chr1_93215477\n",
      "chr1_72871709 chr1_93215477\n",
      "chr1_72871771 chr1_93215477\n",
      "chr1_72871920 chr1_93215477\n",
      "chr1_72872008 chr1_93215477\n",
      "chr1_72872324 chr1_93215477\n",
      "chr1_72872376 chr1_93215477\n",
      "chr1_72872392 chr1_93215477\n",
      "chr1_72872437 chr1_93215477\n",
      "chr1_72872615 chr1_93215477\n",
      "chr1_72882701 chr1_93215477\n",
      "chr1_72882829 chr1_93215477\n",
      "chr1_72883799 chr1_93215477\n",
      "chr1_72883850 chr1_93215477\n",
      "chr1_72890943 chr1_93215477\n",
      "chr1_72891132 chr1_93215477\n",
      "chr1_72891217 chr1_93215477\n",
      "chr1_72891238 chr1_93215477\n",
      "chr1_72891360 chr1_93215477\n",
      "chr1_72891372 chr1_93215477\n",
      "chr1_72891395 chr1_93215477\n",
      "chr1_72891464 chr1_93215477\n",
      "chr1_72891470 chr1_93215477\n",
      "chr1_72891513 chr1_93215477\n",
      "chr1_73734371 chr1_93215477\n",
      "chr1_73734389 chr1_93215477\n",
      "chr1_73734419 chr1_93215477\n",
      "chr1_73734513 chr1_93215477\n",
      "chr1_73734607 chr1_93215477\n",
      "chr1_73737522 chr1_93215477\n",
      "chr1_73737549 chr1_93215477\n",
      "chr1_73737610 chr1_93215477\n",
      "chr1_73739336 chr1_93215477\n",
      "chr1_73739381 chr1_93215477\n",
      "chr1_73739530 chr1_93215477\n",
      "chr1_73739558 chr1_93215477\n",
      "chr1_73739680 chr1_93215477\n",
      "chr1_82720351 chr1_93215477\n",
      "chr1_82720377 chr1_93215477\n",
      "chr1_82720405 chr1_93215477\n",
      "chr1_82720543 chr1_93215477\n",
      "chr1_82720566 chr1_93215477\n",
      "chr1_82720617 chr1_93215477\n",
      "chr1_82720622 chr1_93215477\n",
      "chr1_82720638 chr1_93215477\n",
      "chr1_82720659 chr1_93215477\n",
      "chr1_82720684 chr1_93215477\n",
      "chr1_82720796 chr1_93215477\n",
      "chr1_82720802 chr1_93215477\n",
      "chr1_82722154 chr1_93215477\n",
      "chr1_82722210 chr1_93215477\n",
      "chr1_82722625 chr1_93215477\n",
      "chr1_82723240 chr1_93215477\n",
      "chr1_82723943 chr1_93215477\n",
      "chr1_82723968 chr1_93215477\n",
      "chr1_82724185 chr1_93215477\n",
      "chr1_82724210 chr1_93215477\n",
      "chr1_82724219 chr1_93215477\n",
      "chr1_86751356 chr1_93215477\n",
      "chr1_86751428 chr1_93215477\n",
      "chr1_86751445 chr1_93215477\n",
      "chr1_86752542 chr1_93215477\n",
      "chr1_86754732 chr1_93215477\n",
      "chr1_86754740 chr1_93215477\n",
      "chr1_86754789 chr1_93215477\n",
      "chr1_86754810 chr1_93215477\n",
      "chr1_86754867 chr1_93215477\n",
      "chr1_86754879 chr1_93215477\n",
      "chr1_86754894 chr1_93215477\n",
      "chr1_86754900 chr1_93215477\n",
      "chr1_86754928 chr1_93215477\n",
      "chr1_86754937 chr1_93215477\n",
      "chr1_86754953 chr1_93215477\n",
      "chr1_86754969 chr1_93215477\n",
      "chr1_86754981 chr1_93215477\n",
      "chr1_86755336 chr1_93215477\n",
      "chr1_86755356 chr1_93215477\n",
      "chr1_86757875 chr1_93215477\n",
      "chr1_86757970 chr1_93215477\n",
      "chr1_86757985 chr1_93215477\n",
      "chr1_86758000 chr1_93215477\n",
      "chr1_86758058 chr1_93215477\n",
      "chr1_86761620 chr1_93215477\n",
      "chr1_86761670 chr1_93215477\n",
      "chr1_86764652 chr1_93215477\n",
      "chr1_86765098 chr1_93215477\n",
      "chr1_86765180 chr1_93215477\n",
      "chr1_86765327 chr1_93215477\n",
      "chr1_86765373 chr1_93215477\n",
      "chr1_92899718 chr1_93215477\n",
      "chr1_92899723 chr1_93215477\n",
      "chr1_92899770 chr1_93215477\n",
      "chr1_92899776 chr1_93215477\n",
      "chr1_92899790 chr1_93215477\n",
      "chr1_92899844 chr1_93215477\n",
      "chr1_92899870 chr1_93215477\n",
      "chr1_92899890 chr1_93215477\n",
      "chr1_92899902 chr1_93215477\n",
      "chr1_92899973 chr1_93215477\n",
      "chr1_92899989 chr1_93215477\n",
      "chr1_92900009 chr1_93215477\n",
      "chr1_92900028 chr1_93215477\n",
      "chr1_92900052 chr1_93215477\n",
      "chr1_92900071 chr1_93215477\n",
      "chr1_92900079 chr1_93215477\n",
      "chr1_92900093 chr1_93215477\n",
      "chr1_92900114 chr1_93215477\n",
      "chr1_92900131 chr1_93215477\n",
      "chr1_92900149 chr1_93215477\n",
      "chr1_92900197 chr1_93215477\n",
      "chr1_92900231 chr1_93215477\n",
      "chr1_92900248 chr1_93215477\n",
      "chr1_92900257 chr1_93215477\n",
      "chr1_92900267 chr1_93215477\n",
      "chr1_92900293 chr1_93215477\n",
      "chr1_92900304 chr1_93215477\n",
      "chr1_92900316 chr1_93215477\n",
      "chr1_92900393 chr1_93215477\n",
      "chr1_92900398 chr1_93215477\n",
      "chr1_92900404 chr1_93215477\n",
      "chr1_92900409 chr1_93215477\n",
      "chr1_92900431 chr1_93215477\n",
      "chr1_92900515 chr1_93215477\n",
      "chr1_92900552 chr1_93215477\n",
      "chr1_92901842 chr1_93215477\n",
      "chr1_92903367 chr1_93215477\n",
      "chr1_92903445 chr1_93215477\n",
      "chr1_92903484 chr1_93215477\n",
      "chr1_92903553 chr1_93215477\n",
      "chr1_92903756 chr1_93215477\n",
      "chr1_92903806 chr1_93215477\n",
      "chr1_92903840 chr1_93215477\n",
      "chr1_92918777 chr1_93215477\n",
      "chr1_92918807 chr1_93215477\n",
      "chr1_92918827 chr1_93215477\n",
      "chr1_92918856 chr1_93215477\n",
      "chr1_92918883 chr1_93215477\n",
      "chr1_92918888 chr1_93215477\n",
      "chr1_92918910 chr1_93215477\n",
      "chr1_92918925 chr1_93215477\n",
      "chr1_92918933 chr1_93215477\n",
      "chr1_92918947 chr1_93215477\n",
      "chr1_92918958 chr1_93215477\n",
      "chr1_92918965 chr1_93215477\n",
      "chr1_92918971 chr1_93215477\n",
      "chr1_92918980 chr1_93215477\n",
      "chr1_92918981 chr1_93215477\n",
      "chr1_92919044 chr1_93215477\n",
      "chr1_92919052 chr1_93215477\n",
      "chr1_92919073 chr1_93215477\n",
      "chr1_92919090 chr1_93215477\n",
      "chr1_92919123 chr1_93215477\n",
      "chr1_92919129 chr1_93215477\n",
      "chr1_92920383 chr1_93215477\n",
      "chr1_92920393 chr1_93215477\n",
      "chr1_92920402 chr1_93215477\n",
      "chr1_92920408 chr1_93215477\n",
      "chr1_92920409 chr1_93215477\n",
      "chr1_92920413 chr1_93215477\n",
      "chr1_92920429 chr1_93215477\n",
      "chr1_92920446 chr1_93215477\n",
      "chr1_92920451 chr1_93215477\n",
      "chr1_92920460 chr1_93215477\n",
      "chr1_92920469 chr1_93215477\n",
      "chr1_92920487 chr1_93215477\n",
      "chr1_93213083 chr1_93215477\n",
      "chr1_93213103 chr1_93215477\n",
      "chr1_93213113 chr1_93215477\n",
      "chr1_93213275 chr1_93215477\n",
      "chr1_93213310 chr1_93215477\n",
      "chr1_93213334 chr1_93215477\n",
      "chr1_93213348 chr1_93215477\n",
      "chr1_93213371 chr1_93215477\n",
      "chr1_93215477 chr1_93215477\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/Seq.py:2804: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1_93215477;Egu005507-mRNA-1;NA;NA;snp;A;C;TAT;TCT;Y;S\n"
     ]
    }
   ],
   "source": [
    "import pysam\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq, MutableSeq\n",
    "from Bio import SeqIO\n",
    "\n",
    "# Input\n",
    "out_file = \"/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100_batch1_chr1.test.summary\"\n",
    "bed_file = '/bioinfo2/palm/analysis/yield/Apriyanto2023_DE100.exon.bed'\n",
    "# prepare vcf by: bcftools view -R /bioinfo2/palm/analysis/yield/Apriyanto2023_DE100.exon.bed /bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr2_lc200_C5.bcf > Apriyanto2023_DE100.exon.b1ch2.vcf\n",
    "vcf_file = '/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100.exon.b1ch1.vcf'\n",
    "# tab file is prepared by \n",
    "in_file = '/bioinfo2/palm/analysis/batch1/batch1_chr1_nosex.assoc.qassoc.tab'\n",
    "p = 0.5\n",
    "\n",
    "# Main code\n",
    "in_df = pd.read_csv(in_file, sep = '\\t', header=0, dtype={'CHR':str, 'SNP': str})\n",
    "in_df = in_df.drop([col for col in in_df.columns if col.startswith('Unnamed:')], axis=1)\n",
    "in_file_adj = in_file.split('.tab')[0] + '.adjusted.tab'\n",
    "in_df_adj = pd.read_csv(in_file_adj, sep = '\\t', header=0)\n",
    "colnames = list(in_df_adj.columns)\n",
    "in_df_adj = in_df_adj.iloc[:, :-1]\n",
    "in_df_adj.columns = colnames[1:]\n",
    "in_df_adj['CHR'] = in_df_adj['CHR'].astype(str)\n",
    "in_df = in_df.merge(in_df_adj)\n",
    "in_df = in_df[in_df['BONF'] < p]\n",
    "in_df.to_csv('/bioinfo2/palm/analysis/batch1/in_df.txt', sep='\\t', index=False)\n",
    "\n",
    "bed_column_names = ['Seqid', 'Start', 'End', 'Genes']\n",
    "bed_df = pd.read_csv(bed_file, sep='\\t',comment='#', names=bed_column_names)\n",
    "vcf = pysam.VariantFile(vcf_file, \"r\")\n",
    "genome_fa = '/bioinfo2/palm/ref/dura/dura_ref.fasta'\n",
    "anno_file = \"/bioinfo2/palm/ref/dura/dura_orthoDBv11_4447_3193_33090.og.annotations\"\n",
    "anno_df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "anno_df = anno_df.rename(columns={'#query': 'query'})\n",
    "\n",
    "for coord in in_df['SNP']:\n",
    "    # this function takes a coordinate of a SNP in the dura genome and returns some useful info about it\n",
    "    CHR, POS = coord.split('_')\n",
    "    POS = int(POS)\n",
    "    # look up gene from bed file\n",
    "    snp_df = bed_df[(bed_df['Seqid'] == CHR) & (bed_df['Start'] <= POS) & (bed_df['End'] >= POS)]\n",
    "    if len(snp_df) > 0: # if snp is in the region we care\n",
    "        print(coord)\n",
    "        start = int(snp_df['Start'].iloc[0])\n",
    "        end = int(snp_df['End'].iloc[0])\n",
    "        gene = snp_df['Genes'].iloc[0]\n",
    "        anno_df = anno_df[anno_df['query'] == gene]\n",
    "        if len(anno_df) == 1:\n",
    "            og = anno_df['ODB_OG'].iloc[0]\n",
    "            gene_name = anno_df['Description'].iloc[0]\n",
    "        elif len(anno_df) == 0:\n",
    "            og = 'NA'\n",
    "            gene_name = 'NA'\n",
    "        else:\n",
    "            print(gene, \" has more than one og, recording the first one\")\n",
    "            og = anno_df['ODB_OG'].iloc[0]\n",
    "            gene_name = anno_df['Description'].iloc[0]\n",
    "        # look up REF and ALT from vcf\n",
    "        for variant in vcf.fetch():\n",
    "            TYPE = variant.info[\"TYPE\"]\n",
    "            type_out = '|'.join([i for i in TYPE])\n",
    "            print(variant.id, coord)\n",
    "            if variant.id == coord:\n",
    "                REF = variant.ref\n",
    "                ALT = variant.alts[0]\n",
    "                break\n",
    "        # get the codon and amino acid\n",
    "        for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "            if record.id == CHR:\n",
    "                exon_seq = record.seq[start-1:end]\n",
    "                exon_seq_var = MutableSeq(str(exon_seq))\n",
    "                protein_seq = exon_seq.translate()\n",
    "                snp_position = POS - start\n",
    "                # Find the codon's start position\n",
    "                codon_start = snp_position - (snp_position % 3)\n",
    "                # Extract the codons\n",
    "                codon = exon_seq[codon_start: codon_start+3]\n",
    "                exon_seq_var[snp_position: snp_position+len(ALT)] = ALT\n",
    "                modified_codon = exon_seq_var[codon_start: codon_start+3]    \n",
    "                original_aa = codon.translate()\n",
    "                modified_aa = modified_codon.translate()\n",
    "                break\n",
    "        print(';'.join([coord, gene, og, gene_name, type_out, REF, ALT, str(codon), str(modified_codon), str(original_aa), str(modified_aa)]))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3aae181",
   "metadata": {},
   "outputs": [],
   "source": [
    "        for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "            if record.id == CHR:\n",
    "                exon_seq = record.seq[start-1:end]\n",
    "                exon_seq_snp = MutableSeq(str(exon_seq))\n",
    "                protein_seq = exon_seq.translate()\n",
    "                snp_position = POS - start\n",
    "                if TYPE == 'snp'\n",
    "                if exon_seq[snp_position] == REF:\n",
    "                    exon_seq_snp[snp_position] = ALT\n",
    "                else:\n",
    "                    print(\"REF wrong. Break\")\n",
    "                    break\n",
    "                # Find the codon's start position\n",
    "                codon_start = snp_position - (snp_position % 3)\n",
    "                # Extract the codons\n",
    "                codon = exon_seq[codon_start: codon_start+3]\n",
    "                # If needed, replace the SNP in the codon\n",
    "                modified_codon = exon_seq_snp[codon_start: codon_start+3] # modify with SNP\n",
    "                # Translate the original and modified codon to amino acids\n",
    "                original_aa = codon.translate()\n",
    "                modified_aa = modified_codon.translate()\n",
    "                break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "0c8d05ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>R2</th>\n",
       "      <th>T</th>\n",
       "      <th>P</th>\n",
       "      <th>X10</th>\n",
       "      <th>coord</th>\n",
       "      <th>gene</th>\n",
       "      <th>og</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>codon</th>\n",
       "      <th>modified_codon</th>\n",
       "      <th>original_aa</th>\n",
       "      <th>modified_aa</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16</td>\n",
       "      <td>.</td>\n",
       "      <td>20788022</td>\n",
       "      <td>48</td>\n",
       "      <td>17.55</td>\n",
       "      <td>7.255</td>\n",
       "      <td>0.1128</td>\n",
       "      <td>2.418</td>\n",
       "      <td>0.01961</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_20788022</td>\n",
       "      <td>Egu030167-mRNA-1</td>\n",
       "      <td>130847at4447</td>\n",
       "      <td>Heat stress transcription factor B-2c</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>(T, G, G)</td>\n",
       "      <td>[C, G, G]</td>\n",
       "      <td>(W)</td>\n",
       "      <td>[R]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16</td>\n",
       "      <td>.</td>\n",
       "      <td>32010076</td>\n",
       "      <td>47</td>\n",
       "      <td>-18.43</td>\n",
       "      <td>6.919</td>\n",
       "      <td>0.1362</td>\n",
       "      <td>-2.664</td>\n",
       "      <td>0.01069</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_32010076</td>\n",
       "      <td>Egu030617-mRNA-1</td>\n",
       "      <td>128811at4447</td>\n",
       "      <td>transmembrane protein 53-A</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>(G, G, C)</td>\n",
       "      <td>[G, G, T]</td>\n",
       "      <td>(G)</td>\n",
       "      <td>[G]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR SNP        BP  NMISS   BETA     SE      R2      T        P  X10  \\\n",
       "0   16   .  20788022     48  17.55  7.255  0.1128  2.418  0.01961  NaN   \n",
       "1   16   .  32010076     47 -18.43  6.919  0.1362 -2.664  0.01069  NaN   \n",
       "\n",
       "            coord              gene            og  \\\n",
       "0  chr16_20788022  Egu030167-mRNA-1  130847at4447   \n",
       "1  chr16_32010076  Egu030617-mRNA-1  128811at4447   \n",
       "\n",
       "                               gene_name REF ALT      codon modified_codon  \\\n",
       "0  Heat stress transcription factor B-2c   T   C  (T, G, G)      [C, G, G]   \n",
       "1             transmembrane protein 53-A   C   T  (G, G, C)      [G, G, T]   \n",
       "\n",
       "  original_aa modified_aa  \n",
       "0         (W)         [R]  \n",
       "1         (G)         [G]  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "                # if the variant type if SNP\n",
    "                if (len(TYPE) == 1) & (TYPE[0] == 'snp'):\n",
    "                    if exon_seq[snp_position] == REF:\n",
    "                        exon_seq_var[snp_position] = ALT\n",
    "                    else:\n",
    "                        print(\"REF wrong. Break\")\n",
    "                        break\n",
    "                    # Translate the original and modified codon to amino acids\n",
    "                else:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "94d5430e",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[62], line 31\u001b[0m\n\u001b[1;32m     29\u001b[0m             \u001b[38;5;28;01mbreak\u001b[39;00m\n\u001b[1;32m     30\u001b[0m \u001b[38;5;66;03m# get the codon and amino acid\u001b[39;00m\n\u001b[0;32m---> 31\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m record \u001b[38;5;129;01min\u001b[39;00m SeqIO\u001b[38;5;241m.\u001b[39mparse(genome_fa,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfasta\u001b[39m\u001b[38;5;124m'\u001b[39m):\n\u001b[1;32m     32\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m record\u001b[38;5;241m.\u001b[39mid \u001b[38;5;241m==\u001b[39m CHR:\n\u001b[1;32m     33\u001b[0m         exon_seq \u001b[38;5;241m=\u001b[39m record\u001b[38;5;241m.\u001b[39mseq[start\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m:end]\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/Interfaces.py:72\u001b[0m, in \u001b[0;36mSequenceIterator.__next__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Return the next entry.\"\"\"\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m---> 72\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mnext\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecords\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     73\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m:\n\u001b[1;32m     74\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mshould_close_stream:\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/FastaIO.py:238\u001b[0m, in \u001b[0;36mFastaIterator.iterate\u001b[0;34m(self, handle)\u001b[0m\n\u001b[1;32m    236\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m SeqRecord(Seq(sequence), \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mid\u001b[39m, name\u001b[38;5;241m=\u001b[39mname, description\u001b[38;5;241m=\u001b[39mdescr)\n\u001b[1;32m    237\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 238\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m title, sequence \u001b[38;5;129;01min\u001b[39;00m SimpleFastaParser(handle):\n\u001b[1;32m    239\u001b[0m         \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    240\u001b[0m             first_word \u001b[38;5;241m=\u001b[39m title\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m1\u001b[39m)[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/FastaIO.py:65\u001b[0m, in \u001b[0;36mSimpleFastaParser\u001b[0;34m(handle)\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m line \u001b[38;5;129;01min\u001b[39;00m handle:\n\u001b[1;32m     64\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m line[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m>\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m---> 65\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m title, \u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mjoin\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlines\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m \u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m)\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     66\u001b[0m         lines \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m     67\u001b[0m         title \u001b[38;5;241m=\u001b[39m line[\u001b[38;5;241m1\u001b[39m:]\u001b[38;5;241m.\u001b[39mrstrip()\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# this function takes a coordinate of a SNP in the dura genome and returns some useful info about it\n",
    "coord = 'chr3_69323261'\n",
    "CHR, POS = coord.split('_')\n",
    "POS = int(POS)\n",
    "# look up gene from bed file\n",
    "snp_df = bed_df[(bed_df['Seqid'] == CHR) & (bed_df['Start'] <= POS) & (bed_df['End'] >= POS)]\n",
    "start = int(snp_df['Start'].iloc[0])\n",
    "end = int(snp_df['End'].iloc[0])\n",
    "gene = snp_df['Genes'].iloc[0]\n",
    "anno_df = anno_df[anno_df['query'] == gene]\n",
    "if len(anno_df) == 1:\n",
    "    og = anno_df['ODB_OG'].iloc[0]\n",
    "    gene_name = anno_df['Description'].iloc[0]\n",
    "elif len(anno_df) == 0:\n",
    "    og = 'NA'\n",
    "    gene_name = 'NA'\n",
    "else:\n",
    "    print(gene, \" has more than one og, recording the first one\")\n",
    "    og = anno_df['ODB_OG'].iloc[0]\n",
    "    gene_name = anno_df['Description'].iloc[0]\n",
    "# look up REF and ALT from vcf\n",
    "for variant in vcf:\n",
    "    if variant.chrom == CHR:\n",
    "        print\n",
    "        if variant.pos == POS:\n",
    "            REF = variant.ref\n",
    "            ALT = variant.alts[0]\n",
    "            print(REF, ALT)\n",
    "            break\n",
    "# get the codon and amino acid\n",
    "for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "    if record.id == CHR:\n",
    "        exon_seq = record.seq[start-1:end]\n",
    "        exon_seq_snp = MutableSeq(str(exon_seq))\n",
    "        protein_seq = exon_seq.translate()\n",
    "        snp_position = POS - start\n",
    "        if exon_seq[snp_position] == REF:\n",
    "            exon_seq_snp[snp_position] = ALT\n",
    "        else:\n",
    "            \"REF wrong. Break\"\n",
    "            break\n",
    "        # Find the codon's start position\n",
    "        codon_start = snp_position - (snp_position % 3)\n",
    "        # Extract the codon\n",
    "        codon = exon_seq[codon_start: codon_start+3]\n",
    "        # If needed, replace the SNP in the codon\n",
    "        modified_codon = exon_seq_snp[codon_start: codon_start+3] # modify with SNP\n",
    "        # Translate the original and modified codon to amino acids\n",
    "        original_aa = codon.translate()\n",
    "        modified_aa = modified_codon.translate()\n",
    "        break\n",
    "print(';'.join([coord, gene, og, gene_name, REF, ALT, str(codon), str(modified_codon), str(original_aa), str(modified_aa)]))\n",
    "return [gene, og, gene_name, REF, ALT, codon, modified_codon, original_aa, modified_aa]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "14c74eed",
   "metadata": {},
   "outputs": [],
   "source": [
    "def coord2codon(coord, bed_df = bed_df, vcf = vcf, genome_fa = genome_fa, anno_df = anno_df):\n",
    "    # this function takes a coordinate of a SNP in the dura genome and returns some useful info about it\n",
    "    CHR, POS = coord.split('_')\n",
    "    POS = int(POS)\n",
    "    # look up gene from bed file\n",
    "    snp_df = bed_df[(bed_df['Seqid'] == CHR) & (bed_df['Start'] <= POS) & (bed_df['End'] >= POS)]\n",
    "    start = int(snp_df['Start'].iloc[0])\n",
    "    end = int(snp_df['End'].iloc[0])\n",
    "    gene = snp_df['Genes'].iloc[0]\n",
    "    anno_df = anno_df[anno_df['query'] == gene]\n",
    "    og = str(anno_df['ODB_OG'])\n",
    "    gene_name = str(anno_df['Description'])\n",
    "    # look up REF and ALT from vcf\n",
    "    for variant in vcf:\n",
    "        if variant.chrom == CHR:\n",
    "            if variant.pos == POS:\n",
    "                REF = variant.ref\n",
    "                ALT = variant.alts\n",
    "                break\n",
    "            else:\n",
    "                print(coord, \"Wrong POS\")\n",
    "    # get the codon and amino acid\n",
    "    for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "        if record.id == CHR:\n",
    "            exon_seq = record.seq[start-1:end]\n",
    "            exon_seq_snp = MutableSeq(str(exon_seq))\n",
    "            protein_seq = exon_seq.translate()\n",
    "            snp_position = POS - start\n",
    "            if exon_seq[snp_position] == REF:\n",
    "                exon_seq_snp[snp_position] = ALT[0]\n",
    "            else:\n",
    "                \"REF wrong. Break\"\n",
    "                break\n",
    "            # Find the codon's start position\n",
    "            codon_start = snp_position - (snp_position % 3)\n",
    "            # Extract the codon\n",
    "            codon = exon_seq[codon_start: codon_start+3]\n",
    "            print(coord, codon)\n",
    "            # If needed, replace the SNP in the codon\n",
    "            modified_codon = exon_seq_snp[codon_start: codon_start+3] # modify with SNP\n",
    "            # Translate the original and modified codon to amino acids\n",
    "            original_aa = codon.translate()\n",
    "            modified_aa = modified_codon.translate()\n",
    "            break\n",
    "    return [gene, og, gene_name, REF, ALT, codon, modified_codon, original_aa, modified_aa]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da8dee7e",
   "metadata": {},
   "source": [
    "# Knowledge base from catalogue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2e837461",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr1.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr9.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr2.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr7.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr6.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr3.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr8.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr10.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr5.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr4.snpEff.genes.txt\n"
     ]
    }
   ],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "# Merge snpEff.genes.txt from different chromosomes\n",
    "dfs = []\n",
    "for file in glob.glob('/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr*.snpEff.genes.txt'):\n",
    "    print(file)\n",
    "    df = pd.read_csv(file, header=1, sep=\"\\t\")\n",
    "    # Remove the \"#\" from the column names\n",
    "    df.columns = df.columns.str.replace(\"#\", \"\").str.strip()\n",
    "    dfs.append(df)\n",
    "    \n",
    "result_df = pd.concat(dfs, ignore_index=True, join='outer').fillna(0)\n",
    "result_df.to_csv('/bioinfo2/palm/knowledgebase/chr1to10_snpEff.genes.txt',sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f9212322",
   "metadata": {},
   "outputs": [],
   "source": [
    "trait = 'fruit'\n",
    "name = 'fruit'\n",
    "cata = pd.read_csv('/bioinfo2/palm/knowledgebase/{}/dura_{}_catalogue.txt'.format(trait,name), sep = '\\t',names=['TranscriptId','GeneName'])\n",
    "gene_list = cata['TranscriptId']\n",
    "\n",
    "df_1 = result_df[result_df['TranscriptId'].isin(gene_list)]\n",
    "\n",
    "anno_file = \"/bioinfo2/palm/ref/dura/dura4447.og.annotations\"\n",
    "anno_df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "anno_df = anno_df.rename(columns={'#query': 'TranscriptId'})\n",
    "df_merge = df_1.merge(anno_df)\n",
    "df_merge.to_csv('/bioinfo2/palm/knowledgebase/{}/dura_{}.snpEff.genes.txt'.format(trait, name), sep='\\t', index=False )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dbdc93e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cata = pd.read_csv('/bioinfo2/palm/knowledgebase/RNA_seq/Apriyanto2023_DE100.genes.txt', names=['TranscriptId'])\n",
    "gene_list = cata['TranscriptId']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "787d170b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr1.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr9.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr2.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr7.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr6.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr3.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr8.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr10.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr5.snpEff.genes.txt\n",
      "/bioinfo2/palm/WIL_sequence/3_freebayes/batch1/batch1_chr4.snpEff.genes.txt\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0f7c8832",
   "metadata": {},
   "source": [
    "# knowledge base from .gff or .bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "09f4689d",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pybedtools import BedTool\n",
    "gff_data = BedTool('/bioinfo2/palm/knowledgebase/drought/dura_drought_genes_rbh_49.gff')\n",
    "gff_dic = {}\n",
    "for feature in gff_data:\n",
    "    gff_dic[feature.attrs.get('ID')] = ''\n",
    "gene_list = gff_dic.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ade5cc76",
   "metadata": {},
   "outputs": [],
   "source": [
    "# if reading from a bed file\n",
    "col_names = ['chr', 'start', 'end', 'name','note', 'strand', 'TranscriptId']\n",
    "bed = pd.read_csv('/bioinfo2/palm/knowledgebase/fattyacid/dura_fattyacid_genes.bed', sep = '\\t', names = col_names)\n",
    "gene_list = bed['TranscriptId']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f5a78925",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>GeneId</th>\n",
       "      <th>TranscriptId</th>\n",
       "      <th>BioType</th>\n",
       "      <th>variants_impact_HIGH</th>\n",
       "      <th>variants_impact_LOW</th>\n",
       "      <th>variants_impact_MODERATE</th>\n",
       "      <th>variants_impact_MODIFIER</th>\n",
       "      <th>variants_effect_3_prime_UTR_variant</th>\n",
       "      <th>variants_effect_5_prime_UTR_premature_start_codon_gain_variant</th>\n",
       "      <th>...</th>\n",
       "      <th>ODB_OG</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>GOs_mf</th>\n",
       "      <th>GOs_bp</th>\n",
       "      <th>EC</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>Interpro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Egu004811</td>\n",
       "      <td>Egu004811</td>\n",
       "      <td>Egu004811-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>583</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>118148at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>222.8500</td>\n",
       "      <td>I,C,E,H</td>\n",
       "      <td>Methylcrotonoyl-CoA carboxylase subunit alpha,...</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0000166</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00860,ko04144</td>\n",
       "      <td>IPR000089,IPR001882,IPR005479,IPR005481,IPR005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Egu004811</td>\n",
       "      <td>Egu004811</td>\n",
       "      <td>Egu004811-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>558</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>118148at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>222.3670</td>\n",
       "      <td>I,C,E,H</td>\n",
       "      <td>Methylcrotonoyl-CoA carboxylase subunit alpha,...</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0000166</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00860,ko04144</td>\n",
       "      <td>IPR000089,IPR001882,IPR005479,IPR005481,IPR005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Egu022699</td>\n",
       "      <td>Egu022699</td>\n",
       "      <td>Egu022699-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>638</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>64848at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>184.7670</td>\n",
       "      <td>I,Q,M,C</td>\n",
       "      <td>(3R)-hydroxymyristoyl-</td>\n",
       "      <td>GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...</td>\n",
       "      <td>GO:0009245,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00230,ko00730,ko04075,ko04626,zma01100,zma01110</td>\n",
       "      <td>IPR010084,IPR013114,IPR029069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Egu023815</td>\n",
       "      <td>Egu023815</td>\n",
       "      <td>Egu023815-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>711</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>13185at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>225.2550</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0045300,GO:0016491,GO:0046872</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Egu000360</td>\n",
       "      <td>Egu000360</td>\n",
       "      <td>Egu000360-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>979</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>128492at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>179.5100</td>\n",
       "      <td>M,I,O</td>\n",
       "      <td>Acyl-CoA-binding domain-containing protein 6</td>\n",
       "      <td>GO:0008289,GO:0000062</td>\n",
       "      <td>GO:0006869</td>\n",
       "      <td>-</td>\n",
       "      <td>bdi01100,bdi01110,ko00220,ko00250,ko00520,ko00...</td>\n",
       "      <td>IPR000582,IPR006652,IPR014352,IPR015915,IPR035984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Egu000360</td>\n",
       "      <td>Egu000360</td>\n",
       "      <td>Egu000360-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>979</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>11254at4447</td>\n",
       "      <td>1.540000e-174</td>\n",
       "      <td>103.1600</td>\n",
       "      <td>M,O,T</td>\n",
       "      <td>acyl-CoA-binding domain-containing protein 4</td>\n",
       "      <td>GO:0000062</td>\n",
       "      <td>GO:0006869</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR006652,IPR015915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Egu001574</td>\n",
       "      <td>Egu001574</td>\n",
       "      <td>Egu001574-mRNA-3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>514</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>156112at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>128.8810</td>\n",
       "      <td>-</td>\n",
       "      <td>enoyl-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Egu002030</td>\n",
       "      <td>Egu002030</td>\n",
       "      <td>Egu002030-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>683</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>13185at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>224.8720</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0045300,GO:0016491,GO:0046872</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Egu002186</td>\n",
       "      <td>Egu002186</td>\n",
       "      <td>Egu002186-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>654</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>134714at4447</td>\n",
       "      <td>5.890000e-174</td>\n",
       "      <td>169.5900</td>\n",
       "      <td>-</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016491,GO:0004316,GO:0102131,GO:0102132</td>\n",
       "      <td>GO:0006629,GO:0006631,GO:0055114</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00260,ko03010,ko03013</td>\n",
       "      <td>IPR002347,IPR011284,IPR020904,IPR036291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Egu002499</td>\n",
       "      <td>Egu002499</td>\n",
       "      <td>Egu002499-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>281</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>64848at4447</td>\n",
       "      <td>7.160000e-148</td>\n",
       "      <td>219.0900</td>\n",
       "      <td>I,Q,M,C</td>\n",
       "      <td>(3R)-hydroxymyristoyl-</td>\n",
       "      <td>GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...</td>\n",
       "      <td>GO:0009245,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00230,ko00730,ko04075,ko04626,zma01100,zma01110</td>\n",
       "      <td>IPR010084,IPR013114,IPR029069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Egu003609</td>\n",
       "      <td>Egu003609</td>\n",
       "      <td>Egu003609-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>400</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>129798at4447</td>\n",
       "      <td>7.830000e-226</td>\n",
       "      <td>207.9300</td>\n",
       "      <td>P,F</td>\n",
       "      <td>Stearoyl-</td>\n",
       "      <td>GO:0046872,GO:0045300,GO:0016491</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.1.1.44</td>\n",
       "      <td>ko00030,ko00195,ko00480,ko01200</td>\n",
       "      <td>IPR005067,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Egu018432</td>\n",
       "      <td>Egu018432</td>\n",
       "      <td>Egu018432-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>422</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>119288at4447</td>\n",
       "      <td>1.840000e-218</td>\n",
       "      <td>186.1400</td>\n",
       "      <td>I</td>\n",
       "      <td>Acyl-</td>\n",
       "      <td>GO:0016787,GO:0016790</td>\n",
       "      <td>GO:0006629,GO:0006631</td>\n",
       "      <td>2.3.2.27,3.1.2.-</td>\n",
       "      <td>ko04626,ko03018,ko04120,ko04141</td>\n",
       "      <td>IPR002864,IPR021113,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Egu019609</td>\n",
       "      <td>Egu019609</td>\n",
       "      <td>Egu019609-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>470</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>132388at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>196.8140</td>\n",
       "      <td>I</td>\n",
       "      <td>oleoyl-acyl carrier protein thioesterase, chlo...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>3.1.2.-</td>\n",
       "      <td>ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...</td>\n",
       "      <td>IPR002864,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Egu019621</td>\n",
       "      <td>Egu019621</td>\n",
       "      <td>Egu019621-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>437</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>13185at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>239.4400</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0045300,GO:0016491,GO:0046872</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Egu019696</td>\n",
       "      <td>Egu019696</td>\n",
       "      <td>Egu019696-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>574</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>124146at4447</td>\n",
       "      <td>7.790000e-241</td>\n",
       "      <td>224.6400</td>\n",
       "      <td>I</td>\n",
       "      <td>palmitoyl-acyl carrier protein thioesterase, c...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>3.1.2.-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR002864,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Egu020397</td>\n",
       "      <td>Egu020397</td>\n",
       "      <td>Egu020397-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>530</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>118678at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>238.5600</td>\n",
       "      <td>I,E,C,F</td>\n",
       "      <td>Biotin carboxylase</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>6.3.4.14,6.4.1.2</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR004549,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Egu020397</td>\n",
       "      <td>Egu020397</td>\n",
       "      <td>Egu020397-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>547</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>118678at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>99.6292</td>\n",
       "      <td>I,E,C,F</td>\n",
       "      <td>Biotin carboxylase</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>6.3.4.14,6.4.1.2</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR004549,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Egu020478</td>\n",
       "      <td>Egu020478</td>\n",
       "      <td>Egu020478-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>467</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>148338at4447</td>\n",
       "      <td>3.730000e-287</td>\n",
       "      <td>240.6200</td>\n",
       "      <td>I</td>\n",
       "      <td>palmitoyl-acyl carrier protein thioesterase, c...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.11.1.7</td>\n",
       "      <td>ko00940</td>\n",
       "      <td>IPR002864,IPR021113,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Egu007722</td>\n",
       "      <td>Egu007722</td>\n",
       "      <td>Egu007722-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "      <td>1174</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>38560at4447</td>\n",
       "      <td>2.670000e-285</td>\n",
       "      <td>214.0700</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016747,GO:0016740,GO:0004315</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.1.1.-</td>\n",
       "      <td>ko00330,ko00592,sbi01100,sbi01110,sita01100</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Egu007722</td>\n",
       "      <td>Egu007722</td>\n",
       "      <td>Egu007722-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "      <td>1184</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>38560at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>100.0000</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016747,GO:0016740,GO:0004315</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.1.1.-</td>\n",
       "      <td>ko00330,ko00592,sbi01100,sbi01110,sita01100</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Egu007723</td>\n",
       "      <td>Egu007723</td>\n",
       "      <td>Egu007723-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>24</td>\n",
       "      <td>1374</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>154947at4447</td>\n",
       "      <td>6.400000e-57</td>\n",
       "      <td>98.0400</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>Beta-ketoacyl synthase</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR000794,IPR014030,IPR016039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Egu007724</td>\n",
       "      <td>Egu007724</td>\n",
       "      <td>Egu007724-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>734</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>150042at4447</td>\n",
       "      <td>4.170000e-124</td>\n",
       "      <td>253.3100</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0004315</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR000794,IPR014031,IPR016039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Egu007955</td>\n",
       "      <td>Egu007955</td>\n",
       "      <td>Egu007955-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>27</td>\n",
       "      <td>745</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>112963at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>99.1699</td>\n",
       "      <td>I,C,E,H</td>\n",
       "      <td>Acetyl-CoA carboxylase</td>\n",
       "      <td>GO:0016874,GO:0005524,GO:0000166,GO:0046872,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR000089,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Egu007955</td>\n",
       "      <td>Egu007955</td>\n",
       "      <td>Egu007955-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>28</td>\n",
       "      <td>814</td>\n",
       "      <td>22</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>112963at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>237.9000</td>\n",
       "      <td>I,C,E,H</td>\n",
       "      <td>Acetyl-CoA carboxylase</td>\n",
       "      <td>GO:0016874,GO:0005524,GO:0000166,GO:0046872,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR000089,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Egu008612</td>\n",
       "      <td>Egu008612</td>\n",
       "      <td>Egu008612-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>342</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>117154at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>248.3500</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016747,GO:0016740,GO:0004315</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.2.4.1</td>\n",
       "      <td>ko00010,ko00020,ko00620,ko01200</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Egu009350</td>\n",
       "      <td>Egu009350</td>\n",
       "      <td>Egu009350-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>591</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>119288at4447</td>\n",
       "      <td>1.370000e-273</td>\n",
       "      <td>227.0700</td>\n",
       "      <td>I</td>\n",
       "      <td>Acyl-</td>\n",
       "      <td>GO:0016787,GO:0016790</td>\n",
       "      <td>GO:0006629,GO:0006631</td>\n",
       "      <td>2.3.2.27,3.1.2.-</td>\n",
       "      <td>ko04626,ko03018,ko04120,ko04141</td>\n",
       "      <td>IPR002864,IPR021113,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Egu016836</td>\n",
       "      <td>Egu016836</td>\n",
       "      <td>Egu016836-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1020</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>132388at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>100.0000</td>\n",
       "      <td>I</td>\n",
       "      <td>oleoyl-acyl carrier protein thioesterase, chlo...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>3.1.2.-</td>\n",
       "      <td>ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...</td>\n",
       "      <td>IPR002864,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Egu016836</td>\n",
       "      <td>Egu016836</td>\n",
       "      <td>Egu016836-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>1008</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>132388at4447</td>\n",
       "      <td>3.430000e-207</td>\n",
       "      <td>202.4800</td>\n",
       "      <td>I</td>\n",
       "      <td>oleoyl-acyl carrier protein thioesterase, chlo...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>3.1.2.-</td>\n",
       "      <td>ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...</td>\n",
       "      <td>IPR002864,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Egu016840</td>\n",
       "      <td>Egu016840</td>\n",
       "      <td>Egu016840-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>761</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>13185at4447</td>\n",
       "      <td>1.190000e-251</td>\n",
       "      <td>223.8900</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0045300,GO:0016491,GO:0046872</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Egu017527</td>\n",
       "      <td>Egu017527</td>\n",
       "      <td>Egu017527-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>228</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>924at4447</td>\n",
       "      <td>2.420000e-263</td>\n",
       "      <td>235.2900</td>\n",
       "      <td>I,X,C,P</td>\n",
       "      <td>Fatty acid desaturase DES2</td>\n",
       "      <td>GO:0016717</td>\n",
       "      <td>GO:0006629,GO:0055114</td>\n",
       "      <td>3.4.11.18</td>\n",
       "      <td>ko00561,ko03015,osa01100</td>\n",
       "      <td>IPR005804,IPR021863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Egu017579</td>\n",
       "      <td>Egu017579</td>\n",
       "      <td>Egu017579-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>488</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>132738at4447</td>\n",
       "      <td>2.860000e-237</td>\n",
       "      <td>217.9700</td>\n",
       "      <td>-</td>\n",
       "      <td>enoyl-</td>\n",
       "      <td>GO:0004318,GO:0016491</td>\n",
       "      <td>GO:0055114</td>\n",
       "      <td>3.4.19.12,2.4.2.-,3.1.3.2</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR002347,IPR014358,IPR036291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Egu017579</td>\n",
       "      <td>Egu017579</td>\n",
       "      <td>Egu017579-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>489</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>156112at4447</td>\n",
       "      <td>5.220000e-43</td>\n",
       "      <td>77.9500</td>\n",
       "      <td>-</td>\n",
       "      <td>enoyl-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Egu018339</td>\n",
       "      <td>Egu018339</td>\n",
       "      <td>Egu018339-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>206</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>146876at4447</td>\n",
       "      <td>1.190000e-56</td>\n",
       "      <td>74.5100</td>\n",
       "      <td>I,E,F,C</td>\n",
       "      <td>ATP-grasp fold</td>\n",
       "      <td>GO:0005524,GO:0000166,GO:0046872</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR005479,IPR005482,IPR011054,IPR011761,IPR011764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Egu018339</td>\n",
       "      <td>Egu018339</td>\n",
       "      <td>Egu018339-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>205</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>118678at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>235.4800</td>\n",
       "      <td>I,E,C,F</td>\n",
       "      <td>Biotin carboxylase</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>6.3.4.14,6.4.1.2</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR004549,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     GeneName     GeneId      TranscriptId         BioType  \\\n",
       "0   Egu004811  Egu004811  Egu004811-mRNA-1  protein_coding   \n",
       "1   Egu004811  Egu004811  Egu004811-mRNA-2  protein_coding   \n",
       "2   Egu022699  Egu022699  Egu022699-mRNA-1  protein_coding   \n",
       "3   Egu023815  Egu023815  Egu023815-mRNA-1  protein_coding   \n",
       "4   Egu000360  Egu000360  Egu000360-mRNA-1  protein_coding   \n",
       "5   Egu000360  Egu000360  Egu000360-mRNA-2  protein_coding   \n",
       "6   Egu001574  Egu001574  Egu001574-mRNA-3  protein_coding   \n",
       "7   Egu002030  Egu002030  Egu002030-mRNA-1  protein_coding   \n",
       "8   Egu002186  Egu002186  Egu002186-mRNA-1  protein_coding   \n",
       "9   Egu002499  Egu002499  Egu002499-mRNA-1  protein_coding   \n",
       "10  Egu003609  Egu003609  Egu003609-mRNA-1  protein_coding   \n",
       "11  Egu018432  Egu018432  Egu018432-mRNA-1  protein_coding   \n",
       "12  Egu019609  Egu019609  Egu019609-mRNA-1  protein_coding   \n",
       "13  Egu019621  Egu019621  Egu019621-mRNA-1  protein_coding   \n",
       "14  Egu019696  Egu019696  Egu019696-mRNA-1  protein_coding   \n",
       "15  Egu020397  Egu020397  Egu020397-mRNA-1  protein_coding   \n",
       "16  Egu020397  Egu020397  Egu020397-mRNA-2  protein_coding   \n",
       "17  Egu020478  Egu020478  Egu020478-mRNA-1  protein_coding   \n",
       "18  Egu007722  Egu007722  Egu007722-mRNA-1  protein_coding   \n",
       "19  Egu007722  Egu007722  Egu007722-mRNA-2  protein_coding   \n",
       "20  Egu007723  Egu007723  Egu007723-mRNA-1  protein_coding   \n",
       "21  Egu007724  Egu007724  Egu007724-mRNA-1  protein_coding   \n",
       "22  Egu007955  Egu007955  Egu007955-mRNA-1  protein_coding   \n",
       "23  Egu007955  Egu007955  Egu007955-mRNA-2  protein_coding   \n",
       "24  Egu008612  Egu008612  Egu008612-mRNA-1  protein_coding   \n",
       "25  Egu009350  Egu009350  Egu009350-mRNA-1  protein_coding   \n",
       "26  Egu016836  Egu016836  Egu016836-mRNA-1  protein_coding   \n",
       "27  Egu016836  Egu016836  Egu016836-mRNA-2  protein_coding   \n",
       "28  Egu016840  Egu016840  Egu016840-mRNA-1  protein_coding   \n",
       "29  Egu017527  Egu017527  Egu017527-mRNA-1  protein_coding   \n",
       "30  Egu017579  Egu017579  Egu017579-mRNA-1  protein_coding   \n",
       "31  Egu017579  Egu017579  Egu017579-mRNA-2  protein_coding   \n",
       "32  Egu018339  Egu018339  Egu018339-mRNA-1  protein_coding   \n",
       "33  Egu018339  Egu018339  Egu018339-mRNA-2  protein_coding   \n",
       "\n",
       "    variants_impact_HIGH  variants_impact_LOW  variants_impact_MODERATE  \\\n",
       "0                      0                    8                         9   \n",
       "1                      0                    8                         8   \n",
       "2                      0                    5                        10   \n",
       "3                      0                   11                         4   \n",
       "4                      1                    3                         7   \n",
       "5                      1                    3                         7   \n",
       "6                      3                   10                        10   \n",
       "7                      0                    6                         3   \n",
       "8                      1                    9                         7   \n",
       "9                      1                    3                         2   \n",
       "10                     0                   11                         8   \n",
       "11                     0                    4                         2   \n",
       "12                     0                    6                         9   \n",
       "13                     0                    4                         2   \n",
       "14                     0                    4                         2   \n",
       "15                     0                   14                         8   \n",
       "16                     0                   15                         8   \n",
       "17                     0                    2                         0   \n",
       "18                     0                   25                         4   \n",
       "19                     0                   25                         4   \n",
       "20                     7                   10                        24   \n",
       "21                     1                   12                         8   \n",
       "22                     0                   45                        27   \n",
       "23                     0                   49                        28   \n",
       "24                     0                    8                         7   \n",
       "25                     0                    4                         2   \n",
       "26                     0                    8                         7   \n",
       "27                     0                    8                         7   \n",
       "28                     0                    6                         5   \n",
       "29                     0                    0                         2   \n",
       "30                     0                    9                         2   \n",
       "31                     0                    8                         2   \n",
       "32                     0                    0                         0   \n",
       "33                     0                    0                         1   \n",
       "\n",
       "    variants_impact_MODIFIER  variants_effect_3_prime_UTR_variant  \\\n",
       "0                        583                                   10   \n",
       "1                        558                                    0   \n",
       "2                        638                                    4   \n",
       "3                        711                                   17   \n",
       "4                        979                                    1   \n",
       "5                        979                                    1   \n",
       "6                        514                                    1   \n",
       "7                        683                                   13   \n",
       "8                        654                                   12   \n",
       "9                        281                                    9   \n",
       "10                       400                                   11   \n",
       "11                       422                                    2   \n",
       "12                       470                                    5   \n",
       "13                       437                                    4   \n",
       "14                       574                                    3   \n",
       "15                       530                                   18   \n",
       "16                       547                                    3   \n",
       "17                       467                                    6   \n",
       "18                      1174                                    4   \n",
       "19                      1184                                    8   \n",
       "20                      1374                                    0   \n",
       "21                       734                                    0   \n",
       "22                       745                                   22   \n",
       "23                       814                                   22   \n",
       "24                       342                                    1   \n",
       "25                       591                                    4   \n",
       "26                      1020                                   11   \n",
       "27                      1008                                    7   \n",
       "28                       761                                    5   \n",
       "29                       228                                    0   \n",
       "30                       488                                   17   \n",
       "31                       489                                   17   \n",
       "32                       206                                    4   \n",
       "33                       205                                    4   \n",
       "\n",
       "    variants_effect_5_prime_UTR_premature_start_codon_gain_variant  ...  \\\n",
       "0                                                   0               ...   \n",
       "1                                                   0               ...   \n",
       "2                                                   0               ...   \n",
       "3                                                   0               ...   \n",
       "4                                                   0               ...   \n",
       "5                                                   0               ...   \n",
       "6                                                   6               ...   \n",
       "7                                                   0               ...   \n",
       "8                                                   0               ...   \n",
       "9                                                   0               ...   \n",
       "10                                                  0               ...   \n",
       "11                                                  0               ...   \n",
       "12                                                  0               ...   \n",
       "13                                                  0               ...   \n",
       "14                                                  0               ...   \n",
       "15                                                  3               ...   \n",
       "16                                                  3               ...   \n",
       "17                                                  1               ...   \n",
       "18                                                  1               ...   \n",
       "19                                                  1               ...   \n",
       "20                                                  0               ...   \n",
       "21                                                  0               ...   \n",
       "22                                                  0               ...   \n",
       "23                                                  2               ...   \n",
       "24                                                  0               ...   \n",
       "25                                                  1               ...   \n",
       "26                                                  0               ...   \n",
       "27                                                  0               ...   \n",
       "28                                                  1               ...   \n",
       "29                                                  0               ...   \n",
       "30                                                  1               ...   \n",
       "31                                                  1               ...   \n",
       "32                                                  0               ...   \n",
       "33                                                  0               ...   \n",
       "\n",
       "          ODB_OG         evalue     score  COG_category  \\\n",
       "0   118148at4447   0.000000e+00  222.8500       I,C,E,H   \n",
       "1   118148at4447   0.000000e+00  222.3670       I,C,E,H   \n",
       "2    64848at4447   0.000000e+00  184.7670       I,Q,M,C   \n",
       "3    13185at4447   0.000000e+00  225.2550           P,F   \n",
       "4   128492at4447   0.000000e+00  179.5100         M,I,O   \n",
       "5    11254at4447  1.540000e-174  103.1600         M,O,T   \n",
       "6   156112at4447   0.000000e+00  128.8810             -   \n",
       "7    13185at4447   0.000000e+00  224.8720           P,F   \n",
       "8   134714at4447  5.890000e-174  169.5900             -   \n",
       "9    64848at4447  7.160000e-148  219.0900       I,Q,M,C   \n",
       "10  129798at4447  7.830000e-226  207.9300           P,F   \n",
       "11  119288at4447  1.840000e-218  186.1400             I   \n",
       "12  132388at4447   0.000000e+00  196.8140             I   \n",
       "13   13185at4447   0.000000e+00  239.4400           P,F   \n",
       "14  124146at4447  7.790000e-241  224.6400             I   \n",
       "15  118678at4447   0.000000e+00  238.5600       I,E,C,F   \n",
       "16  118678at4447   0.000000e+00   99.6292       I,E,C,F   \n",
       "17  148338at4447  3.730000e-287  240.6200             I   \n",
       "18   38560at4447  2.670000e-285  214.0700         Q,I,C   \n",
       "19   38560at4447   0.000000e+00  100.0000         Q,I,C   \n",
       "20  154947at4447   6.400000e-57   98.0400         Q,I,C   \n",
       "21  150042at4447  4.170000e-124  253.3100         Q,I,C   \n",
       "22  112963at4447   0.000000e+00   99.1699       I,C,E,H   \n",
       "23  112963at4447   0.000000e+00  237.9000       I,C,E,H   \n",
       "24  117154at4447   0.000000e+00  248.3500         Q,I,C   \n",
       "25  119288at4447  1.370000e-273  227.0700             I   \n",
       "26  132388at4447   0.000000e+00  100.0000             I   \n",
       "27  132388at4447  3.430000e-207  202.4800             I   \n",
       "28   13185at4447  1.190000e-251  223.8900           P,F   \n",
       "29     924at4447  2.420000e-263  235.2900       I,X,C,P   \n",
       "30  132738at4447  2.860000e-237  217.9700             -   \n",
       "31  156112at4447   5.220000e-43   77.9500             -   \n",
       "32  146876at4447   1.190000e-56   74.5100       I,E,F,C   \n",
       "33  118678at4447   0.000000e+00  235.4800       I,E,C,F   \n",
       "\n",
       "                                          Description  \\\n",
       "0   Methylcrotonoyl-CoA carboxylase subunit alpha,...   \n",
       "1   Methylcrotonoyl-CoA carboxylase subunit alpha,...   \n",
       "2                              (3R)-hydroxymyristoyl-   \n",
       "3                                           stearoyl-   \n",
       "4        Acyl-CoA-binding domain-containing protein 6   \n",
       "5        acyl-CoA-binding domain-containing protein 4   \n",
       "6                                              enoyl-   \n",
       "7                                           stearoyl-   \n",
       "8                                          3-oxoacyl-   \n",
       "9                              (3R)-hydroxymyristoyl-   \n",
       "10                                          Stearoyl-   \n",
       "11                                              Acyl-   \n",
       "12  oleoyl-acyl carrier protein thioesterase, chlo...   \n",
       "13                                          stearoyl-   \n",
       "14  palmitoyl-acyl carrier protein thioesterase, c...   \n",
       "15                                 Biotin carboxylase   \n",
       "16                                 Biotin carboxylase   \n",
       "17  palmitoyl-acyl carrier protein thioesterase, c...   \n",
       "18                                         3-oxoacyl-   \n",
       "19                                         3-oxoacyl-   \n",
       "20                             Beta-ketoacyl synthase   \n",
       "21                                         3-oxoacyl-   \n",
       "22                             Acetyl-CoA carboxylase   \n",
       "23                             Acetyl-CoA carboxylase   \n",
       "24                                         3-oxoacyl-   \n",
       "25                                              Acyl-   \n",
       "26  oleoyl-acyl carrier protein thioesterase, chlo...   \n",
       "27  oleoyl-acyl carrier protein thioesterase, chlo...   \n",
       "28                                          stearoyl-   \n",
       "29                         Fatty acid desaturase DES2   \n",
       "30                                             enoyl-   \n",
       "31                                             enoyl-   \n",
       "32                                     ATP-grasp fold   \n",
       "33                                 Biotin carboxylase   \n",
       "\n",
       "                                               GOs_mf  \\\n",
       "0         GO:0046872,GO:0016874,GO:0005524,GO:0000166   \n",
       "1         GO:0046872,GO:0016874,GO:0005524,GO:0000166   \n",
       "2   GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...   \n",
       "3                    GO:0045300,GO:0016491,GO:0046872   \n",
       "4                               GO:0008289,GO:0000062   \n",
       "5                                          GO:0000062   \n",
       "6                                                   -   \n",
       "7                    GO:0045300,GO:0016491,GO:0046872   \n",
       "8         GO:0016491,GO:0004316,GO:0102131,GO:0102132   \n",
       "9   GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...   \n",
       "10                   GO:0046872,GO:0045300,GO:0016491   \n",
       "11                              GO:0016787,GO:0016790   \n",
       "12                              GO:0016790,GO:0016787   \n",
       "13                   GO:0045300,GO:0016491,GO:0046872   \n",
       "14                              GO:0016790,GO:0016787   \n",
       "15  GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...   \n",
       "16  GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...   \n",
       "17                              GO:0016790,GO:0016787   \n",
       "18                   GO:0016747,GO:0016740,GO:0004315   \n",
       "19                   GO:0016747,GO:0016740,GO:0004315   \n",
       "20                                                  -   \n",
       "21                                         GO:0004315   \n",
       "22  GO:0016874,GO:0005524,GO:0000166,GO:0046872,GO...   \n",
       "23  GO:0016874,GO:0005524,GO:0000166,GO:0046872,GO...   \n",
       "24                   GO:0016747,GO:0016740,GO:0004315   \n",
       "25                              GO:0016787,GO:0016790   \n",
       "26                              GO:0016790,GO:0016787   \n",
       "27                              GO:0016790,GO:0016787   \n",
       "28                   GO:0045300,GO:0016491,GO:0046872   \n",
       "29                                         GO:0016717   \n",
       "30                              GO:0004318,GO:0016491   \n",
       "31                                                  -   \n",
       "32                   GO:0005524,GO:0000166,GO:0046872   \n",
       "33  GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...   \n",
       "\n",
       "                              GOs_bp                         EC  \\\n",
       "0                                  -                          -   \n",
       "1                                  -                          -   \n",
       "2              GO:0009245,GO:0006629                          -   \n",
       "3   GO:0006631,GO:0006629,GO:0055114                  1.14.19.-   \n",
       "4                         GO:0006869                          -   \n",
       "5                         GO:0006869                          -   \n",
       "6                                  -                          -   \n",
       "7   GO:0006631,GO:0006629,GO:0055114                  1.14.19.-   \n",
       "8   GO:0006629,GO:0006631,GO:0055114                          -   \n",
       "9              GO:0009245,GO:0006629                          -   \n",
       "10  GO:0006631,GO:0006629,GO:0055114                   1.1.1.44   \n",
       "11             GO:0006629,GO:0006631           2.3.2.27,3.1.2.-   \n",
       "12             GO:0006631,GO:0006629                    3.1.2.-   \n",
       "13  GO:0006631,GO:0006629,GO:0055114                  1.14.19.-   \n",
       "14             GO:0006631,GO:0006629                    3.1.2.-   \n",
       "15  GO:2001295,GO:0006631,GO:0006629           6.3.4.14,6.4.1.2   \n",
       "16  GO:2001295,GO:0006631,GO:0006629           6.3.4.14,6.4.1.2   \n",
       "17             GO:0006631,GO:0006629                   1.11.1.7   \n",
       "18             GO:0006631,GO:0006629                    1.1.1.-   \n",
       "19             GO:0006631,GO:0006629                    1.1.1.-   \n",
       "20                                 -                          -   \n",
       "21                                 -                          -   \n",
       "22  GO:2001295,GO:0006631,GO:0006629                          -   \n",
       "23  GO:2001295,GO:0006631,GO:0006629                          -   \n",
       "24             GO:0006631,GO:0006629                    1.2.4.1   \n",
       "25             GO:0006629,GO:0006631           2.3.2.27,3.1.2.-   \n",
       "26             GO:0006631,GO:0006629                    3.1.2.-   \n",
       "27             GO:0006631,GO:0006629                    3.1.2.-   \n",
       "28  GO:0006631,GO:0006629,GO:0055114                  1.14.19.-   \n",
       "29             GO:0006629,GO:0055114                  3.4.11.18   \n",
       "30                        GO:0055114  3.4.19.12,2.4.2.-,3.1.3.2   \n",
       "31                                 -                          -   \n",
       "32                                 -                          -   \n",
       "33  GO:2001295,GO:0006631,GO:0006629           6.3.4.14,6.4.1.2   \n",
       "\n",
       "                                              KEGG_ko  \\\n",
       "0                                     ko00860,ko04144   \n",
       "1                                     ko00860,ko04144   \n",
       "2   ko00230,ko00730,ko04075,ko04626,zma01100,zma01110   \n",
       "3                                             ko04626   \n",
       "4   bdi01100,bdi01110,ko00220,ko00250,ko00520,ko00...   \n",
       "5                                                   -   \n",
       "6                                                   -   \n",
       "7                                             ko04626   \n",
       "8                             ko00260,ko03010,ko03013   \n",
       "9   ko00230,ko00730,ko04075,ko04626,zma01100,zma01110   \n",
       "10                    ko00030,ko00195,ko00480,ko01200   \n",
       "11                    ko04626,ko03018,ko04120,ko04141   \n",
       "12  ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...   \n",
       "13                                            ko04626   \n",
       "14                                                  -   \n",
       "15                                                  -   \n",
       "16                                                  -   \n",
       "17                                            ko00940   \n",
       "18        ko00330,ko00592,sbi01100,sbi01110,sita01100   \n",
       "19        ko00330,ko00592,sbi01100,sbi01110,sita01100   \n",
       "20                                                  -   \n",
       "21                                                  -   \n",
       "22                                                  -   \n",
       "23                                                  -   \n",
       "24                    ko00010,ko00020,ko00620,ko01200   \n",
       "25                    ko04626,ko03018,ko04120,ko04141   \n",
       "26  ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...   \n",
       "27  ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...   \n",
       "28                                            ko04626   \n",
       "29                           ko00561,ko03015,osa01100   \n",
       "30                                            ko04626   \n",
       "31                                                  -   \n",
       "32                                                  -   \n",
       "33                                                  -   \n",
       "\n",
       "                                             Interpro  \n",
       "0   IPR000089,IPR001882,IPR005479,IPR005481,IPR005...  \n",
       "1   IPR000089,IPR001882,IPR005479,IPR005481,IPR005...  \n",
       "2                       IPR010084,IPR013114,IPR029069  \n",
       "3             IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "4   IPR000582,IPR006652,IPR014352,IPR015915,IPR035984  \n",
       "5                                 IPR006652,IPR015915  \n",
       "6                                                   -  \n",
       "7             IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "8             IPR002347,IPR011284,IPR020904,IPR036291  \n",
       "9                       IPR010084,IPR013114,IPR029069  \n",
       "10                      IPR005067,IPR009078,IPR012348  \n",
       "11            IPR002864,IPR021113,IPR029069,IPR045023  \n",
       "12                      IPR002864,IPR029069,IPR045023  \n",
       "13            IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "14                      IPR002864,IPR029069,IPR045023  \n",
       "15  IPR004549,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "16  IPR004549,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "17            IPR002864,IPR021113,IPR029069,IPR045023  \n",
       "18  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "19  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "20                      IPR000794,IPR014030,IPR016039  \n",
       "21                      IPR000794,IPR014031,IPR016039  \n",
       "22  IPR000089,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "23  IPR000089,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "24  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "25            IPR002864,IPR021113,IPR029069,IPR045023  \n",
       "26                      IPR002864,IPR029069,IPR045023  \n",
       "27                      IPR002864,IPR029069,IPR045023  \n",
       "28            IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "29                                IPR005804,IPR021863  \n",
       "30                      IPR002347,IPR014358,IPR036291  \n",
       "31                                                  -  \n",
       "32  IPR005479,IPR005482,IPR011054,IPR011761,IPR011764  \n",
       "33  IPR004549,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "\n",
       "[34 rows x 43 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "73b20bc1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3332468/894072683.py:2: DtypeWarning: Columns (4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv('/bioinfo2/palm/knowledgebase/chr12679.snpEff.genes.txt', sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('/bioinfo2/palm/knowledgebase/chr12679.snpEff.genes.txt', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "24919880",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>GeneId</th>\n",
       "      <th>TranscriptId</th>\n",
       "      <th>BioType</th>\n",
       "      <th>variants_impact_HIGH</th>\n",
       "      <th>variants_impact_LOW</th>\n",
       "      <th>variants_impact_MODERATE</th>\n",
       "      <th>variants_impact_MODIFIER</th>\n",
       "      <th>variants_effect_3_prime_UTR_variant</th>\n",
       "      <th>variants_effect_5_prime_UTR_premature_start_codon_gain_variant</th>\n",
       "      <th>...</th>\n",
       "      <th>variants_effect_splice_acceptor_variant</th>\n",
       "      <th>variants_effect_splice_donor_variant</th>\n",
       "      <th>variants_effect_splice_region_variant</th>\n",
       "      <th>variants_effect_start_lost</th>\n",
       "      <th>variants_effect_start_retained_variant</th>\n",
       "      <th>variants_effect_stop_gained</th>\n",
       "      <th>variants_effect_stop_lost</th>\n",
       "      <th>variants_effect_stop_retained_variant</th>\n",
       "      <th>variants_effect_synonymous_variant</th>\n",
       "      <th>variants_effect_upstream_gene_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Egu003662</td>\n",
       "      <td>Egu003662</td>\n",
       "      <td>Egu003662-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>625</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>353.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Egu003663</td>\n",
       "      <td>Egu003663</td>\n",
       "      <td>Egu003663-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>476</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>192.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Egu003664</td>\n",
       "      <td>Egu003664</td>\n",
       "      <td>Egu003664-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>414</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>186.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Egu003664</td>\n",
       "      <td>Egu003664</td>\n",
       "      <td>Egu003664-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>414</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>186.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Egu003665</td>\n",
       "      <td>Egu003665</td>\n",
       "      <td>Egu003665-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>690</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>475.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17946</th>\n",
       "      <td>Egu024114</td>\n",
       "      <td>Egu024114</td>\n",
       "      <td>Egu024114-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>264</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>176</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17947</th>\n",
       "      <td>Egu024115</td>\n",
       "      <td>Egu024115</td>\n",
       "      <td>Egu024115-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>1477</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17948</th>\n",
       "      <td>Egu024116</td>\n",
       "      <td>Egu024116</td>\n",
       "      <td>Egu024116-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>294</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17949</th>\n",
       "      <td>Egu024116</td>\n",
       "      <td>Egu024116</td>\n",
       "      <td>Egu024116-mRNA-2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>290</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17950</th>\n",
       "      <td>Egu024117</td>\n",
       "      <td>Egu024117</td>\n",
       "      <td>Egu024117-mRNA-1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>624</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17951 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneName     GeneId      TranscriptId         BioType  \\\n",
       "0      Egu003662  Egu003662  Egu003662-mRNA-1  protein_coding   \n",
       "1      Egu003663  Egu003663  Egu003663-mRNA-1  protein_coding   \n",
       "2      Egu003664  Egu003664  Egu003664-mRNA-1  protein_coding   \n",
       "3      Egu003664  Egu003664  Egu003664-mRNA-2  protein_coding   \n",
       "4      Egu003665  Egu003665  Egu003665-mRNA-1  protein_coding   \n",
       "...          ...        ...               ...             ...   \n",
       "17946  Egu024114  Egu024114  Egu024114-mRNA-1  protein_coding   \n",
       "17947  Egu024115  Egu024115  Egu024115-mRNA-1  protein_coding   \n",
       "17948  Egu024116  Egu024116  Egu024116-mRNA-1  protein_coding   \n",
       "17949  Egu024116  Egu024116  Egu024116-mRNA-2  protein_coding   \n",
       "17950  Egu024117  Egu024117  Egu024117-mRNA-1  protein_coding   \n",
       "\n",
       "      variants_impact_HIGH variants_impact_LOW variants_impact_MODERATE  \\\n",
       "0                        0                   6                        1   \n",
       "1                        0                   0                        2   \n",
       "2                        0                   2                        2   \n",
       "3                        0                   2                        2   \n",
       "4                        0                   6                        8   \n",
       "...                    ...                 ...                      ...   \n",
       "17946                    1                   0                        0   \n",
       "17947                    2                   7                       12   \n",
       "17948                    0                   0                        0   \n",
       "17949                    0                   0                        0   \n",
       "17950                    7                   5                       17   \n",
       "\n",
       "      variants_impact_MODIFIER variants_effect_3_prime_UTR_variant  \\\n",
       "0                          625                                   1   \n",
       "1                          476                                   6   \n",
       "2                          414                                   2   \n",
       "3                          414                                   2   \n",
       "4                          690                                   3   \n",
       "...                        ...                                 ...   \n",
       "17946                      264                                   0   \n",
       "17947                     1477                                   0   \n",
       "17948                      294                                   0   \n",
       "17949                      290                                   0   \n",
       "17950                      624                                   0   \n",
       "\n",
       "      variants_effect_5_prime_UTR_premature_start_codon_gain_variant  ...  \\\n",
       "0                                                      0              ...   \n",
       "1                                                      0              ...   \n",
       "2                                                      0              ...   \n",
       "3                                                      0              ...   \n",
       "4                                                      0              ...   \n",
       "...                                                  ...              ...   \n",
       "17946                                                  0              ...   \n",
       "17947                                                  0              ...   \n",
       "17948                                                  0              ...   \n",
       "17949                                                  0              ...   \n",
       "17950                                                  0              ...   \n",
       "\n",
       "      variants_effect_splice_acceptor_variant  \\\n",
       "0                                           0   \n",
       "1                                           0   \n",
       "2                                           0   \n",
       "3                                           0   \n",
       "4                                           0   \n",
       "...                                       ...   \n",
       "17946                                       1   \n",
       "17947                                       7   \n",
       "17948                                       0   \n",
       "17949                                       0   \n",
       "17950                                       2   \n",
       "\n",
       "      variants_effect_splice_donor_variant  \\\n",
       "0                                        0   \n",
       "1                                        0   \n",
       "2                                        0   \n",
       "3                                        0   \n",
       "4                                        0   \n",
       "...                                    ...   \n",
       "17946                                    0   \n",
       "17947                                    0   \n",
       "17948                                    0   \n",
       "17949                                    0   \n",
       "17950                                    1   \n",
       "\n",
       "      variants_effect_splice_region_variant variants_effect_start_lost  \\\n",
       "0                                         1                          0   \n",
       "1                                         0                          0   \n",
       "2                                         2                          0   \n",
       "3                                         2                          0   \n",
       "4                                         0                          0   \n",
       "...                                     ...                        ...   \n",
       "17946                                     0                          0   \n",
       "17947                                     0                          1   \n",
       "17948                                     0                          0   \n",
       "17949                                     0                          0   \n",
       "17950                                     0                          4   \n",
       "\n",
       "      variants_effect_start_retained_variant variants_effect_stop_gained  \\\n",
       "0                                          0                           0   \n",
       "1                                          0                           0   \n",
       "2                                          0                           0   \n",
       "3                                          0                           0   \n",
       "4                                          0                           0   \n",
       "...                                      ...                         ...   \n",
       "17946                                      1                           0   \n",
       "17947                                      0                           0   \n",
       "17948                                      0                           0   \n",
       "17949                                      0                           0   \n",
       "17950                                      2                           0   \n",
       "\n",
       "      variants_effect_stop_lost variants_effect_stop_retained_variant  \\\n",
       "0                             0                                     0   \n",
       "1                             0                                     0   \n",
       "2                             0                                     0   \n",
       "3                             0                                     0   \n",
       "4                             0                                     0   \n",
       "...                         ...                                   ...   \n",
       "17946                         0                                   176   \n",
       "17947                         4                                   258   \n",
       "17948                         0                                     9   \n",
       "17949                         0                                     9   \n",
       "17950                         5                                   219   \n",
       "\n",
       "      variants_effect_synonymous_variant variants_effect_upstream_gene_variant  \n",
       "0                                      5                                 353.0  \n",
       "1                                      0                                 192.0  \n",
       "2                                      0                                 186.0  \n",
       "3                                      0                                 186.0  \n",
       "4                                      6                                 475.0  \n",
       "...                                  ...                                   ...  \n",
       "17946                                NaN                                   NaN  \n",
       "17947                                NaN                                   NaN  \n",
       "17948                                NaN                                   NaN  \n",
       "17949                                NaN                                   NaN  \n",
       "17950                                NaN                                   NaN  \n",
       "\n",
       "[17951 rows x 33 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "75b5ba10",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam\n",
    "import pandas as pd\n",
    "vcf = pysam.VariantFile(\"/bioinfo2/palm/analysis/yield/Apriyanto2023/Apriyanto2023_DE100.vcf\", \"r\") \n",
    "for variant in vcf:\n",
    "    if variant.chrom == CHR:\n",
    "        if variant.pos == POS:\n",
    "            REF = variant.ref\n",
    "            ALT = variant.alts\n",
    "            break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c633ca56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/xiaorong/anaconda3/envs/bioinfo/bin/python'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import sys\n",
    "sys.executable"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06d431dc",
   "metadata": {},
   "source": [
    "## Get the gene name and amino acid give the coord of a SNP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "fbbe2632",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "warning: this exon is not dividable by 3\n",
      "GAA E AAA K\n"
     ]
    }
   ],
   "source": [
    "# first let's work with just one coord\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq, MutableSeq\n",
    "\n",
    "# eventually this will be a script that only takes the coord, and go look the exon and REF ALT by it self.\n",
    "# * means stop codon. When this is changed we need to look further down the sequence until another stop codon right?\n",
    "\n",
    "coord = 'chr7_37218457'\n",
    "bed_file = '/bioinfo2/palm/analysis/yield/Apriyanto2023_DE100.exon.bed'\n",
    "vcf_file = '/bioinfo2/palm/analysis/' \n",
    "\n",
    "CHR, POS = coord.split('_')\n",
    "POS = int(POS)\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "filtered_df = df_gff[(df_gff['Type'] == \"exon\") & (df['Seqid'] == CHR) & (df['Start'] <= POS) & (df['End'] >= POS)]\n",
    "gene = filtered_df['ID'].iloc[0]\n",
    "\n",
    "REF = 'G'\n",
    "ALT = 'A'\n",
    "\n",
    "filtered_rows = df_gff[df_gff['ID']==gene]\n",
    "start = int(filtered_rows['Start'])\n",
    "end = int(filtered_rows['End'])\n",
    "            \n",
    "for record in SeqIO.parse('/bioinfo2/palm/ref/dura/dura_ref.fasta','fasta'):\n",
    "    if record.id == CHR:\n",
    "        seq = record.seq[start-1:end]\n",
    "        pos_ref = record.seq[int(POS)-1]\n",
    "        # check if the length of this exon is divisible by three\n",
    "        if len(seq)%3 != 0:\n",
    "            print(\"warning: this exon is not dividable by 3\")\n",
    "            # check if the position matches with the REF provided\n",
    "        if pos_ref == REF:\n",
    "            # get the coding region containing this SNP\n",
    "            if (int(POS)-start)%3 == 0:\n",
    "                codon = record.seq[int(POS)-1:int(POS)+2]\n",
    "                codon_m = MutableSeq(str(codon))\n",
    "                codon_m[0] = ALT \n",
    "            elif (int(POS)-start)%3 == 1:\n",
    "                codon = record.seq[int(POS)-2:int(POS)+1]\n",
    "                codon_m = MutableSeq(str(codon))\n",
    "                codon_m[1] = ALT\n",
    "            elif (int(POS)-start)%3 == 2:\n",
    "                codon = record.seq[int(POS)-3:int(POS)]\n",
    "                codon_m = MutableSeq(str(codon))\n",
    "                codon_m[2] = ALT\n",
    "            # translate these codon\n",
    "            aa = codon.translate()\n",
    "            aa_m = codon_m.translate()\n",
    "            print(codon, aa, codon_m, aa_m)\n",
    "            break\n",
    "        else:\n",
    "            print(\"This coord in the genome is different from what was called in the vcf\")\n",
    "            break\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a496bd99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Seqid</th>\n",
       "      <th>Source</th>\n",
       "      <th>Type</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Phase</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>_AED</th>\n",
       "      <th>_QI</th>\n",
       "      <th>_eAED</th>\n",
       "      <th>score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>gene</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0...</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0|0|0|0.5|1|1|2|0|472</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>11686</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032749-mRNA-1:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615982</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>281718</td>\n",
       "      <td>281844</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615983</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>280181</td>\n",
       "      <td>280249</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615984</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>279893</td>\n",
       "      <td>280085</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615985</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>278640</td>\n",
       "      <td>279072</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615986</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>278427</td>\n",
       "      <td>278639</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032211-mRNA-2:three_prime_utr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>615987 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Seqid Source             Type   Start     End Score Strand  \\\n",
       "0       egu.contig.1973  maker             gene   10123   11712     .      +   \n",
       "1       egu.contig.1973  maker             mRNA   10123   11712     .      +   \n",
       "2       egu.contig.1973  maker             exon   10123   11511     .      +   \n",
       "3       egu.contig.1973  maker             exon   11686   11712     .      +   \n",
       "4       egu.contig.1973  maker              CDS   10123   11511     .      +   \n",
       "...                 ...    ...              ...     ...     ...   ...    ...   \n",
       "615982  egu.contig.1149  maker              CDS  281718  281844     .      -   \n",
       "615983  egu.contig.1149  maker              CDS  280181  280249     .      -   \n",
       "615984  egu.contig.1149  maker              CDS  279893  280085     .      -   \n",
       "615985  egu.contig.1149  maker              CDS  278640  279072     .      -   \n",
       "615986  egu.contig.1149  maker  three_prime_UTR  278427  278639     .      -   \n",
       "\n",
       "       Phase                                ID              Name  \\\n",
       "0          .                         Egu032749         Egu032749   \n",
       "1          .                  Egu032749-mRNA-1  Egu032749-mRNA-1   \n",
       "2          .                Egu032749-mRNA-1:1               NaN   \n",
       "3          .                Egu032749-mRNA-1:2               NaN   \n",
       "4          0              Egu032749-mRNA-1:cds               NaN   \n",
       "...      ...                               ...               ...   \n",
       "615982     0              Egu032211-mRNA-2:cds               NaN   \n",
       "615983     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615984     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615985     1              Egu032211-mRNA-2:cds               NaN   \n",
       "615986     .  Egu032211-mRNA-2:three_prime_utr               NaN   \n",
       "\n",
       "                                                    Alias     \\\n",
       "0          snap_masked-egu.contig.1973-processed-gene-0.0      \n",
       "1       snap_masked-egu.contig.1973-processed-gene-0.0...      \n",
       "2                                                     NaN      \n",
       "3                                                     NaN      \n",
       "4                                                     NaN      \n",
       "...                                                   ... ..   \n",
       "615982                                                NaN      \n",
       "615983                                                NaN      \n",
       "615984                                                NaN      \n",
       "615985                                                NaN      \n",
       "615986                                                NaN      \n",
       "\n",
       "                  Parent  _AED                    _QI _eAED score  \n",
       "0                    NaN   NaN                    NaN   NaN   NaN  \n",
       "1              Egu032749  0.06  0|0|0|0.5|1|1|2|0|472  0.04   NaN  \n",
       "2       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "3       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "4       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "...                  ...   ...                    ...   ...   ...  \n",
       "615982  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615983  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615984  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615985  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615986  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "\n",
       "[615987 rows x 17 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "1bd32dad",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Seq import Seq, MutableSeq\n",
    "\n",
    "seq = Seq('GACGTTAGGAGGAGCATATACTAACTTGTTTTTATAA')\n",
    "seq.translate()\n",
    "seq1 = MutableSeq(str(seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "4904af90",
   "metadata": {},
   "outputs": [],
   "source": [
    "aa = Seq('MHLGHRHGHEKGKSPATTMPPKGWVGIRVGQEGEEQQRFEVPVDYLKHPLFMALLHQAKEEFGYEQSGAITIPCGVDHFRHVRDTINRDSAATAAAHHHSHLPHLAGLTLGGAYTNLFL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "ca45f75d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('TLGGAYTNLFL*')"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq = Seq('ACGTTAGGAGGAGCATATACTAACTTGTTTTTATAA')\n",
    "seq.translate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "c7395684",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq = Seq('ATGCACCTGGGCCACAGGCACGGGCATGAGAAGGGAAAGAGTCCGGCGACGACGATGCCACCTAAAGGGTGGGTGGGGATAAGGGTGGGGCAGGAAGGGGAGGAGCAGCAGCGGTTCGAGGTGCCTGTGGACTATCTGAAGCACCCGCTCTTCATGGCCTTGCTGCACCAGGCGAAGGAGGAGTTCGGATATGAGCAGAGCGGAGCCATCACCATTCCCTGCGGCGTCGATCACTTCCGTCATGTCCGGGACACCATTAACCGTGACTCCGCGGCCACCGCCGCCGCGCACCACCACAGCCACCTCCCTCACCTTGCCGGCT')\n",
    "a = seq.translate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "d9c8a778",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "322"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "344739b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('HSHLPHLAGC')"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a[-10:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "0b532e0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'gene': 32778,\n",
       "         'mRNA': 45938,\n",
       "         'exon': 196546,\n",
       "         'CDS': 271952,\n",
       "         'three_prime_UTR': 36830,\n",
       "         'five_prime_UTR': 31943})"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "Counter(df_gff['Type'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "20cc00ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'maker': 615987})"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(df_gff['Source'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "4819ff37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'snap_masked-egu.contig.1973-processed-gene-0.0': 1,\n",
       "         'snap_masked-egu.contig.1973-processed-gene-0.0-mRNA-1': 1,\n",
       "         nan: 537271,\n",
       "         'maker-egu.contig.1477-snap-gene-0.0': 1,\n",
       "         'maker-egu.contig.1477-snap-gene-0.0-mRNA-1': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.4': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.4-mRNA-1': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.5': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.5-mRNA-1': 1,\n",
       "         'snap_masked-egu.contig.1415-processed-gene-0.4': 1,\n",
       "         'snap_masked-egu.contig.1415-processed-gene-0.4-mRNA-1': 1,\n",
       "         'snap_masked-egu.contig.1962-processed-gene-0.2': 1,\n",
       "         'snap_masked-egu.contig.1962-processed-gene-0.2-mRNA-1': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.2': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.2-mRNA-1': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.0': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.0-mRNA-1': 1,\n",
       "         'maker-egu.contig.1588-snap-gene-0.1': 1,\n",
       "         'maker-egu.contig.1588-snap-gene-0.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.1': 1,\n",
       "         'maker-chr9-snap-gene-0.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.1-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-0.2': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-4': 1,\n",
       "         'snap_masked-chr9-processed-gene-0.11': 1,\n",
       "         'snap_masked-chr9-processed-gene-0.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.0': 1,\n",
       "         'maker-chr9-snap-gene-0.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.0-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-0.0-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-0.3': 1,\n",
       "         'maker-chr9-snap-gene-0.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.24': 1,\n",
       "         'maker-chr9-snap-gene-1.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.23': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-1.26': 1,\n",
       "         'maker-chr9-snap-gene-1.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.28': 1,\n",
       "         'maker-chr9-snap-gene-1.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.28-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-1.27': 1,\n",
       "         'maker-chr9-snap-gene-1.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.25': 1,\n",
       "         'maker-chr9-snap-gene-1.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.25-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.1': 1,\n",
       "         'maker-chr9-snap-gene-2.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.4': 1,\n",
       "         'maker-chr9-snap-gene-2.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.12': 1,\n",
       "         'maker-chr9-snap-gene-2.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.12-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.13': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-2.10': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-2.6': 1,\n",
       "         'maker-chr9-snap-gene-2.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-2.30': 1,\n",
       "         'snap_masked-chr9-processed-gene-2.30-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.9': 1,\n",
       "         'maker-chr9-snap-gene-2.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.9-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.5': 1,\n",
       "         'maker-chr9-snap-gene-2.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.7': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-7': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-8': 1,\n",
       "         'maker-chr9-snap-gene-3.31': 1,\n",
       "         'maker-chr9-snap-gene-3.31-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.12-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.2-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.30': 1,\n",
       "         'maker-chr9-snap-gene-3.30-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.30-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.24': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.34': 1,\n",
       "         'maker-chr9-snap-gene-3.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.37': 1,\n",
       "         'maker-chr9-snap-gene-3.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.32': 1,\n",
       "         'maker-chr9-snap-gene-3.32-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.28': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.35': 1,\n",
       "         'maker-chr9-snap-gene-3.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.24': 1,\n",
       "         'maker-chr9-snap-gene-3.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.8': 1,\n",
       "         'maker-chr9-snap-gene-4.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-4.1': 1,\n",
       "         'maker-chr9-snap-gene-4.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.2': 1,\n",
       "         'maker-chr9-snap-gene-4.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.4': 1,\n",
       "         'maker-chr9-snap-gene-4.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.3': 1,\n",
       "         'maker-chr9-snap-gene-4.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.3-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.35': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.7': 1,\n",
       "         'maker-chr9-snap-gene-4.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.5': 1,\n",
       "         'maker-chr9-snap-gene-4.5-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.37': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.37-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.29': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.11': 1,\n",
       "         'maker-chr9-snap-gene-4.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.34': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.4': 1,\n",
       "         'maker-chr9-snap-gene-5.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.0': 1,\n",
       "         'maker-chr9-snap-gene-5.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.40': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.40-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.6': 1,\n",
       "         'maker-chr9-snap-gene-5.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.34': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.17': 1,\n",
       "         'maker-chr9-snap-gene-5.17-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.44': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.44-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.27-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.7': 1,\n",
       "         'maker-chr9-snap-gene-5.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.14': 1,\n",
       "         'maker-chr9-snap-gene-5.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.14-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.48': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.48-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.52': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.52-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.11': 1,\n",
       "         'maker-chr9-snap-gene-5.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.50': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.50-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.15': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-5.13': 1,\n",
       "         'maker-chr9-snap-gene-5.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.12': 1,\n",
       "         'maker-chr9-snap-gene-5.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.8': 1,\n",
       "         'maker-chr9-snap-gene-5.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-5.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-5.9': 1,\n",
       "         'maker-chr9-snap-gene-5.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.0': 1,\n",
       "         'maker-chr9-snap-gene-6.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.16': 1,\n",
       "         'maker-chr9-snap-gene-6.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.1': 1,\n",
       "         'maker-chr9-snap-gene-6.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.5': 1,\n",
       "         'maker-chr9-snap-gene-6.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.5-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-6.8': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-6.4': 1,\n",
       "         'maker-chr9-snap-gene-6.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.6': 1,\n",
       "         'maker-chr9-snap-gene-6.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.6-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-6.6-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-6.7': 1,\n",
       "         'maker-chr9-snap-gene-6.7-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.17': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.17-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.14': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.14-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.30': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.30-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.31': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.10': 1,\n",
       "         'maker-chr9-snap-gene-6.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.9': 1,\n",
       "         'maker-chr9-snap-gene-6.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.11': 1,\n",
       "         'maker-chr9-snap-gene-6.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.35': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.12': 1,\n",
       "         'maker-chr9-snap-gene-6.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.27': 1,\n",
       "         'maker-chr9-snap-gene-6.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.27-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.12': 1,\n",
       "         'maker-chr9-snap-gene-7.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.5': 1,\n",
       "         'maker-chr9-snap-gene-7.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.9': 1,\n",
       "         'maker-chr9-snap-gene-7.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.13': 1,\n",
       "         'maker-chr9-snap-gene-7.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.4': 1,\n",
       "         'maker-chr9-snap-gene-7.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.10': 1,\n",
       "         'maker-chr9-snap-gene-7.10-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.11': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.11': 1,\n",
       "         'maker-chr9-snap-gene-7.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.32': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.8': 1,\n",
       "         'maker-chr9-snap-gene-7.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.1': 1,\n",
       "         'maker-chr9-snap-gene-7.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.0': 1,\n",
       "         'maker-chr9-snap-gene-7.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.41': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.40': 1,\n",
       "         'maker-chr9-snap-gene-7.40-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.37': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.2': 1,\n",
       "         'maker-chr9-snap-gene-7.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.7': 1,\n",
       "         'maker-chr9-snap-gene-7.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.6': 1,\n",
       "         'maker-chr9-snap-gene-7.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.17': 1,\n",
       "         'maker-chr9-snap-gene-7.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.15': 1,\n",
       "         'maker-chr9-snap-gene-7.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.15-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.14': 1,\n",
       "         'maker-chr9-snap-gene-7.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.18': 1,\n",
       "         'maker-chr9-snap-gene-7.18-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.18-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.22': 1,\n",
       "         'maker-chr9-snap-gene-7.22-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.45': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.45-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.21': 1,\n",
       "         'maker-chr9-snap-gene-7.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.21-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.21-mRNA-3': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.44': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.44-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.52': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.52-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.19': 1,\n",
       "         'maker-chr9-snap-gene-7.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.19-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.20': 1,\n",
       "         'maker-chr9-snap-gene-7.20-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.38': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.16': 1,\n",
       "         'maker-chr9-snap-gene-7.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.16-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.56': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.56-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.28': 1,\n",
       "         'maker-chr9-snap-gene-8.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.36': 1,\n",
       "         'maker-chr9-snap-gene-8.36-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.29': 1,\n",
       "         'maker-chr9-snap-gene-8.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.29-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.29': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.3': 1,\n",
       "         'maker-chr9-snap-gene-8.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.3-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-8.32': 1,\n",
       "         'maker-chr9-snap-gene-8.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.27': 1,\n",
       "         'maker-chr9-snap-gene-8.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.27-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.31': 1,\n",
       "         'maker-chr9-snap-gene-8.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.33': 1,\n",
       "         'maker-chr9-snap-gene-8.33-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.42': 1,\n",
       "         'maker-chr9-snap-gene-8.42-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.44': 1,\n",
       "         'maker-chr9-snap-gene-8.44-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.41': 1,\n",
       "         'maker-chr9-snap-gene-8.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.37': 1,\n",
       "         'maker-chr9-snap-gene-8.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.12': 1,\n",
       "         'maker-chr9-snap-gene-8.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.38': 1,\n",
       "         'maker-chr9-snap-gene-8.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.43': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-4': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.25': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.45': 1,\n",
       "         'maker-chr9-snap-gene-8.45-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.45-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-8.45-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-8.40': 1,\n",
       "         'maker-chr9-snap-gene-8.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.27': 1,\n",
       "         'maker-chr9-snap-gene-9.27-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.4': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.3': 1,\n",
       "         'maker-chr9-snap-gene-9.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.3-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-9.3-mRNA-3': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.31': 1,\n",
       "         'maker-chr9-snap-gene-9.31-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.34': 1,\n",
       "         'maker-chr9-snap-gene-9.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.38': 1,\n",
       "         'maker-chr9-snap-gene-9.38-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.37': 1,\n",
       "         'maker-chr9-snap-gene-9.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.37-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-9.32': 1,\n",
       "         'maker-chr9-snap-gene-9.32-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.25': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.25-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.18': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.18-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.15': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.15-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.20': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.33': 1,\n",
       "         'maker-chr9-snap-gene-9.33-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.16': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.35': 1,\n",
       "         'maker-chr9-snap-gene-9.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.2': 1,\n",
       "         'maker-chr9-snap-gene-10.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.1': 1,\n",
       "         'maker-chr9-snap-gene-10.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.1-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.0': 1,\n",
       "         'maker-chr9-snap-gene-10.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.17': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.17-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.20': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.4': 1,\n",
       "         'maker-chr9-snap-gene-10.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.3': 1,\n",
       "         'maker-chr9-snap-gene-10.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.5': 1,\n",
       "         'maker-chr9-snap-gene-10.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.11': 1,\n",
       "         'maker-chr9-snap-gene-10.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.11-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.10': 1,\n",
       "         'maker-chr9-snap-gene-10.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.6': 1,\n",
       "         'maker-chr9-snap-gene-10.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.6-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.31': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.31-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.8': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-10.9': 1,\n",
       "         'maker-chr9-snap-gene-10.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.7': 1,\n",
       "         'maker-chr9-snap-gene-10.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.24': 1,\n",
       "         'maker-chr9-snap-gene-11.24-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.0': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.12-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.3': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.3-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.20': 1,\n",
       "         'maker-chr9-snap-gene-11.20-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.4': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.7': 1,\n",
       "         'maker-chr9-snap-gene-11.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-11.7-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-11.25': 1,\n",
       "         'maker-chr9-snap-gene-11.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.25-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.11': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.5': 1,\n",
       "         'maker-chr9-snap-gene-11.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.5-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-12.12': 1,\n",
       "         'maker-chr9-snap-gene-12.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-12.1': 1,\n",
       "         'maker-chr9-snap-gene-12.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-12.1-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.0': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.1': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.31': 1,\n",
       "         'maker-chr9-snap-gene-13.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-13.33': 1,\n",
       "         'maker-chr9-snap-gene-13.33-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.10-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.2-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.9': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.9-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.8': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.8-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.39': 1,\n",
       "         'maker-chr9-snap-gene-13.39-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.20': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.20-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.28': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.28-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.19': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.37': 1,\n",
       "         'maker-chr9-snap-gene-13.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.37-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-13.34': 1,\n",
       "         'maker-chr9-snap-gene-13.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.34-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-13.38': 1,\n",
       "         'maker-chr9-snap-gene-13.38-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.35': 1,\n",
       "         'maker-chr9-snap-gene-13.35-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.16': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.40': 1,\n",
       "         'maker-chr9-snap-gene-13.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.2': 1,\n",
       "         'maker-chr9-snap-gene-13.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.8': 1,\n",
       "         'maker-chr9-snap-gene-14.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-14.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-14.7': 1,\n",
       "         'maker-chr9-snap-gene-14.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.0': 1,\n",
       "         'maker-chr9-snap-gene-14.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.9': 1,\n",
       "         'maker-chr9-snap-gene-14.9-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.4': 1,\n",
       "         'maker-chr9-snap-gene-14.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.6': 1,\n",
       "         'maker-chr9-snap-gene-14.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.31': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.3': 1,\n",
       "         'maker-chr9-snap-gene-14.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.25': 1,\n",
       "         'maker-chr9-snap-gene-14.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.16': 1,\n",
       "         'maker-chr9-snap-gene-14.16-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.46': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.46-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.20': 1,\n",
       "         'maker-chr9-snap-gene-14.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.21': 1,\n",
       "         'maker-chr9-snap-gene-14.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.18': 1,\n",
       "         'maker-chr9-snap-gene-14.18-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.10': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-7': 1,\n",
       "         'maker-chr9-snap-gene-14.14': 1,\n",
       "         'maker-chr9-snap-gene-14.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.17': 1,\n",
       "         'maker-chr9-snap-gene-14.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.22': 1,\n",
       "         'maker-chr9-snap-gene-14.22-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.13': 1,\n",
       "         'maker-chr9-snap-gene-14.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.13-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-14.15': 1,\n",
       "         'maker-chr9-snap-gene-14.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.23': 1,\n",
       "         'maker-chr9-snap-gene-14.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.11': 1,\n",
       "         'maker-chr9-snap-gene-14.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.12': 1,\n",
       "         'maker-chr9-snap-gene-14.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.24': 1,\n",
       "         'maker-chr9-snap-gene-14.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.19': 1,\n",
       "         'maker-chr9-snap-gene-15.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.19-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.4': 1,\n",
       "         'maker-chr9-snap-gene-15.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.4-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.3': 1,\n",
       "         'maker-chr9-snap-gene-15.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.0': 1,\n",
       "         'maker-chr9-snap-gene-15.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.1': 1,\n",
       "         'maker-chr9-snap-gene-15.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.1-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-15.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-15.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.8': 1,\n",
       "         'maker-chr9-snap-gene-15.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.9': 1,\n",
       "         'maker-chr9-snap-gene-15.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.12': 1,\n",
       "         'maker-chr9-snap-gene-15.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.12-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.17': 1,\n",
       "         'maker-chr9-snap-gene-15.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.16': 1,\n",
       "         'maker-chr9-snap-gene-15.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.6': 1,\n",
       "         'maker-chr9-snap-gene-15.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.13': 1,\n",
       "         'maker-chr9-snap-gene-15.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.5': 1,\n",
       "         'maker-chr9-snap-gene-15.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.7': 1,\n",
       "         'maker-chr9-snap-gene-15.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.10': 1,\n",
       "         'maker-chr9-snap-gene-15.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.10-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.10-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-15.15': 1,\n",
       "         'maker-chr9-snap-gene-15.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.14': 1,\n",
       "         'maker-chr9-snap-gene-15.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.1': 1,\n",
       "         'maker-chr9-snap-gene-16.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.6': 1,\n",
       "         'maker-chr9-snap-gene-16.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.3': 1,\n",
       "         'maker-chr9-snap-gene-16.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.3-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.5': 1,\n",
       "         'maker-chr9-snap-gene-16.5-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.16': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.0': 1,\n",
       "         'maker-chr9-snap-gene-16.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.4': 1,\n",
       "         'maker-chr9-snap-gene-16.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.2': 1,\n",
       "         'maker-chr9-snap-gene-16.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.2-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.2-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-16.8': 1,\n",
       "         'maker-chr9-snap-gene-16.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.9': 1,\n",
       "         'maker-chr9-snap-gene-16.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.9-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-16.7': 1,\n",
       "         'maker-chr9-snap-gene-16.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.11': 1,\n",
       "         'maker-chr9-snap-gene-16.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.10': 1,\n",
       "         'maker-chr9-snap-gene-16.10-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.34': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.14': 1,\n",
       "         'maker-chr9-snap-gene-16.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.14-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.14-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-16.13': 1,\n",
       "         'maker-chr9-snap-gene-16.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.12': 1,\n",
       "         'maker-chr9-snap-gene-16.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.4': 1,\n",
       "         'maker-chr9-snap-gene-17.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.1': 1,\n",
       "         'maker-chr9-snap-gene-17.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.1-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.1-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-17.3': 1,\n",
       "         'maker-chr9-snap-gene-17.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.8': 1,\n",
       "         'maker-chr9-snap-gene-17.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.6': 1,\n",
       "         'maker-chr9-snap-gene-17.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.0': 1,\n",
       "         'maker-chr9-snap-gene-17.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.0-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.0-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-17.10': 1,\n",
       "         'maker-chr9-snap-gene-17.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.9': 1,\n",
       "         'maker-chr9-snap-gene-17.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.2': 1,\n",
       "         'maker-chr9-snap-gene-17.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.7': 1,\n",
       "         'maker-chr9-snap-gene-17.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.5': 1,\n",
       "         'maker-chr9-snap-gene-17.5-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-17.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-17.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.31': 1,\n",
       "         'maker-chr9-snap-gene-17.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.13': 1,\n",
       "         'maker-chr9-snap-gene-17.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.20': 1,\n",
       "         'maker-chr9-snap-gene-17.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.12': 1,\n",
       "         'maker-chr9-snap-gene-17.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.11': 1,\n",
       "         'maker-chr9-snap-gene-17.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.15': 1,\n",
       "         'maker-chr9-snap-gene-17.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.14': 1,\n",
       "         'maker-chr9-snap-gene-17.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.19': 1,\n",
       "         'maker-chr9-snap-gene-17.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.19-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.19-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-17.17': 1,\n",
       "         'maker-chr9-snap-gene-17.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.44': 1,\n",
       "         'maker-chr9-snap-gene-17.44-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.5': 1,\n",
       "         'maker-chr9-snap-gene-18.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.1': 1,\n",
       "         'maker-chr9-snap-gene-18.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.3': 1,\n",
       "         'maker-chr9-snap-gene-18.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.2': 1,\n",
       "         'maker-chr9-snap-gene-18.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.4': 1,\n",
       "         'maker-chr9-snap-gene-18.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.4-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.28': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.28-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.14': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.14-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.9': 1,\n",
       "         'maker-chr9-snap-gene-18.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.6': 1,\n",
       "         'maker-chr9-snap-gene-18.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.6-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-18.6-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-18.11': 1,\n",
       "         'maker-chr9-snap-gene-18.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.21': 1,\n",
       "         'maker-chr9-snap-gene-18.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.8': 1,\n",
       "         'maker-chr9-snap-gene-18.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-18.18': 1,\n",
       "         'maker-chr9-snap-gene-18.18-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.18-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-19.30': 1,\n",
       "         'maker-chr9-snap-gene-19.30-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.24': 1,\n",
       "         'maker-chr9-snap-gene-19.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.26': 1,\n",
       "         'maker-chr9-snap-gene-19.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.23': 1,\n",
       "         'maker-chr9-snap-gene-19.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.29': 1,\n",
       "         'maker-chr9-snap-gene-19.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.27': 1,\n",
       "         'maker-chr9-snap-gene-19.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.31': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-4': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.40': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-19.46': 1,\n",
       "         'maker-chr9-snap-gene-19.46-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.35': 1,\n",
       "         'maker-chr9-snap-gene-19.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.33': 1,\n",
       "         'maker-chr9-snap-gene-19.33-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.37': 1,\n",
       "         'maker-chr9-snap-gene-19.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.36': 1,\n",
       "         'maker-chr9-snap-gene-19.36-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.19': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.38': 1,\n",
       "         'maker-chr9-snap-gene-19.38-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.36': 1,\n",
       "         'maker-chr9-snap-gene-20.36-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.38': 1,\n",
       "         'maker-chr9-snap-gene-20.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.38-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.32': 1,\n",
       "         'maker-chr9-snap-gene-20.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.34': 1,\n",
       "         'maker-chr9-snap-gene-20.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.34-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.33': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.33-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.50': 1,\n",
       "         'maker-chr9-snap-gene-20.50-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.43': 1,\n",
       "         'maker-chr9-snap-gene-20.43-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.48': 1,\n",
       "         'maker-chr9-snap-gene-20.48-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.45': 1,\n",
       "         'maker-chr9-snap-gene-20.45-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.45-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.47': 1,\n",
       "         'maker-chr9-snap-gene-20.47-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.47-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.44': 1,\n",
       "         'maker-chr9-snap-gene-20.44-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.40': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-20.42': 1,\n",
       "         'maker-chr9-snap-gene-20.42-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.42-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.15': 1,\n",
       "         'maker-chr9-snap-gene-20.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.15-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.15-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-20.41': 1,\n",
       "         'maker-chr9-snap-gene-20.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.20': 1,\n",
       "         'maker-chr9-snap-gene-21.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.9': 1,\n",
       "         'maker-chr9-snap-gene-21.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.0': 1,\n",
       "         'maker-chr9-snap-gene-21.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.12': 1,\n",
       "         'maker-chr9-snap-gene-21.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.27': 1,\n",
       "         'maker-chr9-snap-gene-21.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.27-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.24': 1,\n",
       "         'maker-chr9-snap-gene-21.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.24-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.26': 1,\n",
       "         'maker-chr9-snap-gene-21.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.26-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.26-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-21.5': 1,\n",
       "         'maker-chr9-snap-gene-21.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.2': 1,\n",
       "         'maker-chr9-snap-gene-21.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.4': 1,\n",
       "         'maker-chr9-snap-gene-21.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.4-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.4-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-21.1': 1,\n",
       "         'maker-chr9-snap-gene-21.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.35': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.23': 1,\n",
       "         'maker-chr9-snap-gene-21.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.23-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.19': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.14': 1,\n",
       "         'maker-chr9-snap-gene-21.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.14-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.16': 1,\n",
       "         'maker-chr9-snap-gene-21.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.13': 1,\n",
       "         'maker-chr9-snap-gene-21.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.13-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.47': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.47-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.46': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.46-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.15': 1,\n",
       "         'maker-chr9-snap-gene-21.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.34': 1,\n",
       "         'maker-chr9-snap-gene-22.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.31': 1,\n",
       "         'maker-chr9-snap-gene-22.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.31-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.32': 1,\n",
       "         'maker-chr9-snap-gene-22.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.32-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.32-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.35': 1,\n",
       "         'maker-chr9-snap-gene-22.35-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-22.3': 1,\n",
       "         'snap_masked-chr9-processed-gene-22.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.33': 1,\n",
       "         'maker-chr9-snap-gene-22.33-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.33-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.41': 1,\n",
       "         'maker-chr9-snap-gene-22.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.39': 1,\n",
       "         'maker-chr9-snap-gene-22.39-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.43': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-22.40': 1,\n",
       "         'maker-chr9-snap-gene-22.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.40-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.40-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.37': 1,\n",
       "         'maker-chr9-snap-gene-22.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.37-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.38': 1,\n",
       "         'maker-chr9-snap-gene-22.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.38-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.38-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-23.4': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-23.0': 1,\n",
       "         'maker-chr9-snap-gene-23.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.1': 1,\n",
       "         'maker-chr9-snap-gene-23.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.3': 1,\n",
       "         'maker-chr9-snap-gene-23.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.2': 1,\n",
       "         'maker-chr9-snap-gene-23.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.2-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-23.5': 1,\n",
       "         'maker-chr9-snap-gene-23.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.7': 1,\n",
       "         'maker-chr9-snap-gene-23.7-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.22': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.22-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.8': 1,\n",
       "         'maker-chr9-snap-gene-23.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.8-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.21': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.27': 1,\n",
       "         'maker-chr9-snap-gene-24.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.27-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.31': 1,\n",
       "         'maker-chr9-snap-gene-24.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.32': 1,\n",
       "         'maker-chr9-snap-gene-24.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.28': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-24.26': 1,\n",
       "         'maker-chr9-snap-gene-24.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.26-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.29': 1,\n",
       "         'maker-chr9-snap-gene-24.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.29-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.29-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-24.25': 1,\n",
       "         'maker-chr9-snap-gene-24.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.25-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.37': 1,\n",
       "         'maker-chr9-snap-gene-24.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.34': 1,\n",
       "         'maker-chr9-snap-gene-24.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.34-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.36': 1,\n",
       "         'maker-chr9-snap-gene-24.36-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.22': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.22-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.24': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.35': 1,\n",
       "         'maker-chr9-snap-gene-24.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.35-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.25': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-25.28': 1,\n",
       "         'maker-chr9-snap-gene-25.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-25.27': 1,\n",
       "         'maker-chr9-snap-gene-25.27-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-25.21': 1,\n",
       "         'maker-chr9-snap-gene-25.21-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.3': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.3-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.4': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.4-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.7-mRNA-1': 1,\n",
       "         ...})"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(df_gff['Alias'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "ea710868",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Seqid</th>\n",
       "      <th>Source</th>\n",
       "      <th>Type</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Phase</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>_AED</th>\n",
       "      <th>_QI</th>\n",
       "      <th>_eAED</th>\n",
       "      <th>score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>gene</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0...</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0|0|0|0.5|1|1|2|0|472</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>11686</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032749-mRNA-1:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615982</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>281718</td>\n",
       "      <td>281844</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615983</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>280181</td>\n",
       "      <td>280249</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615984</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>279893</td>\n",
       "      <td>280085</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615985</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>278640</td>\n",
       "      <td>279072</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615986</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>278427</td>\n",
       "      <td>278639</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032211-mRNA-2:three_prime_utr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>615987 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Seqid Source             Type   Start     End Score Strand  \\\n",
       "0       egu.contig.1973  maker             gene   10123   11712     .      +   \n",
       "1       egu.contig.1973  maker             mRNA   10123   11712     .      +   \n",
       "2       egu.contig.1973  maker             exon   10123   11511     .      +   \n",
       "3       egu.contig.1973  maker             exon   11686   11712     .      +   \n",
       "4       egu.contig.1973  maker              CDS   10123   11511     .      +   \n",
       "...                 ...    ...              ...     ...     ...   ...    ...   \n",
       "615982  egu.contig.1149  maker              CDS  281718  281844     .      -   \n",
       "615983  egu.contig.1149  maker              CDS  280181  280249     .      -   \n",
       "615984  egu.contig.1149  maker              CDS  279893  280085     .      -   \n",
       "615985  egu.contig.1149  maker              CDS  278640  279072     .      -   \n",
       "615986  egu.contig.1149  maker  three_prime_UTR  278427  278639     .      -   \n",
       "\n",
       "       Phase                                ID              Name  \\\n",
       "0          .                         Egu032749         Egu032749   \n",
       "1          .                  Egu032749-mRNA-1  Egu032749-mRNA-1   \n",
       "2          .                Egu032749-mRNA-1:1               NaN   \n",
       "3          .                Egu032749-mRNA-1:2               NaN   \n",
       "4          0              Egu032749-mRNA-1:cds               NaN   \n",
       "...      ...                               ...               ...   \n",
       "615982     0              Egu032211-mRNA-2:cds               NaN   \n",
       "615983     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615984     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615985     1              Egu032211-mRNA-2:cds               NaN   \n",
       "615986     .  Egu032211-mRNA-2:three_prime_utr               NaN   \n",
       "\n",
       "                                                    Alias     \\\n",
       "0          snap_masked-egu.contig.1973-processed-gene-0.0      \n",
       "1       snap_masked-egu.contig.1973-processed-gene-0.0...      \n",
       "2                                                     NaN      \n",
       "3                                                     NaN      \n",
       "4                                                     NaN      \n",
       "...                                                   ... ..   \n",
       "615982                                                NaN      \n",
       "615983                                                NaN      \n",
       "615984                                                NaN      \n",
       "615985                                                NaN      \n",
       "615986                                                NaN      \n",
       "\n",
       "                  Parent  _AED                    _QI _eAED score  \n",
       "0                    NaN   NaN                    NaN   NaN   NaN  \n",
       "1              Egu032749  0.06  0|0|0|0.5|1|1|2|0|472  0.04   NaN  \n",
       "2       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "3       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "4       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "...                  ...   ...                    ...   ...   ...  \n",
       "615982  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615983  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615984  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615985  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615986  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "\n",
       "[615987 rows x 17 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9855549",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "i = 0\n",
    "protein_file = '/bioinfo/palm/ref/dura/dura_proteins.fasta'\n",
    "with open('dura_proteins_{}_at{}.aa'.format(gene_name, level),'w') as fh:\n",
    "    for record in SeqIO.parse(prselected_rowsotein_file,'fasta'):\n",
    "        if record.id in ['query'].values:\n",
    "            cluster = selected_rows.loc[selected_rows['query'] == record.id, 'ODB_OG']\n",
    "            tag = selected_rows.loc[selected_rows['query'] == record.id, 'Description']\n",
    "            if len(tag) == 1:\n",
    "                record.description = ':'.join([cluster.iloc[0], tag.iloc[0]])\n",
    "            else:\n",
    "                print(record.id + 'appeared in more than one cluster?')\n",
    "\n",
    "            SeqIO.write(record, fh, 'fasta')\n",
    "            i += 1\n",
    "print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d7436432",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>ODB_OG</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>GOs_mf</th>\n",
       "      <th>GOs_bp</th>\n",
       "      <th>EC</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>Interpro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Egu020222-mRNA-1</td>\n",
       "      <td>687at4447</td>\n",
       "      <td>2.140000e-26</td>\n",
       "      <td>117.8900</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>6.5.1.1,4.1.1.15</td>\n",
       "      <td>ko00250,ko00410,ko00430,ko00650,ko03030,ko0341...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>Egu026424-mRNA-1</td>\n",
       "      <td>2328at4447</td>\n",
       "      <td>4.810000e-66</td>\n",
       "      <td>174.1300</td>\n",
       "      <td>-</td>\n",
       "      <td>Auxin-responsive protein SAUR36</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>663</th>\n",
       "      <td>Egu020226-mRNA-1</td>\n",
       "      <td>5943at4447</td>\n",
       "      <td>5.960000e-43</td>\n",
       "      <td>180.6500</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>Egu020227-mRNA-1</td>\n",
       "      <td>5943at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>148.8970</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3706</th>\n",
       "      <td>Egu011092-mRNA-1</td>\n",
       "      <td>33316at4447</td>\n",
       "      <td>4.290000e-28</td>\n",
       "      <td>86.7100</td>\n",
       "      <td>-</td>\n",
       "      <td>indole-3-acetic acid-induced protein ARG7</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00010,ko01200</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3707</th>\n",
       "      <td>Egu030210-mRNA-1</td>\n",
       "      <td>33316at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>108.0540</td>\n",
       "      <td>-</td>\n",
       "      <td>indole-3-acetic acid-induced protein ARG7</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00010,ko01200</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5388</th>\n",
       "      <td>Egu009584-mRNA-1</td>\n",
       "      <td>47410at4447</td>\n",
       "      <td>9.090000e-29</td>\n",
       "      <td>110.7600</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR32</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>2.7.2.3</td>\n",
       "      <td>ko00010,ko00230,ko00710,ko00910,ko00941,ko0094...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9270</th>\n",
       "      <td>Egu019713-mRNA-1</td>\n",
       "      <td>75523at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>78.6238</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR41</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9271</th>\n",
       "      <td>Egu031500-mRNA-1</td>\n",
       "      <td>75523at4447</td>\n",
       "      <td>3.150000e-23</td>\n",
       "      <td>73.9000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR41</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9614</th>\n",
       "      <td>Egu020383-mRNA-2</td>\n",
       "      <td>77872at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>110.4580</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00270,ko04145,sita01100,sita01110</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9615</th>\n",
       "      <td>Egu020383-mRNA-3</td>\n",
       "      <td>77872at4447</td>\n",
       "      <td>1.390000e-40</td>\n",
       "      <td>127.1300</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00270,ko04145,sita01100,sita01110</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10304</th>\n",
       "      <td>Egu003951-mRNA-1</td>\n",
       "      <td>83452at4447</td>\n",
       "      <td>1.080000e-41</td>\n",
       "      <td>131.3700</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR77</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko03013</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11924</th>\n",
       "      <td>Egu011765-mRNA-1</td>\n",
       "      <td>111397at4447</td>\n",
       "      <td>5.890000e-71</td>\n",
       "      <td>253.4000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR50</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00531,ko03430,ko04075,sita01100</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11925</th>\n",
       "      <td>Egu028450-mRNA-1</td>\n",
       "      <td>111397at4447</td>\n",
       "      <td>5.670000e-69</td>\n",
       "      <td>247.0900</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR50</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00531,ko03430,ko04075,sita01100</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12698</th>\n",
       "      <td>Egu003994-mRNA-1</td>\n",
       "      <td>113001at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>95.8472</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00561,ko00770,ko04075,ko04145</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12699</th>\n",
       "      <td>Egu015371-mRNA-1</td>\n",
       "      <td>113001at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>120.8720</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00561,ko00770,ko04075,ko04145</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12700</th>\n",
       "      <td>Egu020228-mRNA-1</td>\n",
       "      <td>113001at4447</td>\n",
       "      <td>3.740000e-53</td>\n",
       "      <td>152.0300</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00561,ko00770,ko04075,ko04145</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14162</th>\n",
       "      <td>Egu010459-mRNA-1</td>\n",
       "      <td>115938at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>102.0430</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR21-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>4.2.2.2</td>\n",
       "      <td>ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14163</th>\n",
       "      <td>Egu027752-mRNA-1</td>\n",
       "      <td>115938at4447</td>\n",
       "      <td>7.800000e-30</td>\n",
       "      <td>61.0200</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR21-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>4.2.2.2</td>\n",
       "      <td>ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14164</th>\n",
       "      <td>Egu030300-mRNA-1</td>\n",
       "      <td>115938at4447</td>\n",
       "      <td>3.980000e-70</td>\n",
       "      <td>142.4000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR21-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>4.2.2.2</td>\n",
       "      <td>ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17289</th>\n",
       "      <td>Egu009585-mRNA-1</td>\n",
       "      <td>121712at4447</td>\n",
       "      <td>1.720000e-39</td>\n",
       "      <td>128.5100</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR32</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>ko00511,ko03060</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20054</th>\n",
       "      <td>Egu031852-mRNA-1</td>\n",
       "      <td>126583at4447</td>\n",
       "      <td>1.190000e-66</td>\n",
       "      <td>187.9700</td>\n",
       "      <td>-</td>\n",
       "      <td>SAUR37-auxin-responsive SAUR family member</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>1.-.-.-</td>\n",
       "      <td>ko03010,ko00380,ko04141,ko04712</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22845</th>\n",
       "      <td>Egu001133-mRNA-1</td>\n",
       "      <td>131384at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>132.8280</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>bdi01100,bdi01110,ko00940,ko03015</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22846</th>\n",
       "      <td>Egu017903-mRNA-1</td>\n",
       "      <td>131384at4447</td>\n",
       "      <td>1.310000e-70</td>\n",
       "      <td>179.3200</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>bdi01100,bdi01110,ko00940,ko03015</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24776</th>\n",
       "      <td>Egu007739-mRNA-1</td>\n",
       "      <td>134723at4447</td>\n",
       "      <td>8.810000e-88</td>\n",
       "      <td>193.4100</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR50-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00904,ko03040</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26035</th>\n",
       "      <td>Egu011864-mRNA-1</td>\n",
       "      <td>136815at4447</td>\n",
       "      <td>2.920000e-71</td>\n",
       "      <td>216.1800</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR36</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26036</th>\n",
       "      <td>Egu028534-mRNA-1</td>\n",
       "      <td>136815at4447</td>\n",
       "      <td>9.670000e-37</td>\n",
       "      <td>164.8000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR36</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  query        ODB_OG        evalue     score COG_category  \\\n",
       "75     Egu020222-mRNA-1     687at4447  2.140000e-26  117.8900            -   \n",
       "262    Egu026424-mRNA-1    2328at4447  4.810000e-66  174.1300            -   \n",
       "663    Egu020226-mRNA-1    5943at4447  5.960000e-43  180.6500            -   \n",
       "664    Egu020227-mRNA-1    5943at4447  0.000000e+00  148.8970            -   \n",
       "3706   Egu011092-mRNA-1   33316at4447  4.290000e-28   86.7100            -   \n",
       "3707   Egu030210-mRNA-1   33316at4447  0.000000e+00  108.0540            -   \n",
       "5388   Egu009584-mRNA-1   47410at4447  9.090000e-29  110.7600            -   \n",
       "9270   Egu019713-mRNA-1   75523at4447  0.000000e+00   78.6238            -   \n",
       "9271   Egu031500-mRNA-1   75523at4447  3.150000e-23   73.9000            -   \n",
       "9614   Egu020383-mRNA-2   77872at4447  0.000000e+00  110.4580            -   \n",
       "9615   Egu020383-mRNA-3   77872at4447  1.390000e-40  127.1300            -   \n",
       "10304  Egu003951-mRNA-1   83452at4447  1.080000e-41  131.3700            -   \n",
       "11924  Egu011765-mRNA-1  111397at4447  5.890000e-71  253.4000            -   \n",
       "11925  Egu028450-mRNA-1  111397at4447  5.670000e-69  247.0900            -   \n",
       "12698  Egu003994-mRNA-1  113001at4447  0.000000e+00   95.8472            -   \n",
       "12699  Egu015371-mRNA-1  113001at4447  0.000000e+00  120.8720            -   \n",
       "12700  Egu020228-mRNA-1  113001at4447  3.740000e-53  152.0300            -   \n",
       "14162  Egu010459-mRNA-1  115938at4447  0.000000e+00  102.0430            -   \n",
       "14163  Egu027752-mRNA-1  115938at4447  7.800000e-30   61.0200            -   \n",
       "14164  Egu030300-mRNA-1  115938at4447  3.980000e-70  142.4000            -   \n",
       "17289  Egu009585-mRNA-1  121712at4447  1.720000e-39  128.5100            -   \n",
       "20054  Egu031852-mRNA-1  126583at4447  1.190000e-66  187.9700            -   \n",
       "22845  Egu001133-mRNA-1  131384at4447  0.000000e+00  132.8280            -   \n",
       "22846  Egu017903-mRNA-1  131384at4447  1.310000e-70  179.3200            -   \n",
       "24776  Egu007739-mRNA-1  134723at4447  8.810000e-88  193.4100            -   \n",
       "26035  Egu011864-mRNA-1  136815at4447  2.920000e-71  216.1800            -   \n",
       "26036  Egu028534-mRNA-1  136815at4447  9.670000e-37  164.8000            -   \n",
       "\n",
       "                                      Description GOs_mf      GOs_bp  \\\n",
       "75                auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "262               Auxin-responsive protein SAUR36      -  GO:0009733   \n",
       "663               auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "664               auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "3706    indole-3-acetic acid-induced protein ARG7      -  GO:0009733   \n",
       "3707    indole-3-acetic acid-induced protein ARG7      -  GO:0009733   \n",
       "5388              auxin-responsive protein SAUR32      -  GO:0009733   \n",
       "9270              auxin-responsive protein SAUR41      -  GO:0009733   \n",
       "9271              auxin-responsive protein SAUR41      -  GO:0009733   \n",
       "9614                     auxin-induced protein 6B      -  GO:0009733   \n",
       "9615                     auxin-induced protein 6B      -  GO:0009733   \n",
       "10304             auxin-responsive protein SAUR77      -  GO:0009733   \n",
       "11924             auxin-responsive protein SAUR50      -  GO:0009733   \n",
       "11925             auxin-responsive protein SAUR50      -  GO:0009733   \n",
       "12698             auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "12699             auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "12700             auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "14162        auxin-responsive protein SAUR21-like      -  GO:0009733   \n",
       "14163        auxin-responsive protein SAUR21-like      -  GO:0009733   \n",
       "14164        auxin-responsive protein SAUR21-like      -  GO:0009733   \n",
       "17289             auxin-responsive protein SAUR32      -  GO:0009733   \n",
       "20054  SAUR37-auxin-responsive SAUR family member      -  GO:0009733   \n",
       "22845                    auxin-induced protein 6B      -  GO:0009733   \n",
       "22846                    auxin-induced protein 6B      -  GO:0009733   \n",
       "24776        auxin-responsive protein SAUR50-like      -  GO:0009733   \n",
       "26035             auxin-responsive protein SAUR36      -  GO:0009733   \n",
       "26036             auxin-responsive protein SAUR36      -  GO:0009733   \n",
       "\n",
       "                     EC                                            KEGG_ko  \\\n",
       "75     6.5.1.1,4.1.1.15  ko00250,ko00410,ko00430,ko00650,ko03030,ko0341...   \n",
       "262                   -                                                  -   \n",
       "663                   -                                                  -   \n",
       "664                   -                                                  -   \n",
       "3706                  -                                    ko00010,ko01200   \n",
       "3707                  -                                    ko00010,ko01200   \n",
       "5388            2.7.2.3  ko00010,ko00230,ko00710,ko00910,ko00941,ko0094...   \n",
       "9270                  -                                                  -   \n",
       "9271                  -                                                  -   \n",
       "9614                  -        ko00190,ko00270,ko04145,sita01100,sita01110   \n",
       "9615                  -        ko00190,ko00270,ko04145,sita01100,sita01110   \n",
       "10304                 -                                            ko03013   \n",
       "11924                 -                  ko00531,ko03430,ko04075,sita01100   \n",
       "11925                 -                  ko00531,ko03430,ko04075,sita01100   \n",
       "12698                 -            ko00190,ko00561,ko00770,ko04075,ko04145   \n",
       "12699                 -            ko00190,ko00561,ko00770,ko04075,ko04145   \n",
       "12700                 -            ko00190,ko00561,ko00770,ko04075,ko04145   \n",
       "14162           4.2.2.2  ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...   \n",
       "14163           4.2.2.2  ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...   \n",
       "14164           4.2.2.2  ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...   \n",
       "17289           3.2.1.-                                    ko00511,ko03060   \n",
       "20054           1.-.-.-                    ko03010,ko00380,ko04141,ko04712   \n",
       "22845                 -                  bdi01100,bdi01110,ko00940,ko03015   \n",
       "22846                 -                  bdi01100,bdi01110,ko00940,ko03015   \n",
       "24776                 -                                    ko00904,ko03040   \n",
       "26035                 -                                                  -   \n",
       "26036                 -                                                  -   \n",
       "\n",
       "        Interpro  \n",
       "75     IPR003676  \n",
       "262    IPR003676  \n",
       "663    IPR003676  \n",
       "664    IPR003676  \n",
       "3706   IPR003676  \n",
       "3707   IPR003676  \n",
       "5388   IPR003676  \n",
       "9270   IPR003676  \n",
       "9271   IPR003676  \n",
       "9614   IPR003676  \n",
       "9615   IPR003676  \n",
       "10304  IPR003676  \n",
       "11924  IPR003676  \n",
       "11925  IPR003676  \n",
       "12698  IPR003676  \n",
       "12699  IPR003676  \n",
       "12700  IPR003676  \n",
       "14162  IPR003676  \n",
       "14163  IPR003676  \n",
       "14164  IPR003676  \n",
       "17289  IPR003676  \n",
       "20054  IPR003676  \n",
       "22845  IPR003676  \n",
       "22846  IPR003676  \n",
       "24776  IPR003676  \n",
       "26035  IPR003676  \n",
       "26036  IPR003676  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Extract values from column B and C from the filtered rows\n",
    "result_values = filtered_rows[['Seqid', 'Start','End','ID']]\n",
    "\n",
    "# Convert columns A and B to numeric\n",
    "result_values['Start'] = pd.to_numeric(result_values['Start'])\n",
    "result_values['End'] = pd.to_numeric(result_values['End'])\n",
    "\n",
    "# Perform the operations and apply the conditions\n",
    "result_values['Start'] = result_values['Start'] - flanking\n",
    "result_values['Start'] = result_values['Start'].apply(lambda x: max(0, x))  # Set negative values to 0\n",
    "\n",
    "result_values['End'] = result_values['End'] + flanking\n",
    "bedfile = fileA.split('_blastp.tbl')[0] + '_' + str(flanking) + '.bed'\n",
    "result_values.to_csv(bedfile, sep='\\t', index=False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (bioinfo)",
   "language": "python",
   "name": "bioinfo"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
