{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download all protein sequences in orthoDB v11 for one/multiple species\n",
    "\n",
    "1. odb11v0_OG2genes.tab.gz were downloaded from https://data.orthodb.org/download/\n",
    "wget https://data.orthodb.org/download/odb11v0_OG2genes.tab.gz\n",
    "2. get the OG ids that are related to oil palm (taxid 51953)\n",
    "zcat odb11v0_OG2genes.tab.gz | grep 51953_0 | cut -f 1 | sort | uniq > OG_51953_0.tab\n",
    "3. Download the oil palm protein sequences from the clusters that contains oil palm sequences\n",
    "nohup sh download.sh > logfile 2>&1 < /dev/null &\n",
    "\n",
    "download.sh:\n",
    "#!/bin/bash\n",
    "output_file=\"combined_data.tsv\"\n",
    "\n",
    "while IFS= read -r line; do\n",
    "    curl \"https://data.orthodb.org/current/fasta?id=$line&species=51953\" -L -k >> \"$output_file\"\n",
    "done < \"OG_51953_0.tab\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "curl 'https://data.orthodb.org/current/blast?seq=MGQMGGPDGDGPHHQYHYQALLAAVQNPSQGLHVPLHAGAGAPAAGPGPRPGADADASSTHNANATPHSQPPRAFTDWSASNSAFAAQPAPATTNTPFHYNLSQSYALWTHYMLNKNVSYSTYSTPHEPLRHTHIPDKYSGCAFSLGFDSFTTMSLGPNICANMTPMERSISAKEPENSEDLPTVVRSSDEMDTRNSGDVRRDTVDTLPESKQSHESCASVSNKFDSGEYQVILRKELTKSDVANSGRIVLPKKDAEAGLPPLVQGDPLILQMDDMVLPIIWKFKYRFWPNNKSRMYILEAAGEFVKTHGPSGRGYAHYLQKLRTWQIYYPWGEVHSADNP' -L -o blast.dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!curl 'https://data.orthodb.org/current/blast?seq=MGQMGGPDGDGPHHQYHYQALLAAVQNPSQGLHVPLHAGAGAPAAGPGPRPGADADASSTHNANATPHSQPPRAFTDWSASNSAFAAQPAPATTNTPFHYNLSQSYALWTHYMLNKNVSYSTYSTPHEPLRHTHIPDKYSGCAFSLGFDSFTTMSLGPNICANMTPMERSISAKEPENSEDLPTVVRSSDEMDTRNSGDVRRDTVDTLPESKQSHESCASVSNKFDSGEYQVILRKELTKSDVANSGRIVLPKKDAEAGLPPLVQGDPLILQMDDMVLPIIWKFKYRFWPNNKSRMYILEAAGEFVKTHGPSGRGYAHYLQKLRTWQIYYPWGEVHSADNP' -L -o blast.dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## parse the search.data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "\n",
    "file_path = \"/bioinfo/palm/ref/orthoDBv11/search.dat\"\n",
    "\n",
    "with open(file_path, \"r\") as json_file:\n",
    "    data_dict = json.load(json_file)\n",
    "\n",
    "ortholog_path = \"/bioinfo/palm/ref/orthoDBv11/orthologs.dat\"\n",
    "\n",
    "with open(ortholog_path, \"r\") as json_file:\n",
    "    ortholog_dict = json.load(json_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# parse the downloaded protein sequence to remove the redundent ones"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import json\n",
    "\n",
    "sequence_dic = {}\n",
    "tag_dic = {}\n",
    "level_dic = {}\n",
    "with open('/bioinfo/palm/ref/orthoDBv11/oil_palm_protein.fa','w') as fh:\n",
    "    for record in SeqIO.parse('/bioinfo/palm/ref/orthoDBv11/combined_data.tsv','fasta'):\n",
    "        if record.id in sequence_dic:\n",
    "            # gene already wrote out, check whether the sequence is different\n",
    "            if record.seq != sequence_dic[record.id]:\n",
    "                # need to write to fh if the sequence is different\n",
    "                gene, description = record.description.split(' ', 1)\n",
    "                description = description.strip()\n",
    "                description_dict = json.loads(description)\n",
    "                cluster = description_dict['pub_og_id']\n",
    "                record.id = record.id + '_' + cluster\n",
    "                tag_dic[gene] = cluster\n",
    "                SeqIO.write(record, fh, 'fasta')\n",
    "                print(gene) #nothing was printed, which means the protein sequences always stay the same no matter the node\n",
    "        else:\n",
    "            SeqIO.write(record, fh, 'fasta')\n",
    "        sequence_dic[record.id] = record.seq\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Now a simpler code with the understanding that the protein sequences always stay the same for the same gene no matter the node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import json\n",
    "\n",
    "sequence_dic = {}\n",
    "level_dic = {}\n",
    "with open('/bioinfo/palm/ref/orthoDBv11/oil_palm_protein.fa','w') as fh:\n",
    "    for record in SeqIO.parse('/bioinfo/palm/ref/orthoDBv11/combined_data.tsv','fasta'):\n",
    "        gene, description = record.description.split(' ', 1)\n",
    "        description = description.strip()\n",
    "        description_dict = json.loads(description)\n",
    "        cluster = description_dict['pub_og_id']\n",
    "        level = cluster.split('at')[1]\n",
    "        if level in level_dic:\n",
    "            level_dic[level].append(gene)\n",
    "        else:\n",
    "            level_dic[level] = [gene]\n",
    "        if record.id not in sequence_dic:\n",
    "            SeqIO.write(record, fh, 'fasta')\n",
    "            sequence_dic[record.id] = record.seq\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lengths_dict = {}\n",
    "\n",
    "# Iterate through the dictionary values and calculate lengths\n",
    "for key, value in level_dic.items():\n",
    "    unique_items = set(value)  # Create a set to store unique items\n",
    "    lengths_dict[key] = len(unique_items)\n",
    "\n",
    "print(lengths_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This means one gene is classified into only one cluster at most at one level. \n",
    "\n",
    "hen we used orthomapper to map all dura proteins to orthodb v11. Results are in /bioinfo/tools/orthologer/Results\n",
    "\n",
    "Now let's get the protein sequences that wasn't placed to any orthodb clusters (14632).\n",
    "\n",
    "Note that if a gene was not placed in orthoDBv11 by othomapper, it won't be found in the reciprocal mapping between dura and EG5 inOrthoDBv11 either. \n",
    "\n",
    "Now what do we do? Map to EG5 just to see what they could have done?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "file_path = 'dura.og.annotations'\n",
    "df = pd.read_csv(file_path, sep='\\t', header=0)\n",
    "df.rename(columns={'#query': 'query'}, inplace=True)\n",
    "df_eg5 = pd.read_csv('dura_EG5_orthoDB11.tbl', sep='\\t', header=None)\n",
    "i = 0\n",
    "with open('dura_proteins_not_in_orthodbv11.fasta','w') as fh:\n",
    "    for record in SeqIO.parse('../dura/dura_proteins.fasta','fasta'):\n",
    "        if record.id not in df['query'].values:\n",
    "            if record.id in df_eg5[0]:\n",
    "                print(record.id)\n",
    "            SeqIO.write(record, fh, 'fasta')\n",
    "            i += 1\n",
    "print(i)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fix the ID in EG5.1 cds and protein.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "in_file = '/home/huan/build/orthologer/EG5.1_Genes.V3.na.fa'\n",
    "out_file = '/home/huan/build/orthologer/EG5.1_Genes.V3.newid.na.fa'\n",
    "with open(out_file,'w') as fh:\n",
    "    for record in SeqIO.parse(in_file,'fasta'):\n",
    "        record.id = record.description.split()[1].split('=')[1]\n",
    "        SeqIO.write(record, fh, 'fasta')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge the orthomapper results from all three levels, with priority to lower levels."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique og:\n",
      "15904\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "name = 'EG5.1_v12'\n",
    "\n",
    "file1 = '~/build/orthologer/Results/{}_4447.og.annotations'.format(name)\n",
    "df1 = pd.read_csv(file1, sep='\\t', header=0)\n",
    "df1.rename(columns={'#query': 'query'}, inplace=True)\n",
    "\n",
    "file2 = '~/build/orthologer/Results/{}_3193.og.annotations'.format(name)\n",
    "df2 = pd.read_csv(file2, sep='\\t', header=0)\n",
    "df2.rename(columns={'#query': 'query'}, inplace=True)\n",
    "\n",
    "df2_new = df2[~df2['query'].isin(df1['query'])]\n",
    "df12 = pd.concat([df1, df2_new], ignore_index=True)\n",
    "\n",
    "file3 = '~/build/orthologer/Results/{}_33090.og.annotations'.format(name)\n",
    "df3 = pd.read_csv(file3, sep='\\t', header=0)\n",
    "df3.rename(columns={'#query': 'query'}, inplace=True)\n",
    "\n",
    "df3_new = df3[~df3['query'].isin(df12['query'])]\n",
    "df123 = pd.concat([df12, df3_new], ignore_index=True)\n",
    "\n",
    "print(\"Number of unique og:\")\n",
    "print(len(df123['ODB_OG'].unique()))\n",
    "# change their names to gene names (this only applies to EG5.1)\n",
    "if name == 'EG5.1':\n",
    "    header = pd.read_csv('~/build/orthologer/EG5.1_aa.header', sep=' ', header=None, names=['old','new','len'])\n",
    "    header['old'] = header['old'].str[1:]\n",
    "    header['new'] = header['new'].str.split('=').str[1]\n",
    "    header.drop('len', axis=1, inplace=True)\n",
    "    dictionary = {key: value for key, value in zip(header['old'], header['new'])}\n",
    "    df123['query'] = df123['query'].map(dictionary)\n",
    "\n",
    "df123.to_csv('~/build/orthologer/{}_4447_3193_33090.og.annotations'.format(name), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19379"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>ODB_OG</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>GOs_mf</th>\n",
       "      <th>GOs_bp</th>\n",
       "      <th>EC</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>Interpro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EG16ag044634.1</td>\n",
       "      <td>153499at4447</td>\n",
       "      <td>6.290000e-62</td>\n",
       "      <td>166.0800</td>\n",
       "      <td>X,L,T,J</td>\n",
       "      <td>Harbinger transposase-derived nuclease domain</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0071702</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EGxxag046697.1</td>\n",
       "      <td>194769at4447</td>\n",
       "      <td>2.160000e-251</td>\n",
       "      <td>248.8700</td>\n",
       "      <td>C,P,G</td>\n",
       "      <td>Photosystem II protein D1</td>\n",
       "      <td>GO:0045156,GO:0046872,GO:0016168</td>\n",
       "      <td>GO:0009772,GO:0019684,GO:0015979,GO:0009635</td>\n",
       "      <td>1.10.3.9</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR036854,IPR000484,IPR005867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGxxag046682.1</td>\n",
       "      <td>194769at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>97.2396</td>\n",
       "      <td>C,P,G</td>\n",
       "      <td>Photosystem II protein D1</td>\n",
       "      <td>GO:0045156,GO:0046872,GO:0016168</td>\n",
       "      <td>GO:0009772,GO:0019684,GO:0015979,GO:0009635</td>\n",
       "      <td>1.10.3.9</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR036854,IPR000484,IPR005867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGxxag046689.1</td>\n",
       "      <td>194769at4447</td>\n",
       "      <td>6.080000e-147</td>\n",
       "      <td>192.5300</td>\n",
       "      <td>C,P,G</td>\n",
       "      <td>Photosystem II protein D1</td>\n",
       "      <td>GO:0045156,GO:0046872,GO:0016168</td>\n",
       "      <td>GO:0009772,GO:0019684,GO:0015979,GO:0009635</td>\n",
       "      <td>1.10.3.9</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR036854,IPR000484,IPR005867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGxxag046660.1</td>\n",
       "      <td>3979at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>100.0000</td>\n",
       "      <td>T,X,L</td>\n",
       "      <td>photosystem II protein D2</td>\n",
       "      <td>GO:0016168,GO:0045156,GO:0046872</td>\n",
       "      <td>GO:0015979,GO:0019684,GO:0009772</td>\n",
       "      <td>1.10.3.9</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR000484,IPR036001,IPR036854,IPR005868,IPR000932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84493</th>\n",
       "      <td>EG08ag026314.1</td>\n",
       "      <td>1591460at33090</td>\n",
       "      <td>7.720000e-07</td>\n",
       "      <td>27.0500</td>\n",
       "      <td>-</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84494</th>\n",
       "      <td>EG12ag037158.1</td>\n",
       "      <td>1545579at33090</td>\n",
       "      <td>5.450000e-73</td>\n",
       "      <td>85.4200</td>\n",
       "      <td>-</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84495</th>\n",
       "      <td>EG03ag011169.1</td>\n",
       "      <td>1528171at33090</td>\n",
       "      <td>1.770000e-09</td>\n",
       "      <td>43.5900</td>\n",
       "      <td>-</td>\n",
       "      <td>uncharacterized protein LOC112003632</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84496</th>\n",
       "      <td>EG11ag035356.1</td>\n",
       "      <td>1594842at33090</td>\n",
       "      <td>2.870000e-09</td>\n",
       "      <td>28.6700</td>\n",
       "      <td>-</td>\n",
       "      <td>Reverse transcriptase, RNA-dependent DNA polym...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR013103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84497</th>\n",
       "      <td>EG15ag044518.1</td>\n",
       "      <td>1539595at33090</td>\n",
       "      <td>2.770000e-12</td>\n",
       "      <td>29.4000</td>\n",
       "      <td>-</td>\n",
       "      <td>Sugar transporter SWEET</td>\n",
       "      <td>GO:0051119</td>\n",
       "      <td>GO:0034219,GO:0071702</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR047664,IPR004316</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84498 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                query          ODB_OG         evalue     score COG_category  \\\n",
       "0      EG16ag044634.1    153499at4447   6.290000e-62  166.0800      X,L,T,J   \n",
       "1      EGxxag046697.1    194769at4447  2.160000e-251  248.8700        C,P,G   \n",
       "2      EGxxag046682.1    194769at4447   0.000000e+00   97.2396        C,P,G   \n",
       "3      EGxxag046689.1    194769at4447  6.080000e-147  192.5300        C,P,G   \n",
       "4      EGxxag046660.1      3979at4447   0.000000e+00  100.0000        T,X,L   \n",
       "...               ...             ...            ...       ...          ...   \n",
       "84493  EG08ag026314.1  1591460at33090   7.720000e-07   27.0500            -   \n",
       "84494  EG12ag037158.1  1545579at33090   5.450000e-73   85.4200            -   \n",
       "84495  EG03ag011169.1  1528171at33090   1.770000e-09   43.5900            -   \n",
       "84496  EG11ag035356.1  1594842at33090   2.870000e-09   28.6700            -   \n",
       "84497  EG15ag044518.1  1539595at33090   2.770000e-12   29.4000            -   \n",
       "\n",
       "                                             Description  \\\n",
       "0          Harbinger transposase-derived nuclease domain   \n",
       "1                              Photosystem II protein D1   \n",
       "2                              Photosystem II protein D1   \n",
       "3                              Photosystem II protein D1   \n",
       "4                              photosystem II protein D2   \n",
       "...                                                  ...   \n",
       "84493                               hypothetical protein   \n",
       "84494                               hypothetical protein   \n",
       "84495               uncharacterized protein LOC112003632   \n",
       "84496  Reverse transcriptase, RNA-dependent DNA polym...   \n",
       "84497                            Sugar transporter SWEET   \n",
       "\n",
       "                                 GOs_mf  \\\n",
       "0                                     -   \n",
       "1      GO:0045156,GO:0046872,GO:0016168   \n",
       "2      GO:0045156,GO:0046872,GO:0016168   \n",
       "3      GO:0045156,GO:0046872,GO:0016168   \n",
       "4      GO:0016168,GO:0045156,GO:0046872   \n",
       "...                                 ...   \n",
       "84493                                 -   \n",
       "84494                                 -   \n",
       "84495                                 -   \n",
       "84496                                 -   \n",
       "84497                        GO:0051119   \n",
       "\n",
       "                                            GOs_bp        EC KEGG_ko  \\\n",
       "0                                       GO:0071702         -       -   \n",
       "1      GO:0009772,GO:0019684,GO:0015979,GO:0009635  1.10.3.9       -   \n",
       "2      GO:0009772,GO:0019684,GO:0015979,GO:0009635  1.10.3.9       -   \n",
       "3      GO:0009772,GO:0019684,GO:0015979,GO:0009635  1.10.3.9       -   \n",
       "4                 GO:0015979,GO:0019684,GO:0009772  1.10.3.9       -   \n",
       "...                                            ...       ...     ...   \n",
       "84493                                            -         -       -   \n",
       "84494                                            -         -       -   \n",
       "84495                                            -         -       -   \n",
       "84496                                            -         -       -   \n",
       "84497                        GO:0034219,GO:0071702         -       -   \n",
       "\n",
       "                                                Interpro  \n",
       "0                                                      -  \n",
       "1                          IPR036854,IPR000484,IPR005867  \n",
       "2                          IPR036854,IPR000484,IPR005867  \n",
       "3                          IPR036854,IPR000484,IPR005867  \n",
       "4      IPR000484,IPR036001,IPR036854,IPR005868,IPR000932  \n",
       "...                                                  ...  \n",
       "84493                                                  -  \n",
       "84494                                                  -  \n",
       "84495                                                  -  \n",
       "84496                                          IPR013103  \n",
       "84497                                IPR047664,IPR004316  \n",
       "\n",
       "[84498 rows x 11 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df123"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting Bio\n",
      "  Downloading bio-1.7.1-py3-none-any.whl.metadata (5.7 kB)\n",
      "Collecting biopython>=1.80 (from Bio)\n",
      "  Downloading biopython-1.84-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)\n",
      "Collecting gprofiler-official (from Bio)\n",
      "  Downloading gprofiler_official-1.0.0-py3-none-any.whl.metadata (11 kB)\n",
      "Collecting mygene (from Bio)\n",
      "  Downloading mygene-3.2.2-py2.py3-none-any.whl.metadata (10 kB)\n",
      "Requirement already satisfied: pandas in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from Bio) (2.2.2)\n",
      "Collecting pooch (from Bio)\n",
      "  Downloading pooch-1.8.2-py3-none-any.whl.metadata (10 kB)\n",
      "Requirement already satisfied: requests in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from Bio) (2.31.0)\n",
      "Collecting tqdm (from Bio)\n",
      "  Downloading tqdm-4.66.4-py3-none-any.whl.metadata (57 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m57.6/57.6 kB\u001b[0m \u001b[31m8.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: numpy in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from biopython>=1.80->Bio) (2.0.1)\n",
      "Collecting biothings-client>=0.2.6 (from mygene->Bio)\n",
      "  Downloading biothings_client-0.3.1-py2.py3-none-any.whl.metadata (9.8 kB)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from pandas->Bio) (2.9.0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from pandas->Bio) (2024.1)\n",
      "Requirement already satisfied: tzdata>=2022.7 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from pandas->Bio) (2024.1)\n",
      "Requirement already satisfied: platformdirs>=2.5.0 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from pooch->Bio) (4.2.1)\n",
      "Requirement already satisfied: packaging>=20.0 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from pooch->Bio) (24.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from requests->Bio) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from requests->Bio) (3.7)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from requests->Bio) (2.2.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from requests->Bio) (2024.2.2)\n",
      "Requirement already satisfied: six>=1.5 in /home/huan/miniconda3/envs/jupyter/lib/python3.12/site-packages (from python-dateutil>=2.8.2->pandas->Bio) (1.16.0)\n",
      "Downloading bio-1.7.1-py3-none-any.whl (280 kB)\n",
      "\u001b[2K   \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m281.0/281.0 kB\u001b[0m \u001b[31m25.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading biopython-1.84-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.2 MB)\n",
      "\u001b[2K   \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.2/3.2 MB\u001b[0m \u001b[31m58.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hDownloading gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n",
      "Downloading mygene-3.2.2-py2.py3-none-any.whl (5.4 kB)\n",
      "Downloading pooch-1.8.2-py3-none-any.whl (64 kB)\n",
      "\u001b[2K   \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m64.6/64.6 kB\u001b[0m \u001b[31m12.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading tqdm-4.66.4-py3-none-any.whl (78 kB)\n",
      "\u001b[2K   \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m78.3/78.3 kB\u001b[0m \u001b[31m17.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading biothings_client-0.3.1-py2.py3-none-any.whl (29 kB)\n",
      "Installing collected packages: tqdm, biopython, pooch, gprofiler-official, biothings-client, mygene, Bio\n",
      "Successfully installed Bio-1.7.1 biopython-1.84 biothings-client-0.3.1 gprofiler-official-1.0.0 mygene-3.2.2 pooch-1.8.2 tqdm-4.66.4\n"
     ]
    }
   ],
   "source": [
    "!pip install Bio"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# dura gene-> og -> at gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = pd.read_csv('/home/huan/build/orthologer/EG5.1_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t', header=0)\n",
    "df1.rename(columns={'#query': 'query'}, inplace=True)\n",
    "\n",
    "df2 = pd.read_csv('/home/huan/build/orthologer/maize_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t', header=0)\n",
    "df2.rename(columns={'#query': 'query'}, inplace=True)\n",
    "\n",
    "df3 = pd.read_csv('/home/huan/build/orthologer/rice_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t', header=0)\n",
    "df3.rename(columns={'#query': 'query'}, inplace=True)\n",
    "\n",
    "df4 = pd.read_csv('/home/huan/build/orthologer/Arabidopsis_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t', header=0)\n",
    "df4.rename(columns={'#query': 'query'}, inplace=True)\n",
    "\n",
    "df5 = pd.read_csv('/home/huan/build/orthologer/dura_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t', header=0)\n",
    "df5.rename(columns={'#query': 'query'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df5 = pd.read_csv('/home/huan/build/orthologer/dura_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t', header=0)\n",
    "df5.rename(columns={'#query': 'query'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff = pd.read_csv('/home/huan/build/orthologer/Araport11_GFF3_genes_transposons.symbol.gff.2', sep='\\t', header=None, comment=\"#\", names=['chr','source','type','start','end','score','strand','phase','attributes'])\n",
    "gff_mRNA = gff[gff['type']=='mRNA']\n",
    "\n",
    "# Apply the function to each row in the 'attribute' column\n",
    "parsed_attributes = gff_mRNA['attributes'].apply(parse_attribute).apply(pd.Series)\n",
    "\n",
    "# Join the parsed attributes back to the original DataFrame\n",
    "gff_mRNA = gff_mRNA.join(parsed_attributes)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_light = gff_mRNA[['symbol','ID']]\n",
    "gff_light.rename(columns={'ID': 'query'}, inplace=True)\n",
    "df4_merged = pd.merge(df4, gff_light, on='query',how='inner')\n",
    "df4_light = df4_merged[['ODB_OG','symbol']].drop_duplicates()\n",
    "df4_light = df4_light[df4_light['symbol'] != 'None']\n",
    "df4_light = df4_light.groupby('ODB_OG')['symbol'].agg(lambda x: ';'.join(x.astype(str))).reset_index()\n",
    "df1_merged = pd.merge(df1, df4_light, on='ODB_OG',how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df5_merged = pd.merge(df5, df4_light, on='ODB_OG',how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df4_light.to_csv('/home/huan/build/orthologer/Arabidopsis_OG_symbol.map')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1_merged.to_csv('/home/huan/build/orthologer/EG5.1_orthoDBv11_4447_3193_33090.og.symbol.annotations', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df5_merged.to_csv('/home/huan/build/orthologer/dura_orthoDBv11_4447_3193_33090.og.symbol.annotations', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff = pd.read_csv('/home/huan/build/orthologer/EG5.1_Genes.V3.newid.gff3', sep='\\t', header=None, comment=\"#\", names=['chr','source','type','start','end','score','strand','phase','attributes'])\n",
    "gff_gene = gff[gff['type']=='gene']\n",
    "parsed_attributes = gff_gene['attributes'].apply(parse_attribute).apply(pd.Series)\n",
    "\n",
    "# Join the parsed attributes back to the original DataFrame\n",
    "gff_gene = gff_gene.join(parsed_attributes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# merge snpEff files from different chromosomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "# Define the pattern to match your TSV files\n",
    "file_pattern = \"/home/huan/build/orthologer/EG5.1_batch2_190_chr*.snpEff.genes.txt\"\n",
    "\n",
    "# Use glob to get a list of file paths\n",
    "file_paths = glob.glob(file_pattern)\n",
    "\n",
    "# List to hold each DataFrame\n",
    "dfs = []\n",
    "\n",
    "# Loop through the file paths, read each file into a DataFrame, and add it to the list\n",
    "for file_path in file_paths:\n",
    "    df = pd.read_csv(file_path, sep='\\t', header = 1)\n",
    "    dfs.append(df)\n",
    "\n",
    "# Concatenate all DataFrames in the list\n",
    "combined_df = pd.concat(dfs, ignore_index=True)\n",
    "combined_df.rename(columns={'#GeneName': 'GeneName'}, inplace=True)\n",
    "\n",
    "# Check the combined DataFrame\n",
    "combined_df.to_csv('/home/huan/build/orthologer/EG5.1_batch2_190.snpEff.genes.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plant_hormone_gene_list = ['TIR1','Aux','IAA','ARF','YUCCA','PIN','GAI','SPY','GA20ox','GA3ox','GA2ox','ARR','CRE1','WOL','AHK4','ETR1','CTR1','EIN2','PYR','PYL','RCAR','ABI','BRI1','BIN2','BZR1','COI1','JAZ','LOX', 'AOS', 'AOC','MAX1','D14','D53','SMXLs','NPR1']\n",
    "MADS_gene_list = ['AG','AP1','AP3','PI','SEP1','SOC1','FLC','SVP','SHP2','SEP3']\n",
    "homeobox_gene_list = ['WUS','KNAT','GL2','WOX','HB-','BEL1','PHB','ATH1']\n",
    "phyto_list = ['PHYA','PHYB','PHYC','PHYD','PHYE','PIF','PIL']\n",
    "#PIF\n",
    "gene_list = phyto_list.copy()\n",
    "outbed = '/home/huan/build/orthologer/EG5.1_PIF.bed'\n",
    "outfile = '/home/huan/build/orthologer/PIF.gene.list'\n",
    "# Plant hormone\n",
    "#gene_list = plant_hormone_gene_list.copy()\n",
    "#outbed = '/home/huan/build/orthologer/EG5.1_plant_hormone.bed'\n",
    "#outfile = '/home/huan/build/orthologer/plant_hormone.gene.list'\n",
    "# MADS-box\n",
    "#gene_list = MADS_gene_list.copy()\n",
    "#outbed = '/home/huan/build/orthologer/EG5.1_MADS_box.bed'\n",
    "#outfile = '/home/huan/build/orthologer/MADS_box.gene.list'\n",
    "# homeobox\n",
    "#gene_list = homeobox_gene_list.copy()\n",
    "#outbed = '/home/huan/build/orthologer/EG5.1_homeobox.bed'\n",
    "#outfile = '/home/huan/build/orthologer/Homeobox.gene.list'\n",
    "\n",
    "snpEff = pd.read_csv('/home/huan/build/orthologer/EG5.1_batch2_190.snpEff.genes.csv')\n",
    "snpEff['ID'] =  snpEff['TranscriptId'].str[:-2]\n",
    "snpEff.columns = [col.lstrip('variants_') if col.startswith('variants_') else col for col in snpEff.columns]\n",
    "out_snp = outbed + '.snpEff.csv'\n",
    "with open(outbed,'w') as fh_bed:\n",
    "    with open(outfile, 'w') as fh_file:\n",
    "        fh_file.write(','.join(['Gene', 'OG_ID', 'EG5.1', 'Dura', 'Arabidopsis', 'Rice', 'Maize']) + '\\n')\n",
    "        for gene in gene_list:\n",
    "            ogs = df4_light[df4_light['symbol'] == gene]['ODB_OG']\n",
    "            if ogs.shape[0] == 0:\n",
    "                pattern = rf'{gene}[1-9][0-9]*'\n",
    "                ogs = df4_light[df4_light['symbol'].str.match(pattern)]['ODB_OG']    \n",
    "            for og in ogs:\n",
    "                genes = df4_light[df4_light['ODB_OG'] == og]['symbol']\n",
    "                for gene in genes:\n",
    "                    EG_genes = df1[df1['ODB_OG'] == og]['query']\n",
    "                    gff_sub = gff_gene[gff_gene['ID'].isin(EG_genes)]\n",
    "                    EG_genes = ';'.join(EG_genes)\n",
    "                    EG_genes = ';'.join(df1[df1['ODB_OG'] == og]['query'])\n",
    "                    maize_genes = ';'.join(df2[df2['ODB_OG'] == og]['query'])\n",
    "                    rice_genes = ';'.join(df3[df3['ODB_OG'] == og]['query'])\n",
    "                    at_genes = ';'.join(df4[df4['ODB_OG'] == og]['query'])\n",
    "                    dura_genes = ';'.join(df5[df5['ODB_OG'] == og]['query'])\n",
    "                    fh_file.write(','.join([gene, og, EG_genes, dura_genes, at_genes, rice_genes, maize_genes]) + '\\n')\n",
    "                    for index, row in gff_sub.iterrows():\n",
    "                        fh_bed.write('\\t'.join([row['chr'],str(row['start']),str(row['end']),row['ID'], gene, og]) + '\\n')\n",
    "\n",
    "bed = pd.read_csv(outbed, sep = '\\t', header = None, names= ['chr','start','end','ID','gene','og'])\n",
    "bed_merge = pd.merge(bed, snpEff, on='ID',how='inner').drop(['GeneName','GeneId'], axis=1)\n",
    "bed_merge.to_csv(out_snp, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MADS-box genes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "anno = pd.read_csv('/home/huan/build/orthologer/EG5.1_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t')\n",
    "filtered_df = anno[anno['Description'].str.contains('MADS-box')]\n",
    "IDs = [i.split(',') for i in filtered_df['Interpro']]\n",
    "EG5_ID = set([item for sublist in IDs for item in sublist])\n",
    "anno = pd.read_csv('/home/huan/build/orthologer/dura_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t')\n",
    "filtered_df = anno[anno['Description'].str.contains('MADS-box')]\n",
    "IDs = [i.split(',') for i in filtered_df['Interpro']]\n",
    "dura_ID = set([item for sublist in IDs for item in sublist])\n",
    "anno = pd.read_csv('/home/huan/build/orthologer/rice_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t')\n",
    "filtered_df = anno[anno['Description'].str.contains('MADS-box')]\n",
    "IDs = [i.split(',') for i in filtered_df['Interpro']]\n",
    "rice_ID = set([item for sublist in IDs for item in sublist])\n",
    "anno = pd.read_csv('/home/huan/build/orthologer/maize_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t')\n",
    "filtered_df = anno[anno['Description'].str.contains('MADS-box')]\n",
    "IDs = [i.split(',') for i in filtered_df['Interpro']]\n",
    "maize_ID = set([item for sublist in IDs for item in sublist])\n",
    "anno = pd.read_csv('/home/huan/build/orthologer/Arabidopsis_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t')\n",
    "filtered_df = anno[anno['Description'].str.contains('MADS-box')]\n",
    "IDs = [i.split(',') for i in filtered_df['Interpro']]\n",
    "AT_ID = set([item for sublist in IDs for item in sublist])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "union = list(set(EG5_ID) | set(dura_ID) | set(rice_ID) | set(maize_ID) | set(AT_ID))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MADS-box interpro IDs are: ['IPR033897','IPR002487','IPR002100','IPR031050','IPR036879','IPR033896']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a='MGRGRVQLKRIENKINRQVTFSKRRAGLLKKAHEISVLCDAEVALVVFSHKGKLFEYSTDSCMEKILERYERYSYAERQLIAPESDVNTNWSMEYNRLKAKIELLERNQRHYLGEDLQAMSPKELQNLEQQLDTALKHIRTRKNQLMYESINELQKKEKAIQEQNSMLSKQIKEREKILRAQQEQWDQQNQGHNMPPPLPPQQHQIQHPYMLSHQPSPFLNMGGLYQEDDPMAMRRNDLELTLEPVYNCNLGCFAA'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic = {}\n",
    "with open('/home/huan/build/orthologer/IPR002100.gene') as fh:\n",
    "    for line in fh:\n",
    "        dic[line.rstrip()] = ''\n",
    "with open('/home/huan/build/orthologer/dura_IPR002100_proteins.fasta','w') as outfh:\n",
    "    for record in SeqIO.parse('/home/huan/build/orthologer/dura_proteins.fasta','fasta'):\n",
    "        if record.id in dic:\n",
    "            SeqIO.write(record, outfh, 'fasta')\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(a1, a2, a3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len('SDVNTNWSMEYNRLKAKIELLERNQRHYLGEDLQAMSPKELQNLEQQLDTALKHIRTRKNQLMYESINELQKKEKAIQEQNSMLSKQI')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a[:172]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_of_lists = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]\n",
    "flattened_list = [item for sublist in list_of_lists for item.spli in sublist]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "flattened_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# OK, now I need to write a script where it takes a symbol, and returns the gene name i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example DataFrame\n",
    "df1 = pd.DataFrame({\n",
    "    'attribute': ['ID=AT1G01010.1;Parent=AT1G01010;Name=AT1G01010.1']\n",
    "})\n",
    "\n",
    "\n",
    "# Apply the function to each row in the 'attribute' column\n",
    "parsed_attributes = df1['attribute'].apply(parse_attribute).apply(pd.Series)\n",
    "\n",
    "# Join the parsed attributes back to the original DataFrame\n",
    "df1 = df1.join(parsed_attributes)\n",
    "\n",
    "print(df1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "union = set(df2['ODB_OG']) | set(df3['ODB_OG']) | set(df4['ODB_OG'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(union)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "shared_og_all = set(df1['ODB_OG']) & union"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(shared_og_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "shared_og = set(df1['ODB_OG']) & set(df2['ODB_OG'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(shared_og)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(shared_og)/23056"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(df1['ODB_OG'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df2.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "12593/15674"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Now we need to go from gene name to clusterID, then dura gene ID.\n",
    "Play with the API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess, json\n",
    "\n",
    "gene_name = 'ABI5'\n",
    "level = 4447\n",
    "cmd = \"curl 'https://data.orthodb.org/current/search?query={}&level={}' -L -o {}_at{}.dat\".format(gene_name, level, gene_name, level)\n",
    "print(cmd)\n",
    "subprocess.run(cmd, shell=True, check=True)\n",
    "with open(\"{}_at{}.dat\".format(gene_name, level), \"r\") as json_file:\n",
    "    data_dict = json.load(json_file)\n",
    "anno_file = 'dura_orthoDBv11_4447_3193_33090.og.annotations'\n",
    "df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "selected_rows = df[df['ODB_OG'].isin(data_dict['data'])]\n",
    "selected_rows.rename(columns={'#query': 'query'}, inplace=True)\n",
    "i = 0\n",
    "protein_file = '/bioinfo/palm/ref/dura/dura_proteins.fasta'\n",
    "with open('dura_proteins_{}_at{}.aa'.format(gene_name, level),'w') as fh:\n",
    "    for record in SeqIO.parse(protein_file,'fasta'):\n",
    "        if record.id in selected_rows['query'].values:\n",
    "            cluster = selected_rows.loc[selected_rows['query'] == record.id, 'ODB_OG']\n",
    "            tag = selected_rows.loc[selected_rows['query'] == record.id, 'Description']\n",
    "            if len(tag) == 1:\n",
    "                record.description = ':'.join([cluster.iloc[0], tag.iloc[0]])\n",
    "            else:\n",
    "                print(record.id + 'appeared in more than one cluster?')\n",
    "\n",
    "            SeqIO.write(record, fh, 'fasta')\n",
    "            i += 1\n",
    "print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(selected_rows.loc[selected_rows['query'] == 'Egu023084-mRNA-1', 'ODB_OG'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# We ran orthoDB on four other species and compared them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from collections import Counter\n",
    "level = 4447\n",
    "def og_dic(level, name):\n",
    "    # this function takes a orthoDB annotation, count the cluser compotion and return a pd\n",
    "    anno_file = \"/bioinfo2/palm/ref/orthoDBv11/{}_{}.og.annotations\".format(name, level)\n",
    "    df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "    dic = Counter(df['ODB_OG'])\n",
    "    # Convert dictionaries to DataFrames\n",
    "    df = pd.DataFrame(list(dic.items()), columns=['key', name]).set_index('key')\n",
    "    return(df)\n",
    "\n",
    "df_dura = og_dic(4447, 'dura')\n",
    "df_coco = og_dic(4447, 'coco')\n",
    "df_date = og_dic(4447, 'date')\n",
    "df_picifera = og_dic(4447, 'EG5')\n",
    "df_rice = og_dic(4447, 'rice')\n",
    "\n",
    "df = pd.concat([df_dura, df_coco, df_date, df_picifera, df_rice], axis=1)\n",
    "\n",
    "df.to_csv(\"/bioinfo2/palm/ref/orthoDBv11/combined_og.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "duplicates = df123[df123.duplicated('query')]  # Replace 'column_name' with the actual column name\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "anno_file = 'dura_orthoDBv11_4447_3193_33090.og.annotations'\n",
    "df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "tpm_file = 'counts_transcript_TPM_concise.txt'\n",
    "tpm = pd.read_csv(tpm_file, sep=',', header=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1 = df[['query','Description']]\n",
    "df_2 = pd.merge(df_1, tpm, on='query',how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2.to_csv('output.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Transfer Dura og to EG5.1 og."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "# Function to parse the 'Attributes' column in gff file and convert it into a dictionary of key-value pairs\n",
    "def parse_attributes(attribute_str):\n",
    "    attribute_pairs = attribute_str.split(';')\n",
    "    attribute_dict = {}\n",
    "    for pair in attribute_pairs:\n",
    "        if len(pair.split('=')) == 2:\n",
    "            key, value = pair.split('=')\n",
    "        else:\n",
    "            key = pair.split('=')[0]\n",
    "            value = ''\n",
    "        attribute_dict[key] = value\n",
    "    return attribute_dict\n",
    "\n",
    "def filtered_rows2bedfile(filtered_rows, bedfile):\n",
    "    # takes a list of gene ids and write their regions into a bed file\n",
    "    df = filtered_rows[['Seqid', 'Start','End','ID']]\n",
    "    df_no_duplicates = df.drop_duplicates()\n",
    "    df_sorted = df_no_duplicates.sort_values(by=['Seqid', 'Start'])\n",
    "    # add pound(#) to the first column name - convention of bed file\n",
    "    df_sorted.columns = ['#' + df_sorted.columns[0] if i == 0 else col for i, col in enumerate(df_sorted.columns)]\n",
    "    df_sorted.to_csv(bedfile, sep='\\t', index=False)\n",
    "    return df_sorted\n",
    "\n",
    "# read in the Dura annotation\n",
    "dura_df = pd.read_csv('/home/huan/build/orthologer/dura_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t')\n",
    "# read in the EG5.1 annotation\n",
    "EG51_df = pd.read_csv('/home/huan/build/orthologer/EG5.1_orthoDBv11_4447_3193_33090.og.annotations', sep='\\t')\n",
    "## read the gff file\n",
    "gff_file = '/home/huan/Documents/palm/EG5.1_Genes.V3.gff3'\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "\n",
    "trait_list = ['drought','height','fa','fattyacid','fruit','SAUR']    \n",
    "\n",
    "for trait in trait_list:\n",
    "    # read in the catalog file\n",
    "    cat_file = '/home/huan/build/orthologer/dura_{}_catalogue.txt'.format(trait)\n",
    "    cat_df = pd.read_csv(cat_file, sep='\\t', header=None, names=['query','gene'])\n",
    "    # cat_dura_df = dura_df[dura_df['query'].isin(cat_df['query'])]\n",
    "    cat_dura_df = pd.merge(cat_df, dura_df, on='query',how='inner')\n",
    "    cat_dura_df.rename(columns={'query': 'dura_gene'}, inplace=True)\n",
    "    # sorted_df = cat_dura_df.sort_values(by='ODB_OG')\n",
    "    cat_EG51_df = EG51_df[EG51_df['ODB_OG'].isin(cat_dura_df['ODB_OG'])]\n",
    "    shared_columns = set(EG51_df.columns).intersection(set(cat_dura_df.columns))\n",
    "    columns_to_keep = ['ODB_OG'] + [col for col in cat_dura_df.columns if col not in shared_columns]\n",
    "    cat_dura_df_reduced = cat_dura_df[columns_to_keep]\n",
    "    \n",
    "    # Perform the merge\n",
    "    cat_EG51_df = pd.merge(cat_dura_df_reduced, EG51_df, on='ODB_OG', how='inner')\n",
    "    selected_df = cat_EG51_df[['ODB_OG', 'gene', 'dura_gene', 'query']]\n",
    "    selected_df.rename(columns={'query': 'EG5.1_gene', 'gene': 'gene_name'}, inplace=True)\n",
    "    \n",
    "    selected_df.to_csv('~/build/orthologer/EG5.1_{}_catalogue.txt'.format(trait), sep='\\t', index=False)\\\n",
    "    \n",
    "    \n",
    "    \n",
    "    ## read in the orthoDB annotation file\n",
    "    selected_rows = EG51_df[EG51_df['ODB_OG'].isin(selected_df['ODB_OG'])]\n",
    "    selected_rows = selected_rows.rename(columns={'#query': 'query'})\n",
    "    \n",
    "    \n",
    "    # prepare bed file gene region\n",
    "    filtered_rows = df_gff[df_gff['ID'].isin(selected_rows['query'])]\n",
    "    bedfile = '~/build/orthologer/EG5.1_{}_exon.bed'.format(trait)\n",
    "    filtered_rows2bedfile(filtered_rows, bedfile) \n",
    "\n",
    "# prepare bed file for exons only: 9 should be enough? No it is not. can easily go beyond that.\n",
    "# pattern = '|'.join([f\"{item}:[1-9]$\" for item in selected_rows['query']])\n",
    "# pattern = r'^(?:' + '|'.join(selected_rows['query']) + r'):\\d+$'\n",
    "\n",
    "# Filtering rows\n",
    "# filtered_rows = df_gff[df_gff['ID'].str.contains(pattern, regex=True)]\n",
    "# bedfile = gene_name + '_at' + level + '_dura.exon.bed'\n",
    "# filtered_rows2bedfile(filtered_rows, bedfile) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "# Synteny"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames = ['pic_chr','pic_len','p_start','p_end','orientation','dur_chr','dur_len','d_start','d_end','A','B','C','D','E','AS','nn','tp','cm','s1','s2','df','zd','rl','cg']\n",
    "syn_chr1 = pd.read_csv('/home/huan/Documents/palm/pic_chr1_dura.paf', sep='\\t', names=colnames)\n",
    "df_sorted = syn_chr1.sort_values(by='p_start')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# misssed jobs in freebayes?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import bisect\n",
    "i = 0\n",
    "dic = {}\n",
    "with open('/home/huan/Downloads/EG5.1_batch2_190_chr1.pos') as fh:\n",
    "    for line in fh:\n",
    "        j = int(line.rstrip())\n",
    "        dis = j-i\n",
    "        dic[i] = dis\n",
    "        i = j\n",
    "i = 0\n",
    "sorted_keys = sorted(dic.keys())\n",
    "\n",
    "with open('/home/huan/Downloads/EG5.1_batch2_262_chr1.pos') as fh:\n",
    "    for line in fh:\n",
    "        j = int(line.rstrip())\n",
    "        if j - i > 10000:\n",
    "            if i in dic:\n",
    "                if dic[i] < 10000:\n",
    "                    print(i,j)\n",
    "            else:\n",
    "                # Find the left and right indices\n",
    "                left_index = bisect.bisect_left(sorted_keys, i)\n",
    "                right_index = bisect.bisect_right(sorted_keys, j)\n",
    "                # Calculate the count\n",
    "                count = right_index - left_index\n",
    "                if count > 1:\n",
    "                    print(i,j)\n",
    "                    break\n",
    "        i = j\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# From gff to bed (using ID instead of transciptID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to parse the attribute column\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "def parse_attribute(attribute):\n",
    "    pairs = attribute.split(';')\n",
    "    attr_dict = {}\n",
    "    for pair in pairs:\n",
    "        # Check if the element contains more attributes separated by ','\n",
    "        if pair.count('=') > 1:\n",
    "            if ',' in pair:\n",
    "                sub_pairs = pair.split(',')\n",
    "                for sub_pair in sub_pairs:\n",
    "                    if '=' in sub_pair:\n",
    "                        print(sub_pair)\n",
    "                        key, value = sub_pair.split('=')\n",
    "                        attr_dict[key] = value\n",
    "        else:\n",
    "            if '=' in pair:\n",
    "                key, value = pair.split('=')\n",
    "                attr_dict[key] = value\n",
    "    return attr_dict\n",
    "\n",
    "def gff2bed(gff_file, type = None, bed_level = 'ID'):\n",
    "    gff = pd.read_csv(gff_file, sep='\\t', header=None, comment=\"#\", names=['chr','source','type','start','end','score','strand','phase','attributes'])\n",
    "    if type:\n",
    "        gff = gff[gff['type']==type]\n",
    "    # Apply the function to each row in the 'attribute' column\n",
    "    parsed_attributes = gff['attributes'].apply(parse_attribute).apply(pd.Series)    \n",
    "    # Join the parsed attributes back to the original DataFrame\n",
    "    gff = gff.join(parsed_attributes)\n",
    "    # note that gff is 1-based and bed is 0-based, so start needs to be adjusted\n",
    "    gff['start'] = gff['start'].astype(int) - 1\n",
    "    bed = gff[['chr','start','end',bed_level]].copy()\n",
    "    bed.rename(columns={'chr': '#chr'}, inplace=True)\n",
    "    root, ext = os.path.splitext(gff_file)\n",
    "    bed.to_csv(root+'.bed', sep = '\\t', index=False)\n",
    "def idlist2seqfile(ids, seqfile, outfile):\n",
    "    from Bio import SeqIO\n",
    "    ids = set(ids)\n",
    "    with open(outfile, 'w') as out_fh:\n",
    "        for record in SeqIO.parse(seqfile, 'fasta'):\n",
    "            if record.id in ids:\n",
    "                SeqIO.write(record, out_fh, 'fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff2bed('/home/huan/Documents/Palm/EG5.1_p5.00_sc00060_p0019.gff3')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff2bed('/home/huan/Documents/Palm/dura_Egu023247.gff')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get rRNA genes from EG5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff = pd.read_csv('/home/huan/Documents/Palm/genomic.gff', sep='\\t', header=None, comment=\"#\", names=['chr','source','type','start','end','score','strand','phase','attributes'])\n",
    "gff_rRNA = gff[gff['type']=='rRNA']\n",
    "\n",
    "# Apply the function to each row in the 'attribute' column\n",
    "parsed_attributes = gff_rRNA['attributes'].apply(parse_attribute).apply(pd.Series)\n",
    "\n",
    "# Join the parsed attributes back to the original DataFrame\n",
    "gff_rRNA = gff_rRNA.join(parsed_attributes)\n",
    "gff_rRNA['id'] = gff_rRNA['ID'].strsplit('-')[1]\n",
    "                \n",
    "seqfile = '/home/huan/Documents/Palm/dura_proteins.fasta'\n",
    "outfile = '/home/huan/Documents/Palm/dura_30unique_proteins.fasta'\n",
    "ids = idfile2seqfile(idfile,seqfile, outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_rRNA['id'] = gff_rRNA['ID'].str.split('-').str[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_rRNA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# From at gene ID to og ID to genes in EG5, dura and braker5.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "at_anno = pd.read_csv('/home/huan/build/orthologer/Arabidopsis_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t')\n",
    "at_anno['AtLocusID'] = at_anno['query'].str.split('.').str[0]\n",
    "EG5_anno =  pd.read_csv('/home/huan/build/orthologer/EG5_4447.og.annotations', sep = '\\t').groupby('ODB_OG')['query'].apply(lambda x: \";\".join(x)).reset_index().rename(columns={'query': 'EG5'})\n",
    "EG5_1_anno = pd.read_csv('/home/huan/build/orthologer/EG5.1_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t').groupby('ODB_OG')['query'].apply(lambda x: \";\".join(x)).reset_index().rename(columns={'query': 'EG5.1'})\n",
    "dura_anno = pd.read_csv('/home/huan/build/orthologer/dura_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t').groupby('ODB_OG')['query'].apply(lambda x: \";\".join(x)).reset_index().rename(columns={'query': 'dura'})\n",
    "braker_anno = pd.read_csv('/home/huan/build/orthologer/braker5.0_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t').groupby('ODB_OG')['query'].apply(lambda x: \";\".join(x)).reset_index().rename(columns={'query': 'braker5'})\n",
    "anno_all = pd.merge(at_anno, EG5_anno, on = 'ODB_OG', how = 'outer')\n",
    "at_anno_B = pd.merge(at_anno_B, EG5_1_anno, on = 'ODB_OG', how = 'left')\n",
    "at_anno_B = pd.merge(at_anno_B, dura_anno, on = 'ODB_OG', how = 'left')\n",
    "at_anno_B = pd.merge(at_anno_B, braker_anno, on = 'ODB_OG', how = 'left').drop(['query'], axis=1)\n",
    "Bourgis2011 = pd.read_csv('/home/huan/Documents/Palm/Bourgis2011.S2A.csv', sep = ',')\n",
    "at_anno_B = pd.merge(at_anno, Bourgis2011, on='AtLocusID', how = 'right')\n",
    "at_anno_B.to_csv('Bourgis2011.ODB_OG.txt', sep = '\\t', index=False)\n",
    "\n",
    "# OK so there are some cases that we had a At gene, and it matches with an ogID, but that ogID is not found in one of the oil annotations.\n",
    "# e.g. BCCP1 and 2. However in the category file Kevin gives, those two geens matches with some other ogID and we have them in some of the oil palm annotations. Check what is happening.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "EG5_1 = pd.read_csv('/home/huan/build/orthologer/EG5.1_orthoDBv11_4447_3193_33090.og.annotations', sep = '\\t').rename(columns={'query': 'gene_id'})\n",
    "FAB_genes = pd.read_excel('/home/huan/Documents/Palm/FAB_genes.xlsx')\n",
    "FAB_genes_og = pd.merge(FAB_genes, EG5_1, on = 'gene_id', how = 'left')\n",
    "FAB_genes_og.to_csv('FAB_genes_og.txt', sep = '\\t', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#</th>\n",
       "      <th>OPFAB_gene_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>enzyme</th>\n",
       "      <th>ec_num</th>\n",
       "      <th>pfam_domain</th>\n",
       "      <th>catalytic_residue</th>\n",
       "      <th>motif</th>\n",
       "      <th>reference</th>\n",
       "      <th>ODB_OG</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>GOs_mf</th>\n",
       "      <th>GOs_bp</th>\n",
       "      <th>EC</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>Interpro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>EgACC_1</td>\n",
       "      <td>p5.00_sc00054_p0048</td>\n",
       "      <td>Multifunctional form of Acetyl-CoA carboxylase...</td>\n",
       "      <td>EC 6.4.1.2</td>\n",
       "      <td>ACC_central (PF08326), Carboxyl_trans (PF01039...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>112963at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>242.970</td>\n",
       "      <td>I,C,E,H</td>\n",
       "      <td>Acetyl-CoA carboxylase</td>\n",
       "      <td>GO:0016874,GO:0005524,GO:0000166,GO:0046872,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR000089,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>EgACC_2</td>\n",
       "      <td>p5.00_sc00095_p0046</td>\n",
       "      <td>Multifunctional form of Acetyl-CoA carboxylase...</td>\n",
       "      <td>EC 6.4.1.2</td>\n",
       "      <td>CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>118148at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>222.700</td>\n",
       "      <td>I,C,E,H</td>\n",
       "      <td>Methylcrotonoyl-CoA carboxylase subunit alpha,...</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0000166</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00860,ko04144</td>\n",
       "      <td>IPR000089,IPR001882,IPR005479,IPR005481,IPR005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>EgCT_1</td>\n",
       "      <td>p5.00_sc00197_p0009</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (CT)</td>\n",
       "      <td>-</td>\n",
       "      <td>ACCA (PF03255)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>124466at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>175.034</td>\n",
       "      <td>I,L</td>\n",
       "      <td>acetyl-coenzyme A carboxylase carboxyl transfe...</td>\n",
       "      <td>GO:0016874,GO:0016740,GO:0005524,GO:0003989,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR001095,IPR011763,IPR029045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>EgCT_2</td>\n",
       "      <td>p5.00_sc00020_p0225</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (CT)</td>\n",
       "      <td>-</td>\n",
       "      <td>ACCA (PF03255)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>124466at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>194.260</td>\n",
       "      <td>I,L</td>\n",
       "      <td>acetyl-coenzyme A carboxylase carboxyl transfe...</td>\n",
       "      <td>GO:0016874,GO:0016740,GO:0005524,GO:0003989,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR001095,IPR011763,IPR029045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>EgCT_3</td>\n",
       "      <td>p5.00_sc00017_p0152</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (CT)</td>\n",
       "      <td>-</td>\n",
       "      <td>ACCA (PF03255)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>124466at4447</td>\n",
       "      <td>3.500000e-306</td>\n",
       "      <td>151.060</td>\n",
       "      <td>I,L</td>\n",
       "      <td>acetyl-coenzyme A carboxylase carboxyl transfe...</td>\n",
       "      <td>GO:0016874,GO:0016740,GO:0005524,GO:0003989,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR001095,IPR011763,IPR029045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>EgBCCP_1</td>\n",
       "      <td>p5.00_sc00011_p0007</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (B...</td>\n",
       "      <td>-</td>\n",
       "      <td>Biotin_lipoyl (PF00364)</td>\n",
       "      <td>Ser-Met-Lys</td>\n",
       "      <td>(A/I/V)-M-K-(L/V/M/A/T)</td>\n",
       "      <td>Barber et al. 2005</td>\n",
       "      <td>136974at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>109.859</td>\n",
       "      <td>C,I,H,E</td>\n",
       "      <td>Biotin carboxyl carrier protein of acetyl-CoA ...</td>\n",
       "      <td>GO:0003989</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>IPR000089,IPR001249,IPR001882,IPR011053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>EgBCCP_2</td>\n",
       "      <td>p5.00_sc00227_p0009</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (B...</td>\n",
       "      <td>-</td>\n",
       "      <td>Biotin_lipoyl (PF00364)</td>\n",
       "      <td>Ser-Met-Lys</td>\n",
       "      <td>(A/I/V)-M-K-(L/V/M/A/T)</td>\n",
       "      <td>Barber et al. 2005</td>\n",
       "      <td>136974at4447</td>\n",
       "      <td>2.570000e-108</td>\n",
       "      <td>141.720</td>\n",
       "      <td>C,I,H,E</td>\n",
       "      <td>Biotin carboxyl carrier protein of acetyl-CoA ...</td>\n",
       "      <td>GO:0003989</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>ko04141</td>\n",
       "      <td>IPR000089,IPR001249,IPR001882,IPR011053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>EgBC_1</td>\n",
       "      <td>p5.00_sc00019_p0245</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (BC)</td>\n",
       "      <td>-</td>\n",
       "      <td>CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...</td>\n",
       "      <td>His-Glu, His-Glu, Glu, Ile-Glu, Arg</td>\n",
       "      <td>-</td>\n",
       "      <td>Waldrop et al. 1994</td>\n",
       "      <td>118678at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>235.260</td>\n",
       "      <td>I,E,C,F</td>\n",
       "      <td>Biotin carboxylase</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>6.3.4.14,6.4.1.2</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR004549,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>EgBC_2</td>\n",
       "      <td>p5.00_sc00077_p0033</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (BC)</td>\n",
       "      <td>-</td>\n",
       "      <td>CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...</td>\n",
       "      <td>His-Glu, His-Glu, Glu, Ile-Glu, Arg</td>\n",
       "      <td>-</td>\n",
       "      <td>Waldrop et al. 1994</td>\n",
       "      <td>118678at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>238.430</td>\n",
       "      <td>I,E,C,F</td>\n",
       "      <td>Biotin carboxylase</td>\n",
       "      <td>GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...</td>\n",
       "      <td>GO:2001295,GO:0006631,GO:0006629</td>\n",
       "      <td>6.3.4.14,6.4.1.2</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR004549,IPR005479,IPR005481,IPR005482,IPR011...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>EgFABD_1</td>\n",
       "      <td>p5.00_sc00007_p0027</td>\n",
       "      <td>Malonyl-CoA:ACPtransacylase (FABD)</td>\n",
       "      <td>EC 2.3.1.39</td>\n",
       "      <td>Acyl_transf_1 (PF00698)</td>\n",
       "      <td>Gln-Ser-Arg-His-Gln</td>\n",
       "      <td>GLSLGEY</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>133140at4447</td>\n",
       "      <td>1.030000e-214</td>\n",
       "      <td>191.840</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>malonyl CoA-acyl carrier protein transacylase</td>\n",
       "      <td>GO:0016740,GO:0004314,GO:0016419</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>bdi01110,ko00901</td>\n",
       "      <td>IPR001227,IPR004410,IPR014043,IPR016035,IPR016036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>EgFABH_1</td>\n",
       "      <td>p5.00_sc00035_p0050</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase III (FABH)</td>\n",
       "      <td>EC 2.3.1.180</td>\n",
       "      <td>ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)</td>\n",
       "      <td>Cys-His-Arg-Asn</td>\n",
       "      <td>GNTSAAS</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>151244at4447</td>\n",
       "      <td>6.400000e-27</td>\n",
       "      <td>46.840</td>\n",
       "      <td>-</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00061,ko01212,zma01100</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>EgFABH_2</td>\n",
       "      <td>p5.00_sc00049_p0018</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase III (FABH)</td>\n",
       "      <td>EC 2.3.1.180</td>\n",
       "      <td>ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)</td>\n",
       "      <td>Cys-His-Arg-Asn</td>\n",
       "      <td>GNTSAAS</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>119951at4447</td>\n",
       "      <td>1.630000e-246</td>\n",
       "      <td>215.490</td>\n",
       "      <td>I,Q,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016740,GO:0004315</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00062,ko00250,ko00260,ko00270,ko00280,ko0201...</td>\n",
       "      <td>IPR004655,IPR013747,IPR013751,IPR016039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>EgFABB_1</td>\n",
       "      <td>p5.00_sc00001_p0491</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>117154at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>255.320</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016747,GO:0016740,GO:0004315</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.2.4.1</td>\n",
       "      <td>ko00010,ko00020,ko00620,ko01200</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>EgFABB_2</td>\n",
       "      <td>p5.00_sc00004_p0248</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>38560at4447</td>\n",
       "      <td>4.240000e-285</td>\n",
       "      <td>215.240</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016747,GO:0016740,GO:0004315</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.1.1.-</td>\n",
       "      <td>ko00330,ko00592,sbi01100,sbi01110,sita01100</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>EgFABB_3</td>\n",
       "      <td>p5.00_sc00064_p0074</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>38560at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>204.786</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016747,GO:0016740,GO:0004315</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.1.1.-</td>\n",
       "      <td>ko00330,ko00592,sbi01100,sbi01110,sita01100</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>EgFABB_4</td>\n",
       "      <td>p5.00_sc00064_p0077</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>38560at4447</td>\n",
       "      <td>2.740000e-283</td>\n",
       "      <td>214.180</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016747,GO:0016740,GO:0004315</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.1.1.-</td>\n",
       "      <td>ko00330,ko00592,sbi01100,sbi01110,sita01100</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>EgFABF_1</td>\n",
       "      <td>p5.00_sc00004_p0242</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase II (FABF)</td>\n",
       "      <td>EC 2.3.1.179</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>48534at4447</td>\n",
       "      <td>8.250000e-305</td>\n",
       "      <td>191.050</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016740,GO:0004315,GO:0016747</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.11.1.7</td>\n",
       "      <td>ko03010,ko04144</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>EgFABF_2</td>\n",
       "      <td>p5.00_sc00036_p0047</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase II (FABF)</td>\n",
       "      <td>EC 2.3.1.179</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>48534at4447</td>\n",
       "      <td>2.230000e-263</td>\n",
       "      <td>169.030</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016740,GO:0004315,GO:0016747</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.11.1.7</td>\n",
       "      <td>ko03010,ko04144</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>EgFABF_3</td>\n",
       "      <td>p5.00_sc00035_p0142</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase II (FABF)</td>\n",
       "      <td>EC 2.3.1.179</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>48534at4447</td>\n",
       "      <td>8.250000e-266</td>\n",
       "      <td>170.470</td>\n",
       "      <td>Q,I,C</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016740,GO:0004315,GO:0016747</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.11.1.7</td>\n",
       "      <td>ko03010,ko04144</td>\n",
       "      <td>IPR000794,IPR014030,IPR014031,IPR016039,IPR017...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>EgFABG_1</td>\n",
       "      <td>p5.00_sc00006_p0025</td>\n",
       "      <td>Beta-ketoacyl-ACP reductase (FABG)</td>\n",
       "      <td>EC 1.1.1.100</td>\n",
       "      <td>adh_short (PF00106)</td>\n",
       "      <td>Arg-Ser-Tyr-Lys-Arg</td>\n",
       "      <td>YX3K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>134714at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>139.587</td>\n",
       "      <td>-</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016491,GO:0004316,GO:0102131,GO:0102132</td>\n",
       "      <td>GO:0006629,GO:0006631,GO:0055114</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00260,ko03010,ko03013</td>\n",
       "      <td>IPR002347,IPR011284,IPR020904,IPR036291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>EgFABG_2</td>\n",
       "      <td>p5.00_sc00034_p0180</td>\n",
       "      <td>Beta-ketoacyl-ACP reductase (FABG)</td>\n",
       "      <td>EC 1.1.1.100</td>\n",
       "      <td>adh_short (PF00106)</td>\n",
       "      <td>Arg-Ser-Tyr-Lys-Arg</td>\n",
       "      <td>YX3K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>134714at4447</td>\n",
       "      <td>1.290000e-167</td>\n",
       "      <td>178.930</td>\n",
       "      <td>-</td>\n",
       "      <td>3-oxoacyl-</td>\n",
       "      <td>GO:0016491,GO:0004316,GO:0102131,GO:0102132</td>\n",
       "      <td>GO:0006629,GO:0006631,GO:0055114</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00260,ko03010,ko03013</td>\n",
       "      <td>IPR002347,IPR011284,IPR020904,IPR036291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>EgFABZ_1</td>\n",
       "      <td>p5.00_sc00271_p0017</td>\n",
       "      <td>Beta-hydroxyacyl-ACP dehydrogenase (FABZ)</td>\n",
       "      <td>EC 4.2.1.59</td>\n",
       "      <td>FabA (PF07977)</td>\n",
       "      <td>His-Asp</td>\n",
       "      <td>LPHRFPFLLVDRV</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>64848at4447</td>\n",
       "      <td>2.950000e-155</td>\n",
       "      <td>230.960</td>\n",
       "      <td>I,Q,M,C</td>\n",
       "      <td>(3R)-hydroxymyristoyl-</td>\n",
       "      <td>GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...</td>\n",
       "      <td>GO:0009245,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00230,ko00730,ko04075,ko04626,zma01100,zma01110</td>\n",
       "      <td>IPR010084,IPR013114,IPR029069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>EgFABZ_2</td>\n",
       "      <td>p5.00_sc00012_p0164</td>\n",
       "      <td>Beta-hydroxyacyl-ACP dehydrogenase (FABZ)</td>\n",
       "      <td>EC 4.2.1.59</td>\n",
       "      <td>FabA (PF07977)</td>\n",
       "      <td>His-Asp</td>\n",
       "      <td>LPHRFPFLLVDRV</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>64848at4447</td>\n",
       "      <td>6.000000e-105</td>\n",
       "      <td>169.670</td>\n",
       "      <td>I,Q,M,C</td>\n",
       "      <td>(3R)-hydroxymyristoyl-</td>\n",
       "      <td>GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...</td>\n",
       "      <td>GO:0009245,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00230,ko00730,ko04075,ko04626,zma01100,zma01110</td>\n",
       "      <td>IPR010084,IPR013114,IPR029069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>EgFABI_1</td>\n",
       "      <td>p5.00_sc00037_p0126</td>\n",
       "      <td>Enoyl-ACP reductase (FABI)</td>\n",
       "      <td>EC 1.3.1.9</td>\n",
       "      <td>Adh_short_C2 (PF13561)</td>\n",
       "      <td>Tyr-Tyr-Lys</td>\n",
       "      <td>YX6K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>132738at4447</td>\n",
       "      <td>2.090000e-237</td>\n",
       "      <td>217.810</td>\n",
       "      <td>-</td>\n",
       "      <td>enoyl-</td>\n",
       "      <td>GO:0004318,GO:0016491</td>\n",
       "      <td>GO:0055114</td>\n",
       "      <td>3.4.19.12,2.4.2.-,3.1.3.2</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR002347,IPR014358,IPR036291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>EgFABI_2</td>\n",
       "      <td>p5.00_sc00131_p0012</td>\n",
       "      <td>Enoyl-ACP reductase (FABI)</td>\n",
       "      <td>EC 1.3.1.9</td>\n",
       "      <td>Adh_short_C2 (PF13561)</td>\n",
       "      <td>Tyr-Tyr-Lys</td>\n",
       "      <td>YX6K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>132738at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>177.550</td>\n",
       "      <td>-</td>\n",
       "      <td>enoyl-</td>\n",
       "      <td>GO:0004318,GO:0016491</td>\n",
       "      <td>GO:0055114</td>\n",
       "      <td>3.4.19.12,2.4.2.-,3.1.3.2</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR002347,IPR014358,IPR036291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>EgFAB2_1</td>\n",
       "      <td>p5.00_sc00091_p0032</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>13185at4447</td>\n",
       "      <td>1.710000e-234</td>\n",
       "      <td>222.180</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0045300,GO:0016491,GO:0046872</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>EgFAB2_2</td>\n",
       "      <td>p5.00_sc00051_p0096</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>129798at4447</td>\n",
       "      <td>4.040000e-226</td>\n",
       "      <td>207.920</td>\n",
       "      <td>P,F</td>\n",
       "      <td>Stearoyl-</td>\n",
       "      <td>GO:0046872,GO:0045300,GO:0016491</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.1.1.44</td>\n",
       "      <td>ko00030,ko00195,ko00480,ko01200</td>\n",
       "      <td>IPR005067,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>EgFAB2_3</td>\n",
       "      <td>p5.00_sc00125_p0015</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>198320at3193</td>\n",
       "      <td>3.340000e-105</td>\n",
       "      <td>184.990</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0016491,GO:0045300</td>\n",
       "      <td>GO:0006629,GO:0006631</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko03010</td>\n",
       "      <td>IPR005067,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>EgFAB2_4</td>\n",
       "      <td>p5.00_sc00147_p0015</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>122758at4447</td>\n",
       "      <td>4.450000e-298</td>\n",
       "      <td>259.070</td>\n",
       "      <td>P,F</td>\n",
       "      <td>Acyl-</td>\n",
       "      <td>GO:0046872,GO:0045300,GO:0016491</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00220,ko00511,ko00520,ko00531,ko00604,ko01230</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>EgFAB2_5</td>\n",
       "      <td>p5.00_sc00229_p0026</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>13185at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>224.254</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0045300,GO:0016491,GO:0046872</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31</td>\n",
       "      <td>EgFAB2_6</td>\n",
       "      <td>p5.00_sc00078_p0031</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>13185at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>209.103</td>\n",
       "      <td>P,F</td>\n",
       "      <td>stearoyl-</td>\n",
       "      <td>GO:0045300,GO:0016491,GO:0046872</td>\n",
       "      <td>GO:0006631,GO:0006629,GO:0055114</td>\n",
       "      <td>1.14.19.-</td>\n",
       "      <td>ko04626</td>\n",
       "      <td>IPR005067,IPR005803,IPR009078,IPR012348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32</td>\n",
       "      <td>EgFAD2_1</td>\n",
       "      <td>p5.00_sc00037_p0124</td>\n",
       "      <td>Oleoyl-Phosphatidylcholine desaturase (FAD2)</td>\n",
       "      <td>EC 1.3.1.35</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>924at4447</td>\n",
       "      <td>1.450000e-262</td>\n",
       "      <td>234.350</td>\n",
       "      <td>I,X,C,P</td>\n",
       "      <td>Fatty acid desaturase DES2</td>\n",
       "      <td>GO:0016717</td>\n",
       "      <td>GO:0006629,GO:0055114</td>\n",
       "      <td>3.4.11.18</td>\n",
       "      <td>ko00561,ko03015,osa01100</td>\n",
       "      <td>IPR005804,IPR021863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33</td>\n",
       "      <td>EgFAD2_2</td>\n",
       "      <td>p5.00_sc00131_p0010</td>\n",
       "      <td>Oleoyl-Phosphatidylcholine desaturase (FAD2)</td>\n",
       "      <td>EC 1.3.1.35</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>924at4447</td>\n",
       "      <td>7.650000e-221</td>\n",
       "      <td>201.280</td>\n",
       "      <td>I,X,C,P</td>\n",
       "      <td>Fatty acid desaturase DES2</td>\n",
       "      <td>GO:0016717</td>\n",
       "      <td>GO:0006629,GO:0055114</td>\n",
       "      <td>3.4.11.18</td>\n",
       "      <td>ko00561,ko03015,osa01100</td>\n",
       "      <td>IPR005804,IPR021863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34</td>\n",
       "      <td>EgFAD3_1</td>\n",
       "      <td>p5.00_sc00121_p0039</td>\n",
       "      <td>Linoleoyl-Phosphatidylcholine desaturase (FAD3)</td>\n",
       "      <td>EC 1.14.19.3</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>112519at4447</td>\n",
       "      <td>9.120000e-256</td>\n",
       "      <td>200.890</td>\n",
       "      <td>I,X,C,P</td>\n",
       "      <td>omega-3 fatty acid desaturase, chloroplastic</td>\n",
       "      <td>GO:0016717</td>\n",
       "      <td>GO:0006629,GO:0055114</td>\n",
       "      <td>1.10.3.2</td>\n",
       "      <td>ko04075</td>\n",
       "      <td>IPR005804,IPR021863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>35</td>\n",
       "      <td>EgFAD3_2</td>\n",
       "      <td>p5.00_sc00033_p0032</td>\n",
       "      <td>Linoleoyl-Phosphatidylcholine desaturase (FAD3)</td>\n",
       "      <td>EC 1.14.19.3</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>112519at4447</td>\n",
       "      <td>5.240000e-247</td>\n",
       "      <td>197.190</td>\n",
       "      <td>I,X,C,P</td>\n",
       "      <td>omega-3 fatty acid desaturase, chloroplastic</td>\n",
       "      <td>GO:0016717</td>\n",
       "      <td>GO:0006629,GO:0055114</td>\n",
       "      <td>1.10.3.2</td>\n",
       "      <td>ko04075</td>\n",
       "      <td>IPR005804,IPR021863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36</td>\n",
       "      <td>EgFAD3_3</td>\n",
       "      <td>p5.00_sc00003_p0184</td>\n",
       "      <td>Linoleoyl-Phosphatidylcholine desaturase (FAD3)</td>\n",
       "      <td>EC 1.14.19.3</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>128946at4447</td>\n",
       "      <td>1.550000e-238</td>\n",
       "      <td>208.010</td>\n",
       "      <td>I,X,C,P</td>\n",
       "      <td>Fatty acid desaturase DES3</td>\n",
       "      <td>GO:0016717</td>\n",
       "      <td>GO:0006629,GO:0055114</td>\n",
       "      <td>2.4.1.-</td>\n",
       "      <td>ko00400,ko01230,ko04144,sbi01100,sbi01110</td>\n",
       "      <td>IPR005804,IPR021863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>37</td>\n",
       "      <td>EgFATB_1</td>\n",
       "      <td>p5.00_sc00018_p0248</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>148338at4447</td>\n",
       "      <td>3.890000e-287</td>\n",
       "      <td>240.340</td>\n",
       "      <td>I</td>\n",
       "      <td>palmitoyl-acyl carrier protein thioesterase, c...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>1.11.1.7</td>\n",
       "      <td>ko00940</td>\n",
       "      <td>IPR002864,IPR021113,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>38</td>\n",
       "      <td>EgFATB_2</td>\n",
       "      <td>p5.00_sc00039_p0052</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>119288at4447</td>\n",
       "      <td>2.710000e-218</td>\n",
       "      <td>185.800</td>\n",
       "      <td>I</td>\n",
       "      <td>Acyl-</td>\n",
       "      <td>GO:0016787,GO:0016790</td>\n",
       "      <td>GO:0006629,GO:0006631</td>\n",
       "      <td>2.3.2.27,3.1.2.-</td>\n",
       "      <td>ko04626,ko03018,ko04120,ko04141</td>\n",
       "      <td>IPR002864,IPR021113,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>EgFATB_3</td>\n",
       "      <td>p5.00_sc00104_p0027</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>119288at4447</td>\n",
       "      <td>2.020000e-273</td>\n",
       "      <td>226.680</td>\n",
       "      <td>I</td>\n",
       "      <td>Acyl-</td>\n",
       "      <td>GO:0016787,GO:0016790</td>\n",
       "      <td>GO:0006629,GO:0006631</td>\n",
       "      <td>2.3.2.27,3.1.2.-</td>\n",
       "      <td>ko04626,ko03018,ko04120,ko04141</td>\n",
       "      <td>IPR002864,IPR021113,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>EgFATB_4</td>\n",
       "      <td>p5.00_sc00052_p0027</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>124146at4447</td>\n",
       "      <td>8.110000e-241</td>\n",
       "      <td>224.340</td>\n",
       "      <td>I</td>\n",
       "      <td>palmitoyl-acyl carrier protein thioesterase, c...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>3.1.2.-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR002864,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>41</td>\n",
       "      <td>EgFATA_1</td>\n",
       "      <td>p5.00_sc00078_p0041</td>\n",
       "      <td>Oleoyl-ACP thioesterase (FATA)</td>\n",
       "      <td>EC 3.1.2.14</td>\n",
       "      <td>Acyl-ACP_TE (PF001643)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>132388at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>196.271</td>\n",
       "      <td>I</td>\n",
       "      <td>oleoyl-acyl carrier protein thioesterase, chlo...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>3.1.2.-</td>\n",
       "      <td>ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...</td>\n",
       "      <td>IPR002864,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>42</td>\n",
       "      <td>EgFATA_2</td>\n",
       "      <td>p5.00_sc00091_p0027</td>\n",
       "      <td>Oleoyl-ACP thioesterase (FATA)</td>\n",
       "      <td>EC 3.1.2.14</td>\n",
       "      <td>Acyl-ACP_TE (PF001643)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>132388at4447</td>\n",
       "      <td>3.570000e-207</td>\n",
       "      <td>202.200</td>\n",
       "      <td>I</td>\n",
       "      <td>oleoyl-acyl carrier protein thioesterase, chlo...</td>\n",
       "      <td>GO:0016790,GO:0016787</td>\n",
       "      <td>GO:0006631,GO:0006629</td>\n",
       "      <td>3.1.2.-</td>\n",
       "      <td>ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...</td>\n",
       "      <td>IPR002864,IPR029069,IPR045023</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     # OPFAB_gene_id              gene_id  \\\n",
       "0    1       EgACC_1  p5.00_sc00054_p0048   \n",
       "1    2       EgACC_2  p5.00_sc00095_p0046   \n",
       "2    3        EgCT_1  p5.00_sc00197_p0009   \n",
       "3    4        EgCT_2  p5.00_sc00020_p0225   \n",
       "4    5        EgCT_3  p5.00_sc00017_p0152   \n",
       "5    6      EgBCCP_1  p5.00_sc00011_p0007   \n",
       "6    7      EgBCCP_2  p5.00_sc00227_p0009   \n",
       "7    8        EgBC_1  p5.00_sc00019_p0245   \n",
       "8    9        EgBC_2  p5.00_sc00077_p0033   \n",
       "9   10      EgFABD_1  p5.00_sc00007_p0027   \n",
       "10  11      EgFABH_1  p5.00_sc00035_p0050   \n",
       "11  12      EgFABH_2  p5.00_sc00049_p0018   \n",
       "12  13      EgFABB_1  p5.00_sc00001_p0491   \n",
       "13  14      EgFABB_2  p5.00_sc00004_p0248   \n",
       "14  15      EgFABB_3  p5.00_sc00064_p0074   \n",
       "15  16      EgFABB_4  p5.00_sc00064_p0077   \n",
       "16  17      EgFABF_1  p5.00_sc00004_p0242   \n",
       "17  18      EgFABF_2  p5.00_sc00036_p0047   \n",
       "18  19      EgFABF_3  p5.00_sc00035_p0142   \n",
       "19  20      EgFABG_1  p5.00_sc00006_p0025   \n",
       "20  21      EgFABG_2  p5.00_sc00034_p0180   \n",
       "21  22      EgFABZ_1  p5.00_sc00271_p0017   \n",
       "22  23      EgFABZ_2  p5.00_sc00012_p0164   \n",
       "23  24      EgFABI_1  p5.00_sc00037_p0126   \n",
       "24  25      EgFABI_2  p5.00_sc00131_p0012   \n",
       "25  26      EgFAB2_1  p5.00_sc00091_p0032   \n",
       "26  27      EgFAB2_2  p5.00_sc00051_p0096   \n",
       "27  28      EgFAB2_3  p5.00_sc00125_p0015   \n",
       "28  29      EgFAB2_4  p5.00_sc00147_p0015   \n",
       "29  30      EgFAB2_5  p5.00_sc00229_p0026   \n",
       "30  31      EgFAB2_6  p5.00_sc00078_p0031   \n",
       "31  32      EgFAD2_1  p5.00_sc00037_p0124   \n",
       "32  33      EgFAD2_2  p5.00_sc00131_p0010   \n",
       "33  34      EgFAD3_1  p5.00_sc00121_p0039   \n",
       "34  35      EgFAD3_2  p5.00_sc00033_p0032   \n",
       "35  36      EgFAD3_3  p5.00_sc00003_p0184   \n",
       "36  37      EgFATB_1  p5.00_sc00018_p0248   \n",
       "37  38      EgFATB_2  p5.00_sc00039_p0052   \n",
       "38  39      EgFATB_3  p5.00_sc00104_p0027   \n",
       "39  40      EgFATB_4  p5.00_sc00052_p0027   \n",
       "40  41      EgFATA_1  p5.00_sc00078_p0041   \n",
       "41  42      EgFATA_2  p5.00_sc00091_p0027   \n",
       "\n",
       "                                               enzyme        ec_num  \\\n",
       "0   Multifunctional form of Acetyl-CoA carboxylase...    EC 6.4.1.2   \n",
       "1   Multifunctional form of Acetyl-CoA carboxylase...    EC 6.4.1.2   \n",
       "2    Multisubunit form of Acetyl-CoA carboxylase (CT)             -   \n",
       "3    Multisubunit form of Acetyl-CoA carboxylase (CT)             -   \n",
       "4    Multisubunit form of Acetyl-CoA carboxylase (CT)             -   \n",
       "5   Multisubunit form of Acetyl-CoA carboxylase (B...             -   \n",
       "6   Multisubunit form of Acetyl-CoA carboxylase (B...             -   \n",
       "7    Multisubunit form of Acetyl-CoA carboxylase (BC)             -   \n",
       "8    Multisubunit form of Acetyl-CoA carboxylase (BC)             -   \n",
       "9                  Malonyl-CoA:ACPtransacylase (FABD)   EC 2.3.1.39   \n",
       "10              Beta-ketoacyl-ACP synthase III (FABH)  EC 2.3.1.180   \n",
       "11              Beta-ketoacyl-ACP synthase III (FABH)  EC 2.3.1.180   \n",
       "12                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "13                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "14                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "15                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "16               Beta-ketoacyl-ACP synthase II (FABF)  EC 2.3.1.179   \n",
       "17               Beta-ketoacyl-ACP synthase II (FABF)  EC 2.3.1.179   \n",
       "18               Beta-ketoacyl-ACP synthase II (FABF)  EC 2.3.1.179   \n",
       "19                 Beta-ketoacyl-ACP reductase (FABG)  EC 1.1.1.100   \n",
       "20                 Beta-ketoacyl-ACP reductase (FABG)  EC 1.1.1.100   \n",
       "21          Beta-hydroxyacyl-ACP dehydrogenase (FABZ)   EC 4.2.1.59   \n",
       "22          Beta-hydroxyacyl-ACP dehydrogenase (FABZ)   EC 4.2.1.59   \n",
       "23                         Enoyl-ACP reductase (FABI)    EC 1.3.1.9   \n",
       "24                         Enoyl-ACP reductase (FABI)    EC 1.3.1.9   \n",
       "25                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "26                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "27                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "28                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "29                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "30                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "31       Oleoyl-Phosphatidylcholine desaturase (FAD2)   EC 1.3.1.35   \n",
       "32       Oleoyl-Phosphatidylcholine desaturase (FAD2)   EC 1.3.1.35   \n",
       "33    Linoleoyl-Phosphatidylcholine desaturase (FAD3)  EC 1.14.19.3   \n",
       "34    Linoleoyl-Phosphatidylcholine desaturase (FAD3)  EC 1.14.19.3   \n",
       "35    Linoleoyl-Phosphatidylcholine desaturase (FAD3)  EC 1.14.19.3   \n",
       "36                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "37                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "38                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "39                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "40                     Oleoyl-ACP thioesterase (FATA)   EC 3.1.2.14   \n",
       "41                     Oleoyl-ACP thioesterase (FATA)   EC 3.1.2.14   \n",
       "\n",
       "                                          pfam_domain  \\\n",
       "0   ACC_central (PF08326), Carboxyl_trans (PF01039...   \n",
       "1   CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...   \n",
       "2                                      ACCA (PF03255)   \n",
       "3                                      ACCA (PF03255)   \n",
       "4                                      ACCA (PF03255)   \n",
       "5                             Biotin_lipoyl (PF00364)   \n",
       "6                             Biotin_lipoyl (PF00364)   \n",
       "7   CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...   \n",
       "8   CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...   \n",
       "9                             Acyl_transf_1 (PF00698)   \n",
       "10     ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)   \n",
       "11     ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)   \n",
       "12  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "13  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "14  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "15  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "16  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "17  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "18  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "19                                adh_short (PF00106)   \n",
       "20                                adh_short (PF00106)   \n",
       "21                                     FabA (PF07977)   \n",
       "22                                     FabA (PF07977)   \n",
       "23                             Adh_short_C2 (PF13561)   \n",
       "24                             Adh_short_C2 (PF13561)   \n",
       "25                         FA_desaturase_C2 (PF03405)   \n",
       "26                         FA_desaturase_C2 (PF03405)   \n",
       "27                         FA_desaturase_C2 (PF03405)   \n",
       "28                         FA_desaturase_C2 (PF03405)   \n",
       "29                         FA_desaturase_C2 (PF03405)   \n",
       "30                         FA_desaturase_C2 (PF03405)   \n",
       "31         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "32         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "33         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "34         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "35         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "36      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "37      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "38      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "39      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "40                             Acyl-ACP_TE (PF001643)   \n",
       "41                             Acyl-ACP_TE (PF001643)   \n",
       "\n",
       "                      catalytic_residue  \\\n",
       "0                                     -   \n",
       "1                                     -   \n",
       "2                                     -   \n",
       "3                                     -   \n",
       "4                                     -   \n",
       "5                           Ser-Met-Lys   \n",
       "6                           Ser-Met-Lys   \n",
       "7   His-Glu, His-Glu, Glu, Ile-Glu, Arg   \n",
       "8   His-Glu, His-Glu, Glu, Ile-Glu, Arg   \n",
       "9                   Gln-Ser-Arg-His-Gln   \n",
       "10                      Cys-His-Arg-Asn   \n",
       "11                      Cys-His-Arg-Asn   \n",
       "12                          Cys-His-His   \n",
       "13                          Cys-His-His   \n",
       "14                          Cys-His-His   \n",
       "15                          Cys-His-His   \n",
       "16                          Cys-His-His   \n",
       "17                          Cys-His-His   \n",
       "18                          Cys-His-His   \n",
       "19                  Arg-Ser-Tyr-Lys-Arg   \n",
       "20                  Arg-Ser-Tyr-Lys-Arg   \n",
       "21                              His-Asp   \n",
       "22                              His-Asp   \n",
       "23                          Tyr-Tyr-Lys   \n",
       "24                          Tyr-Tyr-Lys   \n",
       "25                                    -   \n",
       "26                                    -   \n",
       "27                                    -   \n",
       "28                                    -   \n",
       "29                                    -   \n",
       "30                                    -   \n",
       "31                                    -   \n",
       "32                                    -   \n",
       "33                                    -   \n",
       "34                                    -   \n",
       "35                                    -   \n",
       "36                          Asn-His-Cys   \n",
       "37                          Asn-His-Cys   \n",
       "38                          Asn-His-Cys   \n",
       "39                          Asn-His-Cys   \n",
       "40                          Asn-His-Cys   \n",
       "41                          Asn-His-Cys   \n",
       "\n",
       "                                        motif                    reference  \\\n",
       "0                                           -                            -   \n",
       "1                                           -                            -   \n",
       "2                                           -                            -   \n",
       "3                                           -                            -   \n",
       "4                                           -                            -   \n",
       "5                     (A/I/V)-M-K-(L/V/M/A/T)           Barber et al. 2005   \n",
       "6                     (A/I/V)-M-K-(L/V/M/A/T)           Barber et al. 2005   \n",
       "7                                           -          Waldrop et al. 1994   \n",
       "8                                           -          Waldrop et al. 1994   \n",
       "9                                     GLSLGEY               Li et al. 2009   \n",
       "10                                    GNTSAAS               Li et al. 2009   \n",
       "11                                    GNTSAAS               Li et al. 2009   \n",
       "12                                          -               Li et al. 2009   \n",
       "13                                          -               Li et al. 2009   \n",
       "14                                          -               Li et al. 2009   \n",
       "15                                          -               Li et al. 2009   \n",
       "16                                          -               Li et al. 2009   \n",
       "17                                          -               Li et al. 2009   \n",
       "18                                          -               Li et al. 2009   \n",
       "19                                       YX3K               Li et al. 2009   \n",
       "20                                       YX3K               Li et al. 2009   \n",
       "21                              LPHRFPFLLVDRV               Li et al. 2009   \n",
       "22                              LPHRFPFLLVDRV               Li et al. 2009   \n",
       "23                                       YX6K               Li et al. 2009   \n",
       "24                                       YX6K               Li et al. 2009   \n",
       "25                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "26                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "27                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "28                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "29                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "30                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "31  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "32  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "33  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "34  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "35  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "36                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "37                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "38                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "39                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "40                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "41                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "\n",
       "          ODB_OG         evalue    score COG_category  \\\n",
       "0   112963at4447   0.000000e+00  242.970      I,C,E,H   \n",
       "1   118148at4447   0.000000e+00  222.700      I,C,E,H   \n",
       "2   124466at4447   0.000000e+00  175.034          I,L   \n",
       "3   124466at4447   0.000000e+00  194.260          I,L   \n",
       "4   124466at4447  3.500000e-306  151.060          I,L   \n",
       "5   136974at4447   0.000000e+00  109.859      C,I,H,E   \n",
       "6   136974at4447  2.570000e-108  141.720      C,I,H,E   \n",
       "7   118678at4447   0.000000e+00  235.260      I,E,C,F   \n",
       "8   118678at4447   0.000000e+00  238.430      I,E,C,F   \n",
       "9   133140at4447  1.030000e-214  191.840        Q,I,C   \n",
       "10  151244at4447   6.400000e-27   46.840            -   \n",
       "11  119951at4447  1.630000e-246  215.490        I,Q,C   \n",
       "12  117154at4447   0.000000e+00  255.320        Q,I,C   \n",
       "13   38560at4447  4.240000e-285  215.240        Q,I,C   \n",
       "14   38560at4447   0.000000e+00  204.786        Q,I,C   \n",
       "15   38560at4447  2.740000e-283  214.180        Q,I,C   \n",
       "16   48534at4447  8.250000e-305  191.050        Q,I,C   \n",
       "17   48534at4447  2.230000e-263  169.030        Q,I,C   \n",
       "18   48534at4447  8.250000e-266  170.470        Q,I,C   \n",
       "19  134714at4447   0.000000e+00  139.587            -   \n",
       "20  134714at4447  1.290000e-167  178.930            -   \n",
       "21   64848at4447  2.950000e-155  230.960      I,Q,M,C   \n",
       "22   64848at4447  6.000000e-105  169.670      I,Q,M,C   \n",
       "23  132738at4447  2.090000e-237  217.810            -   \n",
       "24  132738at4447   0.000000e+00  177.550            -   \n",
       "25   13185at4447  1.710000e-234  222.180          P,F   \n",
       "26  129798at4447  4.040000e-226  207.920          P,F   \n",
       "27  198320at3193  3.340000e-105  184.990          P,F   \n",
       "28  122758at4447  4.450000e-298  259.070          P,F   \n",
       "29   13185at4447   0.000000e+00  224.254          P,F   \n",
       "30   13185at4447   0.000000e+00  209.103          P,F   \n",
       "31     924at4447  1.450000e-262  234.350      I,X,C,P   \n",
       "32     924at4447  7.650000e-221  201.280      I,X,C,P   \n",
       "33  112519at4447  9.120000e-256  200.890      I,X,C,P   \n",
       "34  112519at4447  5.240000e-247  197.190      I,X,C,P   \n",
       "35  128946at4447  1.550000e-238  208.010      I,X,C,P   \n",
       "36  148338at4447  3.890000e-287  240.340            I   \n",
       "37  119288at4447  2.710000e-218  185.800            I   \n",
       "38  119288at4447  2.020000e-273  226.680            I   \n",
       "39  124146at4447  8.110000e-241  224.340            I   \n",
       "40  132388at4447   0.000000e+00  196.271            I   \n",
       "41  132388at4447  3.570000e-207  202.200            I   \n",
       "\n",
       "                                          Description  \\\n",
       "0                              Acetyl-CoA carboxylase   \n",
       "1   Methylcrotonoyl-CoA carboxylase subunit alpha,...   \n",
       "2   acetyl-coenzyme A carboxylase carboxyl transfe...   \n",
       "3   acetyl-coenzyme A carboxylase carboxyl transfe...   \n",
       "4   acetyl-coenzyme A carboxylase carboxyl transfe...   \n",
       "5   Biotin carboxyl carrier protein of acetyl-CoA ...   \n",
       "6   Biotin carboxyl carrier protein of acetyl-CoA ...   \n",
       "7                                  Biotin carboxylase   \n",
       "8                                  Biotin carboxylase   \n",
       "9       malonyl CoA-acyl carrier protein transacylase   \n",
       "10                                         3-oxoacyl-   \n",
       "11                                         3-oxoacyl-   \n",
       "12                                         3-oxoacyl-   \n",
       "13                                         3-oxoacyl-   \n",
       "14                                         3-oxoacyl-   \n",
       "15                                         3-oxoacyl-   \n",
       "16                                         3-oxoacyl-   \n",
       "17                                         3-oxoacyl-   \n",
       "18                                         3-oxoacyl-   \n",
       "19                                         3-oxoacyl-   \n",
       "20                                         3-oxoacyl-   \n",
       "21                             (3R)-hydroxymyristoyl-   \n",
       "22                             (3R)-hydroxymyristoyl-   \n",
       "23                                             enoyl-   \n",
       "24                                             enoyl-   \n",
       "25                                          stearoyl-   \n",
       "26                                          Stearoyl-   \n",
       "27                                          stearoyl-   \n",
       "28                                              Acyl-   \n",
       "29                                          stearoyl-   \n",
       "30                                          stearoyl-   \n",
       "31                         Fatty acid desaturase DES2   \n",
       "32                         Fatty acid desaturase DES2   \n",
       "33       omega-3 fatty acid desaturase, chloroplastic   \n",
       "34       omega-3 fatty acid desaturase, chloroplastic   \n",
       "35                         Fatty acid desaturase DES3   \n",
       "36  palmitoyl-acyl carrier protein thioesterase, c...   \n",
       "37                                              Acyl-   \n",
       "38                                              Acyl-   \n",
       "39  palmitoyl-acyl carrier protein thioesterase, c...   \n",
       "40  oleoyl-acyl carrier protein thioesterase, chlo...   \n",
       "41  oleoyl-acyl carrier protein thioesterase, chlo...   \n",
       "\n",
       "                                               GOs_mf  \\\n",
       "0   GO:0016874,GO:0005524,GO:0000166,GO:0046872,GO...   \n",
       "1         GO:0046872,GO:0016874,GO:0005524,GO:0000166   \n",
       "2   GO:0016874,GO:0016740,GO:0005524,GO:0003989,GO...   \n",
       "3   GO:0016874,GO:0016740,GO:0005524,GO:0003989,GO...   \n",
       "4   GO:0016874,GO:0016740,GO:0005524,GO:0003989,GO...   \n",
       "5                                          GO:0003989   \n",
       "6                                          GO:0003989   \n",
       "7   GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...   \n",
       "8   GO:0046872,GO:0016874,GO:0005524,GO:0004075,GO...   \n",
       "9                    GO:0016740,GO:0004314,GO:0016419   \n",
       "10                                                  -   \n",
       "11                              GO:0016740,GO:0004315   \n",
       "12                   GO:0016747,GO:0016740,GO:0004315   \n",
       "13                   GO:0016747,GO:0016740,GO:0004315   \n",
       "14                   GO:0016747,GO:0016740,GO:0004315   \n",
       "15                   GO:0016747,GO:0016740,GO:0004315   \n",
       "16                   GO:0016740,GO:0004315,GO:0016747   \n",
       "17                   GO:0016740,GO:0004315,GO:0016747   \n",
       "18                   GO:0016740,GO:0004315,GO:0016747   \n",
       "19        GO:0016491,GO:0004316,GO:0102131,GO:0102132   \n",
       "20        GO:0016491,GO:0004316,GO:0102131,GO:0102132   \n",
       "21  GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...   \n",
       "22  GO:0047451,GO:0019171,GO:0016836,GO:0016829,GO...   \n",
       "23                              GO:0004318,GO:0016491   \n",
       "24                              GO:0004318,GO:0016491   \n",
       "25                   GO:0045300,GO:0016491,GO:0046872   \n",
       "26                   GO:0046872,GO:0045300,GO:0016491   \n",
       "27                              GO:0016491,GO:0045300   \n",
       "28                   GO:0046872,GO:0045300,GO:0016491   \n",
       "29                   GO:0045300,GO:0016491,GO:0046872   \n",
       "30                   GO:0045300,GO:0016491,GO:0046872   \n",
       "31                                         GO:0016717   \n",
       "32                                         GO:0016717   \n",
       "33                                         GO:0016717   \n",
       "34                                         GO:0016717   \n",
       "35                                         GO:0016717   \n",
       "36                              GO:0016790,GO:0016787   \n",
       "37                              GO:0016787,GO:0016790   \n",
       "38                              GO:0016787,GO:0016790   \n",
       "39                              GO:0016790,GO:0016787   \n",
       "40                              GO:0016790,GO:0016787   \n",
       "41                              GO:0016790,GO:0016787   \n",
       "\n",
       "                              GOs_bp                         EC  \\\n",
       "0   GO:2001295,GO:0006631,GO:0006629                          -   \n",
       "1                                  -                          -   \n",
       "2   GO:2001295,GO:0006631,GO:0006629                          -   \n",
       "3   GO:2001295,GO:0006631,GO:0006629                          -   \n",
       "4   GO:2001295,GO:0006631,GO:0006629                          -   \n",
       "5              GO:0006631,GO:0006629                          -   \n",
       "6              GO:0006631,GO:0006629                          -   \n",
       "7   GO:2001295,GO:0006631,GO:0006629           6.3.4.14,6.4.1.2   \n",
       "8   GO:2001295,GO:0006631,GO:0006629           6.3.4.14,6.4.1.2   \n",
       "9                                  -                          -   \n",
       "10                                 -                          -   \n",
       "11                                 -                          -   \n",
       "12             GO:0006631,GO:0006629                    1.2.4.1   \n",
       "13             GO:0006631,GO:0006629                    1.1.1.-   \n",
       "14             GO:0006631,GO:0006629                    1.1.1.-   \n",
       "15             GO:0006631,GO:0006629                    1.1.1.-   \n",
       "16             GO:0006631,GO:0006629                   1.11.1.7   \n",
       "17             GO:0006631,GO:0006629                   1.11.1.7   \n",
       "18             GO:0006631,GO:0006629                   1.11.1.7   \n",
       "19  GO:0006629,GO:0006631,GO:0055114                          -   \n",
       "20  GO:0006629,GO:0006631,GO:0055114                          -   \n",
       "21             GO:0009245,GO:0006629                          -   \n",
       "22             GO:0009245,GO:0006629                          -   \n",
       "23                        GO:0055114  3.4.19.12,2.4.2.-,3.1.3.2   \n",
       "24                        GO:0055114  3.4.19.12,2.4.2.-,3.1.3.2   \n",
       "25  GO:0006631,GO:0006629,GO:0055114                  1.14.19.-   \n",
       "26  GO:0006631,GO:0006629,GO:0055114                   1.1.1.44   \n",
       "27             GO:0006629,GO:0006631                  1.14.19.-   \n",
       "28             GO:0006631,GO:0006629                          -   \n",
       "29  GO:0006631,GO:0006629,GO:0055114                  1.14.19.-   \n",
       "30  GO:0006631,GO:0006629,GO:0055114                  1.14.19.-   \n",
       "31             GO:0006629,GO:0055114                  3.4.11.18   \n",
       "32             GO:0006629,GO:0055114                  3.4.11.18   \n",
       "33             GO:0006629,GO:0055114                   1.10.3.2   \n",
       "34             GO:0006629,GO:0055114                   1.10.3.2   \n",
       "35             GO:0006629,GO:0055114                    2.4.1.-   \n",
       "36             GO:0006631,GO:0006629                   1.11.1.7   \n",
       "37             GO:0006629,GO:0006631           2.3.2.27,3.1.2.-   \n",
       "38             GO:0006629,GO:0006631           2.3.2.27,3.1.2.-   \n",
       "39             GO:0006631,GO:0006629                    3.1.2.-   \n",
       "40             GO:0006631,GO:0006629                    3.1.2.-   \n",
       "41             GO:0006631,GO:0006629                    3.1.2.-   \n",
       "\n",
       "                                              KEGG_ko  \\\n",
       "0                                                   -   \n",
       "1                                     ko00860,ko04144   \n",
       "2                                                   -   \n",
       "3                                                   -   \n",
       "4                                                   -   \n",
       "5                                             ko04141   \n",
       "6                                             ko04141   \n",
       "7                                                   -   \n",
       "8                                                   -   \n",
       "9                                    bdi01110,ko00901   \n",
       "10                           ko00061,ko01212,zma01100   \n",
       "11  ko00062,ko00250,ko00260,ko00270,ko00280,ko0201...   \n",
       "12                    ko00010,ko00020,ko00620,ko01200   \n",
       "13        ko00330,ko00592,sbi01100,sbi01110,sita01100   \n",
       "14        ko00330,ko00592,sbi01100,sbi01110,sita01100   \n",
       "15        ko00330,ko00592,sbi01100,sbi01110,sita01100   \n",
       "16                                    ko03010,ko04144   \n",
       "17                                    ko03010,ko04144   \n",
       "18                                    ko03010,ko04144   \n",
       "19                            ko00260,ko03010,ko03013   \n",
       "20                            ko00260,ko03010,ko03013   \n",
       "21  ko00230,ko00730,ko04075,ko04626,zma01100,zma01110   \n",
       "22  ko00230,ko00730,ko04075,ko04626,zma01100,zma01110   \n",
       "23                                            ko04626   \n",
       "24                                            ko04626   \n",
       "25                                            ko04626   \n",
       "26                    ko00030,ko00195,ko00480,ko01200   \n",
       "27                                            ko03010   \n",
       "28    ko00220,ko00511,ko00520,ko00531,ko00604,ko01230   \n",
       "29                                            ko04626   \n",
       "30                                            ko04626   \n",
       "31                           ko00561,ko03015,osa01100   \n",
       "32                           ko00561,ko03015,osa01100   \n",
       "33                                            ko04075   \n",
       "34                                            ko04075   \n",
       "35          ko00400,ko01230,ko04144,sbi01100,sbi01110   \n",
       "36                                            ko00940   \n",
       "37                    ko04626,ko03018,ko04120,ko04141   \n",
       "38                    ko04626,ko03018,ko04120,ko04141   \n",
       "39                                                  -   \n",
       "40  ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...   \n",
       "41  ko00053,ko00260,ko00561,ko00564,ko00630,ko0120...   \n",
       "\n",
       "                                             Interpro  \n",
       "0   IPR000089,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "1   IPR000089,IPR001882,IPR005479,IPR005481,IPR005...  \n",
       "2                       IPR001095,IPR011763,IPR029045  \n",
       "3                       IPR001095,IPR011763,IPR029045  \n",
       "4                       IPR001095,IPR011763,IPR029045  \n",
       "5             IPR000089,IPR001249,IPR001882,IPR011053  \n",
       "6             IPR000089,IPR001249,IPR001882,IPR011053  \n",
       "7   IPR004549,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "8   IPR004549,IPR005479,IPR005481,IPR005482,IPR011...  \n",
       "9   IPR001227,IPR004410,IPR014043,IPR016035,IPR016036  \n",
       "10                                                  -  \n",
       "11            IPR004655,IPR013747,IPR013751,IPR016039  \n",
       "12  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "13  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "14  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "15  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "16  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "17  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "18  IPR000794,IPR014030,IPR014031,IPR016039,IPR017...  \n",
       "19            IPR002347,IPR011284,IPR020904,IPR036291  \n",
       "20            IPR002347,IPR011284,IPR020904,IPR036291  \n",
       "21                      IPR010084,IPR013114,IPR029069  \n",
       "22                      IPR010084,IPR013114,IPR029069  \n",
       "23                      IPR002347,IPR014358,IPR036291  \n",
       "24                      IPR002347,IPR014358,IPR036291  \n",
       "25            IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "26                      IPR005067,IPR009078,IPR012348  \n",
       "27                      IPR005067,IPR009078,IPR012348  \n",
       "28            IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "29            IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "30            IPR005067,IPR005803,IPR009078,IPR012348  \n",
       "31                                IPR005804,IPR021863  \n",
       "32                                IPR005804,IPR021863  \n",
       "33                                IPR005804,IPR021863  \n",
       "34                                IPR005804,IPR021863  \n",
       "35                                IPR005804,IPR021863  \n",
       "36            IPR002864,IPR021113,IPR029069,IPR045023  \n",
       "37            IPR002864,IPR021113,IPR029069,IPR045023  \n",
       "38            IPR002864,IPR021113,IPR029069,IPR045023  \n",
       "39                      IPR002864,IPR029069,IPR045023  \n",
       "40                      IPR002864,IPR029069,IPR045023  \n",
       "41                      IPR002864,IPR029069,IPR045023  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FAB_genes_og"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "ename": "UnboundLocalError",
     "evalue": "cannot access local variable 'lidx' where it is not associated with a value",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[80], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmerge\u001b[49m\u001b[43m(\u001b[49m\u001b[43mat_anno_B\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mFAB_genes_og\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mon\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mEG5.1\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhow\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43moutter\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/jupyter/lib/python3.12/site-packages/pandas/core/reshape/merge.py:184\u001b[0m, in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m    169\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    170\u001b[0m     op \u001b[38;5;241m=\u001b[39m _MergeOperation(\n\u001b[1;32m    171\u001b[0m         left_df,\n\u001b[1;32m    172\u001b[0m         right_df,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    182\u001b[0m         validate\u001b[38;5;241m=\u001b[39mvalidate,\n\u001b[1;32m    183\u001b[0m     )\n\u001b[0;32m--> 184\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mop\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_result\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/jupyter/lib/python3.12/site-packages/pandas/core/reshape/merge.py:886\u001b[0m, in \u001b[0;36m_MergeOperation.get_result\u001b[0;34m(self, copy)\u001b[0m\n\u001b[1;32m    883\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindicator:\n\u001b[1;32m    884\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mleft, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mright \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_indicator_pre_merge(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mleft, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mright)\n\u001b[0;32m--> 886\u001b[0m join_index, left_indexer, right_indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_join_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    888\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_reindex_and_concat(\n\u001b[1;32m    889\u001b[0m     join_index, left_indexer, right_indexer, copy\u001b[38;5;241m=\u001b[39mcopy\n\u001b[1;32m    890\u001b[0m )\n\u001b[1;32m    891\u001b[0m result \u001b[38;5;241m=\u001b[39m result\u001b[38;5;241m.\u001b[39m__finalize__(\u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_merge_type)\n",
      "File \u001b[0;32m~/miniconda3/envs/jupyter/lib/python3.12/site-packages/pandas/core/reshape/merge.py:1151\u001b[0m, in \u001b[0;36m_MergeOperation._get_join_info\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1147\u001b[0m     join_index, right_indexer, left_indexer \u001b[38;5;241m=\u001b[39m _left_join_on_index(\n\u001b[1;32m   1148\u001b[0m         right_ax, left_ax, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mright_join_keys, sort\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msort\n\u001b[1;32m   1149\u001b[0m     )\n\u001b[1;32m   1150\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 1151\u001b[0m     (left_indexer, right_indexer) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_join_indexers\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1153\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mright_index:\n\u001b[1;32m   1154\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mleft) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "File \u001b[0;32m~/miniconda3/envs/jupyter/lib/python3.12/site-packages/pandas/core/reshape/merge.py:1125\u001b[0m, in \u001b[0;36m_MergeOperation._get_join_indexers\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1123\u001b[0m \u001b[38;5;66;03m# make mypy happy\u001b[39;00m\n\u001b[1;32m   1124\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhow \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124masof\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1125\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mget_join_indexers\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1126\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mleft_join_keys\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mright_join_keys\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msort\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msort\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhow\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhow\u001b[49m\n\u001b[1;32m   1127\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/jupyter/lib/python3.12/site-packages/pandas/core/reshape/merge.py:1759\u001b[0m, in \u001b[0;36mget_join_indexers\u001b[0;34m(left_keys, right_keys, sort, how)\u001b[0m\n\u001b[1;32m   1757\u001b[0m     _, lidx, ridx \u001b[38;5;241m=\u001b[39m left\u001b[38;5;241m.\u001b[39mjoin(right, how\u001b[38;5;241m=\u001b[39mhow, return_indexers\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, sort\u001b[38;5;241m=\u001b[39msort)\n\u001b[1;32m   1758\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 1759\u001b[0m     lidx, ridx \u001b[38;5;241m=\u001b[39m \u001b[43mget_join_indexers_non_unique\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1760\u001b[0m \u001b[43m        \u001b[49m\u001b[43mleft\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_values\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mright\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_values\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msort\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhow\u001b[49m\n\u001b[1;32m   1761\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1763\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m lidx \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m is_range_indexer(lidx, \u001b[38;5;28mlen\u001b[39m(left)):\n\u001b[1;32m   1764\u001b[0m     lidx \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/envs/jupyter/lib/python3.12/site-packages/pandas/core/reshape/merge.py:1802\u001b[0m, in \u001b[0;36mget_join_indexers_non_unique\u001b[0;34m(left, right, sort, how)\u001b[0m\n\u001b[1;32m   1800\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m how \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mouter\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m   1801\u001b[0m     lidx, ridx \u001b[38;5;241m=\u001b[39m libjoin\u001b[38;5;241m.\u001b[39mfull_outer_join(lkey, rkey, count)\n\u001b[0;32m-> 1802\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mlidx\u001b[49m, ridx\n",
      "\u001b[0;31mUnboundLocalError\u001b[0m: cannot access local variable 'lidx' where it is not associated with a value"
     ]
    }
   ],
   "source": [
    "df = pd.merge(at_anno_B, FAB_genes_og, on = 'EG5.1', how = 'outter')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#</th>\n",
       "      <th>OPFAB_gene_id</th>\n",
       "      <th>EG5.1</th>\n",
       "      <th>enzyme</th>\n",
       "      <th>ec_num</th>\n",
       "      <th>pfam_domain</th>\n",
       "      <th>catalytic_residue</th>\n",
       "      <th>motif</th>\n",
       "      <th>reference</th>\n",
       "      <th>ODB_OG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>EgACC_1</td>\n",
       "      <td>p5.00_sc00054_p0048</td>\n",
       "      <td>Multifunctional form of Acetyl-CoA carboxylase...</td>\n",
       "      <td>EC 6.4.1.2</td>\n",
       "      <td>ACC_central (PF08326), Carboxyl_trans (PF01039...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>112963at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>EgACC_2</td>\n",
       "      <td>p5.00_sc00095_p0046</td>\n",
       "      <td>Multifunctional form of Acetyl-CoA carboxylase...</td>\n",
       "      <td>EC 6.4.1.2</td>\n",
       "      <td>CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>118148at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>EgCT_1</td>\n",
       "      <td>p5.00_sc00197_p0009</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (CT)</td>\n",
       "      <td>-</td>\n",
       "      <td>ACCA (PF03255)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>EgCT_2</td>\n",
       "      <td>p5.00_sc00020_p0225</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (CT)</td>\n",
       "      <td>-</td>\n",
       "      <td>ACCA (PF03255)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>EgCT_3</td>\n",
       "      <td>p5.00_sc00017_p0152</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (CT)</td>\n",
       "      <td>-</td>\n",
       "      <td>ACCA (PF03255)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>EgBCCP_1</td>\n",
       "      <td>p5.00_sc00011_p0007</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (B...</td>\n",
       "      <td>-</td>\n",
       "      <td>Biotin_lipoyl (PF00364)</td>\n",
       "      <td>Ser-Met-Lys</td>\n",
       "      <td>(A/I/V)-M-K-(L/V/M/A/T)</td>\n",
       "      <td>Barber et al. 2005</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>EgBCCP_2</td>\n",
       "      <td>p5.00_sc00227_p0009</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (B...</td>\n",
       "      <td>-</td>\n",
       "      <td>Biotin_lipoyl (PF00364)</td>\n",
       "      <td>Ser-Met-Lys</td>\n",
       "      <td>(A/I/V)-M-K-(L/V/M/A/T)</td>\n",
       "      <td>Barber et al. 2005</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>EgBC_1</td>\n",
       "      <td>p5.00_sc00019_p0245</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (BC)</td>\n",
       "      <td>-</td>\n",
       "      <td>CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...</td>\n",
       "      <td>His-Glu, His-Glu, Glu, Ile-Glu, Arg</td>\n",
       "      <td>-</td>\n",
       "      <td>Waldrop et al. 1994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>EgBC_2</td>\n",
       "      <td>p5.00_sc00077_p0033</td>\n",
       "      <td>Multisubunit form of Acetyl-CoA carboxylase (BC)</td>\n",
       "      <td>-</td>\n",
       "      <td>CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...</td>\n",
       "      <td>His-Glu, His-Glu, Glu, Ile-Glu, Arg</td>\n",
       "      <td>-</td>\n",
       "      <td>Waldrop et al. 1994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>EgFABD_1</td>\n",
       "      <td>p5.00_sc00007_p0027</td>\n",
       "      <td>Malonyl-CoA:ACPtransacylase (FABD)</td>\n",
       "      <td>EC 2.3.1.39</td>\n",
       "      <td>Acyl_transf_1 (PF00698)</td>\n",
       "      <td>Gln-Ser-Arg-His-Gln</td>\n",
       "      <td>GLSLGEY</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>133140at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>EgFABH_1</td>\n",
       "      <td>p5.00_sc00035_p0050</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase III (FABH)</td>\n",
       "      <td>EC 2.3.1.180</td>\n",
       "      <td>ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)</td>\n",
       "      <td>Cys-His-Arg-Asn</td>\n",
       "      <td>GNTSAAS</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>EgFABH_2</td>\n",
       "      <td>p5.00_sc00049_p0018</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase III (FABH)</td>\n",
       "      <td>EC 2.3.1.180</td>\n",
       "      <td>ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)</td>\n",
       "      <td>Cys-His-Arg-Asn</td>\n",
       "      <td>GNTSAAS</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>119951at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>EgFABB_1</td>\n",
       "      <td>p5.00_sc00001_p0491</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>117154at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>EgFABB_2</td>\n",
       "      <td>p5.00_sc00004_p0248</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>EgFABB_3</td>\n",
       "      <td>p5.00_sc00064_p0074</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>EgFABB_4</td>\n",
       "      <td>p5.00_sc00064_p0077</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase I (FABB)</td>\n",
       "      <td>EC 2.3.1.41</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>EgFABF_1</td>\n",
       "      <td>p5.00_sc00004_p0242</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase II (FABF)</td>\n",
       "      <td>EC 2.3.1.179</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>EgFABF_2</td>\n",
       "      <td>p5.00_sc00036_p0047</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase II (FABF)</td>\n",
       "      <td>EC 2.3.1.179</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>EgFABF_3</td>\n",
       "      <td>p5.00_sc00035_p0142</td>\n",
       "      <td>Beta-ketoacyl-ACP synthase II (FABF)</td>\n",
       "      <td>EC 2.3.1.179</td>\n",
       "      <td>Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...</td>\n",
       "      <td>Cys-His-His</td>\n",
       "      <td>-</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>EgFABG_1</td>\n",
       "      <td>p5.00_sc00006_p0025</td>\n",
       "      <td>Beta-ketoacyl-ACP reductase (FABG)</td>\n",
       "      <td>EC 1.1.1.100</td>\n",
       "      <td>adh_short (PF00106)</td>\n",
       "      <td>Arg-Ser-Tyr-Lys-Arg</td>\n",
       "      <td>YX3K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>EgFABG_2</td>\n",
       "      <td>p5.00_sc00034_p0180</td>\n",
       "      <td>Beta-ketoacyl-ACP reductase (FABG)</td>\n",
       "      <td>EC 1.1.1.100</td>\n",
       "      <td>adh_short (PF00106)</td>\n",
       "      <td>Arg-Ser-Tyr-Lys-Arg</td>\n",
       "      <td>YX3K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>EgFABZ_1</td>\n",
       "      <td>p5.00_sc00271_p0017</td>\n",
       "      <td>Beta-hydroxyacyl-ACP dehydrogenase (FABZ)</td>\n",
       "      <td>EC 4.2.1.59</td>\n",
       "      <td>FabA (PF07977)</td>\n",
       "      <td>His-Asp</td>\n",
       "      <td>LPHRFPFLLVDRV</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>EgFABZ_2</td>\n",
       "      <td>p5.00_sc00012_p0164</td>\n",
       "      <td>Beta-hydroxyacyl-ACP dehydrogenase (FABZ)</td>\n",
       "      <td>EC 4.2.1.59</td>\n",
       "      <td>FabA (PF07977)</td>\n",
       "      <td>His-Asp</td>\n",
       "      <td>LPHRFPFLLVDRV</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>EgFABI_1</td>\n",
       "      <td>p5.00_sc00037_p0126</td>\n",
       "      <td>Enoyl-ACP reductase (FABI)</td>\n",
       "      <td>EC 1.3.1.9</td>\n",
       "      <td>Adh_short_C2 (PF13561)</td>\n",
       "      <td>Tyr-Tyr-Lys</td>\n",
       "      <td>YX6K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>EgFABI_2</td>\n",
       "      <td>p5.00_sc00131_p0012</td>\n",
       "      <td>Enoyl-ACP reductase (FABI)</td>\n",
       "      <td>EC 1.3.1.9</td>\n",
       "      <td>Adh_short_C2 (PF13561)</td>\n",
       "      <td>Tyr-Tyr-Lys</td>\n",
       "      <td>YX6K</td>\n",
       "      <td>Li et al. 2009</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>EgFAB2_1</td>\n",
       "      <td>p5.00_sc00091_p0032</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>EgFAB2_2</td>\n",
       "      <td>p5.00_sc00051_p0096</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>129798at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>EgFAB2_3</td>\n",
       "      <td>p5.00_sc00125_p0015</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>EgFAB2_4</td>\n",
       "      <td>p5.00_sc00147_p0015</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>122758at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>EgFAB2_5</td>\n",
       "      <td>p5.00_sc00229_p0026</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31</td>\n",
       "      <td>EgFAB2_6</td>\n",
       "      <td>p5.00_sc00078_p0031</td>\n",
       "      <td>Stearoyl-ACP desaturase (FAB2)</td>\n",
       "      <td>EC 1.14.19.2</td>\n",
       "      <td>FA_desaturase_C2 (PF03405)</td>\n",
       "      <td>-</td>\n",
       "      <td>EENRH and DEKRH</td>\n",
       "      <td>Haralampidis &amp; Milioni 1998</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32</td>\n",
       "      <td>EgFAD2_1</td>\n",
       "      <td>p5.00_sc00037_p0124</td>\n",
       "      <td>Oleoyl-Phosphatidylcholine desaturase (FAD2)</td>\n",
       "      <td>EC 1.3.1.35</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33</td>\n",
       "      <td>EgFAD2_2</td>\n",
       "      <td>p5.00_sc00131_p0010</td>\n",
       "      <td>Oleoyl-Phosphatidylcholine desaturase (FAD2)</td>\n",
       "      <td>EC 1.3.1.35</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34</td>\n",
       "      <td>EgFAD3_1</td>\n",
       "      <td>p5.00_sc00121_p0039</td>\n",
       "      <td>Linoleoyl-Phosphatidylcholine desaturase (FAD3)</td>\n",
       "      <td>EC 1.14.19.3</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>35</td>\n",
       "      <td>EgFAD3_2</td>\n",
       "      <td>p5.00_sc00033_p0032</td>\n",
       "      <td>Linoleoyl-Phosphatidylcholine desaturase (FAD3)</td>\n",
       "      <td>EC 1.14.19.3</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36</td>\n",
       "      <td>EgFAD3_3</td>\n",
       "      <td>p5.00_sc00003_p0184</td>\n",
       "      <td>Linoleoyl-Phosphatidylcholine desaturase (FAD3)</td>\n",
       "      <td>EC 1.14.19.3</td>\n",
       "      <td>FA_desaturase (PF00487), DUF3474 (PF11960)</td>\n",
       "      <td>-</td>\n",
       "      <td>Histidine boxes (HXXXH, HXXHH, and HXXHH)</td>\n",
       "      <td>Shanklin et al. 1994</td>\n",
       "      <td>128946at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>37</td>\n",
       "      <td>EgFATB_1</td>\n",
       "      <td>p5.00_sc00018_p0248</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>148338at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>38</td>\n",
       "      <td>EgFATB_2</td>\n",
       "      <td>p5.00_sc00039_p0052</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>EgFATB_3</td>\n",
       "      <td>p5.00_sc00104_p0027</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>EgFATB_4</td>\n",
       "      <td>p5.00_sc00052_p0027</td>\n",
       "      <td>Acyl-ACP thioesterase (FATB)</td>\n",
       "      <td>EC 3.1.2.-</td>\n",
       "      <td>Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>124146at4447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>41</td>\n",
       "      <td>EgFATA_1</td>\n",
       "      <td>p5.00_sc00078_p0041</td>\n",
       "      <td>Oleoyl-ACP thioesterase (FATA)</td>\n",
       "      <td>EC 3.1.2.14</td>\n",
       "      <td>Acyl-ACP_TE (PF001643)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>42</td>\n",
       "      <td>EgFATA_2</td>\n",
       "      <td>p5.00_sc00091_p0027</td>\n",
       "      <td>Oleoyl-ACP thioesterase (FATA)</td>\n",
       "      <td>EC 3.1.2.14</td>\n",
       "      <td>Acyl-ACP_TE (PF001643)</td>\n",
       "      <td>Asn-His-Cys</td>\n",
       "      <td>NQHVNN and YRRECG</td>\n",
       "      <td>Yuan et al. 1996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     # OPFAB_gene_id                EG5.1  \\\n",
       "0    1       EgACC_1  p5.00_sc00054_p0048   \n",
       "1    2       EgACC_2  p5.00_sc00095_p0046   \n",
       "2    3        EgCT_1  p5.00_sc00197_p0009   \n",
       "3    4        EgCT_2  p5.00_sc00020_p0225   \n",
       "4    5        EgCT_3  p5.00_sc00017_p0152   \n",
       "5    6      EgBCCP_1  p5.00_sc00011_p0007   \n",
       "6    7      EgBCCP_2  p5.00_sc00227_p0009   \n",
       "7    8        EgBC_1  p5.00_sc00019_p0245   \n",
       "8    9        EgBC_2  p5.00_sc00077_p0033   \n",
       "9   10      EgFABD_1  p5.00_sc00007_p0027   \n",
       "10  11      EgFABH_1  p5.00_sc00035_p0050   \n",
       "11  12      EgFABH_2  p5.00_sc00049_p0018   \n",
       "12  13      EgFABB_1  p5.00_sc00001_p0491   \n",
       "13  14      EgFABB_2  p5.00_sc00004_p0248   \n",
       "14  15      EgFABB_3  p5.00_sc00064_p0074   \n",
       "15  16      EgFABB_4  p5.00_sc00064_p0077   \n",
       "16  17      EgFABF_1  p5.00_sc00004_p0242   \n",
       "17  18      EgFABF_2  p5.00_sc00036_p0047   \n",
       "18  19      EgFABF_3  p5.00_sc00035_p0142   \n",
       "19  20      EgFABG_1  p5.00_sc00006_p0025   \n",
       "20  21      EgFABG_2  p5.00_sc00034_p0180   \n",
       "21  22      EgFABZ_1  p5.00_sc00271_p0017   \n",
       "22  23      EgFABZ_2  p5.00_sc00012_p0164   \n",
       "23  24      EgFABI_1  p5.00_sc00037_p0126   \n",
       "24  25      EgFABI_2  p5.00_sc00131_p0012   \n",
       "25  26      EgFAB2_1  p5.00_sc00091_p0032   \n",
       "26  27      EgFAB2_2  p5.00_sc00051_p0096   \n",
       "27  28      EgFAB2_3  p5.00_sc00125_p0015   \n",
       "28  29      EgFAB2_4  p5.00_sc00147_p0015   \n",
       "29  30      EgFAB2_5  p5.00_sc00229_p0026   \n",
       "30  31      EgFAB2_6  p5.00_sc00078_p0031   \n",
       "31  32      EgFAD2_1  p5.00_sc00037_p0124   \n",
       "32  33      EgFAD2_2  p5.00_sc00131_p0010   \n",
       "33  34      EgFAD3_1  p5.00_sc00121_p0039   \n",
       "34  35      EgFAD3_2  p5.00_sc00033_p0032   \n",
       "35  36      EgFAD3_3  p5.00_sc00003_p0184   \n",
       "36  37      EgFATB_1  p5.00_sc00018_p0248   \n",
       "37  38      EgFATB_2  p5.00_sc00039_p0052   \n",
       "38  39      EgFATB_3  p5.00_sc00104_p0027   \n",
       "39  40      EgFATB_4  p5.00_sc00052_p0027   \n",
       "40  41      EgFATA_1  p5.00_sc00078_p0041   \n",
       "41  42      EgFATA_2  p5.00_sc00091_p0027   \n",
       "\n",
       "                                               enzyme        ec_num  \\\n",
       "0   Multifunctional form of Acetyl-CoA carboxylase...    EC 6.4.1.2   \n",
       "1   Multifunctional form of Acetyl-CoA carboxylase...    EC 6.4.1.2   \n",
       "2    Multisubunit form of Acetyl-CoA carboxylase (CT)             -   \n",
       "3    Multisubunit form of Acetyl-CoA carboxylase (CT)             -   \n",
       "4    Multisubunit form of Acetyl-CoA carboxylase (CT)             -   \n",
       "5   Multisubunit form of Acetyl-CoA carboxylase (B...             -   \n",
       "6   Multisubunit form of Acetyl-CoA carboxylase (B...             -   \n",
       "7    Multisubunit form of Acetyl-CoA carboxylase (BC)             -   \n",
       "8    Multisubunit form of Acetyl-CoA carboxylase (BC)             -   \n",
       "9                  Malonyl-CoA:ACPtransacylase (FABD)   EC 2.3.1.39   \n",
       "10              Beta-ketoacyl-ACP synthase III (FABH)  EC 2.3.1.180   \n",
       "11              Beta-ketoacyl-ACP synthase III (FABH)  EC 2.3.1.180   \n",
       "12                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "13                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "14                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "15                Beta-ketoacyl-ACP synthase I (FABB)   EC 2.3.1.41   \n",
       "16               Beta-ketoacyl-ACP synthase II (FABF)  EC 2.3.1.179   \n",
       "17               Beta-ketoacyl-ACP synthase II (FABF)  EC 2.3.1.179   \n",
       "18               Beta-ketoacyl-ACP synthase II (FABF)  EC 2.3.1.179   \n",
       "19                 Beta-ketoacyl-ACP reductase (FABG)  EC 1.1.1.100   \n",
       "20                 Beta-ketoacyl-ACP reductase (FABG)  EC 1.1.1.100   \n",
       "21          Beta-hydroxyacyl-ACP dehydrogenase (FABZ)   EC 4.2.1.59   \n",
       "22          Beta-hydroxyacyl-ACP dehydrogenase (FABZ)   EC 4.2.1.59   \n",
       "23                         Enoyl-ACP reductase (FABI)    EC 1.3.1.9   \n",
       "24                         Enoyl-ACP reductase (FABI)    EC 1.3.1.9   \n",
       "25                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "26                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "27                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "28                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "29                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "30                     Stearoyl-ACP desaturase (FAB2)  EC 1.14.19.2   \n",
       "31       Oleoyl-Phosphatidylcholine desaturase (FAD2)   EC 1.3.1.35   \n",
       "32       Oleoyl-Phosphatidylcholine desaturase (FAD2)   EC 1.3.1.35   \n",
       "33    Linoleoyl-Phosphatidylcholine desaturase (FAD3)  EC 1.14.19.3   \n",
       "34    Linoleoyl-Phosphatidylcholine desaturase (FAD3)  EC 1.14.19.3   \n",
       "35    Linoleoyl-Phosphatidylcholine desaturase (FAD3)  EC 1.14.19.3   \n",
       "36                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "37                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "38                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "39                       Acyl-ACP thioesterase (FATB)    EC 3.1.2.-   \n",
       "40                     Oleoyl-ACP thioesterase (FATA)   EC 3.1.2.14   \n",
       "41                     Oleoyl-ACP thioesterase (FATA)   EC 3.1.2.14   \n",
       "\n",
       "                                          pfam_domain  \\\n",
       "0   ACC_central (PF08326), Carboxyl_trans (PF01039...   \n",
       "1   CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...   \n",
       "2                                      ACCA (PF03255)   \n",
       "3                                      ACCA (PF03255)   \n",
       "4                                      ACCA (PF03255)   \n",
       "5                             Biotin_lipoyl (PF00364)   \n",
       "6                             Biotin_lipoyl (PF00364)   \n",
       "7   CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...   \n",
       "8   CPSase_L_D2 (PF02786), CPSase_L_chain (PF00289...   \n",
       "9                             Acyl_transf_1 (PF00698)   \n",
       "10     ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)   \n",
       "11     ACP_syn_III_C (PF08541), ACP_syn_III (PF08545)   \n",
       "12  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "13  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "14  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "15  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "16  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "17  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "18  Ketoacyl-synt (PF00109), Ketoacyl-synt_C (PF02...   \n",
       "19                                adh_short (PF00106)   \n",
       "20                                adh_short (PF00106)   \n",
       "21                                     FabA (PF07977)   \n",
       "22                                     FabA (PF07977)   \n",
       "23                             Adh_short_C2 (PF13561)   \n",
       "24                             Adh_short_C2 (PF13561)   \n",
       "25                         FA_desaturase_C2 (PF03405)   \n",
       "26                         FA_desaturase_C2 (PF03405)   \n",
       "27                         FA_desaturase_C2 (PF03405)   \n",
       "28                         FA_desaturase_C2 (PF03405)   \n",
       "29                         FA_desaturase_C2 (PF03405)   \n",
       "30                         FA_desaturase_C2 (PF03405)   \n",
       "31         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "32         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "33         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "34         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "35         FA_desaturase (PF00487), DUF3474 (PF11960)   \n",
       "36      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "37      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "38      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "39      Acyl-ACP_TE (PF001643), Acyl-thio_N (PF12590)   \n",
       "40                             Acyl-ACP_TE (PF001643)   \n",
       "41                             Acyl-ACP_TE (PF001643)   \n",
       "\n",
       "                      catalytic_residue  \\\n",
       "0                                     -   \n",
       "1                                     -   \n",
       "2                                     -   \n",
       "3                                     -   \n",
       "4                                     -   \n",
       "5                           Ser-Met-Lys   \n",
       "6                           Ser-Met-Lys   \n",
       "7   His-Glu, His-Glu, Glu, Ile-Glu, Arg   \n",
       "8   His-Glu, His-Glu, Glu, Ile-Glu, Arg   \n",
       "9                   Gln-Ser-Arg-His-Gln   \n",
       "10                      Cys-His-Arg-Asn   \n",
       "11                      Cys-His-Arg-Asn   \n",
       "12                          Cys-His-His   \n",
       "13                          Cys-His-His   \n",
       "14                          Cys-His-His   \n",
       "15                          Cys-His-His   \n",
       "16                          Cys-His-His   \n",
       "17                          Cys-His-His   \n",
       "18                          Cys-His-His   \n",
       "19                  Arg-Ser-Tyr-Lys-Arg   \n",
       "20                  Arg-Ser-Tyr-Lys-Arg   \n",
       "21                              His-Asp   \n",
       "22                              His-Asp   \n",
       "23                          Tyr-Tyr-Lys   \n",
       "24                          Tyr-Tyr-Lys   \n",
       "25                                    -   \n",
       "26                                    -   \n",
       "27                                    -   \n",
       "28                                    -   \n",
       "29                                    -   \n",
       "30                                    -   \n",
       "31                                    -   \n",
       "32                                    -   \n",
       "33                                    -   \n",
       "34                                    -   \n",
       "35                                    -   \n",
       "36                          Asn-His-Cys   \n",
       "37                          Asn-His-Cys   \n",
       "38                          Asn-His-Cys   \n",
       "39                          Asn-His-Cys   \n",
       "40                          Asn-His-Cys   \n",
       "41                          Asn-His-Cys   \n",
       "\n",
       "                                        motif                    reference  \\\n",
       "0                                           -                            -   \n",
       "1                                           -                            -   \n",
       "2                                           -                            -   \n",
       "3                                           -                            -   \n",
       "4                                           -                            -   \n",
       "5                     (A/I/V)-M-K-(L/V/M/A/T)           Barber et al. 2005   \n",
       "6                     (A/I/V)-M-K-(L/V/M/A/T)           Barber et al. 2005   \n",
       "7                                           -          Waldrop et al. 1994   \n",
       "8                                           -          Waldrop et al. 1994   \n",
       "9                                     GLSLGEY               Li et al. 2009   \n",
       "10                                    GNTSAAS               Li et al. 2009   \n",
       "11                                    GNTSAAS               Li et al. 2009   \n",
       "12                                          -               Li et al. 2009   \n",
       "13                                          -               Li et al. 2009   \n",
       "14                                          -               Li et al. 2009   \n",
       "15                                          -               Li et al. 2009   \n",
       "16                                          -               Li et al. 2009   \n",
       "17                                          -               Li et al. 2009   \n",
       "18                                          -               Li et al. 2009   \n",
       "19                                       YX3K               Li et al. 2009   \n",
       "20                                       YX3K               Li et al. 2009   \n",
       "21                              LPHRFPFLLVDRV               Li et al. 2009   \n",
       "22                              LPHRFPFLLVDRV               Li et al. 2009   \n",
       "23                                       YX6K               Li et al. 2009   \n",
       "24                                       YX6K               Li et al. 2009   \n",
       "25                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "26                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "27                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "28                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "29                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "30                            EENRH and DEKRH  Haralampidis & Milioni 1998   \n",
       "31  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "32  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "33  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "34  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "35  Histidine boxes (HXXXH, HXXHH, and HXXHH)         Shanklin et al. 1994   \n",
       "36                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "37                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "38                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "39                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "40                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "41                          NQHVNN and YRRECG             Yuan et al. 1996   \n",
       "\n",
       "          ODB_OG  \n",
       "0   112963at4447  \n",
       "1   118148at4447  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4            NaN  \n",
       "5            NaN  \n",
       "6            NaN  \n",
       "7            NaN  \n",
       "8            NaN  \n",
       "9   133140at4447  \n",
       "10           NaN  \n",
       "11  119951at4447  \n",
       "12  117154at4447  \n",
       "13           NaN  \n",
       "14           NaN  \n",
       "15           NaN  \n",
       "16           NaN  \n",
       "17           NaN  \n",
       "18           NaN  \n",
       "19           NaN  \n",
       "20           NaN  \n",
       "21           NaN  \n",
       "22           NaN  \n",
       "23           NaN  \n",
       "24           NaN  \n",
       "25           NaN  \n",
       "26  129798at4447  \n",
       "27           NaN  \n",
       "28  122758at4447  \n",
       "29           NaN  \n",
       "30           NaN  \n",
       "31           NaN  \n",
       "32           NaN  \n",
       "33           NaN  \n",
       "34           NaN  \n",
       "35  128946at4447  \n",
       "36  148338at4447  \n",
       "37           NaN  \n",
       "38           NaN  \n",
       "39  124146at4447  \n",
       "40           NaN  \n",
       "41           NaN  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FAB_genes_og"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
