{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "441dbb2a",
   "metadata": {},
   "source": [
    "## This is just a playground. See finished standalone python script at ~/GitHub/YBD/genename_to_bedfile.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "bcf7c786",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curl 'https://data.orthodb.org/current/search?query=SAUR&level=4447' -L -o SAUR_at4447.dat\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  5896    0  5896    0     0   3234      0 --:--:--  0:00:01 --:--:--  3234\n"
     ]
    }
   ],
   "source": [
    "import subprocess, json\n",
    "import pandas as pd\n",
    "\n",
    "gene_name = 'SAUR'\n",
    "level = 4447\n",
    "# Function to parse the 'Attributes' column in gff file and convert it into a dictionary of key-value pairs\n",
    "def parse_attributes(attribute_str):\n",
    "    attribute_pairs = attribute_str.split(';')\n",
    "    attribute_dict = {}\n",
    "    for pair in attribute_pairs:\n",
    "        if len(pair.split('=')) == 2:\n",
    "            key, value = pair.split('=')\n",
    "        else:\n",
    "            key = pair.split('=')[0]\n",
    "            value = ''\n",
    "        attribute_dict[key] = value\n",
    "    return attribute_dict\n",
    "anno_file = \"/bioinfo2/palm/ref/orthoDBv11/dura{}_Results/dura{}.og.annotations\".format(level, level)\n",
    "gff_file = '/bioinfo2/palm/ref/dura/dura_ref.gff'\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "cmd = \"curl 'https://data.orthodb.org/current/search?query={}&level={}' -L -o {}_at{}.dat\".format(gene_name, level, gene_name, level)\n",
    "print(cmd)\n",
    "subprocess.run(cmd, shell=True, check=True)\n",
    "with open(\"{}_at{}.dat\".format(gene_name, level), \"r\") as json_file:\n",
    "    data_dict = json.load(json_file)\n",
    "\n",
    "df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "selected_rows = df[df['ODB_OG'].isin(data_dict['data'])]\n",
    "selected_rows = selected_rows.rename(columns={'#query': 'query'})\n",
    "selected_rows['gene'] = selected_rows['query'].str.split('-').str[0]\n",
    "\n",
    "# Filter rows where value from column A is in the list\n",
    "filtered_rows = df_gff[df_gff['ID'].isin(selected_rows['gene'])]\n",
    "\n",
    "# write bed file\n",
    "df = filtered_rows[['Seqid', 'Start','End','ID']]\n",
    "df_no_duplicates = df.drop_duplicates()\n",
    "df_sorted = df_no_duplicates.sort_values(by=['Seqid', 'Start'])\n",
    "# add pound(#) to the first column name - convention of bed file\n",
    "df_sorted.columns = ['#' + df_sorted.columns[0] if i == 0 else col for i, col in enumerate(df_sorted.columns)]\n",
    "bedfile = gene_name + '_at' + level + '_dura.bed'\n",
    "df_sorted.to_csv(bedfile, sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c575e40f",
   "metadata": {},
   "source": [
    "# Write bed file based on a list of transcript names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b5f4774d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "def parse_attributes(attribute_str):\n",
    "    attribute_pairs = attribute_str.split(';')\n",
    "    attribute_dict = {}\n",
    "    for pair in attribute_pairs:\n",
    "        if len(pair.split('=')) == 2:\n",
    "            key, value = pair.split('=')\n",
    "        else:\n",
    "            key = pair.split('=')[0]\n",
    "            value = ''\n",
    "        attribute_dict[key] = value\n",
    "    return attribute_dict\n",
    "with open('/bioinfo2/palm/accession/Apriyanto2023/DE_table.genes', 'r') as f:\n",
    "    genes = [line.strip() for line in f]\n",
    "gff_file = '/bioinfo2/palm/ref/dura/dura_ref.gff'\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "\n",
    "# Filter rows where value from column A is in the gene list\n",
    "filtered_rows = df_gff[df_gff['ID'].isin(genes)]\n",
    "\n",
    "# write bed file\n",
    "df = filtered_rows[['Seqid', 'Start','End','ID']]\n",
    "df_no_duplicates = df.drop_duplicates()\n",
    "df_sorted = df_no_duplicates.sort_values(by=['Seqid', 'Start'])\n",
    "# add pound(#) to the first column name - convention of bed file\n",
    "df_sorted.columns = ['#' + df_sorted.columns[0] if i == 0 else col for i, col in enumerate(df_sorted.columns)]\n",
    "bedfile = 'Apriyanto2023_DE100.bed'\n",
    "df_sorted.to_csv(bedfile, sep='\\t', index=False)\n",
    "# the rest of the analysis please see https://www.notion.so/Association-studies-e780ce66173e44849149291fed4c0e52?pvs=4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acd430c8",
   "metadata": {},
   "source": [
    "# To look up any SNP from the genome (slow)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d12a2eb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# first let's work with just one coord\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq, MutableSeq\n",
    "\n",
    "# eventually this will be a script that only takes the coord, and go look the exon and REF ALT by it self.\n",
    "# * means stop codon. When this is changed we need to look further down the sequence until another stop codon right?\n",
    "\n",
    "coord = 'chr1_8180009'\n",
    "gff_file = '/bioinfo2/palm/ref/dura/dura_ref.gff'\n",
    "\n",
    "CHR, POS = coord.split('_')\n",
    "POS = int(POS)\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "filtered_df = df_gff[(df_gff['Type'] == \"exon\") & (df['Seqid'] == CHR) & (df['Start'] <= POS) & (df['End'] >= POS)]\n",
    "gene = filtered_df['ID'].iloc[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "192544ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "in_file = '/bioinfo2/palm/analysis/batch1/batch1_chr1_nosex.assoc.qassoc.tab'\n",
    "in_df = pd.read_csv(in_file, sep = '\\t', header=0)\n",
    "in_df['coord'] = 'chr' + in_df['CHR'].astype(str) + '_' + in_df['BP'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "138df701",
   "metadata": {},
   "outputs": [],
   "source": [
    "in_df1 = in_df[in_df['P'] < 0.01]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "8def7048",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4c3daee0>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex', 'ins', 'del')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del', 'mnp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins', 'ins', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins', 'del', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('mnp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('del', 'ins', 'del')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp', 'complex')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('complex', 'snp')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('ins', 'ins')\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd4cb9ea00>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n",
      "<pysam.libcbcf.VariantRecordInfo object at 0x7fbd96c08940>\n",
      "['AB', 'ABP', 'AC', 'AF', 'AN', 'AO', 'CIGAR', 'DP', 'DPB', 'DPRA', 'EPP', 'EPPR', 'GTI', 'LEN', 'MEANALT', 'MQM', 'MQMR', 'NS', 'NUMALT', 'ODDS', 'PAIRED', 'PAIREDR', 'PAO', 'PQA', 'PQR', 'PRO', 'QA', 'QR', 'RO', 'RPL', 'RPP', 'RPPR', 'RPR', 'RUN', 'SAF', 'SAP', 'SAR', 'SRF', 'SRP', 'SRR', 'TYPE', 'technology.NovaSeqXPlus']\n",
      "('snp',)\n"
     ]
    }
   ],
   "source": [
    "vcf_file = '/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100.exon.b1ch1.vcf'\n",
    "vcf = pysam.VariantFile(vcf_file)\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "8241d1b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('del',)\n",
      "('ins', 'ins')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex', 'ins', 'del')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex', 'snp', 'ins')\n",
      "('snp',)\n",
      "('del', 'mnp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins', 'snp')\n",
      "('snp',)\n",
      "('ins', 'ins', 'ins', 'snp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins', 'ins', 'del', 'ins')\n",
      "('del',)\n",
      "('snp',)\n",
      "('del',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('snp',)\n",
      "('complex', 'snp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('mnp',)\n",
      "('complex',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex', 'snp')\n",
      "('snp',)\n",
      "('del',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('complex',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('del', 'ins', 'del')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp', 'complex')\n",
      "('complex', 'snp')\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins',)\n",
      "('snp',)\n",
      "('snp',)\n",
      "('ins', 'ins')\n",
      "('snp',)\n",
      "('snp',)\n"
     ]
    }
   ],
   "source": [
    "for variant in vcf.fetch():\n",
    "    print(variant.info[\"TYPE\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "560818be",
   "metadata": {},
   "outputs": [],
   "source": [
    "a='TCTTCTTCTTCTTCTCCTCCTGCTTCTCTCCCTCTCCTCGGCAATCTCACTCCTCGCTGCCCTTATATTTTTTCTCTCTCTCTCTCTATCTCTATTCCCATTTTGCCCCTCCAACGGATACATCGATAATAGAAGAGAGATAGTGACTCCGGGGAAAGGGAAGGGGGAGAAAGGGGAACAAAAAGAGAGAGAGAAGAGTTTGAGTGTGTGCGAGTGGGGGAGAGAGTTTTTGAGGGTGGTACCTGTGAGAAGGGAGGGGAGGGGGTTTATTTGGTTGCCGAGAGGAGGGGGAAAATAATAAGAAGAAAAGAAGGGGATTTTTATTTTATTTTCATCTGGTGAGGGATTTTTTTCTTTTTTTTTTCTTCTTTTCTTCTTCTTCTTGTTTGAGGGAGGGGAAGAGAGGAGAAGAAGGGAAAGAAGAGGGGCCATGGCGGCGCACGTTAGCGTGAACCATGGCTTGCTGGCGCTGTTACTGGTGCTGATCTGCTTATCGTGCGCCGAGGGTGCGGTGCGGGTCGCGAGGCAGCGGCTGGAGGTGCGGCGCCACCTGAAGCGATTCAATAAACCCGCCGTCAAGAGCATCAAG'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "7df1a264",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_seq_var = MutableSeq(str(a))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "72931e2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_seq_var[38:39] = 'G'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "772ef95e",
   "metadata": {},
   "source": [
    "# Look up SNPs from a designated bed file (fast)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "a9a16937",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1_82720377\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/Seq.py:2804: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1_82720377_Egu005022-mRNA-1_110056at4447_carboxyl-terminal peptidase precursor_snp_C_G_CTC_CTG_L_L\n",
      "chr1_86754740\n",
      "chr1_86754740_Egu005191-mRNA-2_136145at4447_jacalin-related lectin 3_snp_C_G_CCC_CCG_P_P\n",
      "chr1_86758000\n",
      "chr1_86758000_Egu005191-mRNA-3_136145at4447_jacalin-related lectin 3_snp_C_G_CAG_GAG_Q_E\n",
      "chr1_93213103\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[137], line 76\u001b[0m\n\u001b[1;32m     73\u001b[0m col_names \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgene\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mog\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgene_name\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mType\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mREF\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mALT\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mcodon\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmodified_codon\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124moriginal_aa\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmodified_aa\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m     75\u001b[0m \u001b[38;5;66;03m# Generate lists and split them into separate columns\u001b[39;00m\n\u001b[0;32m---> 76\u001b[0m in_df[col_names] \u001b[38;5;241m=\u001b[39m \u001b[43min_df\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mcoord\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcoord2codon\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mapply(pd\u001b[38;5;241m.\u001b[39mSeries)\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/pandas/core/series.py:4771\u001b[0m, in \u001b[0;36mSeries.apply\u001b[0;34m(self, func, convert_dtype, args, **kwargs)\u001b[0m\n\u001b[1;32m   4661\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mapply\u001b[39m(\n\u001b[1;32m   4662\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m   4663\u001b[0m     func: AggFuncType,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   4666\u001b[0m     \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs,\n\u001b[1;32m   4667\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m DataFrame \u001b[38;5;241m|\u001b[39m Series:\n\u001b[1;32m   4668\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   4669\u001b[0m \u001b[38;5;124;03m    Invoke function on values of Series.\u001b[39;00m\n\u001b[1;32m   4670\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   4769\u001b[0m \u001b[38;5;124;03m    dtype: float64\u001b[39;00m\n\u001b[1;32m   4770\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 4771\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mSeriesApply\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfunc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mconvert_dtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/pandas/core/apply.py:1123\u001b[0m, in \u001b[0;36mSeriesApply.apply\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1120\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mapply_str()\n\u001b[1;32m   1122\u001b[0m \u001b[38;5;66;03m# self.f is Callable\u001b[39;00m\n\u001b[0;32m-> 1123\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply_standard\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/pandas/core/apply.py:1174\u001b[0m, in \u001b[0;36mSeriesApply.apply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1172\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1173\u001b[0m         values \u001b[38;5;241m=\u001b[39m obj\u001b[38;5;241m.\u001b[39mastype(\u001b[38;5;28mobject\u001b[39m)\u001b[38;5;241m.\u001b[39m_values\n\u001b[0;32m-> 1174\u001b[0m         mapped \u001b[38;5;241m=\u001b[39m \u001b[43mlib\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmap_infer\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1175\u001b[0m \u001b[43m            \u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1176\u001b[0m \u001b[43m            \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1177\u001b[0m \u001b[43m            \u001b[49m\u001b[43mconvert\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconvert_dtype\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1178\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1180\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(mapped) \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(mapped[\u001b[38;5;241m0\u001b[39m], ABCSeries):\n\u001b[1;32m   1181\u001b[0m     \u001b[38;5;66;03m# GH#43986 Need to do list(mapped) in order to get treated as nested\u001b[39;00m\n\u001b[1;32m   1182\u001b[0m     \u001b[38;5;66;03m#  See also GH#25959 regarding EA support\u001b[39;00m\n\u001b[1;32m   1183\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m obj\u001b[38;5;241m.\u001b[39m_constructor_expanddim(\u001b[38;5;28mlist\u001b[39m(mapped), index\u001b[38;5;241m=\u001b[39mobj\u001b[38;5;241m.\u001b[39mindex)\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/pandas/_libs/lib.pyx:2924\u001b[0m, in \u001b[0;36mpandas._libs.lib.map_infer\u001b[0;34m()\u001b[0m\n",
      "Cell \u001b[0;32mIn[137], line 53\u001b[0m, in \u001b[0;36mcoord2codon\u001b[0;34m(coord, bed_df, vcf, genome_fa, anno_df)\u001b[0m\n\u001b[1;32m     51\u001b[0m             \u001b[38;5;28;01mbreak\u001b[39;00m\n\u001b[1;32m     52\u001b[0m \u001b[38;5;66;03m# get the codon and amino acid\u001b[39;00m\n\u001b[0;32m---> 53\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m record \u001b[38;5;129;01min\u001b[39;00m SeqIO\u001b[38;5;241m.\u001b[39mparse(genome_fa,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfasta\u001b[39m\u001b[38;5;124m'\u001b[39m):\n\u001b[1;32m     54\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m record\u001b[38;5;241m.\u001b[39mid \u001b[38;5;241m==\u001b[39m CHR:\n\u001b[1;32m     55\u001b[0m         exon_seq \u001b[38;5;241m=\u001b[39m record\u001b[38;5;241m.\u001b[39mseq[start\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m:end]\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/Interfaces.py:72\u001b[0m, in \u001b[0;36mSequenceIterator.__next__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Return the next entry.\"\"\"\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m---> 72\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mnext\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecords\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     73\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m:\n\u001b[1;32m     74\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mshould_close_stream:\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/FastaIO.py:238\u001b[0m, in \u001b[0;36mFastaIterator.iterate\u001b[0;34m(self, handle)\u001b[0m\n\u001b[1;32m    236\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m SeqRecord(Seq(sequence), \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mid\u001b[39m, name\u001b[38;5;241m=\u001b[39mname, description\u001b[38;5;241m=\u001b[39mdescr)\n\u001b[1;32m    237\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 238\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m title, sequence \u001b[38;5;129;01min\u001b[39;00m SimpleFastaParser(handle):\n\u001b[1;32m    239\u001b[0m         \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    240\u001b[0m             first_word \u001b[38;5;241m=\u001b[39m title\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m1\u001b[39m)[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/FastaIO.py:63\u001b[0m, in \u001b[0;36mSimpleFastaParser\u001b[0;34m(handle)\u001b[0m\n\u001b[1;32m     58\u001b[0m \u001b[38;5;66;03m# Main logic\u001b[39;00m\n\u001b[1;32m     59\u001b[0m \u001b[38;5;66;03m# Note, remove trailing whitespace, and any internal spaces\u001b[39;00m\n\u001b[1;32m     60\u001b[0m \u001b[38;5;66;03m# (and any embedded \\r which are possible in mangled files\u001b[39;00m\n\u001b[1;32m     61\u001b[0m \u001b[38;5;66;03m# when not opened in universal read lines mode)\u001b[39;00m\n\u001b[1;32m     62\u001b[0m lines \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m---> 63\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m line \u001b[38;5;129;01min\u001b[39;00m handle:\n\u001b[1;32m     64\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m line[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m>\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m     65\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m title, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(lines)\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m \u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m)\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/codecs.py:319\u001b[0m, in \u001b[0;36mBufferedIncrementalDecoder.decode\u001b[0;34m(self, input, final)\u001b[0m\n\u001b[1;32m    314\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_buffer_decode\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;28minput\u001b[39m, errors, final):\n\u001b[1;32m    315\u001b[0m     \u001b[38;5;66;03m# Overwrite this method in subclasses: It must decode input\u001b[39;00m\n\u001b[1;32m    316\u001b[0m     \u001b[38;5;66;03m# and return an (output, length consumed) tuple\u001b[39;00m\n\u001b[1;32m    317\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mNotImplementedError\u001b[39;00m\n\u001b[0;32m--> 319\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mdecode\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;28minput\u001b[39m, final\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m):\n\u001b[1;32m    320\u001b[0m     \u001b[38;5;66;03m# decode input (taking the buffer into account)\u001b[39;00m\n\u001b[1;32m    321\u001b[0m     data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuffer \u001b[38;5;241m+\u001b[39m \u001b[38;5;28minput\u001b[39m\n\u001b[1;32m    322\u001b[0m     (result, consumed) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_buffer_decode(data, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39merrors, final)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import pysam\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq, MutableSeq\n",
    "from Bio import SeqIO\n",
    "\n",
    "out_file = \"/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100_batch1.summary\"\n",
    "bed_file = '/bioinfo2/palm/analysis/yield/Apriyanto2023_DE100.exon.bed'\n",
    "vcf_file = '/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100.exon.b1ch1.vcf'\n",
    "in_file = '/bioinfo2/palm/analysis/batch1/Apriyanto2023_DE100.exon.b1ch1.assoc.qassoc.tab'\n",
    "in_df = pd.read_csv(in_file, sep = '\\t', header=0)\n",
    "in_df = in_df[in_df['P'] < 0.01]\n",
    "in_df['coord'] = 'chr' + in_df['CHR'].astype(str) + '_' + in_df['BP'].astype(str)\n",
    "bed_column_names = ['Seqid', 'Start', 'End', 'Genes']\n",
    "bed_df = pd.read_csv(bed_file, sep='\\t',comment='#', names=bed_column_names)\n",
    "vcf = pysam.VariantFile(vcf_file, \"r\")\n",
    "genome_fa = '/bioinfo2/palm/ref/dura/dura_ref.fasta'\n",
    "anno_file = \"/bioinfo2/palm/ref/dura/dura_orthoDBv11_4447_3193_33090.og.annotations\"\n",
    "anno_df = pd.read_csv(anno_file, sep='\\t', header=0)\n",
    "anno_df = anno_df.rename(columns={'#query': 'query'})\n",
    "\n",
    "def coord2codon(coord, bed_df = bed_df, vcf = vcf, genome_fa = genome_fa, anno_df = anno_df):\n",
    "    # this function takes a coordinate of a SNP in the dura genome and returns some useful info about it\n",
    "    print(coord)\n",
    "    CHR, POS = coord.split('_')\n",
    "    POS = int(POS)\n",
    "    # look up gene from bed file\n",
    "    snp_df = bed_df[(bed_df['Seqid'] == CHR) & (bed_df['Start'] <= POS) & (bed_df['End'] >= POS)]\n",
    "    if len(snp_df) > 0: # if snp is in the region we care\n",
    "        start = int(snp_df['Start'].iloc[0])\n",
    "        end = int(snp_df['End'].iloc[0])\n",
    "        gene = snp_df['Genes'].iloc[0]\n",
    "        anno_df = anno_df[anno_df['query'] == gene]\n",
    "        if len(anno_df) == 1:\n",
    "            og = anno_df['ODB_OG'].iloc[0]\n",
    "            gene_name = anno_df['Description'].iloc[0]\n",
    "        elif len(anno_df) == 0:\n",
    "            og = 'NA'\n",
    "            gene_name = 'NA'\n",
    "        else:\n",
    "            print(gene, \" has more than one og, recording the first one\")\n",
    "            og = anno_df['ODB_OG'].iloc[0]\n",
    "            gene_name = anno_df['Description'].iloc[0]\n",
    "        # look up REF and ALT from vcf\n",
    "        for variant in vcf.fetch():\n",
    "            TYPE = variant.info[\"TYPE\"]\n",
    "            type_out = '|'.join([i for i in TYPE])\n",
    "            if variant.chrom == CHR:\n",
    "                if variant.pos == POS:\n",
    "                    REF = variant.ref\n",
    "                    ALT = variant.alts[0]\n",
    "                    break\n",
    "        # get the codon and amino acid\n",
    "        for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "            if record.id == CHR:\n",
    "                exon_seq = record.seq[start-1:end]\n",
    "                exon_seq_var = MutableSeq(str(exon_seq))\n",
    "                protein_seq = exon_seq.translate()\n",
    "                snp_position = POS - start\n",
    "                # Find the codon's start position\n",
    "                codon_start = snp_position - (snp_position % 3)\n",
    "                # Extract the codons\n",
    "                codon = exon_seq[codon_start: codon_start+3]\n",
    "                exon_seq_var[snp_position: snp_position+len(ALT)] = ALT\n",
    "                modified_codon = exon_seq_var[codon_start: codon_start+3]    \n",
    "                original_aa = codon.translate()\n",
    "                modified_aa = modified_codon.translate()\n",
    "                break\n",
    "\n",
    "        print(';'.join([coord, gene, og, gene_name, type_out, REF, ALT, str(codon), str(modified_codon), str(original_aa), str(modified_aa)]))\n",
    "        return [gene, og, gene_name, type_out, REF, ALT, str(codon), str(modified_codon), str(original_aa), str(modified_aa)]\n",
    "\n",
    "# new column names\n",
    "col_names = ['gene', 'og', 'gene_name', 'Type','REF', 'ALT', 'codon', 'modified_codon', 'original_aa', 'modified_aa']\n",
    "\n",
    "# Generate lists and split them into separate columns\n",
    "in_df[col_names] = in_df['coord'].apply(coord2codon).apply(pd.Series)\n",
    "    \n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7cded14",
   "metadata": {},
   "outputs": [],
   "source": [
    "        for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "            if record.id == CHR:\n",
    "                exon_seq = record.seq[start-1:end]\n",
    "                exon_seq_snp = MutableSeq(str(exon_seq))\n",
    "                protein_seq = exon_seq.translate()\n",
    "                snp_position = POS - start\n",
    "                if TYPE == 'snp'\n",
    "                if exon_seq[snp_position] == REF:\n",
    "                    exon_seq_snp[snp_position] = ALT\n",
    "                else:\n",
    "                    print(\"REF wrong. Break\")\n",
    "                    break\n",
    "                # Find the codon's start position\n",
    "                codon_start = snp_position - (snp_position % 3)\n",
    "                # Extract the codons\n",
    "                codon = exon_seq[codon_start: codon_start+3]\n",
    "                # If needed, replace the SNP in the codon\n",
    "                modified_codon = exon_seq_snp[codon_start: codon_start+3] # modify with SNP\n",
    "                # Translate the original and modified codon to amino acids\n",
    "                original_aa = codon.translate()\n",
    "                modified_aa = modified_codon.translate()\n",
    "                break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "fb570dd9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>R2</th>\n",
       "      <th>T</th>\n",
       "      <th>P</th>\n",
       "      <th>X10</th>\n",
       "      <th>coord</th>\n",
       "      <th>gene</th>\n",
       "      <th>og</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>codon</th>\n",
       "      <th>modified_codon</th>\n",
       "      <th>original_aa</th>\n",
       "      <th>modified_aa</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16</td>\n",
       "      <td>.</td>\n",
       "      <td>20788022</td>\n",
       "      <td>48</td>\n",
       "      <td>17.55</td>\n",
       "      <td>7.255</td>\n",
       "      <td>0.1128</td>\n",
       "      <td>2.418</td>\n",
       "      <td>0.01961</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_20788022</td>\n",
       "      <td>Egu030167-mRNA-1</td>\n",
       "      <td>130847at4447</td>\n",
       "      <td>Heat stress transcription factor B-2c</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>(T, G, G)</td>\n",
       "      <td>[C, G, G]</td>\n",
       "      <td>(W)</td>\n",
       "      <td>[R]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16</td>\n",
       "      <td>.</td>\n",
       "      <td>32010076</td>\n",
       "      <td>47</td>\n",
       "      <td>-18.43</td>\n",
       "      <td>6.919</td>\n",
       "      <td>0.1362</td>\n",
       "      <td>-2.664</td>\n",
       "      <td>0.01069</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_32010076</td>\n",
       "      <td>Egu030617-mRNA-1</td>\n",
       "      <td>128811at4447</td>\n",
       "      <td>transmembrane protein 53-A</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>(G, G, C)</td>\n",
       "      <td>[G, G, T]</td>\n",
       "      <td>(G)</td>\n",
       "      <td>[G]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR SNP        BP  NMISS   BETA     SE      R2      T        P  X10  \\\n",
       "0   16   .  20788022     48  17.55  7.255  0.1128  2.418  0.01961  NaN   \n",
       "1   16   .  32010076     47 -18.43  6.919  0.1362 -2.664  0.01069  NaN   \n",
       "\n",
       "            coord              gene            og  \\\n",
       "0  chr16_20788022  Egu030167-mRNA-1  130847at4447   \n",
       "1  chr16_32010076  Egu030617-mRNA-1  128811at4447   \n",
       "\n",
       "                               gene_name REF ALT      codon modified_codon  \\\n",
       "0  Heat stress transcription factor B-2c   T   C  (T, G, G)      [C, G, G]   \n",
       "1             transmembrane protein 53-A   C   T  (G, G, C)      [G, G, T]   \n",
       "\n",
       "  original_aa modified_aa  \n",
       "0         (W)         [R]  \n",
       "1         (G)         [G]  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "                # if the variant type if SNP\n",
    "                if (len(TYPE) == 1) & (TYPE[0] == 'snp'):\n",
    "                    if exon_seq[snp_position] == REF:\n",
    "                        exon_seq_var[snp_position] = ALT\n",
    "                    else:\n",
    "                        print(\"REF wrong. Break\")\n",
    "                        break\n",
    "                    # Translate the original and modified codon to amino acids\n",
    "                else:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "633f6c14",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[62], line 31\u001b[0m\n\u001b[1;32m     29\u001b[0m             \u001b[38;5;28;01mbreak\u001b[39;00m\n\u001b[1;32m     30\u001b[0m \u001b[38;5;66;03m# get the codon and amino acid\u001b[39;00m\n\u001b[0;32m---> 31\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m record \u001b[38;5;129;01min\u001b[39;00m SeqIO\u001b[38;5;241m.\u001b[39mparse(genome_fa,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfasta\u001b[39m\u001b[38;5;124m'\u001b[39m):\n\u001b[1;32m     32\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m record\u001b[38;5;241m.\u001b[39mid \u001b[38;5;241m==\u001b[39m CHR:\n\u001b[1;32m     33\u001b[0m         exon_seq \u001b[38;5;241m=\u001b[39m record\u001b[38;5;241m.\u001b[39mseq[start\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m:end]\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/Interfaces.py:72\u001b[0m, in \u001b[0;36mSequenceIterator.__next__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Return the next entry.\"\"\"\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m---> 72\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mnext\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecords\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     73\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m:\n\u001b[1;32m     74\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mshould_close_stream:\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/FastaIO.py:238\u001b[0m, in \u001b[0;36mFastaIterator.iterate\u001b[0;34m(self, handle)\u001b[0m\n\u001b[1;32m    236\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m SeqRecord(Seq(sequence), \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mid\u001b[39m, name\u001b[38;5;241m=\u001b[39mname, description\u001b[38;5;241m=\u001b[39mdescr)\n\u001b[1;32m    237\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 238\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m title, sequence \u001b[38;5;129;01min\u001b[39;00m SimpleFastaParser(handle):\n\u001b[1;32m    239\u001b[0m         \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    240\u001b[0m             first_word \u001b[38;5;241m=\u001b[39m title\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m1\u001b[39m)[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[0;32m/home/xiaorong/anaconda3/envs/bioinfo/lib/python3.9/site-packages/Bio/SeqIO/FastaIO.py:65\u001b[0m, in \u001b[0;36mSimpleFastaParser\u001b[0;34m(handle)\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m line \u001b[38;5;129;01min\u001b[39;00m handle:\n\u001b[1;32m     64\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m line[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m>\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m---> 65\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m title, \u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mjoin\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlines\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m \u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m)\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     66\u001b[0m         lines \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m     67\u001b[0m         title \u001b[38;5;241m=\u001b[39m line[\u001b[38;5;241m1\u001b[39m:]\u001b[38;5;241m.\u001b[39mrstrip()\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# this function takes a coordinate of a SNP in the dura genome and returns some useful info about it\n",
    "coord = 'chr3_69323261'\n",
    "CHR, POS = coord.split('_')\n",
    "POS = int(POS)\n",
    "# look up gene from bed file\n",
    "snp_df = bed_df[(bed_df['Seqid'] == CHR) & (bed_df['Start'] <= POS) & (bed_df['End'] >= POS)]\n",
    "start = int(snp_df['Start'].iloc[0])\n",
    "end = int(snp_df['End'].iloc[0])\n",
    "gene = snp_df['Genes'].iloc[0]\n",
    "anno_df = anno_df[anno_df['query'] == gene]\n",
    "if len(anno_df) == 1:\n",
    "    og = anno_df['ODB_OG'].iloc[0]\n",
    "    gene_name = anno_df['Description'].iloc[0]\n",
    "elif len(anno_df) == 0:\n",
    "    og = 'NA'\n",
    "    gene_name = 'NA'\n",
    "else:\n",
    "    print(gene, \" has more than one og, recording the first one\")\n",
    "    og = anno_df['ODB_OG'].iloc[0]\n",
    "    gene_name = anno_df['Description'].iloc[0]\n",
    "# look up REF and ALT from vcf\n",
    "for variant in vcf:\n",
    "    if variant.chrom == CHR:\n",
    "        print\n",
    "        if variant.pos == POS:\n",
    "            REF = variant.ref\n",
    "            ALT = variant.alts[0]\n",
    "            print(REF, ALT)\n",
    "            break\n",
    "# get the codon and amino acid\n",
    "for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "    if record.id == CHR:\n",
    "        exon_seq = record.seq[start-1:end]\n",
    "        exon_seq_snp = MutableSeq(str(exon_seq))\n",
    "        protein_seq = exon_seq.translate()\n",
    "        snp_position = POS - start\n",
    "        if exon_seq[snp_position] == REF:\n",
    "            exon_seq_snp[snp_position] = ALT\n",
    "        else:\n",
    "            \"REF wrong. Break\"\n",
    "            break\n",
    "        # Find the codon's start position\n",
    "        codon_start = snp_position - (snp_position % 3)\n",
    "        # Extract the codon\n",
    "        codon = exon_seq[codon_start: codon_start+3]\n",
    "        # If needed, replace the SNP in the codon\n",
    "        modified_codon = exon_seq_snp[codon_start: codon_start+3] # modify with SNP\n",
    "        # Translate the original and modified codon to amino acids\n",
    "        original_aa = codon.translate()\n",
    "        modified_aa = modified_codon.translate()\n",
    "        break\n",
    "print(';'.join([coord, gene, og, gene_name, REF, ALT, str(codon), str(modified_codon), str(original_aa), str(modified_aa)]))\n",
    "return [gene, og, gene_name, REF, ALT, codon, modified_codon, original_aa, modified_aa]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "c4ae8ac7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>R2</th>\n",
       "      <th>T</th>\n",
       "      <th>P</th>\n",
       "      <th>X10</th>\n",
       "      <th>coord</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>72869566</td>\n",
       "      <td>49</td>\n",
       "      <td>-25.33</td>\n",
       "      <td>7.081</td>\n",
       "      <td>0.21400</td>\n",
       "      <td>-3.577</td>\n",
       "      <td>0.000819</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_72869566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>72869593</td>\n",
       "      <td>50</td>\n",
       "      <td>18.00</td>\n",
       "      <td>8.701</td>\n",
       "      <td>0.08189</td>\n",
       "      <td>2.069</td>\n",
       "      <td>0.043940</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_72869593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>73734371</td>\n",
       "      <td>50</td>\n",
       "      <td>29.14</td>\n",
       "      <td>10.540</td>\n",
       "      <td>0.13750</td>\n",
       "      <td>2.766</td>\n",
       "      <td>0.008038</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_73734371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>82723240</td>\n",
       "      <td>50</td>\n",
       "      <td>27.56</td>\n",
       "      <td>8.345</td>\n",
       "      <td>0.18520</td>\n",
       "      <td>3.303</td>\n",
       "      <td>0.001815</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_82723240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>92918856</td>\n",
       "      <td>52</td>\n",
       "      <td>26.25</td>\n",
       "      <td>9.942</td>\n",
       "      <td>0.12240</td>\n",
       "      <td>2.641</td>\n",
       "      <td>0.011010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_92918856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>15</td>\n",
       "      <td>.</td>\n",
       "      <td>63227901</td>\n",
       "      <td>51</td>\n",
       "      <td>-13.79</td>\n",
       "      <td>6.668</td>\n",
       "      <td>0.08024</td>\n",
       "      <td>-2.068</td>\n",
       "      <td>0.043980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr15_63227901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>15</td>\n",
       "      <td>.</td>\n",
       "      <td>63228447</td>\n",
       "      <td>48</td>\n",
       "      <td>-19.94</td>\n",
       "      <td>6.750</td>\n",
       "      <td>0.15950</td>\n",
       "      <td>-2.955</td>\n",
       "      <td>0.004920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr15_63228447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>16</td>\n",
       "      <td>.</td>\n",
       "      <td>14913334</td>\n",
       "      <td>37</td>\n",
       "      <td>26.96</td>\n",
       "      <td>7.669</td>\n",
       "      <td>0.26090</td>\n",
       "      <td>3.515</td>\n",
       "      <td>0.001235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_14913334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>16</td>\n",
       "      <td>.</td>\n",
       "      <td>20788022</td>\n",
       "      <td>48</td>\n",
       "      <td>17.55</td>\n",
       "      <td>7.255</td>\n",
       "      <td>0.11280</td>\n",
       "      <td>2.418</td>\n",
       "      <td>0.019610</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_20788022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>16</td>\n",
       "      <td>.</td>\n",
       "      <td>32010076</td>\n",
       "      <td>47</td>\n",
       "      <td>-18.43</td>\n",
       "      <td>6.919</td>\n",
       "      <td>0.13620</td>\n",
       "      <td>-2.664</td>\n",
       "      <td>0.010690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_32010076</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR SNP        BP  NMISS   BETA      SE       R2      T         P  X10  \\\n",
       "0     1   .  72869566     49 -25.33   7.081  0.21400 -3.577  0.000819  NaN   \n",
       "1     1   .  72869593     50  18.00   8.701  0.08189  2.069  0.043940  NaN   \n",
       "2     1   .  73734371     50  29.14  10.540  0.13750  2.766  0.008038  NaN   \n",
       "3     1   .  82723240     50  27.56   8.345  0.18520  3.303  0.001815  NaN   \n",
       "4     1   .  92918856     52  26.25   9.942  0.12240  2.641  0.011010  NaN   \n",
       "..  ...  ..       ...    ...    ...     ...      ...    ...       ...  ...   \n",
       "64   15   .  63227901     51 -13.79   6.668  0.08024 -2.068  0.043980  NaN   \n",
       "65   15   .  63228447     48 -19.94   6.750  0.15950 -2.955  0.004920  NaN   \n",
       "66   16   .  14913334     37  26.96   7.669  0.26090  3.515  0.001235  NaN   \n",
       "67   16   .  20788022     48  17.55   7.255  0.11280  2.418  0.019610  NaN   \n",
       "68   16   .  32010076     47 -18.43   6.919  0.13620 -2.664  0.010690  NaN   \n",
       "\n",
       "             coord  \n",
       "0    chr1_72869566  \n",
       "1    chr1_72869593  \n",
       "2    chr1_73734371  \n",
       "3    chr1_82723240  \n",
       "4    chr1_92918856  \n",
       "..             ...  \n",
       "64  chr15_63227901  \n",
       "65  chr15_63228447  \n",
       "66  chr16_14913334  \n",
       "67  chr16_20788022  \n",
       "68  chr16_32010076  \n",
       "\n",
       "[69 rows x 11 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "in_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "f2dcec27",
   "metadata": {},
   "outputs": [],
   "source": [
    "def coord2codon(coord, bed_df = bed_df, vcf = vcf, genome_fa = genome_fa, anno_df = anno_df):\n",
    "    # this function takes a coordinate of a SNP in the dura genome and returns some useful info about it\n",
    "    CHR, POS = coord.split('_')\n",
    "    POS = int(POS)\n",
    "    # look up gene from bed file\n",
    "    snp_df = bed_df[(bed_df['Seqid'] == CHR) & (bed_df['Start'] <= POS) & (bed_df['End'] >= POS)]\n",
    "    start = int(snp_df['Start'].iloc[0])\n",
    "    end = int(snp_df['End'].iloc[0])\n",
    "    gene = snp_df['Genes'].iloc[0]\n",
    "    anno_df = anno_df[anno_df['query'] == gene]\n",
    "    og = str(anno_df['ODB_OG'])\n",
    "    gene_name = str(anno_df['Description'])\n",
    "    # look up REF and ALT from vcf\n",
    "    for variant in vcf:\n",
    "        if variant.chrom == CHR:\n",
    "            if variant.pos == POS:\n",
    "                REF = variant.ref\n",
    "                ALT = variant.alts\n",
    "                break\n",
    "            else:\n",
    "                print(coord, \"Wrong POS\")\n",
    "    # get the codon and amino acid\n",
    "    for record in SeqIO.parse(genome_fa,'fasta'):\n",
    "        if record.id == CHR:\n",
    "            exon_seq = record.seq[start-1:end]\n",
    "            exon_seq_snp = MutableSeq(str(exon_seq))\n",
    "            protein_seq = exon_seq.translate()\n",
    "            snp_position = POS - start\n",
    "            if exon_seq[snp_position] == REF:\n",
    "                exon_seq_snp[snp_position] = ALT[0]\n",
    "            else:\n",
    "                \"REF wrong. Break\"\n",
    "                break\n",
    "            # Find the codon's start position\n",
    "            codon_start = snp_position - (snp_position % 3)\n",
    "            # Extract the codon\n",
    "            codon = exon_seq[codon_start: codon_start+3]\n",
    "            print(coord, codon)\n",
    "            # If needed, replace the SNP in the codon\n",
    "            modified_codon = exon_seq_snp[codon_start: codon_start+3] # modify with SNP\n",
    "            # Translate the original and modified codon to amino acids\n",
    "            original_aa = codon.translate()\n",
    "            modified_aa = modified_codon.translate()\n",
    "            break\n",
    "    return [gene, og, gene_name, REF, ALT, codon, modified_codon, original_aa, modified_aa]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "92314d1d",
   "metadata": {},
   "outputs": [
    {
     "ename": "UnboundLocalError",
     "evalue": "local variable 'REF' referenced before assignment",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[41], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m a \u001b[38;5;241m=\u001b[39m \u001b[43mcoord2codon\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mchr1_72869566\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[40], line 29\u001b[0m, in \u001b[0;36mcoord2codon\u001b[0;34m(coord, bed_df, vcf, genome_fa, anno_df)\u001b[0m\n\u001b[1;32m     27\u001b[0m protein_seq \u001b[38;5;241m=\u001b[39m exon_seq\u001b[38;5;241m.\u001b[39mtranslate()\n\u001b[1;32m     28\u001b[0m snp_position \u001b[38;5;241m=\u001b[39m POS \u001b[38;5;241m-\u001b[39m start\n\u001b[0;32m---> 29\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m exon_seq[snp_position] \u001b[38;5;241m==\u001b[39m \u001b[43mREF\u001b[49m:\n\u001b[1;32m     30\u001b[0m     exon_seq_snp[snp_position] \u001b[38;5;241m=\u001b[39m ALT[\u001b[38;5;241m0\u001b[39m]\n\u001b[1;32m     31\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mUnboundLocalError\u001b[0m: local variable 'REF' referenced before assignment"
     ]
    }
   ],
   "source": [
    "a = coord2codon('chr1_72869566')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "d09a5122",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GGT'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a[51:54]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "75b5ba10",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam\n",
    "import pandas as pd\n",
    "vcf = pysam.VariantFile(\"/bioinfo2/palm/analysis/yield/Apriyanto2023/Apriyanto2023_DE100.vcf\", \"r\") \n",
    "for variant in vcf:\n",
    "    if variant.chrom == CHR:\n",
    "        if variant.pos == POS:\n",
    "            REF = variant.ref\n",
    "            ALT = variant.alts\n",
    "            break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c633ca56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/xiaorong/anaconda3/envs/bioinfo/bin/python'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import sys\n",
    "sys.executable"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06d431dc",
   "metadata": {},
   "source": [
    "## Get the gene name and amino acid give the coord of a SNP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "fbbe2632",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "warning: this exon is not dividable by 3\n",
      "GAA E AAA K\n"
     ]
    }
   ],
   "source": [
    "# first let's work with just one coord\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq, MutableSeq\n",
    "\n",
    "# eventually this will be a script that only takes the coord, and go look the exon and REF ALT by it self.\n",
    "# * means stop codon. When this is changed we need to look further down the sequence until another stop codon right?\n",
    "\n",
    "coord = 'chr7_37218457'\n",
    "bed_file = '/bioinfo2/palm/analysis/yield/Apriyanto2023_DE100.exon.bed'\n",
    "vcf_file = '/bioinfo2/palm/analysis/' \n",
    "\n",
    "CHR, POS = coord.split('_')\n",
    "POS = int(POS)\n",
    "gff_column_names = ['Seqid', 'Source', 'Type', 'Start', 'End', 'Score', 'Strand', 'Phase', 'Attributes']\n",
    "df = pd.read_csv(gff_file, sep='\\t',comment='#', names=gff_column_names)\n",
    "df1 = df['Attributes'].apply(parse_attributes).apply(pd.Series)\n",
    "df_gff = pd.concat([df.drop('Attributes', axis=1), df1], axis=1)\n",
    "filtered_df = df_gff[(df_gff['Type'] == \"exon\") & (df['Seqid'] == CHR) & (df['Start'] <= POS) & (df['End'] >= POS)]\n",
    "gene = filtered_df['ID'].iloc[0]\n",
    "\n",
    "REF = 'G'\n",
    "ALT = 'A'\n",
    "\n",
    "filtered_rows = df_gff[df_gff['ID']==gene]\n",
    "start = int(filtered_rows['Start'])\n",
    "end = int(filtered_rows['End'])\n",
    "            \n",
    "for record in SeqIO.parse('/bioinfo2/palm/ref/dura/dura_ref.fasta','fasta'):\n",
    "    if record.id == CHR:\n",
    "        seq = record.seq[start-1:end]\n",
    "        pos_ref = record.seq[int(POS)-1]\n",
    "        # check if the length of this exon is divisible by three\n",
    "        if len(seq)%3 != 0:\n",
    "            print(\"warning: this exon is not dividable by 3\")\n",
    "            # check if the position matches with the REF provided\n",
    "        if pos_ref == REF:\n",
    "            # get the coding region containing this SNP\n",
    "            if (int(POS)-start)%3 == 0:\n",
    "                codon = record.seq[int(POS)-1:int(POS)+2]\n",
    "                codon_m = MutableSeq(str(codon))\n",
    "                codon_m[0] = ALT \n",
    "            elif (int(POS)-start)%3 == 1:\n",
    "                codon = record.seq[int(POS)-2:int(POS)+1]\n",
    "                codon_m = MutableSeq(str(codon))\n",
    "                codon_m[1] = ALT\n",
    "            elif (int(POS)-start)%3 == 2:\n",
    "                codon = record.seq[int(POS)-3:int(POS)]\n",
    "                codon_m = MutableSeq(str(codon))\n",
    "                codon_m[2] = ALT\n",
    "            # translate these codon\n",
    "            aa = codon.translate()\n",
    "            aa_m = codon_m.translate()\n",
    "            print(codon, aa, codon_m, aa_m)\n",
    "            break\n",
    "        else:\n",
    "            print(\"This coord in the genome is different from what was called in the vcf\")\n",
    "            break\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a496bd99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Seqid</th>\n",
       "      <th>Source</th>\n",
       "      <th>Type</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Phase</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>_AED</th>\n",
       "      <th>_QI</th>\n",
       "      <th>_eAED</th>\n",
       "      <th>score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>gene</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0...</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0|0|0|0.5|1|1|2|0|472</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>11686</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032749-mRNA-1:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615982</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>281718</td>\n",
       "      <td>281844</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615983</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>280181</td>\n",
       "      <td>280249</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615984</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>279893</td>\n",
       "      <td>280085</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615985</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>278640</td>\n",
       "      <td>279072</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615986</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>278427</td>\n",
       "      <td>278639</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032211-mRNA-2:three_prime_utr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>615987 rows  17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Seqid Source             Type   Start     End Score Strand  \\\n",
       "0       egu.contig.1973  maker             gene   10123   11712     .      +   \n",
       "1       egu.contig.1973  maker             mRNA   10123   11712     .      +   \n",
       "2       egu.contig.1973  maker             exon   10123   11511     .      +   \n",
       "3       egu.contig.1973  maker             exon   11686   11712     .      +   \n",
       "4       egu.contig.1973  maker              CDS   10123   11511     .      +   \n",
       "...                 ...    ...              ...     ...     ...   ...    ...   \n",
       "615982  egu.contig.1149  maker              CDS  281718  281844     .      -   \n",
       "615983  egu.contig.1149  maker              CDS  280181  280249     .      -   \n",
       "615984  egu.contig.1149  maker              CDS  279893  280085     .      -   \n",
       "615985  egu.contig.1149  maker              CDS  278640  279072     .      -   \n",
       "615986  egu.contig.1149  maker  three_prime_UTR  278427  278639     .      -   \n",
       "\n",
       "       Phase                                ID              Name  \\\n",
       "0          .                         Egu032749         Egu032749   \n",
       "1          .                  Egu032749-mRNA-1  Egu032749-mRNA-1   \n",
       "2          .                Egu032749-mRNA-1:1               NaN   \n",
       "3          .                Egu032749-mRNA-1:2               NaN   \n",
       "4          0              Egu032749-mRNA-1:cds               NaN   \n",
       "...      ...                               ...               ...   \n",
       "615982     0              Egu032211-mRNA-2:cds               NaN   \n",
       "615983     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615984     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615985     1              Egu032211-mRNA-2:cds               NaN   \n",
       "615986     .  Egu032211-mRNA-2:three_prime_utr               NaN   \n",
       "\n",
       "                                                    Alias     \\\n",
       "0          snap_masked-egu.contig.1973-processed-gene-0.0      \n",
       "1       snap_masked-egu.contig.1973-processed-gene-0.0...      \n",
       "2                                                     NaN      \n",
       "3                                                     NaN      \n",
       "4                                                     NaN      \n",
       "...                                                   ... ..   \n",
       "615982                                                NaN      \n",
       "615983                                                NaN      \n",
       "615984                                                NaN      \n",
       "615985                                                NaN      \n",
       "615986                                                NaN      \n",
       "\n",
       "                  Parent  _AED                    _QI _eAED score  \n",
       "0                    NaN   NaN                    NaN   NaN   NaN  \n",
       "1              Egu032749  0.06  0|0|0|0.5|1|1|2|0|472  0.04   NaN  \n",
       "2       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "3       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "4       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "...                  ...   ...                    ...   ...   ...  \n",
       "615982  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615983  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615984  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615985  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615986  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "\n",
       "[615987 rows x 17 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "1bd32dad",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Seq import Seq, MutableSeq\n",
    "\n",
    "seq = Seq('GACGTTAGGAGGAGCATATACTAACTTGTTTTTATAA')\n",
    "seq.translate()\n",
    "seq1 = MutableSeq(str(seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "4904af90",
   "metadata": {},
   "outputs": [],
   "source": [
    "aa = Seq('MHLGHRHGHEKGKSPATTMPPKGWVGIRVGQEGEEQQRFEVPVDYLKHPLFMALLHQAKEEFGYEQSGAITIPCGVDHFRHVRDTINRDSAATAAAHHHSHLPHLAGLTLGGAYTNLFL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "ca45f75d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('TLGGAYTNLFL*')"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq = Seq('ACGTTAGGAGGAGCATATACTAACTTGTTTTTATAA')\n",
    "seq.translate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "c7395684",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq = Seq('ATGCACCTGGGCCACAGGCACGGGCATGAGAAGGGAAAGAGTCCGGCGACGACGATGCCACCTAAAGGGTGGGTGGGGATAAGGGTGGGGCAGGAAGGGGAGGAGCAGCAGCGGTTCGAGGTGCCTGTGGACTATCTGAAGCACCCGCTCTTCATGGCCTTGCTGCACCAGGCGAAGGAGGAGTTCGGATATGAGCAGAGCGGAGCCATCACCATTCCCTGCGGCGTCGATCACTTCCGTCATGTCCGGGACACCATTAACCGTGACTCCGCGGCCACCGCCGCCGCGCACCACCACAGCCACCTCCCTCACCTTGCCGGCT')\n",
    "a = seq.translate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "d9c8a778",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "322"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "344739b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('HSHLPHLAGC')"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a[-10:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "0b532e0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'gene': 32778,\n",
       "         'mRNA': 45938,\n",
       "         'exon': 196546,\n",
       "         'CDS': 271952,\n",
       "         'three_prime_UTR': 36830,\n",
       "         'five_prime_UTR': 31943})"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "Counter(df_gff['Type'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "20cc00ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'maker': 615987})"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(df_gff['Source'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d7436432",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>ODB_OG</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>GOs_mf</th>\n",
       "      <th>GOs_bp</th>\n",
       "      <th>EC</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>Interpro</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Egu020222-mRNA-1</td>\n",
       "      <td>687at4447</td>\n",
       "      <td>2.140000e-26</td>\n",
       "      <td>117.8900</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>6.5.1.1,4.1.1.15</td>\n",
       "      <td>ko00250,ko00410,ko00430,ko00650,ko03030,ko0341...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>Egu026424-mRNA-1</td>\n",
       "      <td>2328at4447</td>\n",
       "      <td>4.810000e-66</td>\n",
       "      <td>174.1300</td>\n",
       "      <td>-</td>\n",
       "      <td>Auxin-responsive protein SAUR36</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>663</th>\n",
       "      <td>Egu020226-mRNA-1</td>\n",
       "      <td>5943at4447</td>\n",
       "      <td>5.960000e-43</td>\n",
       "      <td>180.6500</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>Egu020227-mRNA-1</td>\n",
       "      <td>5943at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>148.8970</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3706</th>\n",
       "      <td>Egu011092-mRNA-1</td>\n",
       "      <td>33316at4447</td>\n",
       "      <td>4.290000e-28</td>\n",
       "      <td>86.7100</td>\n",
       "      <td>-</td>\n",
       "      <td>indole-3-acetic acid-induced protein ARG7</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00010,ko01200</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3707</th>\n",
       "      <td>Egu030210-mRNA-1</td>\n",
       "      <td>33316at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>108.0540</td>\n",
       "      <td>-</td>\n",
       "      <td>indole-3-acetic acid-induced protein ARG7</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00010,ko01200</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5388</th>\n",
       "      <td>Egu009584-mRNA-1</td>\n",
       "      <td>47410at4447</td>\n",
       "      <td>9.090000e-29</td>\n",
       "      <td>110.7600</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR32</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>2.7.2.3</td>\n",
       "      <td>ko00010,ko00230,ko00710,ko00910,ko00941,ko0094...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9270</th>\n",
       "      <td>Egu019713-mRNA-1</td>\n",
       "      <td>75523at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>78.6238</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR41</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9271</th>\n",
       "      <td>Egu031500-mRNA-1</td>\n",
       "      <td>75523at4447</td>\n",
       "      <td>3.150000e-23</td>\n",
       "      <td>73.9000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR41</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9614</th>\n",
       "      <td>Egu020383-mRNA-2</td>\n",
       "      <td>77872at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>110.4580</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00270,ko04145,sita01100,sita01110</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9615</th>\n",
       "      <td>Egu020383-mRNA-3</td>\n",
       "      <td>77872at4447</td>\n",
       "      <td>1.390000e-40</td>\n",
       "      <td>127.1300</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00270,ko04145,sita01100,sita01110</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10304</th>\n",
       "      <td>Egu003951-mRNA-1</td>\n",
       "      <td>83452at4447</td>\n",
       "      <td>1.080000e-41</td>\n",
       "      <td>131.3700</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR77</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko03013</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11924</th>\n",
       "      <td>Egu011765-mRNA-1</td>\n",
       "      <td>111397at4447</td>\n",
       "      <td>5.890000e-71</td>\n",
       "      <td>253.4000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR50</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00531,ko03430,ko04075,sita01100</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11925</th>\n",
       "      <td>Egu028450-mRNA-1</td>\n",
       "      <td>111397at4447</td>\n",
       "      <td>5.670000e-69</td>\n",
       "      <td>247.0900</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR50</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00531,ko03430,ko04075,sita01100</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12698</th>\n",
       "      <td>Egu003994-mRNA-1</td>\n",
       "      <td>113001at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>95.8472</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00561,ko00770,ko04075,ko04145</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12699</th>\n",
       "      <td>Egu015371-mRNA-1</td>\n",
       "      <td>113001at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>120.8720</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00561,ko00770,ko04075,ko04145</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12700</th>\n",
       "      <td>Egu020228-mRNA-1</td>\n",
       "      <td>113001at4447</td>\n",
       "      <td>3.740000e-53</td>\n",
       "      <td>152.0300</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR71</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00190,ko00561,ko00770,ko04075,ko04145</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14162</th>\n",
       "      <td>Egu010459-mRNA-1</td>\n",
       "      <td>115938at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>102.0430</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR21-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>4.2.2.2</td>\n",
       "      <td>ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14163</th>\n",
       "      <td>Egu027752-mRNA-1</td>\n",
       "      <td>115938at4447</td>\n",
       "      <td>7.800000e-30</td>\n",
       "      <td>61.0200</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR21-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>4.2.2.2</td>\n",
       "      <td>ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14164</th>\n",
       "      <td>Egu030300-mRNA-1</td>\n",
       "      <td>115938at4447</td>\n",
       "      <td>3.980000e-70</td>\n",
       "      <td>142.4000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR21-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>4.2.2.2</td>\n",
       "      <td>ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17289</th>\n",
       "      <td>Egu009585-mRNA-1</td>\n",
       "      <td>121712at4447</td>\n",
       "      <td>1.720000e-39</td>\n",
       "      <td>128.5100</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR32</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>3.2.1.-</td>\n",
       "      <td>ko00511,ko03060</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20054</th>\n",
       "      <td>Egu031852-mRNA-1</td>\n",
       "      <td>126583at4447</td>\n",
       "      <td>1.190000e-66</td>\n",
       "      <td>187.9700</td>\n",
       "      <td>-</td>\n",
       "      <td>SAUR37-auxin-responsive SAUR family member</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>1.-.-.-</td>\n",
       "      <td>ko03010,ko00380,ko04141,ko04712</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22845</th>\n",
       "      <td>Egu001133-mRNA-1</td>\n",
       "      <td>131384at4447</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>132.8280</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>bdi01100,bdi01110,ko00940,ko03015</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22846</th>\n",
       "      <td>Egu017903-mRNA-1</td>\n",
       "      <td>131384at4447</td>\n",
       "      <td>1.310000e-70</td>\n",
       "      <td>179.3200</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-induced protein 6B</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>bdi01100,bdi01110,ko00940,ko03015</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24776</th>\n",
       "      <td>Egu007739-mRNA-1</td>\n",
       "      <td>134723at4447</td>\n",
       "      <td>8.810000e-88</td>\n",
       "      <td>193.4100</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR50-like</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00904,ko03040</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26035</th>\n",
       "      <td>Egu011864-mRNA-1</td>\n",
       "      <td>136815at4447</td>\n",
       "      <td>2.920000e-71</td>\n",
       "      <td>216.1800</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR36</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26036</th>\n",
       "      <td>Egu028534-mRNA-1</td>\n",
       "      <td>136815at4447</td>\n",
       "      <td>9.670000e-37</td>\n",
       "      <td>164.8000</td>\n",
       "      <td>-</td>\n",
       "      <td>auxin-responsive protein SAUR36</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0009733</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IPR003676</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  query        ODB_OG        evalue     score COG_category  \\\n",
       "75     Egu020222-mRNA-1     687at4447  2.140000e-26  117.8900            -   \n",
       "262    Egu026424-mRNA-1    2328at4447  4.810000e-66  174.1300            -   \n",
       "663    Egu020226-mRNA-1    5943at4447  5.960000e-43  180.6500            -   \n",
       "664    Egu020227-mRNA-1    5943at4447  0.000000e+00  148.8970            -   \n",
       "3706   Egu011092-mRNA-1   33316at4447  4.290000e-28   86.7100            -   \n",
       "3707   Egu030210-mRNA-1   33316at4447  0.000000e+00  108.0540            -   \n",
       "5388   Egu009584-mRNA-1   47410at4447  9.090000e-29  110.7600            -   \n",
       "9270   Egu019713-mRNA-1   75523at4447  0.000000e+00   78.6238            -   \n",
       "9271   Egu031500-mRNA-1   75523at4447  3.150000e-23   73.9000            -   \n",
       "9614   Egu020383-mRNA-2   77872at4447  0.000000e+00  110.4580            -   \n",
       "9615   Egu020383-mRNA-3   77872at4447  1.390000e-40  127.1300            -   \n",
       "10304  Egu003951-mRNA-1   83452at4447  1.080000e-41  131.3700            -   \n",
       "11924  Egu011765-mRNA-1  111397at4447  5.890000e-71  253.4000            -   \n",
       "11925  Egu028450-mRNA-1  111397at4447  5.670000e-69  247.0900            -   \n",
       "12698  Egu003994-mRNA-1  113001at4447  0.000000e+00   95.8472            -   \n",
       "12699  Egu015371-mRNA-1  113001at4447  0.000000e+00  120.8720            -   \n",
       "12700  Egu020228-mRNA-1  113001at4447  3.740000e-53  152.0300            -   \n",
       "14162  Egu010459-mRNA-1  115938at4447  0.000000e+00  102.0430            -   \n",
       "14163  Egu027752-mRNA-1  115938at4447  7.800000e-30   61.0200            -   \n",
       "14164  Egu030300-mRNA-1  115938at4447  3.980000e-70  142.4000            -   \n",
       "17289  Egu009585-mRNA-1  121712at4447  1.720000e-39  128.5100            -   \n",
       "20054  Egu031852-mRNA-1  126583at4447  1.190000e-66  187.9700            -   \n",
       "22845  Egu001133-mRNA-1  131384at4447  0.000000e+00  132.8280            -   \n",
       "22846  Egu017903-mRNA-1  131384at4447  1.310000e-70  179.3200            -   \n",
       "24776  Egu007739-mRNA-1  134723at4447  8.810000e-88  193.4100            -   \n",
       "26035  Egu011864-mRNA-1  136815at4447  2.920000e-71  216.1800            -   \n",
       "26036  Egu028534-mRNA-1  136815at4447  9.670000e-37  164.8000            -   \n",
       "\n",
       "                                      Description GOs_mf      GOs_bp  \\\n",
       "75                auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "262               Auxin-responsive protein SAUR36      -  GO:0009733   \n",
       "663               auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "664               auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "3706    indole-3-acetic acid-induced protein ARG7      -  GO:0009733   \n",
       "3707    indole-3-acetic acid-induced protein ARG7      -  GO:0009733   \n",
       "5388              auxin-responsive protein SAUR32      -  GO:0009733   \n",
       "9270              auxin-responsive protein SAUR41      -  GO:0009733   \n",
       "9271              auxin-responsive protein SAUR41      -  GO:0009733   \n",
       "9614                     auxin-induced protein 6B      -  GO:0009733   \n",
       "9615                     auxin-induced protein 6B      -  GO:0009733   \n",
       "10304             auxin-responsive protein SAUR77      -  GO:0009733   \n",
       "11924             auxin-responsive protein SAUR50      -  GO:0009733   \n",
       "11925             auxin-responsive protein SAUR50      -  GO:0009733   \n",
       "12698             auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "12699             auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "12700             auxin-responsive protein SAUR71      -  GO:0009733   \n",
       "14162        auxin-responsive protein SAUR21-like      -  GO:0009733   \n",
       "14163        auxin-responsive protein SAUR21-like      -  GO:0009733   \n",
       "14164        auxin-responsive protein SAUR21-like      -  GO:0009733   \n",
       "17289             auxin-responsive protein SAUR32      -  GO:0009733   \n",
       "20054  SAUR37-auxin-responsive SAUR family member      -  GO:0009733   \n",
       "22845                    auxin-induced protein 6B      -  GO:0009733   \n",
       "22846                    auxin-induced protein 6B      -  GO:0009733   \n",
       "24776        auxin-responsive protein SAUR50-like      -  GO:0009733   \n",
       "26035             auxin-responsive protein SAUR36      -  GO:0009733   \n",
       "26036             auxin-responsive protein SAUR36      -  GO:0009733   \n",
       "\n",
       "                     EC                                            KEGG_ko  \\\n",
       "75     6.5.1.1,4.1.1.15  ko00250,ko00410,ko00430,ko00650,ko03030,ko0341...   \n",
       "262                   -                                                  -   \n",
       "663                   -                                                  -   \n",
       "664                   -                                                  -   \n",
       "3706                  -                                    ko00010,ko01200   \n",
       "3707                  -                                    ko00010,ko01200   \n",
       "5388            2.7.2.3  ko00010,ko00230,ko00710,ko00910,ko00941,ko0094...   \n",
       "9270                  -                                                  -   \n",
       "9271                  -                                                  -   \n",
       "9614                  -        ko00190,ko00270,ko04145,sita01100,sita01110   \n",
       "9615                  -        ko00190,ko00270,ko04145,sita01100,sita01110   \n",
       "10304                 -                                            ko03013   \n",
       "11924                 -                  ko00531,ko03430,ko04075,sita01100   \n",
       "11925                 -                  ko00531,ko03430,ko04075,sita01100   \n",
       "12698                 -            ko00190,ko00561,ko00770,ko04075,ko04145   \n",
       "12699                 -            ko00190,ko00561,ko00770,ko04075,ko04145   \n",
       "12700                 -            ko00190,ko00561,ko00770,ko04075,ko04145   \n",
       "14162           4.2.2.2  ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...   \n",
       "14163           4.2.2.2  ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...   \n",
       "14164           4.2.2.2  ko00071,ko00511,ko00520,ko00592,ko00600,ko0104...   \n",
       "17289           3.2.1.-                                    ko00511,ko03060   \n",
       "20054           1.-.-.-                    ko03010,ko00380,ko04141,ko04712   \n",
       "22845                 -                  bdi01100,bdi01110,ko00940,ko03015   \n",
       "22846                 -                  bdi01100,bdi01110,ko00940,ko03015   \n",
       "24776                 -                                    ko00904,ko03040   \n",
       "26035                 -                                                  -   \n",
       "26036                 -                                                  -   \n",
       "\n",
       "        Interpro  \n",
       "75     IPR003676  \n",
       "262    IPR003676  \n",
       "663    IPR003676  \n",
       "664    IPR003676  \n",
       "3706   IPR003676  \n",
       "3707   IPR003676  \n",
       "5388   IPR003676  \n",
       "9270   IPR003676  \n",
       "9271   IPR003676  \n",
       "9614   IPR003676  \n",
       "9615   IPR003676  \n",
       "10304  IPR003676  \n",
       "11924  IPR003676  \n",
       "11925  IPR003676  \n",
       "12698  IPR003676  \n",
       "12699  IPR003676  \n",
       "12700  IPR003676  \n",
       "14162  IPR003676  \n",
       "14163  IPR003676  \n",
       "14164  IPR003676  \n",
       "17289  IPR003676  \n",
       "20054  IPR003676  \n",
       "22845  IPR003676  \n",
       "22846  IPR003676  \n",
       "24776  IPR003676  \n",
       "26035  IPR003676  \n",
       "26036  IPR003676  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Extract values from column B and C from the filtered rows\n",
    "result_values = filtered_rows[['Seqid', 'Start','End','ID']]\n",
    "\n",
    "# Convert columns A and B to numeric\n",
    "result_values['Start'] = pd.to_numeric(result_values['Start'])\n",
    "result_values['End'] = pd.to_numeric(result_values['End'])\n",
    "\n",
    "# Perform the operations and apply the conditions\n",
    "result_values['Start'] = result_values['Start'] - flanking\n",
    "result_values['Start'] = result_values['Start'].apply(lambda x: max(0, x))  # Set negative values to 0\n",
    "\n",
    "result_values['End'] = result_values['End'] + flanking\n",
    "bedfile = fileA.split('_blastp.tbl')[0] + '_' + str(flanking) + '.bed'\n",
    "result_values.to_csv(bedfile, sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "4819ff37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'snap_masked-egu.contig.1973-processed-gene-0.0': 1,\n",
       "         'snap_masked-egu.contig.1973-processed-gene-0.0-mRNA-1': 1,\n",
       "         nan: 537271,\n",
       "         'maker-egu.contig.1477-snap-gene-0.0': 1,\n",
       "         'maker-egu.contig.1477-snap-gene-0.0-mRNA-1': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.4': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.4-mRNA-1': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.5': 1,\n",
       "         'maker-egu.contig.1784-snap-gene-0.5-mRNA-1': 1,\n",
       "         'snap_masked-egu.contig.1415-processed-gene-0.4': 1,\n",
       "         'snap_masked-egu.contig.1415-processed-gene-0.4-mRNA-1': 1,\n",
       "         'snap_masked-egu.contig.1962-processed-gene-0.2': 1,\n",
       "         'snap_masked-egu.contig.1962-processed-gene-0.2-mRNA-1': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.2': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.2-mRNA-1': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.0': 1,\n",
       "         'maker-egu.contig.1962-snap-gene-0.0-mRNA-1': 1,\n",
       "         'maker-egu.contig.1588-snap-gene-0.1': 1,\n",
       "         'maker-egu.contig.1588-snap-gene-0.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.1': 1,\n",
       "         'maker-chr9-snap-gene-0.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.1-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-0.2': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-0.2-mRNA-4': 1,\n",
       "         'snap_masked-chr9-processed-gene-0.11': 1,\n",
       "         'snap_masked-chr9-processed-gene-0.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.0': 1,\n",
       "         'maker-chr9-snap-gene-0.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-0.0-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-0.0-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-0.3': 1,\n",
       "         'maker-chr9-snap-gene-0.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.24': 1,\n",
       "         'maker-chr9-snap-gene-1.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.23': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-1.23-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-1.26': 1,\n",
       "         'maker-chr9-snap-gene-1.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.28': 1,\n",
       "         'maker-chr9-snap-gene-1.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.28-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-1.27': 1,\n",
       "         'maker-chr9-snap-gene-1.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.25': 1,\n",
       "         'maker-chr9-snap-gene-1.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-1.25-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.1': 1,\n",
       "         'maker-chr9-snap-gene-2.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.4': 1,\n",
       "         'maker-chr9-snap-gene-2.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.12': 1,\n",
       "         'maker-chr9-snap-gene-2.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.12-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.13': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-2.13-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-2.10': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.10-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-2.6': 1,\n",
       "         'maker-chr9-snap-gene-2.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-2.30': 1,\n",
       "         'snap_masked-chr9-processed-gene-2.30-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.9': 1,\n",
       "         'maker-chr9-snap-gene-2.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.9-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.5': 1,\n",
       "         'maker-chr9-snap-gene-2.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.7': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-7': 1,\n",
       "         'maker-chr9-snap-gene-2.7-mRNA-8': 1,\n",
       "         'maker-chr9-snap-gene-3.31': 1,\n",
       "         'maker-chr9-snap-gene-3.31-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.12-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.2-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.30': 1,\n",
       "         'maker-chr9-snap-gene-3.30-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.30-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.24': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.34': 1,\n",
       "         'maker-chr9-snap-gene-3.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.37': 1,\n",
       "         'maker-chr9-snap-gene-3.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.32': 1,\n",
       "         'maker-chr9-snap-gene-3.32-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.28': 1,\n",
       "         'snap_masked-chr9-processed-gene-3.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.35': 1,\n",
       "         'maker-chr9-snap-gene-3.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-3.24': 1,\n",
       "         'maker-chr9-snap-gene-3.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.8': 1,\n",
       "         'maker-chr9-snap-gene-4.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-4.1': 1,\n",
       "         'maker-chr9-snap-gene-4.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.2': 1,\n",
       "         'maker-chr9-snap-gene-4.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.4': 1,\n",
       "         'maker-chr9-snap-gene-4.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.3': 1,\n",
       "         'maker-chr9-snap-gene-4.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.3-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.35': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.7': 1,\n",
       "         'maker-chr9-snap-gene-4.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.5': 1,\n",
       "         'maker-chr9-snap-gene-4.5-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.37': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.37-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.29': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-4.11': 1,\n",
       "         'maker-chr9-snap-gene-4.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.34': 1,\n",
       "         'snap_masked-chr9-processed-gene-4.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.4': 1,\n",
       "         'maker-chr9-snap-gene-5.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.0': 1,\n",
       "         'maker-chr9-snap-gene-5.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.40': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.40-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.6': 1,\n",
       "         'maker-chr9-snap-gene-5.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.34': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.17': 1,\n",
       "         'maker-chr9-snap-gene-5.17-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.44': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.44-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.27-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.7': 1,\n",
       "         'maker-chr9-snap-gene-5.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.14': 1,\n",
       "         'maker-chr9-snap-gene-5.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.14-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.48': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.48-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.52': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.52-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.11': 1,\n",
       "         'maker-chr9-snap-gene-5.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.50': 1,\n",
       "         'snap_masked-chr9-processed-gene-5.50-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.15': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-5.15-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-5.13': 1,\n",
       "         'maker-chr9-snap-gene-5.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.12': 1,\n",
       "         'maker-chr9-snap-gene-5.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.8': 1,\n",
       "         'maker-chr9-snap-gene-5.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-5.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-5.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-5.9': 1,\n",
       "         'maker-chr9-snap-gene-5.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.0': 1,\n",
       "         'maker-chr9-snap-gene-6.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.16': 1,\n",
       "         'maker-chr9-snap-gene-6.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.1': 1,\n",
       "         'maker-chr9-snap-gene-6.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.5': 1,\n",
       "         'maker-chr9-snap-gene-6.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.5-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-6.8': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-6.8-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-6.4': 1,\n",
       "         'maker-chr9-snap-gene-6.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.6': 1,\n",
       "         'maker-chr9-snap-gene-6.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.6-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-6.6-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-6.7': 1,\n",
       "         'maker-chr9-snap-gene-6.7-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.17': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.17-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.14': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.14-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.30': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.30-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.31': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.10': 1,\n",
       "         'maker-chr9-snap-gene-6.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.9': 1,\n",
       "         'maker-chr9-snap-gene-6.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.11': 1,\n",
       "         'maker-chr9-snap-gene-6.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.35': 1,\n",
       "         'snap_masked-chr9-processed-gene-6.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.12': 1,\n",
       "         'maker-chr9-snap-gene-6.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.27': 1,\n",
       "         'maker-chr9-snap-gene-6.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-6.27-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.12': 1,\n",
       "         'maker-chr9-snap-gene-7.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.5': 1,\n",
       "         'maker-chr9-snap-gene-7.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.9': 1,\n",
       "         'maker-chr9-snap-gene-7.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.13': 1,\n",
       "         'maker-chr9-snap-gene-7.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.4': 1,\n",
       "         'maker-chr9-snap-gene-7.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.10': 1,\n",
       "         'maker-chr9-snap-gene-7.10-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.11': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.11': 1,\n",
       "         'maker-chr9-snap-gene-7.11-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.32': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.8': 1,\n",
       "         'maker-chr9-snap-gene-7.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.1': 1,\n",
       "         'maker-chr9-snap-gene-7.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.0': 1,\n",
       "         'maker-chr9-snap-gene-7.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.41': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.40': 1,\n",
       "         'maker-chr9-snap-gene-7.40-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.37': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.2': 1,\n",
       "         'maker-chr9-snap-gene-7.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.7': 1,\n",
       "         'maker-chr9-snap-gene-7.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.6': 1,\n",
       "         'maker-chr9-snap-gene-7.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.17': 1,\n",
       "         'maker-chr9-snap-gene-7.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.15': 1,\n",
       "         'maker-chr9-snap-gene-7.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.15-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.14': 1,\n",
       "         'maker-chr9-snap-gene-7.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.18': 1,\n",
       "         'maker-chr9-snap-gene-7.18-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.18-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.22': 1,\n",
       "         'maker-chr9-snap-gene-7.22-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.45': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.45-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.21': 1,\n",
       "         'maker-chr9-snap-gene-7.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.21-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.21-mRNA-3': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.44': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.44-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.52': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.52-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.19': 1,\n",
       "         'maker-chr9-snap-gene-7.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.19-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-7.20': 1,\n",
       "         'maker-chr9-snap-gene-7.20-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.38': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.16': 1,\n",
       "         'maker-chr9-snap-gene-7.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-7.16-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.56': 1,\n",
       "         'snap_masked-chr9-processed-gene-7.56-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.28': 1,\n",
       "         'maker-chr9-snap-gene-8.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.36': 1,\n",
       "         'maker-chr9-snap-gene-8.36-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.29': 1,\n",
       "         'maker-chr9-snap-gene-8.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.29-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.29': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.3': 1,\n",
       "         'maker-chr9-snap-gene-8.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.3-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-8.32': 1,\n",
       "         'maker-chr9-snap-gene-8.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.27': 1,\n",
       "         'maker-chr9-snap-gene-8.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.27-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.31': 1,\n",
       "         'maker-chr9-snap-gene-8.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.33': 1,\n",
       "         'maker-chr9-snap-gene-8.33-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.42': 1,\n",
       "         'maker-chr9-snap-gene-8.42-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.44': 1,\n",
       "         'maker-chr9-snap-gene-8.44-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.41': 1,\n",
       "         'maker-chr9-snap-gene-8.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.37': 1,\n",
       "         'maker-chr9-snap-gene-8.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.12': 1,\n",
       "         'maker-chr9-snap-gene-8.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.38': 1,\n",
       "         'maker-chr9-snap-gene-8.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.43': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-8.43-mRNA-4': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.25': 1,\n",
       "         'snap_masked-chr9-processed-gene-8.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.45': 1,\n",
       "         'maker-chr9-snap-gene-8.45-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-8.45-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-8.45-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-8.40': 1,\n",
       "         'maker-chr9-snap-gene-8.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.27': 1,\n",
       "         'maker-chr9-snap-gene-9.27-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.4': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.3': 1,\n",
       "         'maker-chr9-snap-gene-9.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.3-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-9.3-mRNA-3': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.31': 1,\n",
       "         'maker-chr9-snap-gene-9.31-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.34': 1,\n",
       "         'maker-chr9-snap-gene-9.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.38': 1,\n",
       "         'maker-chr9-snap-gene-9.38-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.37': 1,\n",
       "         'maker-chr9-snap-gene-9.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.37-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-9.32': 1,\n",
       "         'maker-chr9-snap-gene-9.32-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.25': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.25-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.18': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.18-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.15': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.15-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.20': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.33': 1,\n",
       "         'maker-chr9-snap-gene-9.33-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.16': 1,\n",
       "         'snap_masked-chr9-processed-gene-9.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-9.35': 1,\n",
       "         'maker-chr9-snap-gene-9.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.2': 1,\n",
       "         'maker-chr9-snap-gene-10.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.1': 1,\n",
       "         'maker-chr9-snap-gene-10.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.1-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.0': 1,\n",
       "         'maker-chr9-snap-gene-10.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.17': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.17-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.20': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.4': 1,\n",
       "         'maker-chr9-snap-gene-10.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.3': 1,\n",
       "         'maker-chr9-snap-gene-10.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.5': 1,\n",
       "         'maker-chr9-snap-gene-10.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.11': 1,\n",
       "         'maker-chr9-snap-gene-10.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.11-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.10': 1,\n",
       "         'maker-chr9-snap-gene-10.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.6': 1,\n",
       "         'maker-chr9-snap-gene-10.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.6-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.31': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.31-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-10.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.8': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-10.8-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-10.9': 1,\n",
       "         'maker-chr9-snap-gene-10.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-10.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-10.7': 1,\n",
       "         'maker-chr9-snap-gene-10.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.24': 1,\n",
       "         'maker-chr9-snap-gene-11.24-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.0': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.12-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.3': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.3-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.20': 1,\n",
       "         'maker-chr9-snap-gene-11.20-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.4': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.7': 1,\n",
       "         'maker-chr9-snap-gene-11.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-11.7-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-11.25': 1,\n",
       "         'maker-chr9-snap-gene-11.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.25-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.11': 1,\n",
       "         'snap_masked-chr9-processed-gene-11.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.5': 1,\n",
       "         'maker-chr9-snap-gene-11.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-11.5-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.6': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-12.12': 1,\n",
       "         'maker-chr9-snap-gene-12.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-12.1': 1,\n",
       "         'maker-chr9-snap-gene-12.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-12.1-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.0': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.0-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.1': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-12.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.31': 1,\n",
       "         'maker-chr9-snap-gene-13.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-13.33': 1,\n",
       "         'maker-chr9-snap-gene-13.33-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.10-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.2': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.2-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.9': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.9-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.8': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.8-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.39': 1,\n",
       "         'maker-chr9-snap-gene-13.39-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.20': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.20-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.28': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.28-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.19': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.37': 1,\n",
       "         'maker-chr9-snap-gene-13.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.37-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-13.34': 1,\n",
       "         'maker-chr9-snap-gene-13.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.34-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-13.38': 1,\n",
       "         'maker-chr9-snap-gene-13.38-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.35': 1,\n",
       "         'maker-chr9-snap-gene-13.35-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.16': 1,\n",
       "         'snap_masked-chr9-processed-gene-13.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.40': 1,\n",
       "         'maker-chr9-snap-gene-13.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-13.2': 1,\n",
       "         'maker-chr9-snap-gene-13.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.8': 1,\n",
       "         'maker-chr9-snap-gene-14.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-14.8-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-14.7': 1,\n",
       "         'maker-chr9-snap-gene-14.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.0': 1,\n",
       "         'maker-chr9-snap-gene-14.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.9': 1,\n",
       "         'maker-chr9-snap-gene-14.9-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.4': 1,\n",
       "         'maker-chr9-snap-gene-14.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.6': 1,\n",
       "         'maker-chr9-snap-gene-14.6-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.31': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.3': 1,\n",
       "         'maker-chr9-snap-gene-14.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.25': 1,\n",
       "         'maker-chr9-snap-gene-14.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.16': 1,\n",
       "         'maker-chr9-snap-gene-14.16-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.46': 1,\n",
       "         'snap_masked-chr9-processed-gene-14.46-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.20': 1,\n",
       "         'maker-chr9-snap-gene-14.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.21': 1,\n",
       "         'maker-chr9-snap-gene-14.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.18': 1,\n",
       "         'maker-chr9-snap-gene-14.18-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.10': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-6': 1,\n",
       "         'maker-chr9-snap-gene-14.10-mRNA-7': 1,\n",
       "         'maker-chr9-snap-gene-14.14': 1,\n",
       "         'maker-chr9-snap-gene-14.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.17': 1,\n",
       "         'maker-chr9-snap-gene-14.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.22': 1,\n",
       "         'maker-chr9-snap-gene-14.22-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.13': 1,\n",
       "         'maker-chr9-snap-gene-14.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.13-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-14.15': 1,\n",
       "         'maker-chr9-snap-gene-14.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.23': 1,\n",
       "         'maker-chr9-snap-gene-14.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.11': 1,\n",
       "         'maker-chr9-snap-gene-14.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.12': 1,\n",
       "         'maker-chr9-snap-gene-14.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-14.24': 1,\n",
       "         'maker-chr9-snap-gene-14.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.19': 1,\n",
       "         'maker-chr9-snap-gene-15.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.19-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.4': 1,\n",
       "         'maker-chr9-snap-gene-15.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.4-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.3': 1,\n",
       "         'maker-chr9-snap-gene-15.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.0': 1,\n",
       "         'maker-chr9-snap-gene-15.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.1': 1,\n",
       "         'maker-chr9-snap-gene-15.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.1-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-15.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-15.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.8': 1,\n",
       "         'maker-chr9-snap-gene-15.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.9': 1,\n",
       "         'maker-chr9-snap-gene-15.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.12': 1,\n",
       "         'maker-chr9-snap-gene-15.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.12-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.17': 1,\n",
       "         'maker-chr9-snap-gene-15.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.16': 1,\n",
       "         'maker-chr9-snap-gene-15.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.6': 1,\n",
       "         'maker-chr9-snap-gene-15.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.13': 1,\n",
       "         'maker-chr9-snap-gene-15.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.5': 1,\n",
       "         'maker-chr9-snap-gene-15.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.7': 1,\n",
       "         'maker-chr9-snap-gene-15.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.10': 1,\n",
       "         'maker-chr9-snap-gene-15.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.10-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-15.10-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-15.15': 1,\n",
       "         'maker-chr9-snap-gene-15.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-15.14': 1,\n",
       "         'maker-chr9-snap-gene-15.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.1': 1,\n",
       "         'maker-chr9-snap-gene-16.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.6': 1,\n",
       "         'maker-chr9-snap-gene-16.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.3': 1,\n",
       "         'maker-chr9-snap-gene-16.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.3-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.5': 1,\n",
       "         'maker-chr9-snap-gene-16.5-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.16': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.0': 1,\n",
       "         'maker-chr9-snap-gene-16.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.4': 1,\n",
       "         'maker-chr9-snap-gene-16.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.2': 1,\n",
       "         'maker-chr9-snap-gene-16.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.2-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.2-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-16.8': 1,\n",
       "         'maker-chr9-snap-gene-16.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.9': 1,\n",
       "         'maker-chr9-snap-gene-16.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.9-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-16.7': 1,\n",
       "         'maker-chr9-snap-gene-16.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.11': 1,\n",
       "         'maker-chr9-snap-gene-16.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.10': 1,\n",
       "         'maker-chr9-snap-gene-16.10-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.34': 1,\n",
       "         'snap_masked-chr9-processed-gene-16.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.14': 1,\n",
       "         'maker-chr9-snap-gene-16.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.14-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-16.14-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-16.13': 1,\n",
       "         'maker-chr9-snap-gene-16.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-16.12': 1,\n",
       "         'maker-chr9-snap-gene-16.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.4': 1,\n",
       "         'maker-chr9-snap-gene-17.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.1': 1,\n",
       "         'maker-chr9-snap-gene-17.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.1-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.1-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-17.3': 1,\n",
       "         'maker-chr9-snap-gene-17.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.8': 1,\n",
       "         'maker-chr9-snap-gene-17.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.6': 1,\n",
       "         'maker-chr9-snap-gene-17.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.0': 1,\n",
       "         'maker-chr9-snap-gene-17.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.0-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.0-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-17.10': 1,\n",
       "         'maker-chr9-snap-gene-17.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.9': 1,\n",
       "         'maker-chr9-snap-gene-17.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.2': 1,\n",
       "         'maker-chr9-snap-gene-17.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.7': 1,\n",
       "         'maker-chr9-snap-gene-17.7-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.7-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.5': 1,\n",
       "         'maker-chr9-snap-gene-17.5-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-17.23': 1,\n",
       "         'snap_masked-chr9-processed-gene-17.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.31': 1,\n",
       "         'maker-chr9-snap-gene-17.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.13': 1,\n",
       "         'maker-chr9-snap-gene-17.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.20': 1,\n",
       "         'maker-chr9-snap-gene-17.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.12': 1,\n",
       "         'maker-chr9-snap-gene-17.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.11': 1,\n",
       "         'maker-chr9-snap-gene-17.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.15': 1,\n",
       "         'maker-chr9-snap-gene-17.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.14': 1,\n",
       "         'maker-chr9-snap-gene-17.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.19': 1,\n",
       "         'maker-chr9-snap-gene-17.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.19-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-17.19-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-17.17': 1,\n",
       "         'maker-chr9-snap-gene-17.17-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-17.44': 1,\n",
       "         'maker-chr9-snap-gene-17.44-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.5': 1,\n",
       "         'maker-chr9-snap-gene-18.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.1': 1,\n",
       "         'maker-chr9-snap-gene-18.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.3': 1,\n",
       "         'maker-chr9-snap-gene-18.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.2': 1,\n",
       "         'maker-chr9-snap-gene-18.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.4': 1,\n",
       "         'maker-chr9-snap-gene-18.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.4-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.28': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.28-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.14': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.14-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-18.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.9': 1,\n",
       "         'maker-chr9-snap-gene-18.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.6': 1,\n",
       "         'maker-chr9-snap-gene-18.6-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.6-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-18.6-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-18.11': 1,\n",
       "         'maker-chr9-snap-gene-18.11-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.21': 1,\n",
       "         'maker-chr9-snap-gene-18.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.8': 1,\n",
       "         'maker-chr9-snap-gene-18.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.8-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-18.18': 1,\n",
       "         'maker-chr9-snap-gene-18.18-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-18.18-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-19.30': 1,\n",
       "         'maker-chr9-snap-gene-19.30-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.24': 1,\n",
       "         'maker-chr9-snap-gene-19.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.26': 1,\n",
       "         'maker-chr9-snap-gene-19.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.23': 1,\n",
       "         'maker-chr9-snap-gene-19.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.29': 1,\n",
       "         'maker-chr9-snap-gene-19.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.27': 1,\n",
       "         'maker-chr9-snap-gene-19.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.31': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-19.31-mRNA-4': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.12': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.40': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-19.40-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-19.46': 1,\n",
       "         'maker-chr9-snap-gene-19.46-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.35': 1,\n",
       "         'maker-chr9-snap-gene-19.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.33': 1,\n",
       "         'maker-chr9-snap-gene-19.33-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.37': 1,\n",
       "         'maker-chr9-snap-gene-19.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.36': 1,\n",
       "         'maker-chr9-snap-gene-19.36-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.19': 1,\n",
       "         'snap_masked-chr9-processed-gene-19.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-19.38': 1,\n",
       "         'maker-chr9-snap-gene-19.38-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.36': 1,\n",
       "         'maker-chr9-snap-gene-20.36-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.38': 1,\n",
       "         'maker-chr9-snap-gene-20.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.38-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.32': 1,\n",
       "         'maker-chr9-snap-gene-20.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.34': 1,\n",
       "         'maker-chr9-snap-gene-20.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.34-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.33': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.33-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.27': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.50': 1,\n",
       "         'maker-chr9-snap-gene-20.50-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.43': 1,\n",
       "         'maker-chr9-snap-gene-20.43-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.48': 1,\n",
       "         'maker-chr9-snap-gene-20.48-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.45': 1,\n",
       "         'maker-chr9-snap-gene-20.45-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.45-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.47': 1,\n",
       "         'maker-chr9-snap-gene-20.47-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.47-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.44': 1,\n",
       "         'maker-chr9-snap-gene-20.44-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.40': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-20.40-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-20.42': 1,\n",
       "         'maker-chr9-snap-gene-20.42-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.42-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.26': 1,\n",
       "         'snap_masked-chr9-processed-gene-20.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.15': 1,\n",
       "         'maker-chr9-snap-gene-20.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-20.15-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-20.15-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-20.41': 1,\n",
       "         'maker-chr9-snap-gene-20.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.20': 1,\n",
       "         'maker-chr9-snap-gene-21.20-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.9': 1,\n",
       "         'maker-chr9-snap-gene-21.9-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.9-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.0': 1,\n",
       "         'maker-chr9-snap-gene-21.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.12': 1,\n",
       "         'maker-chr9-snap-gene-21.12-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.27': 1,\n",
       "         'maker-chr9-snap-gene-21.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.27-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.24': 1,\n",
       "         'maker-chr9-snap-gene-21.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.24-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.26': 1,\n",
       "         'maker-chr9-snap-gene-21.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.26-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.26-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-21.5': 1,\n",
       "         'maker-chr9-snap-gene-21.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.2': 1,\n",
       "         'maker-chr9-snap-gene-21.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.4': 1,\n",
       "         'maker-chr9-snap-gene-21.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.4-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.4-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-21.1': 1,\n",
       "         'maker-chr9-snap-gene-21.1-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.35': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.23': 1,\n",
       "         'maker-chr9-snap-gene-21.23-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.23-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.19': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.19-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.14': 1,\n",
       "         'maker-chr9-snap-gene-21.14-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.14-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-21.16': 1,\n",
       "         'maker-chr9-snap-gene-21.16-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.13': 1,\n",
       "         'maker-chr9-snap-gene-21.13-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.13-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.47': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.47-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.46': 1,\n",
       "         'snap_masked-chr9-processed-gene-21.46-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-21.15': 1,\n",
       "         'maker-chr9-snap-gene-21.15-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.34': 1,\n",
       "         'maker-chr9-snap-gene-22.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.31': 1,\n",
       "         'maker-chr9-snap-gene-22.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.31-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.32': 1,\n",
       "         'maker-chr9-snap-gene-22.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.32-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.32-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.35': 1,\n",
       "         'maker-chr9-snap-gene-22.35-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-22.3': 1,\n",
       "         'snap_masked-chr9-processed-gene-22.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.33': 1,\n",
       "         'maker-chr9-snap-gene-22.33-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.33-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.41': 1,\n",
       "         'maker-chr9-snap-gene-22.41-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.39': 1,\n",
       "         'maker-chr9-snap-gene-22.39-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.43': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.43-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-22.40': 1,\n",
       "         'maker-chr9-snap-gene-22.40-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.40-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.40-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-22.37': 1,\n",
       "         'maker-chr9-snap-gene-22.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.37-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.38': 1,\n",
       "         'maker-chr9-snap-gene-22.38-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-22.38-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-22.38-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-23.4': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-23.4-mRNA-5': 1,\n",
       "         'maker-chr9-snap-gene-23.0': 1,\n",
       "         'maker-chr9-snap-gene-23.0-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.1': 1,\n",
       "         'maker-chr9-snap-gene-23.1-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.3': 1,\n",
       "         'maker-chr9-snap-gene-23.3-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.2': 1,\n",
       "         'maker-chr9-snap-gene-23.2-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.2-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-23.5': 1,\n",
       "         'maker-chr9-snap-gene-23.5-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.7': 1,\n",
       "         'maker-chr9-snap-gene-23.7-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.22': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.22-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.8': 1,\n",
       "         'maker-chr9-snap-gene-23.8-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-23.8-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.21': 1,\n",
       "         'snap_masked-chr9-processed-gene-23.21-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.27': 1,\n",
       "         'maker-chr9-snap-gene-24.27-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.27-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.31': 1,\n",
       "         'maker-chr9-snap-gene-24.31-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.31-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.32': 1,\n",
       "         'maker-chr9-snap-gene-24.32-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.28': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-24.28-mRNA-4': 1,\n",
       "         'maker-chr9-snap-gene-24.26': 1,\n",
       "         'maker-chr9-snap-gene-24.26-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.26-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.29': 1,\n",
       "         'maker-chr9-snap-gene-24.29-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.29-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.29-mRNA-3': 1,\n",
       "         'maker-chr9-snap-gene-24.25': 1,\n",
       "         'maker-chr9-snap-gene-24.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.25-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.37': 1,\n",
       "         'maker-chr9-snap-gene-24.37-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.34': 1,\n",
       "         'maker-chr9-snap-gene-24.34-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.34-mRNA-2': 1,\n",
       "         'maker-chr9-snap-gene-24.36': 1,\n",
       "         'maker-chr9-snap-gene-24.36-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.22': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.22-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.24': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.24-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.35': 1,\n",
       "         'maker-chr9-snap-gene-24.35-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-24.35-mRNA-2': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.25': 1,\n",
       "         'snap_masked-chr9-processed-gene-24.25-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-25.28': 1,\n",
       "         'maker-chr9-snap-gene-25.28-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-25.27': 1,\n",
       "         'maker-chr9-snap-gene-25.27-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.10': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.10-mRNA-1': 1,\n",
       "         'maker-chr9-snap-gene-25.21': 1,\n",
       "         'maker-chr9-snap-gene-25.21-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.3': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.3-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.4': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.4-mRNA-1': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.7': 1,\n",
       "         'snap_masked-chr9-processed-gene-25.7-mRNA-1': 1,\n",
       "         ...})"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(df_gff['Alias'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "ea710868",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Seqid</th>\n",
       "      <th>Source</th>\n",
       "      <th>Type</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Phase</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>_AED</th>\n",
       "      <th>_QI</th>\n",
       "      <th>_eAED</th>\n",
       "      <th>score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>gene</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>10123</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>snap_masked-egu.contig.1973-processed-gene-0.0...</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0|0|0|0.5|1|1|2|0|472</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>exon</td>\n",
       "      <td>11686</td>\n",
       "      <td>11712</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032749-mRNA-1:2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>egu.contig.1973</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>10123</td>\n",
       "      <td>11511</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032749-mRNA-1:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032749-mRNA-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615982</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>281718</td>\n",
       "      <td>281844</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615983</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>280181</td>\n",
       "      <td>280249</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615984</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>279893</td>\n",
       "      <td>280085</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615985</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>CDS</td>\n",
       "      <td>278640</td>\n",
       "      <td>279072</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>Egu032211-mRNA-2:cds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615986</th>\n",
       "      <td>egu.contig.1149</td>\n",
       "      <td>maker</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>278427</td>\n",
       "      <td>278639</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>Egu032211-mRNA-2:three_prime_utr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>Egu032211-mRNA-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>615987 rows  17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Seqid Source             Type   Start     End Score Strand  \\\n",
       "0       egu.contig.1973  maker             gene   10123   11712     .      +   \n",
       "1       egu.contig.1973  maker             mRNA   10123   11712     .      +   \n",
       "2       egu.contig.1973  maker             exon   10123   11511     .      +   \n",
       "3       egu.contig.1973  maker             exon   11686   11712     .      +   \n",
       "4       egu.contig.1973  maker              CDS   10123   11511     .      +   \n",
       "...                 ...    ...              ...     ...     ...   ...    ...   \n",
       "615982  egu.contig.1149  maker              CDS  281718  281844     .      -   \n",
       "615983  egu.contig.1149  maker              CDS  280181  280249     .      -   \n",
       "615984  egu.contig.1149  maker              CDS  279893  280085     .      -   \n",
       "615985  egu.contig.1149  maker              CDS  278640  279072     .      -   \n",
       "615986  egu.contig.1149  maker  three_prime_UTR  278427  278639     .      -   \n",
       "\n",
       "       Phase                                ID              Name  \\\n",
       "0          .                         Egu032749         Egu032749   \n",
       "1          .                  Egu032749-mRNA-1  Egu032749-mRNA-1   \n",
       "2          .                Egu032749-mRNA-1:1               NaN   \n",
       "3          .                Egu032749-mRNA-1:2               NaN   \n",
       "4          0              Egu032749-mRNA-1:cds               NaN   \n",
       "...      ...                               ...               ...   \n",
       "615982     0              Egu032211-mRNA-2:cds               NaN   \n",
       "615983     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615984     2              Egu032211-mRNA-2:cds               NaN   \n",
       "615985     1              Egu032211-mRNA-2:cds               NaN   \n",
       "615986     .  Egu032211-mRNA-2:three_prime_utr               NaN   \n",
       "\n",
       "                                                    Alias     \\\n",
       "0          snap_masked-egu.contig.1973-processed-gene-0.0      \n",
       "1       snap_masked-egu.contig.1973-processed-gene-0.0...      \n",
       "2                                                     NaN      \n",
       "3                                                     NaN      \n",
       "4                                                     NaN      \n",
       "...                                                   ... ..   \n",
       "615982                                                NaN      \n",
       "615983                                                NaN      \n",
       "615984                                                NaN      \n",
       "615985                                                NaN      \n",
       "615986                                                NaN      \n",
       "\n",
       "                  Parent  _AED                    _QI _eAED score  \n",
       "0                    NaN   NaN                    NaN   NaN   NaN  \n",
       "1              Egu032749  0.06  0|0|0|0.5|1|1|2|0|472  0.04   NaN  \n",
       "2       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "3       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "4       Egu032749-mRNA-1   NaN                    NaN   NaN   NaN  \n",
       "...                  ...   ...                    ...   ...   ...  \n",
       "615982  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615983  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615984  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615985  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "615986  Egu032211-mRNA-2   NaN                    NaN   NaN   NaN  \n",
       "\n",
       "[615987 rows x 17 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9855549",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "i = 0\n",
    "protein_file = '/bioinfo/palm/ref/dura/dura_proteins.fasta'\n",
    "with open('dura_proteins_{}_at{}.aa'.format(gene_name, level),'w') as fh:\n",
    "    for record in SeqIO.parse(prselected_rowsotein_file,'fasta'):\n",
    "        if record.id in ['query'].values:\n",
    "            cluster = selected_rows.loc[selected_rows['query'] == record.id, 'ODB_OG']\n",
    "            tag = selected_rows.loc[selected_rows['query'] == record.id, 'Description']\n",
    "            if len(tag) == 1:\n",
    "                record.description = ':'.join([cluster.iloc[0], tag.iloc[0]])\n",
    "            else:\n",
    "                print(record.id + 'appeared in more than one cluster?')\n",
    "\n",
    "            SeqIO.write(record, fh, 'fasta')\n",
    "            i += 1\n",
    "print(i)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (bioinfo)",
   "language": "python",
   "name": "bioinfo"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
